### Machine learning modeling strategies for high-dimensional -omics data

Biomedical datasets, including those in cancer genomics, often contain more input predictors than data samples [@doi:10.1109/JPROC.2015.2494198; @arxiv:1611.09340].
A genetic study may genotype millions of single nucleotide polymorphisms (SNPs) in thousands of individuals, or a gene expression study may profile the expression of thousands of genes in tens of samples.
Thus, it can be useful to include prior information describing relationships between predictors to inform the representation learned by the model.
This contrasts with non-biological applications of machine learning, where one might fit a model on millions of images [@doi:10.1109/CVPR.2009.5206848] or tens of thousands of documents [@url:https://www.aclweb.org/anthology/P11-1015/], making inclusion of prior information unnecessary.
In this section, we review existing machine learning methods to take into account structure in the input dataset, particularly for large genomics datasets with correlations between the input predictors or features.

#### Incorporating structure into models using gene expression data

Models built from gene expression data can benefit from incorporating gene-level relationships.
One form that this knowledge commonly takes is a database of gene sets, which may represent biological pathways or gene signatures for a biological state of interest.
PLIER [@doi:10.1038/s41592-019-0456-1] uses gene set information from MSigDB [@doi:10.1073/pnas.0506580102] and cell type markers to extract a representation of gene expression data that corresponds to biological processes and reduces technical noise.
The resulting gene set-aligned representation accurately decomposed cell type mixtures.
MultiPLIER [@doi:10.1016/j.cels.2019.04.003] applied PLIER to the recount2 gene expression compendium [@doi:10.1038/nbt.3838] to develop a model that shares information across multiple tissues and diseases, including rare diseases with limited sample sizes.
PASNet [@doi:10.1186/s12859-018-2500-z] uses MSigDB to inform the structure of a neural network for predicting patient outcomes in glioblastoma multiforme (GBM) from gene expression data.
This approach aids interpretation, as pathway nodes in the network with high weights can be inferred to correspond to certain pathways in GBM outcome prediction.

Gene-level relationships can also be represented with networks.
Network nodes typically represent genes and real-valued edges may represent interactions or correlations between genes, often in a tissue or cell type context of interest.
Network-based stratification [@doi:10.1038/nmeth.2651] is an early example of a method for utilizing gene interaction network data in this manner, applied to improve subtyping across several cancer types.
More recently, netNMF-sc [@doi:10.1101/544346] incorporates coexpression networks [@doi:10.1093/nar/gkw868] as a smoothing term for dimension reduction and dropout imputation in single-cell gene expression data.
The coexpression network improves performance for identifying cell types and cell cycle marker genes, as compared to using raw gene expression or other single-cell dimension reduction methods.
Combining gene expression data with a network-derived smoothing term also improved prediction of patient drug response in acute myeloid leukemia [@arxiv:1906.10670] and detection of mutated cancer genes [@doi:10.1038/s41598-017-03141-w].
PIMKL [@doi:10.1038/s41540-019-0086-3] combines network and pathway data to predict disease-free survival from breast cancer cohorts.
This method takes as input both RNA-seq gene expression data and copy number alteration data, but can also be applied to gene expression data alone.

Gene regulatory networks can also augment models for gene expression data.
These networks describe how the expression of genes is modulated by biological regulators such as transcription factors, microRNAs, or small molecules.
creNET [@doi:10.1038/s41598-018-19635-0] integrates a gene regulatory network, derived from STRING [@doi:10.1093/nar/gku1003], with a sparse logistic regression model to predict phenotypic response in clinical trials for ulcerative colitis and acute kidney rejection.
The gene regulatory information allows the model to identify the biological regulators associated with the response, potentially giving mechanistic insight into differential clinical trial response.
GRRANN [@doi:10.1186/s12859-017-1984-2], which was applied to the same data as creNET, uses a gene regulatory network to inform the structure of a neural network.
Several other methods [@doi:10.1093/nar/gkx681; @doi:10.1093/bioinformatics/bty945] have also used gene regulatory network structure to constrain the structure of a neural network, reducing the number of parameters to be fit and facilitating interpretation.

Approaches that incorporate gene set or network structure into genetic studies have a long history [@doi:10.1093/biostatistics/kxl007; @doi:10.1093/bioinformatics/btn081].
Recent applications include expression quantitative trait loci (eQTL) mapping studies, which aim to identify associations between genetic variants and gene expression.
netReg [@doi:10.1093/bioinformatics/btx677] implements a graph-regularized dual LASSO algorithm for eQTL mapping [@doi:10.1093/bioinformatics/btu293] in a publicly available R package.
This model smooths regression coefficients simultaneously based on networks describing associations between genes (target variables in the eQTL regression model) and between variants (predictors in the eQTL regression model).
eQTL information is also used in conjunction with genetic variant information to predict phenotypes, in an approach known as Mendelian randomization (MR).
In [@doi:10.1111/biom.13072], a smoothing term derived from a gene regulatory network is used in an MR model.
The model with the network smoothing term, applied to a human liver dataset, more robustly identifies genes that influence enzyme activity than a network-agnostic model.
As genetic datasets grow, we expect that researchers will continue to develop models that leverage gene set and network databases.

#### Other models incorporating biological structure

Knowledge about biological entities is often organized in an ontology, which is a directed graph that encodes relationships between entities.
The Gene Ontology (GO) [@doi:10.1093/nar/gky1055] describes the relationships between cellular subsystems and other attributes describing proteins or genes.
DCell [@doi:10.1038/nmeth.4627] uses GO to inform the connectivity of a neural network predicting the effects of gene deletions on yeast growth.
DCell performs comparably to an unconstrained neural network for this task.
Additionally, it is easier to interpret: a cellular subsystem with high neuron outputs under a particular gene deletion can be inferred to be strongly affected by the gene deletion, providing a putative genotype-phenotype association.
DeepGO [@doi:10.1093/bioinformatics/btx624] uses a similar approach to predict protein function from amino acid sequence with a neural network constrained by the dependencies of GO.
However, a follow-up paper by the same authors [@doi:10.1093/bioinformatics/btz595] showed that this hierarchy-aware approach can be outperformed by a hierarchy-naive CNN, which uses only amino acid sequence and similarity to labeled training set proteins.
This suggests a tradeoff between interpretability and predictive accuracy for protein function prediction.

Phylogenetic trees, or hierarchies describing the evolutionary relationships between species, can be useful for a similar purpose.
glmmTree [@doi:10.3389/fmicb.2018.01391] uses a phylogenetic tree describing the relationship between microorganisms to improve predictions of age based on gut microbiome data.
The same authors combine a similar phylogeny smoothing strategy with sparse regression to model caffeine intake and smoking status based on microbiome data [@doi:10.3389/fmicb.2018.03112].
Phylogenetic trees can also describe the relationships between subclones of a tumor, which are fundamental to understanding cancer evolution and development.
Using a tumor phylogeny inferred from copy number aberration (CNA) sequencing data as a smoothing term for deconvolving tumor subclones provided more robust predictions than a phylogeny-free model [@doi:10.1007/978-3-030-17083-7_11].
The tree structure of the phylogeny and the subclone mixture model are fit jointly to the CNA data.

Depending on the application, other forms of structure or prior knowledge can inform predictions and interpretation of the model's output.
CYCLOPS [@doi:10.1073/pnas.1619320114] uses a circular node autoencoder [@doi:10.1162/neco.1996.8.2.390] to order periodic gene expression data and estimate circadian rhythms.
The authors validated the method by correctly ordering samples without temporal labels and identifying genes with known circadian expression.
They then applied it to compare gene expression in normal and cancerous liver biopsies, identifying drug targets with circadian expression as candidates for chronotherapy.
NetBiTE [@arxiv:1808.06603] uses drug-gene interaction information from GDSC [@doi:10.1093/nar/gks1111], in addition to protein interaction data, to build a tree ensemble model with splits that are biased toward high-confidence drug-gene interactions.
The model predicts sensitivity to drugs that inhibit critical signaling pathways in cancer, showing improved predictive performance compared to random forests, another commonly used tree ensemble model.

