## Machine learning modeling strategies for -omics data

It can be challenging to distinguish signal from noise in biomedical datasets, and machine learning methods are particularly hampered when the amount of available training data is small.
Incorporating biomedical knowledge into machine learning models can reveal patterns in noisy data [@doi:10.1038/nrg.2017.38] and aid model interpretation [@doi:10.1016/j.cell.2018.05.056].
Biological knowledge can take many forms, including genomic sequences, pathway databases, gene interaction networks, and knowledge hierarchies such as the Gene Ontology [@doi:10.1093/nar/gky1055].
However, there is often no canonical way to encode these structures as real-valued predictors.
Modelers must creatively decide how to encode biological knowledge that they expect will be relevant to the task.

Biomedical datasets often contain more input predictors than data samples [@doi:10.1109/JPROC.2015.2494198; @arxiv:1611.09340].
A genetic study may genotype millions of single nucleotide polymorphisms (SNPs) in thousands of individuals, or a gene expression study may profile the expression of thousands of genes in tens of samples.
Thus, it can be useful to include prior information describing relationships between predictors to inform the representation learned by the model.
This contrasts with non-biological applications of machine learning, where one might fit a model on millions of images [@doi:10.1109/CVPR.2009.5206848] or tens of thousands of documents [@url:https://www.aclweb.org/anthology/P11-1015/], making inclusion of prior information unnecessary.
There are many complementary ways to incorporate heterogeneous sources of biomedical data into the learning process, which have been covered in review papers elsewhere [@doi:10.3389/fgene.2019.00381; @doi:10.1016/j.inffus.2018.09.012].
These include feature extraction or representation learning prior to modeling and/or other data integration methods that do not necessarily involve customizing the model itself.

### Sequence models

Early neural network models primarily used hand-engineered sequence features as input to a fully connected neural network [@doi:10.1093/nar/gku1058; @doi:10.1126/science.1254806] (Figure {@fig:sequence_features}).
As convolutional neural network (CNN) approaches matured for image processing and computer vision, researchers leveraged biological sequence proximity similarly.
CNNs are a neural network variant that groups input data by spatial context to extract features for prediction.

The definition of "spatial context" is specific to the input: one might group image pixels that are nearby in 2D space, or genomic base pairs that are nearby in the linear genome.
In this way, CNNs consider context without making strong assumptions about exactly how much context is needed or how it should be encoded; the data informs the encoding.
A detailed description of how CNNs are applied to sequences can be found in Angermueller et al. [@doi:10.15252/msb.20156651].

![
    Contrasting approaches to extracting features from DNA or RNA sequence data.
    Early models defined features of interest by hand based on prior knowledge about the prediction or clustering problem of interest, such as GC content or sequence melting point, as depicted in the left branch in the figure.
    Convolutional models, depicted in the right branch, use sequence convolutions to derive features directly from sequence proximity, without requiring quantities of interest to be identified before the model is trained.
    Red or blue emphasis denotes inputs to the predictive model (either the hand-defined numeric features on the left or the outputs of convolutional filters on the right).
](images/biopriors/sequence_features_revised.svg){#fig:sequence_features .white}

#### Applications in regulatory biology

Many early applications of deep learning to biological sequences were in regulatory biology.
Early CNNs for sequence data predicted binding protein sequence specificity from DNA or RNA sequence [@doi:10.1038/nbt.3300], variant effects from noncoding DNA sequence [@doi:10.1038/nmeth.3547], and chromatin accessibility from DNA sequence [@doi:10.1101/gr.200535.115].

Recent sequence models take advantage of hardware advances and methodological innovation to incorporate more sequence context and rely on fewer modeling assumptions.
BPNet, a CNN that predicts transcription factor binding profiles from DNA sequences, accurately mapped known locations of binding motifs in mouse embryonic stem cells [@doi:10.1101/737981].
BPNet considers 1000 base pairs of context around each position when predicting binding probabilities with a technique called dilated convolutions [@arxiv:1511.07122], which is particularly important because motif spacing and periodicity can influence binding.
cDeepbind [@doi:10.1101/345140] combines RNA sequences with information about secondary structure to predict RNA binding protein affinities.
Its convolutional model acts on a feature vector combining sequence and structural information, using context for both to inform predictions.
APARENT [@doi:10.1016/j.cell.2019.04.046] is a CNN that predicts alternative polyadenylation (APA) from a training set of over 3 million synthetic APA reporter sequences.
These diverse applications underscore the power of modern deep learning models to synthesize large sequence datasets.

Models that consider sequence context have also been applied to epigenetic data.
DeepSignal [@doi:10.1093/bioinformatics/btz276] is a CNN that uses contextual electrical signals from Oxford Nanopore single-molecule sequencing data to predict 5mC or 6mA DNA methylation status.
MRCNN [@doi:10.1186/s12864-019-5488-5] uses sequences of length 400, centered at CpG sites, to predict 5mC methylation status.
Deep learning models have also been used to predict gene expression from histone modifications [@doi:10.1101/329334; @doi:10.1093/bioinformatics/bty612].
Here, a neural network model consisting of long short-term memory (LSTM) units was used to encode the long-distance interactions of histone marks in both the 3' and 5' genomic directions.
In each of these cases, proximity in the linear genome helped model the complex interactions between DNA sequence and epigenome.

#### Applications in variant calling and mutation detection

Identification of genetic variants also benefits from models that include sequence context.
DeepVariant [@doi:10.1038/nbt.4235] applies a CNN to images of sequence read pileups, using read data around each candidate variant to accurately distinguish true variants from sequencing errors.
CNNs have also been applied to single molecule (PacBio and Oxford Nanopore) sequencing data [@doi:10.1038/s41467-019-09025-z], using a different sequence encoding that results in better performance than DeepVariant on single molecule data.
However, many variant calling models still use hand-engineered sequence features as input to a classifier, including current state-of-the-art approaches to insertion/deletion calling [@doi:10.1101/601450; @doi:10.1101/628222].
Detection of somatic mutations is a distinct but related challenge to detection of germline variants, and has also recently benefitted from use of CNNs [@doi:10.1038/s41467-019-09027-x].

### Network- and pathway-based models

Rather than operating on sequences, many machine learning models in biomedicine operate on inputs that lack intrinsic order.
Models may make use of gene expression data matrices from RNA sequencing or microarray experiments in which rows represent samples and columns represent genes.
To account for relationships between genes, one might incorporate known interactions or correlations when making predictions or generating a low-dimensional representation of the data (Figure {@fig:network_models}).
This is comparable to the manner in which sequence context pushes models to consider nearby base pairs similarly.

![
    The relationships between genes provide structure that can be incorporated into machine learning models.
    One common approach is to use a network or collection of gene sets to embed the data in a lower-dimensional space, in which genes that are in the same gene sets or that are well-connected in the network have a similar representation in the lower-dimensional space.
    The embedded data can then be used for classification or clustering tasks.
    The "x" values in the data table represent gene expression measurements.
](images/biopriors/network_models_revised.svg){#fig:network_models .white}

#### Applications in transcriptomics

Models built from gene expression data can benefit from incorporating gene-level relationships.
One form that this knowledge commonly takes is a database of gene sets, which may represent biological pathways or gene signatures for a biological state of interest.
PLIER [@doi:10.1038/s41592-019-0456-1] uses gene set information from MSigDB [@doi:10.1073/pnas.0506580102] and cell type markers to extract a representation of gene expression data that corresponds to biological processes and reduces technical noise.
The resulting gene set-aligned representation accurately decomposed cell type mixtures.
MultiPLIER [@doi:10.1016/j.cels.2019.04.003] applied PLIER to the recount2 gene expression compendium [@doi:10.1038/nbt.3838] to develop a model that shares information across multiple tissues and diseases, including rare diseases with limited sample sizes.
PASNet [@doi:10.1186/s12859-018-2500-z] uses MSigDB to inform the structure of a neural network for predicting patient outcomes in glioblastoma multiforme (GBM) from gene expression data.
This approach aids interpretation, as pathway nodes in the network with high weights can be inferred to correspond to certain pathways in GBM outcome prediction.

Gene-level relationships can also be represented with networks.
Network nodes typically represent genes and real-valued edges may represent interactions or correlations between genes, often in a tissue or cell type context of interest.
Network-based stratification [@doi:10.1038/nmeth.2651] is an early example of a method for utilizing gene interaction network data in this manner, applied to improve subtyping across several cancer types.
More recently, netNMF-sc [@doi:10.1101/544346] incorporates coexpression networks [@doi:10.1093/nar/gkw868] as a smoothing term for dimension reduction and dropout imputation in single-cell gene expression data.
The coexpression network improves performance for identifying cell types and cell cycle marker genes, as compared to using raw gene expression or other single-cell dimension reduction methods.
Combining gene expression data with a network-derived smoothing term also improved prediction of patient drug response in acute myeloid leukemia [@arxiv:1906.10670] and detection of mutated cancer genes [@doi:10.1038/s41598-017-03141-w].
PIMKL [@doi:10.1038/s41540-019-0086-3] combines network and pathway data to predict disease-free survival from breast cancer cohorts.
This method takes as input both RNA-seq gene expression data and copy number alteration data, but can also be applied to gene expression data alone.

Gene regulatory networks can also augment models for gene expression data.
These networks describe how the expression of genes is modulated by biological regulators such as transcription factors, microRNAs, or small molecules.
creNET [@doi:10.1038/s41598-018-19635-0] integrates a gene regulatory network, derived from STRING [@doi:10.1093/nar/gku1003], with a sparse logistic regression model to predict phenotypic response in clinical trials for ulcerative colitis and acute kidney rejection.
The gene regulatory information allows the model to identify the biological regulators associated with the response, potentially giving mechanistic insight into differential clinical trial response.
GRRANN [@doi:10.1186/s12859-017-1984-2], which was applied to the same data as creNET, uses a gene regulatory network to inform the structure of a neural network.
Several other methods [@doi:10.1093/nar/gkx681; @doi:10.1093/bioinformatics/bty945] have also used gene regulatory network structure to constrain the structure of a neural network, reducing the number of parameters to be fit and facilitating interpretation.

#### Applications in genetics

Approaches that incorporate gene set or network structure into genetic studies have a long history [@doi:10.1093/biostatistics/kxl007; @doi:10.1093/bioinformatics/btn081].
Recent applications include expression quantitative trait loci (eQTL) mapping studies, which aim to identify associations between genetic variants and gene expression.
netReg [@doi:10.1093/bioinformatics/btx677] implements a graph-regularized dual LASSO algorithm for eQTL mapping [@doi:10.1093/bioinformatics/btu293] in a publicly available R package.
This model smooths regression coefficients simultaneously based on networks describing associations between genes (target variables in the eQTL regression model) and between variants (predictors in the eQTL regression model).
eQTL information is also used in conjunction with genetic variant information to predict phenotypes, in an approach known as Mendelian randomization (MR).
In [@doi:10.1111/biom.13072], a smoothing term derived from a gene regulatory network is used in an MR model.
The model with the network smoothing term, applied to a human liver dataset, more robustly identifies genes that influence enzyme activity than a network-agnostic model.
As genetic datasets grow, we expect that researchers will continue to develop models that leverage gene set and network databases.

### Other models incorporating biological structure

Knowledge about biological entities is often organized in an ontology, which is a directed graph that encodes relationships between entities (see Figure {@fig:ontology_models} for a visual example).
The Gene Ontology (GO) [@doi:10.1093/nar/gky1055] describes the relationships between cellular subsystems and other attributes describing proteins or genes.
DCell [@doi:10.1038/nmeth.4627] uses GO to inform the connectivity of a neural network predicting the effects of gene deletions on yeast growth.
DCell performs comparably to an unconstrained neural network for this task.
Additionally, it is easier to interpret: a cellular subsystem with high neuron outputs under a particular gene deletion can be inferred to be strongly affected by the gene deletion, providing a putative genotype-phenotype association.
DeepGO [@doi:10.1093/bioinformatics/btx624] uses a similar approach to predict protein function from amino acid sequence with a neural network constrained by the dependencies of GO.
However, a follow-up paper by the same authors [@doi:10.1093/bioinformatics/btz595] showed that this hierarchy-aware approach can be outperformed by a hierarchy-naive CNN, which uses only amino acid sequence and similarity to labeled training set proteins.
This suggests a tradeoff between interpretability and predictive accuracy for protein function prediction.

![
    Directed graph-structured data, such as an ontology or phylogenetic tree, can be incorporated into machine learning models.
    Here, the connections in the neural network used to predict a set of labels parallel those in the tree graph.
    This type of constraint can also be useful in model interpretation: for example, if the nodes in the right tree branch have high neuron outputs for a given sample, then the subsystem encoded in the right branch of the tree graph is most likely important in making predictions for that sample.
    The "x" values in the data table represent gene expression measurements.
](images/biopriors/ontology_models_revised.svg){#fig:ontology_models .white}

Phylogenetic trees, or hierarchies describing the evolutionary relationships between species, can be useful for a similar purpose.
glmmTree [@doi:10.3389/fmicb.2018.01391] uses a phylogenetic tree describing the relationship between microorganisms to improve predictions of age based on gut microbiome data.
The same authors combine a similar phylogeny smoothing strategy with sparse regression to model caffeine intake and smoking status based on microbiome data [@doi:10.3389/fmicb.2018.03112].
Phylogenetic trees can also describe the relationships between subclones of a tumor, which are fundamental to understanding cancer evolution and development.
Using a tumor phylogeny inferred from copy number aberration (CNA) sequencing data as a smoothing term for deconvolving tumor subclones provided more robust predictions than a phylogeny-free model [@doi:10.1007/978-3-030-17083-7_11].
The tree structure of the phylogeny and the subclone mixture model are fit jointly to the CNA data.

Depending on the application, other forms of structure or prior knowledge can inform predictions and interpretation of the model's output.
CYCLOPS [@doi:10.1073/pnas.1619320114] uses a circular node autoencoder [@doi:10.1162/neco.1996.8.2.390] to order periodic gene expression data and estimate circadian rhythms.
The authors validated the method by correctly ordering samples without temporal labels and identifying genes with known circadian expression.
They then applied it to compare gene expression in normal and cancerous liver biopsies, identifying drug targets with circadian expression as candidates for chronotherapy.
NetBiTE [@arxiv:1808.06603] uses drug-gene interaction information from GDSC [@doi:10.1093/nar/gks1111], in addition to protein interaction data, to build a tree ensemble model with splits that are biased toward high-confidence drug-gene interactions.
The model predicts sensitivity to drugs that inhibit critical signaling pathways in cancer, showing improved predictive performance compared to random forests, another commonly used tree ensemble model.

### Conclusions

As the quantity and richness of biomedical data has increased, sequence repositories and interaction databases have expanded and become more robust.
This raises opportunities to integrate these resources into the structure of machine learning models.
There have been several past attempts to benchmark and compare approaches to integrating structured data into predictive models in biomedicine, including the evaluation in [@doi:10.1371/journal.pone.0034796] and more recent studies in [@arxiv:1905.02295] and [@arxiv:1910.09600].
Extending and broadening benchmarking efforts such as these will be vital, improving our understanding of the suitability of problem domains and datasets for the classes of methods described in this review.
In the future, we foresee that incorporating structured biomedical data will become commonplace for improving model interpretability and boosting performance when sample size is limited.
