### Outline and thesis contributions

In this background section, we have described a wide array of datasets and predictive modeling problems in cancer -omics, which are of both biological and clinical interest.
We have also surveyed methods for taking into account structure that may exist in these datasets when building a model, and identified some examples where this approach works well, in terms of both predictive performance and model interpretability.
In the experimental/original work chapters of this thesis, we will take a somewhat opposite perspective: we will explore caveats and complications of applying machine learning models to large cancer datasets, and areas where more work may be needed to understand and integrate the structure of the data into modeling efforts.

In Chapter 2 of this thesis, we show that the choice of optimization method can affect model selection and tuning, for prediction from cancer transcriptomic data.
Chapter 3 explores the relative information content of -omics types/feature groups in TCGA, showing that gene expression tends to contain the most information relative to cancer driver mutations, but most -omics types can serve as effective, and likely somewhat redundant, readouts.
In Chapter 4, we test generalization across cancer types in TCGA and across datasets (CCLE to TCGA and vice-versa), showing that smaller models do not tend to generalize better across contexts, and cross-validation performance is in most cases a sufficient model selection criterion.
Finally, in Chapter 5, we conclude by summarizing the implications of these results and discussing future directions.

