## Chapter 5: future directions

**Contributions:**
I wrote this chapter for the purpose of this dissertation.

### Introduction

In this dissertation, we have identified complexities of the structure of large cancer -omics datasets that make building predictive models less straightforward than on more uniform datasets often considered in large-scale machine learning applications.
The results of Chapter 2 show that model selection depends heavily on which optimizer is used to fit the model, even when relying on a fairly simple linear modeling strategy, logistic regression.
In Chapter 3, we saw that although different -omics types in TCGA may have slightly variable information content related to cancer function, much of it is redundant in the sense that combining -omics layers generally does not result in a large performance boost.
Our experiments in Chapter 4 showed that generalization across cancer types and model systems can vary highly based on the held-out biological context, and that smaller or simpler models do not in general tend to be more robust across contexts than models that optimize cross-validation performance within a biological context.
Taken together, these results suggest several threads of future directions, both focused on development of methods and datasets, and focused on downstream biological/clinical applications.

### Similarities and differences between cancer types

- Leave-one-domain-out cross-validation
- Methods for effective domain adaptation in biology
    - Basket trials proteomics paper

### Cancer model systems and data

We showed in chapter 4 that when mutation prediction models are trained on cell line data, performance on tumor samples is generally worse than performance on held out cell lines.
We also saw that the converse is not true: on balance, models trained on tumor samples tend to generalize relatively well to cell lines.
This is likely at least in part due to biological differences between cancer cell lines and the tumors they aim to model.
Other studies have also observed that cell line gene expression is less representative of tumor samples than PDX mouse models [@doi:10.1101/2023.04.11.536431] and genetically engineered mouse models [@doi:10.1186/s13073-021-00888-w], although this can vary by cancer type.
In Chapter 2, we noted the general reliance of drug response prediction efforts on public cell line datasets such as CCLE and GDSC, most likely due to the size (both in terms of sample size and number of -omics data types profiled) and accessibility of the datasets.
Together with our results, this underscores the need for larger and more representative compendia of model system data that may be biologically closer to human tumors, including mouse models, tumoroids, and others.

### Multi-omics integration

- Wide world of approaches to multi-omics integration
- Better ways to tease out where multiple views are redundant, and where they can contribute uniquely
  (ideally conditioned on a label e.g. mutation status, cancer type, etc)

### Mutation prediction for subgroup identification in clinical trials

- Using mutation prediction models to define patient subgroups, e.g. for clinical trials
    - Other units of "mutation" prediction (e.g. Ras pathway, etc)
    - Challenge with prospective vs. retrospective evaluation (hard to know what didn't work,
      or what would have worked in a clinical trial that was already run i.e. hard to observe
      counterfactuals)
        - Causal inference methods could help here

### Conclusion
