<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Jake Crawford" />
  <meta name="dcterms.date" content="2023-09-12" />
  <meta name="keywords" content="gene-expression, cancer-genomics, machine-learning, optimization, domain-adaptation" />
  <title>Jake Crawford dissertation title</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1.6em;
      vertical-align: middle;
    }
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/main/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta property="og:type" content="article" />
  <meta name="dc.title" content="Jake Crawford dissertation title" />
  <meta name="citation_title" content="Jake Crawford dissertation title" />
  <meta property="og:title" content="Jake Crawford dissertation title" />
  <meta property="twitter:title" content="Jake Crawford dissertation title" />
  <meta name="dc.date" content="2023-09-12" />
  <meta name="citation_publication_date" content="2023-09-12" />
  <meta property="article:published_time" content="2023-09-12" />
  <meta name="dc.modified" content="2023-09-12T19:30:34+00:00" />
  <meta property="article:modified_time" content="2023-09-12T19:30:34+00:00" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="Jake Crawford" />
  <meta name="citation_author_institution" content="Genomics and Computational Biology Graduate Group, Perelman School of Medicine, University of Pennsylvania, Philadelphia, PA, USA" />
  <meta name="citation_author_orcid" content="0000-0001-6207-0782" />
  <meta name="twitter:creator" content="@jjc2718" />
  <link rel="canonical" href="https://greenelab.github.io/jake_dissertation/" />
  <meta property="og:url" content="https://greenelab.github.io/jake_dissertation/" />
  <meta property="twitter:url" content="https://greenelab.github.io/jake_dissertation/" />
  <meta name="citation_fulltext_html_url" content="https://greenelab.github.io/jake_dissertation/" />
  <meta name="citation_pdf_url" content="https://greenelab.github.io/jake_dissertation/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://greenelab.github.io/jake_dissertation/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://greenelab.github.io/jake_dissertation/v/2773a822f3c45b12491bf5d664d37ba8b1f7f9aa/" />
  <meta name="manubot_html_url_versioned" content="https://greenelab.github.io/jake_dissertation/v/2773a822f3c45b12491bf5d664d37ba8b1f7f9aa/" />
  <meta name="manubot_pdf_url_versioned" content="https://greenelab.github.io/jake_dissertation/v/2773a822f3c45b12491bf5d664d37ba8b1f7f9aa/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Jake Crawford dissertation title</h1>
</header>
<p><small><em>
This manuscript
(<a href="https://greenelab.github.io/jake_dissertation/v/2773a822f3c45b12491bf5d664d37ba8b1f7f9aa/">permalink</a>)
was automatically generated
from <a href="https://github.com/greenelab/jake_dissertation/tree/2773a822f3c45b12491bf5d664d37ba8b1f7f9aa">greenelab/jake_dissertation@2773a82</a>
on September 12, 2023.
</em></small></p>
<h2 id="authors">Authors</h2>
<ul>
<li><strong>Jake Crawford</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-6207-0782">0000-0001-6207-0782</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/jjc2718">jjc2718</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/jjc2718">jjc2718</a>
<br>
<small>
Genomics and Computational Biology Graduate Group, Perelman School of Medicine, University of Pennsylvania, Philadelphia, PA, USA
</small></li>
</ul>
<div id="correspondence">
<p>✉ — Correspondence possible via <a href="https://github.com/greenelab/jake_dissertation/issues">GitHub Issues</a></p>
</div>
<h2 class="page_break_before" id="abstract">Abstract</h2>
<h2 id="chapter-1-background">Chapter 1: background</h2>
<p>This chapter was formatted for this dissertation to provide background information and context for the following chapters. The subsection titled “Machine learning modeling strategies for high-dimensional -omics data” was adapted from a review paper previously published in the <em>Current Opinion in Biotechnology</em> journal, as “Incorporating biological structure into machine learning models in biomedicine” (https://doi.org/10.1016/j.copbio.2019.12.021).</p>
<p><strong>Contributions:</strong>
For the unpublished parts of this chapter, I was the sole author.
For the published parts of this chapter, I wrote the original draft of the review paper, which was edited based on feedback from Casey S. Greene and anonymous reviewers.</p>
<h3 id="introduction">Introduction</h3>
<p>Precision oncology, or the selection of cancer treatments based on molecular or cellular features of patients’ tumors, has become a fundamental part of the standard of care for some cancers <span class="citation" data-cites="fsabqks">[<a href="#ref-fsabqks" role="doc-biblioref">1</a>]</span>.
Although each tumor is unique, the successes of precision oncology reinforce the idea that there are commonalities that can be understood and therapeutically targeted.
Targeted therapies that have been successfully applied across cancer types and patient subsets include <em>HER2</em> (<em>ERBB2</em>) inhibitors in breast and stomach cancer <span class="citation" data-cites="t9NvuXx2">[<a href="#ref-t9NvuXx2" role="doc-biblioref">2</a>]</span>, BTK inhibitors in various hematological malignancies <span class="citation" data-cites="qSH6r3Uo">[<a href="#ref-qSH6r3Uo" role="doc-biblioref">3</a>]</span>, <em>EGFR</em> inhibitors across a variety of carcinomas <span class="citation" data-cites="CkczGwEJ">[<a href="#ref-CkczGwEJ" role="doc-biblioref">4</a>]</span>, and <em>PARP</em> inhibitors for tumors with DNA damage repair defects <span class="citation" data-cites="c9AdwXLE">[<a href="#ref-c9AdwXLE" role="doc-biblioref">5</a>]</span>.
The genes and mutations that drive cancer are often specific to a given cancer type or subtype, but they tend to converge on a few pathways <span class="citation" data-cites="11ZPgRuLk 5N9Iz2gd">[<a href="#ref-11ZPgRuLk" role="doc-biblioref">6</a>,<a href="#ref-5N9Iz2gd" role="doc-biblioref">7</a>]</span>, making more general targeted treatments possible.</p>
<p>The past decade has seen an expansion in the size and diversity of cancer genomics datasets, both publicly available and otherwise.
The Cancer Genome Atlas (TCGA) Pan-Cancer Atlas <span class="citation" data-cites="KrfP0fMx">[<a href="#ref-KrfP0fMx" role="doc-biblioref">8</a>]</span> is a large, public human tumor sample dataset, containing &gt;10,000 samples from 33 different cancer types, each profiled for varying -omics types with associated clinical information.
There are also public datasets containing model system data, including the Cancer Cell Line Encyclopedia (CCLE) containing -omics data from human-derived cancer cell lines <span class="citation" data-cites="kcdvZmNN">[<a href="#ref-kcdvZmNN" role="doc-biblioref">9</a>]</span>, and the Genomics of Drug Sensitivity in Cancer (GDSC) dataset containing drug sensitivity data for thousands of the same cell lines across hundreds of drugs <span class="citation" data-cites="lJFuND9F">[<a href="#ref-lJFuND9F" role="doc-biblioref">10</a>]</span>.
These datasets exhibit heterogeneity on multiple levels.
Overall, they vary in size, with TCGA having about an order of magnitude more tumor samples than the number of cell lines in CCLE.
Going a level deeper, the cancer types within them vary in size as well: TCGA has 1,218 breast cancer samples with gene expression data, but only 265 soft tissue sarcoma samples, and only 45 cholangiocarcinoma samples.</p>
<p>In modern machine learning research using text and images, there is a trend toward bigger models capable of solving broader arrays of tasks.
Foundation models, trained on large datasets to generalize to new tasks with no or minimal task-specific fine-tuning, are in many cases competitive with task-specific models <span class="citation" data-cites="m16z46g3">[<a href="#ref-m16z46g3" role="doc-biblioref">11</a>]</span>, although they are not without unique caveats <span class="citation" data-cites="9Swcx8ai">[<a href="#ref-9Swcx8ai" role="doc-biblioref">12</a>]</span>.
Similarly, in genomics, early examples of foundation models are beginning to appear <span class="citation" data-cites="NMsylj71 r5y0HbhJ Uetau8wh">[<a href="#ref-NMsylj71" role="doc-biblioref">13</a>,<a href="#ref-r5y0HbhJ" role="doc-biblioref">14</a>,<a href="#ref-Uetau8wh" role="doc-biblioref">15</a>]</span>.
Training foundation models on pan-cancer, pan-omics data would be a natural extension of these ideas, which could improve power to detect correlations between biomarkers and phenotypes of interest, or to identify drug susceptibilities in patient sub-populations.</p>
<p>As a whole, this dissertation explores ways in which the structure of large, public pan-cancer datasets can present unexpected challenges and caveats for machine learning.
TCGA and CCLE both contain data from various -omics types (feature groups) and samples from diverse cancer types/tissues of origin (sample groups).
There are additional, less obvious forms of structure in these data such as patient sub-populations and sample collection locations, which we will not address directly in this dissertation but which can affect model training and performance as well.
This chapter, Chapter 1, describes existing work at the intersection of cancer -omics and machine learning, which will provide context for the following chapters.
In particular, we focus in turn on applications of ML to cancer -omics data (first section of Chapter 1), and on ML methods that are currently used to take into account structure in -omics data (second section of Chapter 1).</p>
<h3 id="cancer--omics-data-and-applications">Cancer -omics data and applications</h3>
<h4 id="publicly-available-cancer--omics-data-resources">Publicly available cancer -omics data resources</h4>
<p>A wealth of public cancer genomics and multi-omics human sample resources have been generated in the past decade.
As mentioned in the introduction, the TCGA Pan-Cancer Atlas <span class="citation" data-cites="KrfP0fMx">[<a href="#ref-KrfP0fMx" role="doc-biblioref">8</a>]</span> contains data spanning 33 cancer types and multiple -omics data types, including mutation, CNV, gene expression, miRNA, DNA methylation, reverse phase protein array (RPPA) proteomics data, and clinical outcome data <span class="citation" data-cites="ZB5QpmE">[<a href="#ref-ZB5QpmE" role="doc-biblioref">16</a>]</span>.
The International Cancer Genome Consortium (ICGC) data portal is an initiative to unite and harmonize data from many worldwide cancer projects including TCGA, mostly focused on DNA/somatic mutation data but containing some gene expression and other -omics data <span class="citation" data-cites="IKDYyuvh">[<a href="#ref-IKDYyuvh" role="doc-biblioref">17</a>]</span>.
The Pan-Cancer Analysis of Whole Genomes (PCAWG) project attempts to expand from the whole-exome sequencing provided by TCGA to whole-genome sequencing, providing data and analysis for 2,658 whole genome cancer samples <span class="citation" data-cites="12f5xcS10">[<a href="#ref-12f5xcS10" role="doc-biblioref">18</a>]</span>
The American Association for Cancer Research (AACR)’s Project GENIE (Genomics, Evidence, Neoplasia, Information, Exchange) is another large-scale initiative to share genomic data, with the intention of complementing TCGA and allowing for external validation of methods and biological findings <span class="citation" data-cites="8oOo13r">[<a href="#ref-8oOo13r" role="doc-biblioref">19</a>]</span>.
Unlike TCGA, which contains whole-exome sequencing data, the GENIE dataset is primarily comprised of targeted sequencing panels of a subset of cancer-relevant genes.</p>
<p>In addition to samples derived from human tumors or neoplasms, data from model systems such as cancer cell lines and mouse models are an important element of therapeutic development.
The Cancer Cell Line Encyclopedia (CCLE) contains a variety of uniformly processed -omics data across more than 1000 human-derived cell lines, including somatic mutations, CNV data, gene fusion information, and gene expression <span class="citation" data-cites="kcdvZmNN">[<a href="#ref-kcdvZmNN" role="doc-biblioref">9</a>]</span>.
The Cancer Dependency Map (DepMap) complements CCLE with information about cancer cell line vulnerabilities, derived from CRISPR and RNAi knockout screens <span class="citation" data-cites="RnnKuIPr HXL9NZ9Z">[<a href="#ref-RnnKuIPr" role="doc-biblioref">20</a>,<a href="#ref-HXL9NZ9Z" role="doc-biblioref">21</a>]</span>.
The Connectivity Map (CMap) and LINCS L1000 project aims to catalog the responses of cell lines to both genetic and pharmacological perturbations, identifying the changes to gene expression and protein expression that result <span class="citation" data-cites="F7lIlh2N">[<a href="#ref-F7lIlh2N" role="doc-biblioref">22</a>]</span>.
The GDSC and PRISM drug screening datasets provide cell viability dose-response readings for many of the cell lines in CCLE, after perturbation with small molecules <span class="citation" data-cites="lJFuND9F ZRWXpNGs">[<a href="#ref-lJFuND9F" role="doc-biblioref">10</a>,<a href="#ref-ZRWXpNGs" role="doc-biblioref">23</a>]</span>.
Aside from cell lines, the PDX Encyclopedia is a dataset of patient-derived xenograft (PDX) mouse model data, including more than 1000 models with mutation, CNV, and gene expression data for each <span class="citation" data-cites="BFfDQGl5">[<a href="#ref-BFfDQGl5" role="doc-biblioref">24</a>]</span>.
The National Cancer Institute’s Patient-Derived Models Repository (PDMR) also contains mutation and gene expression profiles for mouse models and patient-derived tumor organoids, or tumoroids <span class="citation" data-cites="Ct0A32uh yoGhmtgc">[<a href="#ref-Ct0A32uh" role="doc-biblioref">25</a>,<a href="#ref-yoGhmtgc" role="doc-biblioref">26</a>]</span>, although it is still under development.</p>
<h4 id="applications-of-machine-learning-in-cancer-genomics">Applications of machine learning in cancer genomics</h4>
<p>Historically, one common use of -omics data in cancer has been to define subtypes, or clinically relevant patient subsets that may have similar prognosis or respond similarly to therapy.
Many studies have sought to distinguish tumor samples from control/normal samples, to identify subtypes of a particular cancer type, or to distinguish samples of a particular cancer type/tissue of origin from samples of other cancer types (e.g. <span class="citation" data-cites="MHNiNgKE 16isU2R76 NgF0DB31 3oNamamF">[<a href="#ref-MHNiNgKE" role="doc-biblioref">27</a>,<a href="#ref-16isU2R76" role="doc-biblioref">28</a>,<a href="#ref-NgF0DB31" role="doc-biblioref">29</a>,<a href="#ref-3oNamamF" role="doc-biblioref">30</a>]</span>).
External validation is difficult, however, since samples in TCGA were taken from patients who had already been clinically diagnosed with a particular cancer type or subtype, i.e. without using machine learning.
Potentially a more clinically relevant way to frame the problem is to classify cancers of unknown primary (CUP), which are metastatic cancers where the primary site cannot be identified in the clinic.
Machine learning approaches have identified cell lineages and developmental trajectories for CUP samples <span class="citation" data-cites="hocatXNe">[<a href="#ref-hocatXNe" role="doc-biblioref">31</a>]</span> and integrated electronic health record (EHR) data and genomic data to suggest targeted therapies for CUP patients <span class="citation" data-cites="jwzDipUG">[<a href="#ref-jwzDipUG" role="doc-biblioref">32</a>]</span>.
Relatedly, distinguishing primary samples from metastatic samples, or predicting metastatic potential of primary samples, is another classification problem which -omics data has been used for <span class="citation" data-cites="RjkBa6QW T2RxGl4r 1BSWdoRe0">[<a href="#ref-RjkBa6QW" role="doc-biblioref">33</a>,<a href="#ref-T2RxGl4r" role="doc-biblioref">34</a>,<a href="#ref-1BSWdoRe0" role="doc-biblioref">35</a>]</span></p>
<p>Prediction of drug response from genomic data, often combined with clinical features or other metadata, is a machine learning problem with clear clinical applications.
Given the availability and uniformity of the cell line data in CCLE, and drug response data in GDSC, PRISM and other cell line datasets, many method development efforts have centered on these data sources.
Examples include prediction of drug response from integrated multi-omics data <span class="citation" data-cites="shldZigM">[<a href="#ref-shldZigM" role="doc-biblioref">36</a>]</span>, prediction of drug response using perturbation modeling via CMap as an intermediate step <span class="citation" data-cites="WRbRNpls">[<a href="#ref-WRbRNpls" role="doc-biblioref">37</a>]</span>, and prediction of drug response via single-cell transcriptomic data <span class="citation" data-cites="D5XNExDI">[<a href="#ref-D5XNExDI" role="doc-biblioref">38</a>]</span>, among many others reviewed in <span class="citation" data-cites="AzsCu0Vu 17Pw4cclY 1FBomJOo7">[<a href="#ref-AzsCu0Vu" role="doc-biblioref">39</a>,<a href="#ref-17Pw4cclY" role="doc-biblioref">40</a>,<a href="#ref-1FBomJOo7" role="doc-biblioref">41</a>]</span>.
Large datasets of human-derived genomic data with associated drug response annotations are more difficult to find.
Still, there have been attempts to develop and/or validate models on human data, including for prediction of immunotherapy response, which benefits from applications across a wide range of cancer types <span class="citation" data-cites="11fRJjEgO 4jSZ7PTT lw3yfGOX">[<a href="#ref-11fRJjEgO" role="doc-biblioref">42</a>,<a href="#ref-4jSZ7PTT" role="doc-biblioref">43</a>,<a href="#ref-lw3yfGOX" role="doc-biblioref">44</a>]</span>.
Prognosis or patient survival prediction from multi-omics data is another area of modeling that leverages widely available clinical metadata, reviewed in detail in many existing papers <span class="citation" data-cites="1DjBaykxF 2ioatXdg 1AopCEsxs OWaqmgwK">[<a href="#ref-1DjBaykxF" role="doc-biblioref">45</a>,<a href="#ref-2ioatXdg" role="doc-biblioref">46</a>,<a href="#ref-1AopCEsxs" role="doc-biblioref">47</a>,<a href="#ref-OWaqmgwK" role="doc-biblioref">48</a>]</span>.</p>
<p>Much of our work, described later in this thesis, stems from the idea of predicting the mutation status in key driver genes of cancer samples, based on functional readouts such as gene expression <span class="citation" data-cites="zPJFCX9h 17XHtmqPY NsW0qxZF MUt3ESlf">[<a href="#ref-zPJFCX9h" role="doc-biblioref">49</a>,<a href="#ref-17XHtmqPY" role="doc-biblioref">50</a>,<a href="#ref-NsW0qxZF" role="doc-biblioref">51</a>,<a href="#ref-MUt3ESlf" role="doc-biblioref">52</a>]</span>.
At first consideration, an accurate mutation status classifier may not seem particularly useful, since for a patient sample a clinician could simply sequence the genome, or select genes in the genome, to determine driver mutation status.
One application of accurate mutation status classifiers, however, is to identify samples with a similar phenotype to those with a driver mutation, but <em>without</em> the mutation being present in DNA sequencing data.
Observed examples of this phenomenon include the “BRCAness” phenotype in tumors without observed <em>BRCA1</em>/<em>BRCA2</em> mutations <span class="citation" data-cites="j19ozmy">[<a href="#ref-j19ozmy" role="doc-biblioref">53</a>]</span>, and the “Ph-like” leukemia phenotype in the absence of the Philadelphia chromosome fusion <span class="citation" data-cites="19oMq1OuY">[<a href="#ref-19oMq1OuY" role="doc-biblioref">54</a>]</span>, among others.</p>
<p>Following this line of reasoning, algorithms have been developed to identify mutations that “phenocopy” known cancer drivers <span class="citation" data-cites="odNCMbto sngWnSGq">[<a href="#ref-odNCMbto" role="doc-biblioref">55</a>,<a href="#ref-sngWnSGq" role="doc-biblioref">56</a>]</span>, and to integrate this information into drug response prediction pipelines to define larger and more accurate patient subgroups <span class="citation" data-cites="7gROlS3 zbMcel6V">[<a href="#ref-7gROlS3" role="doc-biblioref">57</a>,<a href="#ref-zbMcel6V" role="doc-biblioref">58</a>]</span>.
Related machine learning approaches to genomic prediction/phenotype identification include methods for identifying DNA damage repair deficiencies based on genomic and transcriptomic data <span class="citation" data-cites="19JSolFTy 1EZ2xyPvl 1Echl6MYV">[<a href="#ref-19JSolFTy" role="doc-biblioref">59</a>,<a href="#ref-1EZ2xyPvl" role="doc-biblioref">60</a>,<a href="#ref-1Echl6MYV" role="doc-biblioref">61</a>]</span> and for identifying synthetic lethal relationships for use in targeted therapy selection <span class="citation" data-cites="oqXUIdaZ">[<a href="#ref-oqXUIdaZ" role="doc-biblioref">62</a>]</span>.
Such methods could be useful for defining broader and more representative patient groups than would be possible based solely on somatic mutation status, that may pool together similar tumor phenotypes or identify patients that respond to similar therapies.
For example, in “basket” clinical trials where patients are included across cancer types based on the presence or absence of individual molecular markers <span class="citation" data-cites="1FF1hUyo8">[<a href="#ref-1FF1hUyo8" role="doc-biblioref">63</a>]</span>, including “phenocopies” could improve efficacy for some therapies targeted to a particular gene or pathway.</p>
<h3 id="machine-learning-modeling-strategies-for-high-dimensional--omics-data">Machine learning modeling strategies for high-dimensional -omics data</h3>
<p>Biomedical datasets, including those in cancer genomics, often contain more input predictors than data samples <span class="citation" data-cites="1B2Ue9gnv JPGgWnoo">[<a href="#ref-1B2Ue9gnv" role="doc-biblioref">64</a>,<a href="#ref-JPGgWnoo" role="doc-biblioref">65</a>]</span>.
A genetic study may genotype millions of single nucleotide polymorphisms (SNPs) in thousands of individuals, or a gene expression study may profile the expression of thousands of genes in tens of samples.
Thus, it can be useful to include prior information describing relationships between predictors to inform the representation learned by the model.
This contrasts with non-biological applications of machine learning, where one might fit a model on millions of images <span class="citation" data-cites="lt4BNUoG">[<a href="#ref-lt4BNUoG" role="doc-biblioref">66</a>]</span> or tens of thousands of documents <span class="citation" data-cites="yJtbkmny">[<a href="#ref-yJtbkmny" role="doc-biblioref">67</a>]</span>, making inclusion of prior information unnecessary.
In this section, we review existing machine learning methods to take into account structure in the input dataset, particularly for large genomics datasets with correlations between the input predictors or features.</p>
<h4 id="incorporating-structure-into-models-using-gene-expression-data">Incorporating structure into models using gene expression data</h4>
<p>Models built from gene expression data can benefit from incorporating gene-level relationships.
One form that this knowledge commonly takes is a database of gene sets, which may represent biological pathways or gene signatures for a biological state of interest.
PLIER <span class="citation" data-cites="Ki2ij7zE">[<a href="#ref-Ki2ij7zE" role="doc-biblioref">68</a>]</span> uses gene set information from MSigDB <span class="citation" data-cites="15p5LWIVP">[<a href="#ref-15p5LWIVP" role="doc-biblioref">69</a>]</span> and cell type markers to extract a representation of gene expression data that corresponds to biological processes and reduces technical noise.
The resulting gene set-aligned representation accurately decomposed cell type mixtures.
MultiPLIER <span class="citation" data-cites="14rnBunuZ">[<a href="#ref-14rnBunuZ" role="doc-biblioref">70</a>]</span> applied PLIER to the recount2 gene expression compendium <span class="citation" data-cites="6SPTvFXq">[<a href="#ref-6SPTvFXq" role="doc-biblioref">71</a>]</span> to develop a model that shares information across multiple tissues and diseases, including rare diseases with limited sample sizes.
PASNet <span class="citation" data-cites="1Bb8CyeTY">[<a href="#ref-1Bb8CyeTY" role="doc-biblioref">72</a>]</span> uses MSigDB to inform the structure of a neural network for predicting patient outcomes in glioblastoma multiforme (GBM) from gene expression data.
This approach aids interpretation, as pathway nodes in the network with high weights can be inferred to correspond to certain pathways in GBM outcome prediction.</p>
<p>Gene-level relationships can also be represented with networks.
Network nodes typically represent genes and real-valued edges may represent interactions or correlations between genes, often in a tissue or cell type context of interest.
Network-based stratification <span class="citation" data-cites="c2Oc0eIx">[<a href="#ref-c2Oc0eIx" role="doc-biblioref">73</a>]</span> is an early example of a method for utilizing gene interaction network data in this manner, applied to improve subtyping across several cancer types.
More recently, netNMF-sc <span class="citation" data-cites="17fvHtbrH">[<a href="#ref-17fvHtbrH" role="doc-biblioref">74</a>]</span> incorporates coexpression networks <span class="citation" data-cites="3VYPTgXw">[<a href="#ref-3VYPTgXw" role="doc-biblioref">75</a>]</span> as a smoothing term for dimension reduction and dropout imputation in single-cell gene expression data.
The coexpression network improves performance for identifying cell types and cell cycle marker genes, as compared to using raw gene expression or other single-cell dimension reduction methods.
Combining gene expression data with a network-derived smoothing term also improved prediction of patient drug response in acute myeloid leukemia <span class="citation" data-cites="LLInUBEI">[<a href="#ref-LLInUBEI" role="doc-biblioref">76</a>]</span> and detection of mutated cancer genes <span class="citation" data-cites="1BbuXJuIl">[<a href="#ref-1BbuXJuIl" role="doc-biblioref">77</a>]</span>.
PIMKL <span class="citation" data-cites="12cJO5Pse">[<a href="#ref-12cJO5Pse" role="doc-biblioref">78</a>]</span> combines network and pathway data to predict disease-free survival from breast cancer cohorts.
This method takes as input both RNA-seq gene expression data and copy number alteration data, but can also be applied to gene expression data alone.</p>
<p>Gene regulatory networks can also augment models for gene expression data.
These networks describe how the expression of genes is modulated by biological regulators such as transcription factors, microRNAs, or small molecules.
creNET <span class="citation" data-cites="g8OoyIPj">[<a href="#ref-g8OoyIPj" role="doc-biblioref">79</a>]</span> integrates a gene regulatory network, derived from STRING <span class="citation" data-cites="q9Fhy8eq">[<a href="#ref-q9Fhy8eq" role="doc-biblioref">80</a>]</span>, with a sparse logistic regression model to predict phenotypic response in clinical trials for ulcerative colitis and acute kidney rejection.
The gene regulatory information allows the model to identify the biological regulators associated with the response, potentially giving mechanistic insight into differential clinical trial response.
GRRANN <span class="citation" data-cites="19wuAzYvo">[<a href="#ref-19wuAzYvo" role="doc-biblioref">81</a>]</span>, which was applied to the same data as creNET, uses a gene regulatory network to inform the structure of a neural network.
Several other methods <span class="citation" data-cites="owp8L957 e4tSAJkK">[<a href="#ref-owp8L957" role="doc-biblioref">82</a>,<a href="#ref-e4tSAJkK" role="doc-biblioref">83</a>]</span> have also used gene regulatory network structure to constrain the structure of a neural network, reducing the number of parameters to be fit and facilitating interpretation.</p>
<p>Approaches that incorporate gene set or network structure into genetic studies have a long history <span class="citation" data-cites="PuUYD4zV MPgqEKuv">[<a href="#ref-PuUYD4zV" role="doc-biblioref">84</a>,<a href="#ref-MPgqEKuv" role="doc-biblioref">85</a>]</span>.
Recent applications include expression quantitative trait loci (eQTL) mapping studies, which aim to identify associations between genetic variants and gene expression.
netReg <span class="citation" data-cites="1AuHIuXiR">[<a href="#ref-1AuHIuXiR" role="doc-biblioref">86</a>]</span> implements a graph-regularized dual LASSO algorithm for eQTL mapping <span class="citation" data-cites="9SBhyy2x">[<a href="#ref-9SBhyy2x" role="doc-biblioref">87</a>]</span> in a publicly available R package.
This model smooths regression coefficients simultaneously based on networks describing associations between genes (target variables in the eQTL regression model) and between variants (predictors in the eQTL regression model).
eQTL information is also used in conjunction with genetic variant information to predict phenotypes, in an approach known as Mendelian randomization (MR).
In <span class="citation" data-cites="m7CdzEUG">[<a href="#ref-m7CdzEUG" role="doc-biblioref">88</a>]</span>, a smoothing term derived from a gene regulatory network is used in an MR model.
The model with the network smoothing term, applied to a human liver dataset, more robustly identifies genes that influence enzyme activity than a network-agnostic model.
As genetic datasets grow, we expect that researchers will continue to develop models that leverage gene set and network databases.</p>
<h4 id="other-models-incorporating-biological-structure">Other models incorporating biological structure</h4>
<p>Knowledge about biological entities is often organized in an ontology, which is a directed graph that encodes relationships between entities.
The Gene Ontology (GO) <span class="citation" data-cites="eH3LaU5K">[<a href="#ref-eH3LaU5K" role="doc-biblioref">89</a>]</span> describes the relationships between cellular subsystems and other attributes describing proteins or genes.
DCell <span class="citation" data-cites="qQP20moO">[<a href="#ref-qQP20moO" role="doc-biblioref">90</a>]</span> uses GO to inform the connectivity of a neural network predicting the effects of gene deletions on yeast growth.
DCell performs comparably to an unconstrained neural network for this task.
Additionally, it is easier to interpret: a cellular subsystem with high neuron outputs under a particular gene deletion can be inferred to be strongly affected by the gene deletion, providing a putative genotype-phenotype association.
DeepGO <span class="citation" data-cites="TIQTmEOG">[<a href="#ref-TIQTmEOG" role="doc-biblioref">91</a>]</span> uses a similar approach to predict protein function from amino acid sequence with a neural network constrained by the dependencies of GO.
However, a follow-up paper by the same authors <span class="citation" data-cites="Cf5duPBD">[<a href="#ref-Cf5duPBD" role="doc-biblioref">92</a>]</span> showed that this hierarchy-aware approach can be outperformed by a hierarchy-naive CNN, which uses only amino acid sequence and similarity to labeled training set proteins.
This suggests a tradeoff between interpretability and predictive accuracy for protein function prediction.</p>
<p>Phylogenetic trees, or hierarchies describing the evolutionary relationships between species, can be useful for a similar purpose.
glmmTree <span class="citation" data-cites="uQ5z1fAc">[<a href="#ref-uQ5z1fAc" role="doc-biblioref">93</a>]</span> uses a phylogenetic tree describing the relationship between microorganisms to improve predictions of age based on gut microbiome data.
The same authors combine a similar phylogeny smoothing strategy with sparse regression to model caffeine intake and smoking status based on microbiome data <span class="citation" data-cites="VvllhwW1">[<a href="#ref-VvllhwW1" role="doc-biblioref">94</a>]</span>.
Phylogenetic trees can also describe the relationships between subclones of a tumor, which are fundamental to understanding cancer evolution and development.
Using a tumor phylogeny inferred from copy number aberration (CNA) sequencing data as a smoothing term for deconvolving tumor subclones provided more robust predictions than a phylogeny-free model <span class="citation" data-cites="VzWLIpJn">[<a href="#ref-VzWLIpJn" role="doc-biblioref">95</a>]</span>.
The tree structure of the phylogeny and the subclone mixture model are fit jointly to the CNA data.</p>
<p>Depending on the application, other forms of structure or prior knowledge can inform predictions and interpretation of the model’s output.
CYCLOPS <span class="citation" data-cites="1Dk2FpWk6">[<a href="#ref-1Dk2FpWk6" role="doc-biblioref">96</a>]</span> uses a circular node autoencoder <span class="citation" data-cites="15jrSlkXD">[<a href="#ref-15jrSlkXD" role="doc-biblioref">97</a>]</span> to order periodic gene expression data and estimate circadian rhythms.
The authors validated the method by correctly ordering samples without temporal labels and identifying genes with known circadian expression.
They then applied it to compare gene expression in normal and cancerous liver biopsies, identifying drug targets with circadian expression as candidates for chronotherapy.
NetBiTE <span class="citation" data-cites="kKiwlzZq">[<a href="#ref-kKiwlzZq" role="doc-biblioref">98</a>]</span> uses drug-gene interaction information from GDSC <span class="citation" data-cites="lJFuND9F">[<a href="#ref-lJFuND9F" role="doc-biblioref">10</a>]</span>, in addition to protein interaction data, to build a tree ensemble model with splits that are biased toward high-confidence drug-gene interactions.
The model predicts sensitivity to drugs that inhibit critical signaling pathways in cancer, showing improved predictive performance compared to random forests, another commonly used tree ensemble model.</p>
<h3 id="outline-and-thesis-contributions">Outline and thesis contributions</h3>
<p>In this background section, we have described a wide array of datasets and predictive modeling problems in cancer -omics, which are of both biological and clinical interest.
We have also surveyed methods for taking into account structure that may exist in these datasets when building a model, and identified some examples where this approach works well, in terms of both predictive performance and model interpretability.
In the experimental/original work chapters of this thesis, we will take a somewhat opposite perspective: we will explore caveats and complications of applying machine learning models to large cancer datasets, and areas where more work may be needed to understand and integrate the structure of the data into modeling efforts.</p>
<p>In Chapter 2 of this thesis, we show that the choice of optimization method can affect model selection and tuning, for prediction from cancer transcriptomic data.
Chapter 3 explores the relative information content of -omics types/feature groups in TCGA, showing that gene expression tends to contain the most information relative to cancer driver mutations, but most -omics types can serve as effective, and likely somewhat redundant, readouts.
In Chapter 4, we test generalization across cancer types in TCGA and across datasets (CCLE to TCGA and vice-versa), showing that smaller models do not tend to generalize better across contexts, and cross-validation performance is in most cases a sufficient model selection criterion.
Finally, in Chapter 5, we conclude by summarizing the implications of these results and discussing future directions.</p>
<h2 id="chapter-2-optimization-strongly-influences-model-selection-in-transcriptomic-prediction">Chapter 2: optimization strongly influences model selection in transcriptomic prediction</h2>
<p>This chapter has been posted as a preprint on bioRxiv (https://www.biorxiv.org/content/10.1101/2023.06.26.546586v1) and submitted for publication at Bioinformatics Advances as “Optimizer’s dilemma: optimization strongly influences model selection in transcriptomic prediction”.</p>
<p><strong>Contributions</strong>:
I designed and ran the experiments, created the figures, wrote the initial draft of the manuscript, and edited the manuscript. Maria Chikina gave feedback on an initial version of the manuscript, gave guidance on experimental design, and edited the manuscript. Casey S. Greene gave feedback and guidance on experiments, and edited the manuscript.</p>
<h3 id="abstract-1">Abstract</h3>
<h4 id="motivation">Motivation</h4>
<p>Most models can be fit to data using various optimization approaches.
While model choice is frequently reported in machine-learning-based research, optimizers are not often noted.
We applied two different implementations of LASSO logistic regression implemented in Python’s scikit-learn package, using two different optimization approaches (coordinate descent and stochastic gradient descent), to predict driver mutation presence or absence from gene expression across 84 pan-cancer driver genes.
Across varying levels of regularization, we compared performance and model sparsity between optimizers.</p>
<h4 id="results">Results</h4>
<p>After model selection and tuning, we found that coordinate descent (implemented in the <code>liblinear</code> library) and SGD tended to perform comparably.
<code>liblinear</code> models required more extensive tuning of regularization strength, performing best for high model sparsities (more nonzero coefficients), but did not require selection of a learning rate parameter.
SGD models required tuning of the learning rate to perform well, but generally performed more robustly across different model sparsities as regularization strength decreased.
Given these tradeoffs, we believe that the choice of optimizers should be clearly reported as a part of the model selection and validation process, to allow readers and reviewers to better understand the context in which results have been generated.</p>
<h4 id="availability-and-implementation">Availability and implementation</h4>
<p>The code used to carry out the analyses in this study is available at <a href="https://github.com/greenelab/pancancer-evaluation/tree/master/01_stratified_classification" class="uri">https://github.com/greenelab/pancancer-evaluation/tree/master/01_stratified_classification</a>. Performance/regularization strength curves for all genes in the Vogelstein et al. 2013 dataset are available at <a href="https://doi.org/10.6084/m9.figshare.22728644" class="uri">https://doi.org/10.6084/m9.figshare.22728644</a>.</p>
<h3 id="introduction-1">Introduction</h3>
<p>Gene expression profiles are widely used to classify samples or patients into relevant groups or categories, both preclinically <span class="citation" data-cites="1CdDn310M bOvwnoRq">[<a href="#ref-1CdDn310M" role="doc-biblioref">99</a>,<a href="#ref-bOvwnoRq" role="doc-biblioref">100</a>]</span> and clinically <span class="citation" data-cites="lnK82Ey6 OW5P00JC">[<a href="#ref-lnK82Ey6" role="doc-biblioref">101</a>,<a href="#ref-OW5P00JC" role="doc-biblioref">102</a>]</span>.
To extract informative gene features and to perform classification, a diverse array of algorithms exist, and different algorithms perform well across varying datasets and tasks <span class="citation" data-cites="1CdDn310M">[<a href="#ref-1CdDn310M" role="doc-biblioref">99</a>]</span>.
Even within a given model class, multiple optimization methods can often be applied to find well-performing model parameters or to optimize a model’s loss function.
One commonly used example is logistic regression.
The widely used scikit-learn Python package for machine learning <span class="citation" data-cites="43wsMmMv">[<a href="#ref-43wsMmMv" role="doc-biblioref">103</a>]</span> provides two modules for fitting logistic regression classifiers: <code>LogisticRegression</code>, which uses the <code>liblinear</code> coordinate descent method <span class="citation" data-cites="Mm8xIDqq">[<a href="#ref-Mm8xIDqq" role="doc-biblioref">104</a>]</span> to find parameters that optimize the logistic loss function, and <code>SGDClassifier</code>, which uses stochastic gradient descent <span class="citation" data-cites="x24AA6cY">[<a href="#ref-x24AA6cY" role="doc-biblioref">105</a>]</span> to optimize the same loss function.</p>
<p>Using scikit-learn, we compared the <code>liblinear</code> (coordinate descent) and SGD optimization techniques for prediction of driver mutation status in tumor samples, across a wide variety of genes implicated in cancer initiation and development <span class="citation" data-cites="809OyWlC">[<a href="#ref-809OyWlC" role="doc-biblioref">106</a>]</span>.
We applied LASSO (L1-regularized) logistic regression, and tuned the strength of the regularization to compare model selection between optimizers.
We found that across a variety of models (i.e. varying regularization strengths), the training dynamics of the optimizers were considerably different: models fit using <code>liblinear</code> tended to perform best at fairly high regularization strengths (100-1000 nonzero features in the model) and overfit easily with low regularization strengths.
On the other hand, after tuning the learning rate, models fit using SGD tended to perform well across both higher and lower regularization strengths, and overfitting was less common.</p>
<p>Our results caution against viewing optimizer choice as a “black box” component of machine learning modeling.
The observation that LASSO logistic regression models fit using SGD tended to perform well for low levels of regularization, across diverse driver genes, runs counter to conventional wisdom in machine learning for high-dimensional data which generally states that explicit regularization and/or feature selection is necessary.
Comparing optimizers or model implementations directly is rare in applications of machine learning for genomics, and our work shows that this choice can affect generalization and interpretation properties of the model significantly.
Based on our results, we recommend considering the appropriate optimization approach carefully based on the goals of each individual analysis.</p>
<h3 id="methods">Methods</h3>
<h4 id="data-download-and-preprocessing">Data download and preprocessing</h4>
<p>To generate binary mutated/non-mutated gene labels for our machine learning model, we used mutation calls for TCGA Pan-Cancer Atlas samples from MC3 <span class="citation" data-cites="cDnF3a5Q">[<a href="#ref-cDnF3a5Q" role="doc-biblioref">107</a>]</span> and copy number threshold calls from GISTIC2.0 <span class="citation" data-cites="rDV1hhfF">[<a href="#ref-rDV1hhfF" role="doc-biblioref">108</a>]</span>.
MC3 mutation calls were downloaded from the Genomic Data Commons (GDC) of the National Cancer Institute, at <a href="https://gdc.cancer.gov/about-data/publications/pancanatlas" class="uri">https://gdc.cancer.gov/about-data/publications/pancanatlas</a>.
Thresholded copy number calls are from an older version of the GDC data and are available here: <a href="https://figshare.com/articles/dataset/TCGA_PanCanAtlas_Copy_Number_Data/6144122" class="uri">https://figshare.com/articles/dataset/TCGA_PanCanAtlas_Copy_Number_Data/6144122</a>.
We removed hypermutated samples, defined as two or more standard deviations above the mean non-silent somatic mutation count, from our dataset to reduce the number of false positives (i.e., non-driver mutations).
Any sample with either a non-silent somatic variant or a copy number variation (copy number gain in the target gene for oncogenes and copy number loss in the target gene for tumor suppressor genes) was included in the positive set; all remaining samples were considered negative for mutation in the target gene.</p>
<p>RNA sequencing data for TCGA was downloaded from GDC at the same link provided above for the Pan-Cancer Atlas.
We discarded non-protein-coding genes and genes that failed to map and removed tumors that were measured from multiple sites.
After filtering to remove hypermutated samples and taking the intersection of samples with both mutation and gene expression data, 9074 total TCGA samples remained.</p>
<h4 id="cancer-gene-set-construction">Cancer gene set construction</h4>
<p>In order to study mutation status classification for a diverse set of cancer driver genes, we started with the set of 125 frequently altered genes from Vogelstein et al. <span class="citation" data-cites="809OyWlC">[<a href="#ref-809OyWlC" role="doc-biblioref">106</a>]</span> (all genes from Table S2A).
For each target gene, in order to ensure that the training dataset was reasonably balanced (i.e., that there would be enough mutated samples to train an effective classifier), we included only cancer types with at least 15 mutated samples and at least 5% mutated samples, which we refer to here as “valid” cancer types.
In some cases, this resulted in genes with no valid cancer types, which we dropped from the analysis.
Out of the 125 genes originally listed in the Vogelstein et al. cancer gene set, we retained 84 target genes after filtering for valid cancer types.</p>
<h4 id="classifier-setup-and-optimizer-comparison-details">Classifier setup and optimizer comparison details</h4>
<p>We trained logistic regression classifiers to predict whether or not a given sample had a mutational event in a given target gene using gene expression features as explanatory variables.
Based on our previous work, gene expression is generally effective for this problem across many target genes, although other -omics types can be equally effective in many cases <span class="citation" data-cites="cAxtCrBP">[<a href="#ref-cAxtCrBP" role="doc-biblioref">109</a>]</span>.
Our model was trained on gene expression data (X) to predict mutation presence or absence (y) in a target gene.
To control for varying mutation burden per sample and to adjust for potential cancer type-specific expression patterns, we included one-hot encoded cancer type and log<sub>10</sub>(sample mutation count) in the model as covariates.
Since gene expression datasets tend to have many dimensions and comparatively few samples, we used a LASSO penalty to perform feature selection <span class="citation" data-cites="kX2zf6UE">[<a href="#ref-kX2zf6UE" role="doc-biblioref">110</a>]</span>.
LASSO logistic regression has the advantage of generating sparse models (some or most coefficients are 0), as well as having a single tunable hyperparameter which can be easily interpreted as an indicator of regularization strength, or model complexity.</p>
<p>To compare model selection across optimizers, we first split the “valid” cancer types into train (75%) and test (25%) sets.
We then split the training data into “subtrain” (66% of the training set) data to train the model on, and “holdout” (33% of the training set) data to perform model selection, i.e. to use to select the best-performing regularization parameter, and the best-performing learning rate for SGD in the cases where multiple learning rates were considered.
In each case, these splits were stratified by cancer type, i.e. each split had as close as possible to equal proportions of each cancer type included in the dataset for the given driver gene.</p>
<h4 id="lasso-parameter-range-selection-and-comparison-between-optimizers">LASSO parameter range selection and comparison between optimizers</h4>
<p>The scikit-learn implementations of coordinate descent (in <code>liblinear</code>/<code>LogisticRegression</code>) and stochastic gradient descent (in <code>SGDClassifier</code>) use slightly different parameterizations of the LASSO regularization strength parameter. <code>liblinear</code>’s logistic regression solver optimizes the following loss function:</p>
<p><span class="math display">\[\hat{w} = \text{argmin}_{w} \ (C \cdot \ell(X, y; w)) + ||w||_1\]</span></p>
<p>where <span class="math inline">\(\ell(X, y; w)\)</span> denotes the negative log-likelihood of the observed data <span class="math inline">\((X, y)\)</span> given a particular choice of feature weights <span class="math inline">\(w\)</span>.<code>SGDClassifier</code> optimizes the following loss function:</p>
<p><span class="math display">\[\hat{w} = \text{argmin}_{w} \ \ell(X, y; w) + \alpha ||w||_1\]</span></p>
<!--_ -->
<p>which is equivalent with the exception of the LASSO parameter which is formulated slightly differently, as <span class="math inline">\(\alpha = \frac{1}{C}\)</span>.
The result of this slight difference in parameterization is that <code>liblinear</code> <span class="math inline">\(C\)</span> values vary inversely with regularization strength (higher values = less regularization, or greater model complexity) and <code>SGDClassifier</code> <span class="math inline">\(\alpha\)</span> values vary directly with regularization strength (lower values = less regularization, or greater model complexity).</p>
<p>For the <code>liblinear</code> optimizer, we trained models using <span class="math inline">\(C\)</span> values evenly spaced on a logarithmic scale between (10<sup>-3</sup>, 10<sup>7</sup>); i.e. the output of <code>numpy.logspace(-3, 7, 21)</code>.
For the SGD optimizer, we trained models using the inverse range of <span class="math inline">\(\alpha\)</span> values between (10<sup>-7</sup>, 10<sup>3</sup>), or <code>numpy.logspace(-7, 3, 21)</code>.
These hyperparameter ranges were intended to give evenly distributed coverage across genes that included “underfit” models (predicting only the mean or using very few features, poor performance on all datasets), “overfit” models (performing perfectly on training data but comparatively poorly on cross-validation and test data), and a wide variety of models in between that typically included the best fits to the cross-validation and test data.</p>
<p>For ease of visual comparison in our figures, we plot the SGD <span class="math inline">\(\alpha\)</span> parameter directly, and the <code>liblinear</code> <span class="math inline">\(C\)</span> parameter inversely (i.e. <span class="math inline">\(\frac{1}{C}\)</span>).
This orients the x-axes of the relevant plots in the same direction: lower values represent lower regularization strength or higher model complexity, and higher values represent higher regularization strength or lower model complexity, for both optimizers.</p>
<h4 id="sgd-learning-rate-selection">SGD learning rate selection</h4>
<p>scikit-learn’s <code>SGDClassifier</code> provides four built-in approaches to learning rate scheduling: <code>constant</code> (a single, constant learning rate), <code>optimal</code> (a learning rate with an initial value selected using a heuristic based on the regularization parameter and the data loss, that decreases across epochs), <code>invscaling</code> (a learning rate that decreases exponentially by epoch), and <code>adaptive</code> (a learning rate that starts at a constant value, which is divided by 5 each time the training loss fails to decrease for 5 straight epochs).
The <code>optimal</code> learning rate schedule is used by default.</p>
<p>When we compared these four approaches, we used a constant learning rate of 0.0005, and an initial learning rate of 0.1 for the <code>adaptive</code> and <code>invscaling</code> schedules.
We also tested a fifth approach that we called “<code>constant_search</code>”, in which we tested a range of constant learning rates in a grid search on a validation dataset, then evaluated the model on the test data using the best-performing constant learning rate by validation AUPR.
For the grid search, we used the following range of constant learning rates: {0.000005, 0.00001, 0.00005, 0.0001, 0.0005, 0.001, 0.01}.
Unless otherwise specified, results for SGD in the main paper figures used the <code>constant_search</code> approach, which performed the best in our comparison between schedulers.</p>
<h3 id="results-1">Results</h3>
<h4 id="liblinear-and-sgd-lasso-models-perform-comparably-but-liblinear-is-sensitive-to-regularization-strength"><code>liblinear</code> and SGD LASSO models perform comparably, but <code>liblinear</code> is sensitive to regularization strength</h4>
<p>For each of the 125 driver genes from the Vogelstein et al. 2013 paper, we trained models to predict mutation status (presence or absence) from RNA-seq data, derived from the TCGA Pan-Cancer Atlas.
For each optimizer, we trained LASSO logistic regression models across a variety of regularization parameters (see Methods for parameter range details), achieving a variety of different levels of model sparsity (Supplementary Figure <a href="#fig:compare_sparsity">4</a>).
We repeated model fitting/evaluation across 4 cross-validation splits x 2 replicates (random seeds) for a total of 8 different models per parameter.
Cross-validation splits were stratified by cancer type.</p>
<p>Previous work has shown that pan-cancer classifiers of Ras mutation status are accurate and biologically informative <span class="citation" data-cites="17XHtmqPY">[<a href="#ref-17XHtmqPY" role="doc-biblioref">50</a>]</span>.
We first evaluated models for KRAS mutation prediction.
As model complexity increases (more nonzero coefficients) for the <code>liblinear</code> optimizer, we observed that performance increases then decreases, corresponding to overfitting for high model complexities/numbers of nonzero coefficients (Figure <a href="#fig:optimizer_compare_mutations">1</a>A).
On the other hand, for the SGD optimizer, we observed consistent performance as model complexity increases, with models having no nonzero coefficients performing comparably to the best (Figure <a href="#fig:optimizer_compare_mutations">1</a>B).
In this case, top performance for SGD (a regularization parameter of 10<sup>-1</sup>) is slightly better than top performance for <code>liblinear</code> (a regularization parameter of 1 / 3.16 x 10<sup>2</sup>): we observed a mean test AUPR of 0.722 for SGD vs. mean AUPR of 0.692 for <code>liblinear</code>.</p>
<p>To determine how relative performance trends with <code>liblinear</code> tend to compare across the genes in the Vogelstein dataset at large, we looked at the difference in performance between optimizers for the best-performing models for each gene (Figure <a href="#fig:optimizer_compare_mutations">1</a>C).
The distribution is centered around 0 and more or less symmetrical, suggesting that across the gene set, <code>liblinear</code> and SGD tend to perform comparably to one another.
We saw that for 52/84 genes, performance for the best-performing model was better using SGD than <code>liblinear</code>, and for the other 32 genes performance was better using <code>liblinear</code>.
In order to quantify whether the overfitting tendencies (or lack thereof) also hold across the gene set, we plotted the difference in performance between the best-performing model and the largest (least regularized) model; classifiers with a large difference in performance exhibit strong overfitting, and classifiers with a small difference in performance do not overfit (Figure <a href="#fig:optimizer_compare_mutations">1</a>D).
For SGD, the least regularized models tend to perform comparably to the best-performing models, whereas for <code>liblinear</code> the distribution is wider suggesting that overfitting is more common.</p>
<div id="fig:optimizer_compare_mutations" class="fignos">
<figure>
<img src="images/optimizers/figure_1.png" style="width:100.0%" alt="Figure 1: A. Performance vs. inverse regularization parameter for KRAS mutation status prediction, using the liblinear coordinate descent optimizer. Dotted lines indicate top performance value of the opposite optimizer. B. Performance vs. regularization parameter for KRAS mutation status prediction, using the SGD optimizer. “Holdout” dataset is used for SGD learning rate selection, “test” data is completely held out from model selection and used for evaluation. C. Distribution of performance difference between best-performing model for liblinear and SGD optimizers, across all 84 genes in Vogelstein driver gene set. Positive numbers on the x-axis indicate better performance using liblinear, and negative numbers indicate better performance using SGD. D. Distribution of performance difference between best-performing model and largest (least regularized) model, for liblinear and SGD, across all 84 genes. Smaller numbers on the y-axis indicate less overfitting, and larger numbers indicate more overfitting." />
<figcaption aria-hidden="true"><span>Figure 1:</span> <strong>A.</strong> Performance vs. inverse regularization parameter for KRAS mutation status prediction, using the <code>liblinear</code> coordinate descent optimizer. Dotted lines indicate top performance value of the opposite optimizer.
<strong>B.</strong> Performance vs. regularization parameter for KRAS mutation status prediction, using the SGD optimizer. “Holdout” dataset is used for SGD learning rate selection, “test” data is completely held out from model selection and used for evaluation.
<strong>C.</strong> Distribution of performance difference between best-performing model for <code>liblinear</code> and SGD optimizers, across all 84 genes in Vogelstein driver gene set. Positive numbers on the x-axis indicate better performance using <code>liblinear</code>, and negative numbers indicate better performance using SGD.
<strong>D.</strong> Distribution of performance difference between best-performing model and largest (least regularized) model, for <code>liblinear</code> and SGD, across all 84 genes. Smaller numbers on the y-axis indicate less overfitting, and larger numbers indicate more overfitting.</figcaption>
</figure>
</div>
<h4 id="sgd-is-sensitive-to-learning-rate-selection">SGD is sensitive to learning rate selection</h4>
<p>The SGD results shown in Figure <a href="#fig:optimizer_compare_mutations">1</a> select the best-performing learning rate using a grid search on the holdout dataset, independently for each regularization parameter.
We also compared against other learning rate scheduling approaches implemented in scikit-learn (see Methods for implementation details and grid search specifications).
For KRAS mutation prediction, we observed that the choice of initial learning rate and scheduling approach affects performance significantly, and other approaches to selecting the learning rate performed poorly relative to <code>liblinear</code> (black dotted lines in Figure <a href="#fig:sgd_lr_schedulers">2</a>) and to the grid search.
We did not observe an improvement in performance over <code>liblinear</code> or the grid search for learning rate schedulers that decrease across epochs (Figure <a href="#fig:sgd_lr_schedulers">2</a>A, C, and D), nor did we see comparable performance when we selected a single constant learning rate for all levels of regularization without the preceding grid search (Figure <a href="#fig:sgd_lr_schedulers">2</a>B).
Notably, scikit-learn’s default “optimal” learning rate schedule performed relatively poorly for this problem, suggesting that tuning the learning rate and selecting a well-performing scheduler is a critical component of applying SGD successfully for this problem (Figure <a href="#fig:sgd_lr_schedulers">2</a>D).
We observed similar trends across all genes in the Vogelstein gene set, with other learning rate scheduling approaches performing poorly in aggregate relative to both <code>liblinear</code> and SGD with the learning rate grid search (Supplementary Figure <a href="#fig:compare_all_lr">5</a>).</p>
<div id="fig:sgd_lr_schedulers" class="fignos">
<figure>
<img src="images/optimizers/figure_2.png" style="width:100.0%" alt="Figure 2: A. Performance vs. regularization parameter for KRAS mutation prediction, using SGD optimizer with adaptive learning rate scheduler. Dotted line indicates top performance value using liblinear, from Figure 1A. B. Performance vs. regularization parameter, using SGD optimizer with constant learning rate scheduler and a learning rate of 0.0005. C. Performance vs. regularization parameter, using SGD optimizer with inverse scaling learning rate scheduler. D. Performance vs. regularization parameter, using SGD optimizer with “optimal” learning rate scheduler." />
<figcaption aria-hidden="true"><span>Figure 2:</span> <strong>A.</strong> Performance vs. regularization parameter for KRAS mutation prediction, using SGD optimizer with adaptive learning rate scheduler. Dotted line indicates top performance value using <code>liblinear</code>, from Figure <a href="#fig:optimizer_compare_mutations">1</a>A.
<strong>B.</strong> Performance vs. regularization parameter, using SGD optimizer with constant learning rate scheduler and a learning rate of 0.0005.
<strong>C.</strong> Performance vs. regularization parameter, using SGD optimizer with inverse scaling learning rate scheduler.
<strong>D.</strong> Performance vs. regularization parameter, using SGD optimizer with “optimal” learning rate scheduler.</figcaption>
</figure>
</div>
<h4 id="liblinear-and-sgd-result-in-different-models-with-varying-loss-dynamics"><code>liblinear</code> and SGD result in different models, with varying loss dynamics</h4>
<p>We sought to determine whether there was a difference in the sparsity of the models resulting from the different optimization schemes.
In general across all genes, the best-performing SGD models mostly tend to have many nonzero coefficients, but with a distinct positive tail, sometimes having few nonzero coefficients.
By contrast, the <code>liblinear</code> models are generally sparser with fewer than 2500 nonzero coefficients, out of ~16100 total input features, and a much narrower tail (Figure <a href="#fig:optimizer_coefs">3</a>A).
The sum of the coefficient magnitudes, however, tends to be smaller on average across all levels of regularization for SGD than for <code>liblinear</code> (Figure <a href="#fig:optimizer_coefs">3</a>B).
This effect is less pronounced for the other learning rate schedules shown in Figure <a href="#fig:sgd_lr_schedulers">2</a>, with the other options resulting in larger coefficient magnitudes (Supplementary Figure <a href="#fig:coef_weights_lr">6</a>).
These results suggest that the models fit by <code>liblinear</code> and SGD navigate the tradeoff between bias and variance in slightly different ways: <code>liblinear</code> tends to produce sparser models (more zero coefficients) as regularization increases, but if the learning rate is properly tuned, SGD coefficients tend to have smaller overall magnitudes as regularization increases.</p>
<p>We also compared the components of the loss function across different levels of regularization between optimizers.
The LASSO logistic regression loss function can be broken down into a data-dependent component (the log-loss) and a parameter magnitude dependent component (the L1 penalty), which are added to get the total loss that is minimized by each optimizer; see Methods for additional details.
As regularization strength decreases for <code>liblinear</code>, the data loss collapses to near 0, and the L1 penalty dominates the overall loss (Figure <a href="#fig:optimizer_coefs">3</a>C).
For SGD, on the other hand, the data loss decreases slightly as regularization strength decreases but remains relatively high (Figure <a href="#fig:optimizer_coefs">3</a>D).
Other SGD learning rate schedules have similar loss curves to the <code>liblinear</code> results, although this does not result in improved classification performance (Supplementary Figure <a href="#fig:loss_lr">7</a>).</p>
<div id="fig:optimizer_coefs" class="fignos">
<figure>
<img src="images/optimizers/figure_3.png" style="width:100.0%" alt="Figure 3: A. Distribution across genes of the number of nonzero coefficients included in best-performing LASSO logistic regression models. Violin plot density estimations are clipped at the ends of the observed data range, and boxes show the median/IQR. B. Distribution across genes of the sum of model coefficient weights for best-performing LASSO logistic regression models. C. Decomposition of loss function for models fit using liblinear across regularization levels. 0 values on the y-axis are rounded up to machine epsilon; i.e. 2.22 x 10-16. D. Decomposition of loss function for models fit using SGD across regularization levels. 0 values on the y-axis are rounded up to machine epsilon; i.e. 2.22 x 10-16." />
<figcaption aria-hidden="true"><span>Figure 3:</span> <strong>A.</strong> Distribution across genes of the number of nonzero coefficients included in best-performing LASSO logistic regression models. Violin plot density estimations are clipped at the ends of the observed data range, and boxes show the median/IQR.
<strong>B.</strong> Distribution across genes of the sum of model coefficient weights for best-performing LASSO logistic regression models.
<strong>C.</strong> Decomposition of loss function for models fit using <code>liblinear</code> across regularization levels. 0 values on the y-axis are rounded up to machine epsilon; i.e. 2.22 x 10<sup>-16</sup>.
<strong>D.</strong> Decomposition of loss function for models fit using SGD across regularization levels. 0 values on the y-axis are rounded up to machine epsilon; i.e. 2.22 x 10<sup>-16</sup>.</figcaption>
</figure>
</div>
<h3 id="discussion">Discussion</h3>
<p>Our work shows that optimizer choice presents tradeoffs in model selection for cancer transcriptomics.
We observed that LASSO logistic regression models for mutation status prediction fit using stochastic gradient descent were highly sensitive to learning rate tuning, but they tended to perform robustly across diverse levels of regularization and sparsity.
Coordinate descent implemented in <code>liblinear</code> did not require learning rate tuning, but generally performed best for a narrow range of fairly sparse models, overfitting as regularization strength decreased.
Tuning of regularization strength for <code>liblinear</code>, and learning rate (and regularization strength to a lesser degree) for SGD, are critical steps which must be considered as part of analysis pipelines.
The sensitivity we observed to these details highlights the importance of reporting exactly what optimizer was used, and how the relevant hyperparameters were selected, in studies that use machine learning models for transcriptomic data.</p>
<p>To our knowledge, the phenomenon we observed with SGD has not been documented in other applications of machine learning to genomic or transcriptomic data.
In recent years, however, the broader machine learning research community has identified and characterized implicit regularization for SGD in many settings, including overparametrized or feature-rich problems as is often the case in transcriptomics <span class="citation" data-cites="kwgYU3iP DIONdBGr kajx4KkT">[<a href="#ref-kwgYU3iP" role="doc-biblioref">111</a>,<a href="#ref-DIONdBGr" role="doc-biblioref">112</a>,<a href="#ref-kajx4KkT" role="doc-biblioref">113</a>]</span>.
The resistance we observed of SGD-optimized models to decreased performance on held-out data as model complexity increases is often termed “benign overfitting”: overfit models, in the sense that they fit the training data perfectly and perform worse on the test data, can still outperform models that do not fit the training data as well or that have stronger explicit regularization.
Benign overfitting has been attributed to optimization using SGD <span class="citation" data-cites="kajx4KkT 1HdGWY10w">[<a href="#ref-kajx4KkT" role="doc-biblioref">113</a>,<a href="#ref-1HdGWY10w" role="doc-biblioref">114</a>]</span>, and similar patterns have been observed for both linear models and deep neural networks <span class="citation" data-cites="jhZXoLtp 1Cb94HrPD">[<a href="#ref-jhZXoLtp" role="doc-biblioref">115</a>,<a href="#ref-1Cb94HrPD" role="doc-biblioref">116</a>]</span>.</p>
<p>Existing gene expression prediction benchmarks and pipelines typically use a single model implementation, and thus a single optimizer.
We recommend thinking critically about optimizer choice, but this can be challenging for researchers that are inexperienced with machine learning or unfamiliar with how certain models are fit under the hood.
For example, R’s <code>glmnet</code> package uses a cyclical coordinate descent algorithm to fit logistic regression models <span class="citation" data-cites="10aVUDFmA">[<a href="#ref-10aVUDFmA" role="doc-biblioref">117</a>]</span>, which would presumably behave similarly to <code>liblinear</code>, but this is somewhat opaque in the <code>glmnet</code> documentation itself.
Increased transparency and documentation in popular machine learning packages with respect to optimization, especially for models that are difficult to fit or sensitive to hyperparameter settings, would benefit new and unfamiliar users.</p>
<p>Related to what we see in our SGD-optimized models, there exist other problems in gene expression analysis where using all available features is comparable to, or better than, using a subset.
For example, using the full gene set improves correlations between preclinical cancer models and their tissue of origin, as compared to selecting genes based on variability or tissue-specificity <span class="citation" data-cites="lhYg2Yc3">[<a href="#ref-lhYg2Yc3" role="doc-biblioref">118</a>]</span>.
On the other hand, when predicting cell line viability from gene expression profiles, selecting features by Pearson correlation improves performance over using all features, similar to our <code>liblinear</code> classifiers <span class="citation" data-cites="1CcaMqQZg">[<a href="#ref-1CcaMqQZg" role="doc-biblioref">119</a>]</span>.
In future work, it could be useful to explore if the coefficients found by <code>liblinear</code> and SGD emphasize the same pathways or functional gene sets, or if there are patterns to which mutation status classifiers (or other cancer transcriptomics classifiers) perform better with more/fewer nonzero coefficients.</p>
<h3 id="data-and-code-availability">Data and code availability</h3>
<p>The data analyzed during this study were previously published as part of the TCGA Pan-Cancer Atlas project <span class="citation" data-cites="AK17eOgD">[<a href="#ref-AK17eOgD" role="doc-biblioref">120</a>]</span>, and are available from the NIH NCI Genomic Data Commons (GDC).
The scripts used to download and preprocess the datasets for this study are available at <a href="https://github.com/greenelab/pancancer-evaluation/tree/master/00_process_data" class="uri">https://github.com/greenelab/pancancer-evaluation/tree/master/00_process_data</a>, and the code used to carry out the analyses in this study is available at <a href="https://github.com/greenelab/pancancer-evaluation/tree/master/01_stratified_classification" class="uri">https://github.com/greenelab/pancancer-evaluation/tree/master/01_stratified_classification</a>, both under the open-source BSD 3-clause license.
Equivalent versions of Figure <a href="#fig:optimizer_compare_mutations">1</a>A and <a href="#fig:optimizer_compare_mutations">1</a>B for all 84 genes in the Vogelstein et al. 2013 gene set are available on Figshare at <a href="https://doi.org/10.6084/m9.figshare.22728644" class="uri">https://doi.org/10.6084/m9.figshare.22728644</a>, under a CC0 license.
This manuscript was written using Manubot <span class="citation" data-cites="YuJbg3zO">[<a href="#ref-YuJbg3zO" role="doc-biblioref">121</a>]</span> and is available on GitHub at <a href="https://github.com/greenelab/optimizer-manuscript" class="uri">https://github.com/greenelab/optimizer-manuscript</a> under the CC0-1.0 license.
This research was supported in part by the University of Pittsburgh Center for Research Computing through the resources provided. Specifically, this work used the HTC cluster, which is supported by NIH award number S10OD028483.</p>
<h3 id="supplementary-material">Supplementary Material</h3>
<div id="fig:compare_sparsity" class="fignos">
<figure>
<img src="images/optimizers/supp_figure_1.png" style="width:100.0%" alt="Figure 4: Number of nonzero coefficients (model sparsity) across varying regularization parameter settings for KRAS mutation prediction using SGD and liblinear optimizers." />
<figcaption aria-hidden="true"><span>Figure 4:</span> Number of nonzero coefficients (model sparsity) across varying regularization parameter settings for KRAS mutation prediction using SGD and <code>liblinear</code> optimizers.</figcaption>
</figure>
</div>
<div id="fig:compare_all_lr" class="fignos">
<figure>
<img src="images/optimizers/supp_figure_2.png" class="page_break_before" style="width:100.0%" alt="Figure 5: Distribution of performance difference between best-performing model for liblinear and SGD optimizers, across all 84 genes in Vogelstein driver gene set, for varying SGD learning rate schedulers. Positive numbers on the x-axis indicate better performance using liblinear, and negative numbers indicate better performance using SGD." />
<figcaption aria-hidden="true"><span>Figure 5:</span> Distribution of performance difference between best-performing model for <code>liblinear</code> and SGD optimizers, across all 84 genes in Vogelstein driver gene set, for varying SGD learning rate schedulers. Positive numbers on the x-axis indicate better performance using <code>liblinear</code>, and negative numbers indicate better performance using SGD.</figcaption>
</figure>
</div>
<div id="fig:coef_weights_lr" class="fignos">
<figure>
<img src="images/optimizers/supp_figure_3.png" class="page_break_before" style="width:100.0%" alt="Figure 6: Sum of absolute value of coefficients + 1 for KRAS mutation prediction using SGD and liblinear optimizers, with varying learning rate schedules for SGD. Similar to the figures in the main paper, the liblinear x-axis represents the inverse of the C regularization parameter; SGD x-axes represent the untransformed \alpha parameter." />
<figcaption aria-hidden="true"><span>Figure 6:</span> Sum of absolute value of coefficients + 1 for KRAS mutation prediction using SGD and <code>liblinear</code> optimizers, with varying learning rate schedules for SGD. Similar to the figures in the main paper, the <code>liblinear</code> x-axis represents the inverse of the <span class="math inline">\(C\)</span> regularization parameter; SGD x-axes represent the untransformed <span class="math inline">\(\alpha\)</span> parameter.</figcaption>
</figure>
</div>
<div id="fig:loss_lr" class="fignos">
<figure>
<img src="images/optimizers/supp_figure_4.png" class="page_break_before" style="width:100.0%" alt="Figure 7: Decomposition of loss function into data loss and L1 penalty components for KRAS mutation prediction using SGD optimizer, across regularization levels, using varying learning rate schedulers. 0 values on the y-axis are rounded up to machine epsilon, i.e. 2.22 x 10-16." />
<figcaption aria-hidden="true"><span>Figure 7:</span> Decomposition of loss function into data loss and L1 penalty components for KRAS mutation prediction using SGD optimizer, across regularization levels, using varying learning rate schedulers. 0 values on the y-axis are rounded up to machine epsilon, i.e. 2.22 x 10<sup>-16</sup>.</figcaption>
</figure>
</div>
<h2 id="chapter-3-widespread-redundancy-in--omics-profiles-of-cancer-mutation-states">Chapter 3: widespread redundancy in -omics profiles of cancer mutation states</h2>
<p>This chapter has been published in <em>Genome Biology</em> (https://doi.org/10.1186/s13059-022-02705-y) as “Widespread redundancy in -omics profiles of cancer mutation states”.</p>
<p><strong>Contributions:</strong>
JC: conceptualization, methodology, software, visualization, writing - original draft, writing - review and editing
BCC: methodology, writing - review and editing
MC: methodology, writing - review and editing
CSG: conceptualization, funding acquisition, methodology, supervision, writing - review and editing</p>
<h3 id="abstract-2">Abstract</h3>
<h4 id="background">Background</h4>
<p>In studies of cellular function in cancer, researchers are increasingly able to choose from many -omics assays as functional readouts.
Choosing the correct readout for a given study can be difficult, and which layer of cellular function is most suitable to capture the relevant signal remains unclear.</p>
<h4 id="results-2">Results</h4>
<p>We consider prediction of cancer mutation status (presence or absence) from functional -omics data as a representative problem that presents an opportunity to quantify and compare the ability of different -omics readouts to capture signals of dysregulation in cancer.
From the TCGA Pan-Cancer Atlas that contains genetic alteration data, we focus on RNA sequencing, DNA methylation arrays, reverse phase protein arrays (RPPA), microRNA, and somatic mutational signatures as -omics readouts.
Across a collection of genes recurrently mutated in cancer, RNA sequencing tends to be the most effective predictor of mutation state.
We find that one or more other data types for many of the genes are approximately equally effective predictors.
Performance is more variable between mutations than that between data types for the same mutation, and there is little difference between the top data types.
We also find that combining data types into a single multi-omics model provides little or no improvement in predictive ability over the best individual data type.</p>
<h4 id="conclusions">Conclusions</h4>
<p>Based on our results, for the design of studies focused on the functional outcomes of cancer mutations, there are often multiple -omics types that can serve as effective readouts, although gene expression seems to be a reasonable default option.</p>
<h3 id="background-1">Background</h3>
<p>Although cancer can be initiated and driven by many different genetic alterations, these tend to converge on a limited number of pathways or signaling processes <span class="citation" data-cites="11ZPgRuLk">[<a href="#ref-11ZPgRuLk" role="doc-biblioref">6</a>]</span>.
As driver mutation status alone confers limited prognostic information, a comprehensive understanding of how diverse genetic alterations perturb central pathways is vital to precision medicine and biomarker identification efforts <span class="citation" data-cites="1CHUpIalb Lm19TMrZ">[<a href="#ref-1CHUpIalb" role="doc-biblioref">122</a>,<a href="#ref-Lm19TMrZ" role="doc-biblioref">123</a>]</span>.
While many methods exist to distinguish driver mutations from passenger mutations based on genomic sequence characteristics <span class="citation" data-cites="MnWpJEWE YKm7IBTV 17HuO3ZJW">[<a href="#ref-MnWpJEWE" role="doc-biblioref">124</a>,<a href="#ref-YKm7IBTV" role="doc-biblioref">125</a>,<a href="#ref-17HuO3ZJW" role="doc-biblioref">126</a>]</span>, until recently it has been a challenge to connect driver mutations to downstream changes in gene expression and cellular function within individual tumor samples.</p>
<p>The Cancer Genome Atlas (TCGA) Pan-Cancer Atlas provides uniformly processed, multi-platform -omics measurements across tens of thousands of samples from 33 cancer types <span class="citation" data-cites="AK17eOgD">[<a href="#ref-AK17eOgD" role="doc-biblioref">120</a>]</span>.
Enabled by this publicly available data, a growing body of work on linking the presence of driving genetic alterations in cancer to downstream gene expression changes has emerged.
Recent studies have considered Ras pathway alteration status in colorectal cancer <span class="citation" data-cites="zPJFCX9h">[<a href="#ref-zPJFCX9h" role="doc-biblioref">49</a>]</span>, alteration status across many cancer types in Ras genes <span class="citation" data-cites="17XHtmqPY Z5JvAfru">[<a href="#ref-17XHtmqPY" role="doc-biblioref">50</a>,<a href="#ref-Z5JvAfru" role="doc-biblioref">127</a>]</span>, <em>TP53</em> <span class="citation" data-cites="X6oQhIf8">[<a href="#ref-X6oQhIf8" role="doc-biblioref">128</a>]</span>, and <em>PIK3CA</em> <span class="citation" data-cites="MUt3ESlf">[<a href="#ref-MUt3ESlf" role="doc-biblioref">52</a>]</span>, and alteration status across cancer types in frequently mutated genes <span class="citation" data-cites="NsW0qxZF">[<a href="#ref-NsW0qxZF" role="doc-biblioref">51</a>]</span>.
More broadly, other groups have drawn on similar ideas to distinguish between the functional effects of different alterations in the same driver gene <span class="citation" data-cites="ekDcxjcr">[<a href="#ref-ekDcxjcr" role="doc-biblioref">129</a>]</span>, to link alterations with similar gene expression signatures within cancer types <span class="citation" data-cites="odNCMbto">[<a href="#ref-odNCMbto" role="doc-biblioref">55</a>]</span>, and to identify trans-acting expression quantitative trait loci (trans-eQTLs) in germline genetic studies <span class="citation" data-cites="pLo10xih">[<a href="#ref-pLo10xih" role="doc-biblioref">130</a>]</span>.</p>
<p>These studies share a common thread: they each combine genomic (point mutation and copy number variation) data with transcriptomic (RNA sequencing) data within samples to interrogate the functional effects of genetic variation.
RNA sequencing is ubiquitous and cheap, and its experimental and computational methods are relatively mature, making it a vital tool for generating insight into cancer pathology <span class="citation" data-cites="1z8H2EME">[<a href="#ref-1z8H2EME" role="doc-biblioref">131</a>]</span>.
Some driver mutations, however, are known to act indirectly on gene expression through varying mechanisms.
For example, oncogenic <em>IDH1</em> and <em>IDH2</em> mutations in glioma have been shown to interfere with histone demethylation, which results in increased DNA methylation and blocked cell differentiation <span class="citation" data-cites="1Cxnszz2L WN82rY7P Sp8RO1w7 FYj8JHBX">[<a href="#ref-1Cxnszz2L" role="doc-biblioref">132</a>,<a href="#ref-WN82rY7P" role="doc-biblioref">133</a>,<a href="#ref-Sp8RO1w7" role="doc-biblioref">134</a>,<a href="#ref-FYj8JHBX" role="doc-biblioref">135</a>]</span>.
Other genes implicated in aberrant DNA methylation in cancer include the TET family of genes <span class="citation" data-cites="19sDhpROP">[<a href="#ref-19sDhpROP" role="doc-biblioref">136</a>]</span> and <em>SETD2</em> <span class="citation" data-cites="rNm0rAz7">[<a href="#ref-rNm0rAz7" role="doc-biblioref">137</a>]</span>.
Certain driver mutations, such as those in DNA damage repair genes, may lead to detectable patterns of somatic mutation <span class="citation" data-cites="18LSSJo3W">[<a href="#ref-18LSSJo3W" role="doc-biblioref">138</a>]</span>.
Additionally, correlation between gene expression and protein abundance in cancer cell lines is limited, and proteomics data could correspond more directly to certain cancer phenotypes and pathway perturbations <span class="citation" data-cites="pWXLXywn">[<a href="#ref-pWXLXywn" role="doc-biblioref">139</a>]</span>.
In these contexts and others, integrating different data modalities or combining multiple data modalities could be more effective than relying solely on gene expression as a functional signature.</p>
<p>Here, we compare -omics data types profiled in the TCGA Pan-Cancer Atlas to evaluate use as a multivariate functional readout of genetic alterations in cancer.
We focus on gene expression (RNA sequencing data), DNA methylation (27K and 450K probe chips), reverse phase protein array (RPPA), microRNA expression, and mutational signatures data <span class="citation" data-cites="11yRODwNY">[<a href="#ref-11yRODwNY" role="doc-biblioref">140</a>]</span> as possible readouts.
Prior studies have identified univariate correlations of CpG site methylation <span class="citation" data-cites="UTtkLysq YBXY18s4">[<a href="#ref-UTtkLysq" role="doc-biblioref">141</a>,<a href="#ref-YBXY18s4" role="doc-biblioref">142</a>]</span> and correlations of RPPA protein profiles <span class="citation" data-cites="pRTkOXGI">[<a href="#ref-pRTkOXGI" role="doc-biblioref">143</a>]</span> with the presence or absence of certain driver mutations.
Other relevant past work includes linking point mutations and copy number variants (CNVs) with changes in methylation and expression at individual genes <span class="citation" data-cites="ZuUqcptb lQ8oalgu">[<a href="#ref-ZuUqcptb" role="doc-biblioref">144</a>,<a href="#ref-lQ8oalgu" role="doc-biblioref">145</a>]</span> and identifying functional modules that are perturbed by somatic mutations <span class="citation" data-cites="StwDSCo7 1BMNB7BTo">[<a href="#ref-StwDSCo7" role="doc-biblioref">146</a>,<a href="#ref-1BMNB7BTo" role="doc-biblioref">147</a>]</span>.
However, direct comparison among different data types for this application is lacking, particularly in the multivariate case where we consider changes to -omics-derived gene signatures rather than individual genes in isolation.</p>
<p>We select a collection of potential cancer drivers with varying functions and roles in cancer development.
We use mutation status in these genes as labels to train classifiers, using each of the data types listed as training data, in a pan-cancer setting; we follow similar methods to the elastic net logistic regression approach described in Way et al. 2018 <span class="citation" data-cites="17XHtmqPY">[<a href="#ref-17XHtmqPY" role="doc-biblioref">50</a>]</span> and Way et al. 2020 <span class="citation" data-cites="NsW0qxZF">[<a href="#ref-NsW0qxZF" role="doc-biblioref">51</a>]</span>.
We show that there is considerable predictive signal for many genes relative to a cancer-type corrected baseline and that gene expression tends to provide good predictions of mutation state across most genes.
Surprisingly, we find that for a variety of genes, multiple data types are approximately equally effective predictors.
We observe similar results for pan-cancer survival prediction across the same data types with little separation between the top-performing data types.
In addition, we observe that combining data types into a single multi-omics model for mutation prediction provides little, if any, performance benefit over the most performant model using a single data type.
Our results will help to inform the design of future functional genomics studies in cancer, suggesting that for many strong drivers with clear functional signatures, different -omics measurements can provide similar information content.</p>
<h3 id="methods-1">Methods</h3>
<h4 id="mutation-data-download-and-preprocessing">Mutation data download and preprocessing</h4>
<p>To generate binary mutated/non-mutated gene labels for our machine learning model, we used mutation calls for TCGA samples from MC3 <span class="citation" data-cites="cDnF3a5Q">[<a href="#ref-cDnF3a5Q" role="doc-biblioref">107</a>]</span> and copy number threshold calls from GISTIC2.0 <span class="citation" data-cites="rDV1hhfF">[<a href="#ref-rDV1hhfF" role="doc-biblioref">108</a>]</span>.
MC3 mutation calls were downloaded from the Genomic Data Commons (GDC) of the National Cancer Institute, at <a href="https://gdc.cancer.gov/about-data/publications/pancanatlas"><code>https://gdc.cancer.gov/about-data/publications/pancanatlas</code></a>.
Copy number threshold calls are from an older version of the GDC data, and are available here: <a href="https://figshare.com/articles/dataset/TCGA_PanCanAtlas_Copy_Number_Data/6144122"><code>https://figshare.com/articles/dataset/TCGA_PanCanAtlas_Copy_Number_Data/6144122</code></a>.
We removed hypermutated samples (defined as five or more standard deviations above the mean non-silent somatic mutation count) from our dataset to reduce the number of false positives (i.e., non-driver mutations).
After this filtering, 9,074 TCGA samples with mutation and copy number data remained.
Any sample with a non-silent somatic variant in the target gene was included in the positive set.
We also included copy number gains in the target gene for oncogenes and copy number losses in the target gene for tumor suppressor genes in the positive set; all remaining samples were considered negative for mutation in the target gene.</p>
<h4 id="omics-data-download-and-preprocessing">Omics data download and preprocessing</h4>
<p>RNA sequencing, 27K and 450K methylation array, microRNA, and RPPA datasets for TCGA samples were all downloaded from GDC, at the same link provided above.
Mutational signatures information for TCGA samples with whole-exome sequencing data was downloaded from the International Cancer Genome Consortium (ICGC) data portal, at <a href="https://dcc.icgc.org/releases/PCAWG/mutational_signatures/Signatures_in_Samples/SP_Signatures_in_Samples"><code>https://dcc.icgc.org/releases/PCAWG/mutational_signatures/Signatures_in_Samples/SP_Signatures_in_Samples</code></a>.
For our experiments, we used only the “single base signature” (SBS) mutational signatures, generated in <span class="citation" data-cites="11yRODwNY">[<a href="#ref-11yRODwNY" role="doc-biblioref">140</a>]</span>.
In general, before training classifiers or extracting PCA components from all of the data types, we standardized (took z-scores of) each column/feature of all data types.
For the RNA sequencing dataset, we generally used only the top 8,000 gene features by mean absolute deviation as predictors in our single-omics models, except where specified otherwise.
For the RPPA, microRNA, and mutational signatures datasets, all columns/features were used.</p>
<p>To remove missing values from the methylation datasets, we removed the 10 samples with the most missing values, then performed mean imputation for probes with 1 or 2 values missing.
All probes with missing values remaining after sample filtering and imputation were dropped from the analysis.
This left us with 20,040 CpG probes in the 27K methylation dataset and 370,961 CpG probes in the 450K methylation dataset.
For experiments where “raw” methylation data was used, we used the top 100,000 probes in the 450K dataset by mean absolute deviation for computational efficiency, and we used all of the 20,040 probes in the 27K dataset.
For experiments where “compressed” methylation data was used, we used principal component analysis (PCA), as implemented in the <code>scikit-learn</code> Python library <span class="citation" data-cites="lF0Fr9LL">[<a href="#ref-lF0Fr9LL" role="doc-biblioref">148</a>]</span>, to extract the top 5,000 principal components from the methylation datasets.
We initially applied the beta-mixture quantile normalization (BMIQ) method <span class="citation" data-cites="Zg2jmD6p">[<a href="#ref-Zg2jmD6p" role="doc-biblioref">149</a>]</span> to correct for variability in signal intensity between type I and type II probes, but we observed that this had no effect on our results.
We report uncorrected results in the main paper for simplicity.</p>
<h4 id="construction-of-a-set-of-cancer-genes">Construction of a set of cancer genes</h4>
<p>To get a comprehensive picture of classification performance across a wide variety of cancer-related genes, we integrated several curated gene sets from the literature into a single “merged” cancer gene set.
The individual gene sets we integrated were from Vogelstein et al. <span class="citation" data-cites="809OyWlC">[<a href="#ref-809OyWlC" role="doc-biblioref">106</a>]</span> (all genes from Table S2A), Bailey et al. <span class="citation" data-cites="A3VeCeaB">[<a href="#ref-A3VeCeaB" role="doc-biblioref">150</a>]</span> (only genes annotated as “pan-cancer” drivers in Table S1), and the COSMIC Cancer Gene Census <span class="citation" data-cites="DxovERyv">[<a href="#ref-DxovERyv" role="doc-biblioref">151</a>]</span> (all Tier 1 genes annotated as “somatic”).
In addition, the COSMIC CGC dataset contains 3 possible “roles in cancer” for each gene: oncogene, TSG, and fusion gene; for this analysis we dropped genes that are annotated only as fusion genes (i.e. no oncogene or TSG annotation).
These filters resulted in a starting dataset of 511 cancer-related genes, which we reduced further for each experiment based on the number of mutated (i.e. positively labeled) samples as described in the next section.</p>
<h4 id="comparing-data-modalities">Comparing data modalities</h4>
<p>We made three main comparisons in this study: one between different sets of genes using only expression data, one comparing expression and DNA methylation data types, and one comparing all data types.
This choice in comparisons was mainly due to sample size limitations, as running a single comparison using all data types would force us to use only samples that are profiled for every data type, which would discard a large number of samples that lack profiling on only one or a few data types.
Thus, for each of the three comparisons, we used the intersection of TCGA samples having measurements for all of the datasets being compared in that experiment.
This resulted in three distinct sets of samples: 9,074 samples shared between {expression, mutation} data, 7,981 samples shared between {expression, mutation, 27K methylation, 450K methylation}, and 5,226 samples shared between {expression, mutation, 27K methylation, 450K methylation, RPPA, microRNA, mutational signatures}.
When we dropped samples between experiments as progressively more data types were added, we observed that the dropped samples had approximately the same cancer type proportions as the dataset as a whole.
In other words, samples that were profiled for one data type but not another did not tend to come exclusively from one or a few cancer types.
Exceptions included acute myeloid leukemia (LAML) which had no samples profiled in the RPPA data, and ovarian cancer (OV) which had only 8 samples with 450K methylation data.
More detailed information on cancer type proportions profiled for each data type is provided in Additional File 1: Fig. S1 and Additional File 2.</p>
<p>For each target gene, in order to ensure that the training dataset was reasonably balanced (i.e. that there would be enough mutated samples to train an effective classifier), we included only cancer types with at least 15 mutated samples and at least 5% mutated samples, which we refer to here as “valid” cancer types.
After applying these filters, the number of valid cancer types remaining for each gene varied based on the set of samples used: more data types resulted in fewer shared samples, and fewer samples generally meant fewer valid cancer types.
In some cases, this resulted in genes with no valid cancer types, which we dropped from the analysis.
Out of the 511 genes from the “merged” cancer gene set described in the previous section, for the analysis using {expression, mutation} data we retained 268 target genes, for the {expression, mutation, 27k methylation, 450k methylation} analysis we retained 272 genes, and for the analysis using all data types we retained 217 genes.</p>
<p>We additionally explored mutation prediction from gene expression alone using three gene sets of equal size: the cancer-related genes from the merged dataset described previously, a set of frequently mutated genes in TCGA, and a set of random genes with mutations profiled by MC3.
To match the size of the merged cancer gene set, we took the 268 most frequently mutated genes in TCGA as quantified by MC3, all of which had at least one valid cancer type.
For the random gene set, we first filtered to the set of all genes with one or more valid cancer types by the same criteria (15 total samples mutated and at least 5% of samples mutated), then sampled 268 of the resulting 1,348 genes uniformly at random.
Based on the results of the gene expression experiments, we used the merged cancer-related gene set for all subsequent experiments comparing -omics data types.</p>
<h4 id="training-classifiers-to-detect-cancer-mutations">Training classifiers to detect cancer mutations</h4>
<p>We trained logistic regression classifiers to predict whether or not a given sample had a mutational event in a given target gene using data from various -omics datasets as explanatory variables.
Our model was trained on -omics data (<span class="math inline">\(X\)</span>) to predict mutation presence or absence (<span class="math inline">\(y\)</span>) in a target gene.
To control for varying mutation burden per sample and to adjust for potential cancer type-specific expression patterns, we included one-hot encoded cancer type and log<sub>10</sub>(sample mutation count) in the model as covariates.
Since our -omics datasets tend to have many dimensions and comparatively few samples, we used an elastic net penalty to prevent overfitting <span class="citation" data-cites="JZNkB8d7">[<a href="#ref-JZNkB8d7" role="doc-biblioref">152</a>]</span> in line with the approach used in Way et al. 2018 <span class="citation" data-cites="17XHtmqPY">[<a href="#ref-17XHtmqPY" role="doc-biblioref">50</a>]</span> and Way et al. 2020 <span class="citation" data-cites="NsW0qxZF">[<a href="#ref-NsW0qxZF" role="doc-biblioref">51</a>]</span>.
Elastic net logistic regression finds the feature weights <span class="math inline">\(\hat{w} \in \mathbb{R}^{p}\)</span> solving the following optimization problem:</p>
<p><span class="math display">\[\hat{w} = \text{argmin}_{w} \ \ell(X, y; w) + \alpha \lambda||w||_1 + \frac{1}{2}\alpha (1 - \lambda) ||w||_2\]</span></p>
<p>where <span class="math inline">\(i \in \{1, \dots, n\}\)</span> denotes a sample in the dataset, <span class="math inline">\(X_i \in \mathbb{R}^{p}\)</span> denotes features (omics measurements) from the given sample, <span class="math inline">\(y_i \in \{0, 1\}\)</span> denotes the label (mutation presence/absence) for the given sample, and <span class="math inline">\(\ell(\cdot)\)</span> denotes the negative log-likelihood of the observed data given a particular choice of feature weights, i.e.</p>
<p><span class="math display">\[\ell(X, y; w) = -\sum_{i=1}^{n} y_i \log\left(\frac{1}{1 + e^{-w^{\top}X_i}}\right) + (1 - y_i) \log\left(1 - \frac{1}{1 + e^{-w^{\top}X_i}}\right)\]</span></p>
<p>This optimization problem leaves two hyperparameters to select: <span class="math inline">\(\alpha\)</span> (controlling the tradeoff between the data log-likelihood and the penalty on large feature weight values), and <span class="math inline">\(\lambda\)</span> (controlling the tradeoff between the L1 penalty and L2 penalty on the weight values).
Although the elastic net optimization problem does not have a closed form solution, the loss function is convex, and iterative optimization algorithms are commonly used for finding reasonable solutions.
For fixed values of <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\lambda\)</span>, we solved for <span class="math inline">\(\hat{w}\)</span> using stochastic gradient descent, as implemented in <code>scikit-learn</code>’s <code>SGDClassifier</code> method.</p>
<p>Given weight values <span class="math inline">\(\hat{w}\)</span>, it is straightforward to predict the probability of a positive label (mutation in the target gene) <span class="math inline">\(P(y^{*} = 1 \mid X^{*}; \hat{w})\)</span> for a test sample <span class="math inline">\(X^{*}\)</span>:</p>
<p><span class="math display">\[P(y^{*} = 1 \mid X^{*}; \hat{w}) = \frac{1}{1 + e^{-\hat{w}^{\top}X^{*}}}\]</span></p>
<p>and the probability of no mutation in the target gene, <span class="math inline">\(P(y^{*} = 0 \mid X^{*}; \hat{w})\)</span>, is given by (1 - the above quantity).</p>
<p>For each target gene, we evaluated model performance using two replicates of 4-fold cross-validation, where train and test splits were stratified by cancer type and sample type.
That is, each training set/test set combination had equal proportions of each cancer type (BRCA, SKCM, COAD, etc.) and each sample type (primary tumor, recurrent tumor, etc.).
To choose the elastic net hyperparameters, we used 3-fold nested cross-validation, with a grid search over the following hyperparameter ranges: <span class="math inline">\(\lambda\)</span> = [0.0, 0.05, 0.1, 0.3, 0.5, 0.7, 0.9, 1.0] and <span class="math inline">\(\alpha\)</span> = [0.0001, 0.001, 0.01, 0.1, 1, 10].
Using the grid search results, for each evaluation fold we selected the set of hyperparameters with the optimal area under the precision-recall curve (AUPR), averaged over the three inner folds.</p>
<h4 id="evaluating-mutation-prediction-classifiers">Evaluating mutation prediction classifiers</h4>
<p>Area under the receiver-operator curve (AUROC) <span class="citation" data-cites="WTSzUd5Y">[<a href="#ref-WTSzUd5Y" role="doc-biblioref">153</a>]</span> and area under the precision-recall curve (AUPR) <span class="citation" data-cites="pazxvNcz">[<a href="#ref-pazxvNcz" role="doc-biblioref">154</a>]</span> are metrics that are frequently used to quantify classification performance for a continuous or probabilistic output, such as that provided by logistic regression.
These metrics summarize performance across a variety of binary label thresholds, rather than requiring choice of a single threshold to determine positive or negative predictions.
In the main text, we report results using AUPR, summarized using average precision.
AUPR has been shown to distinguish between models more accurately than AUROC when there are few positively labeled samples <span class="citation" data-cites="LB4mDas6 lWf6zHNO">[<a href="#ref-LB4mDas6" role="doc-biblioref">155</a>,<a href="#ref-lWf6zHNO" role="doc-biblioref">156</a>]</span>.
As an additional correction for imbalanced labels, in many of the results in the main text we report the difference in AUPR between a classifier fit to true mutation labels and a classifier fit to data where the mutation labels are randomly permuted.
In cases where mutation labels are highly imbalanced (very few mutated samples and many non-mutated samples), a classifier with permuted labels may perform well simply by chance, e.g. by predicting the negative/non-mutated class for most samples.
To maintain the same label balance for the classifiers with permuted labels as the classifiers with the true labels, we permuted labels separately in the train and test sets for each cross-validation split.
Additionally, to maintain the same label proportions within each cancer type after permuting the labels, we permuted labels independently for each cancer type.</p>
<p>Recall that for each target gene and each -omics dataset, we ran two replicates of 4-fold cross-validation, for a total of eight performance results.
To make a statistical comparison between two models using these performance distributions, we used paired-sample <em>t</em>-tests, where performance measurements derived from the same cross-validation fold are considered paired measurements.
We used this approach to compare a model trained on true labels with a model trained on permuted labels (addressing the question, “for the given gene using the given data type, can we predict mutation status better than random”), and to compare a model trained on data type A with a model trained on data type B (addressing the question, “for the given gene, can we make more effective mutation status predictions using data type A or data type B”).</p>
<p>We corrected for multiple tests using a Benjamini-Hochberg false discovery rate correction.
For experiments where we chose a binary threshold for accepting/rejecting <span class="math inline">\(H_0\)</span> we set a conservative corrected threshold of <span class="math inline">\(p = 0.001\)</span>; we were able to estimate the number of false positives by examining genes with better performance for permuted mutation labels than true labels.
We chose this threshold to ensure that none of the observed false positive genes were considered significant, since we would never expect permuting labels to improve performance.
However, our results were not sensitive to the choice of this threshold, and we display cutoffs of <span class="math inline">\(p = 0.05\)</span> and <span class="math inline">\(p = 0.01\)</span> in many of our plots as well.</p>
<h4 id="survival-prediction-using--omics-datasets">Survival prediction using -omics datasets</h4>
<p>As a complementary comparison to mutation prediction, we constructed predictors of patient survival using the clinical data available from the GDC, in the <code>TCGA-CDR-SupplementalTableS1.xlsx</code> file.
Following the methods described in <span class="citation" data-cites="CZHe0A4l">[<a href="#ref-CZHe0A4l" role="doc-biblioref">157</a>]</span>, as the clinical endpoint we used overall survival (OS), except in nine cancer types with few deaths observed where we used progression-free intervals (PFI) as the clinical endpoint (BRCA, DLBC, LGG, PCPG, PRAD, READ, TGCT, THCA and THYM).
For prediction, we used Cox regression as implemented in the <code>scikit-survival</code> Python package <span class="citation" data-cites="wiw7Q7M7">[<a href="#ref-wiw7Q7M7" role="doc-biblioref">158</a>]</span>, with patient age at diagnosis and log<sub>10</sub>(sample mutation count) included as covariates, as well as a one-hot encoded variable for cancer type in the pan-cancer case.
To ensure that the per-feature information content was comparable between -omics data types, we preprocessed the -omics datasets using PCA and extracted the top <span class="math inline">\(k\)</span> principal components; in the case where the number of features in the original dataset was less than <span class="math inline">\(k\)</span> we used all available PCs (that is, we set <span class="math inline">\(k = \min(p, k)\)</span> where <span class="math inline">\(p\)</span> is the number of features in the unprocessed dataset).
For the pan-cancer models we plot results over multiple values of <span class="math inline">\(k\)</span>: <span class="math inline">\(k \in \{10, 100, 500, 1000, 5000\}\)</span>; for the individual cancer type models we set <span class="math inline">\(k = 10\)</span>.</p>
<p>To model pan-cancer survival (results shown in main paper), we used the elastic net Cox regression implementation in <code>scikit-survival</code> (i.e. the <code>CoxnetSurvivalAnalysis</code> method).
To select hyperparameters for the elastic net Cox regression model, we performed a grid search over <span class="math inline">\(\lambda\)</span> = [0.0, 0.05, 0.1, 0.3, 0.5, 0.7, 0.9, 1.0] and <span class="math inline">\(\alpha\)</span> = [0, 1e-5, 1e-4, 5e-4, 0.001, 0.005, 0.01, 0.05, 0.1, 0.5, 1, 10, 100, 1000].
To select the regularization parameter <span class="math inline">\(\alpha\)</span>, we used the default selection procedure implemented in <code>scikit-survival</code> to determine a range of potential <span class="math inline">\(\alpha\)</span> values based on the data.
This procedure begins by deriving the maximum <span class="math inline">\(\alpha\)</span> value as the smallest value for which all coefficients are 0 (call this <span class="math inline">\(\alpha_{\text{max}}\)</span>), then it selects 100 possibilities for alpha spaced evenly on a log scale between <span class="math inline">\(\alpha_{\text{max}}\)</span> and <span class="math inline">\(0.01 \cdot \alpha_{\text{max}}\)</span>.
We found that for individual cancer types, this data-driven procedure resulted in more consistent and stable model convergence than choosing a fixed set of alphas to search over as in the pan-cancer survival prediction experiments.</p>
<p>We measured survival prediction performance using the censored concordance index (c-index) <span class="citation" data-cites="pubmed:8668867">[<a href="#ref-pubmed:8668867" role="doc-biblioref"><strong>pubmed:8668867?</strong></a>]</span>, which quantifies agreement between the order of survival time predictions and true outcomes for a held-out dataset; higher c-index values indicate more accurate survival prediction performance.
Similar to the mutation prediction experiments, we calculated c-index values on held-out subsets of the data for two replicates of 4-fold cross-validation, resulting in eight performance measurements for each model.
As a baseline, for both the pan-cancer and cancer type specific datasets, we constructed survival models using only non-omics covariates.
For the pan-cancer data, covariates included patient age at diagnosis, log<sub>10</sub>(sample mutation count), and a one-hot encoded variable for sample cancer type.
The cancer type-specific baseline models were the same, without the cancer type indicator, since all train and test samples were derived from the same cancer type.</p>
<h4 id="multi-omics-mutation-prediction-experiments">Multi-omics mutation prediction experiments</h4>
<p>To predict mutation presence or absence in cancer genes using multiple data types simultaneously, we concatenated individual datasets into a large feature matrix, then used the same elastic net logistic regression method described previously.
For this task, we considered only the gene expression, 27K methylation, and 450K methylation datasets.
We used only these data types to limit the number of multi-omics combinations; the expression and methylation datasets resulted in the best overall performance across the single-omics experiments, so we limited combinations to those datasets.
In the main text, we report results using the top 5,000 principal components for each dataset , which ensures that most variance is captured (approximately 95-98% of variance for each data type).
In Additional File 1: Fig. S6, we also report results using “raw” features: for gene expression we used all 15,639 genes available in our RNA sequencing dataset, and for the 27K and 450K methylation datasets we used the top 20,000 CpG probes by mean absolute deviation.</p>
<p>To construct the multi-omics models, we considered each of the pairwise combinations of the datasets listed above, as well as a combination of all 3 datasets.
When combining multiple datasets, we concatenated along the column axis and included covariates for cancer type and sample mutation burden as before.
For all multi-omics experiments, we used only the samples from TCGA with data for all three data types (i.e. the same 7,981 samples used in the single-omics experiments comparing expression and methylation data types).
We considered a limited subset of well-performing genes from the merged cancer gene set as target genes, including <em>EGFR</em>, <em>IDH1</em>, <em>KRAS</em>, <em>PIK3CA</em>, <em>SETD2</em>, and <em>TP53</em>.
We selected these genes because we had previously observed that they have good predictive performance and because they represent a combination of alterations that have strong gene expression signatures (<em>KRAS</em>, <em>EGFR</em>, <em>IDH1</em>, <em>TP53</em>) and strong DNA methylation signatures (<em>IDH1</em>, <em>SETD2</em>, <em>TP53</em>).</p>
<p>For the experiments predicting mutation status using a 3-layer fully connected neural network, described in the Results section and Additional File 1: Fig. S7, we used the top 5,000 principal components as input for each data type.
We selected hyperparameters for each of the 8 outer cross-validation splits using a single inner train/validation split and a random search over 20 hyperparameter combinations.
The hyperparameter ranges that we sampled from in the random search were: <code>learning_rate: [0.1, 0.01, 0.001, 5e-4, 1e-4], h1_size: [1000, 500, 250], dropout: [0.5, 0.75, 0.9], weight_decay: [0, 0.1, 1, 100]</code>.
Here, <code>h1_size</code> refers to the size of the first hidden layer, and the size of the second hidden layer was always set to <code>h1_size / 2</code>.
As in the elastic net grid search, we chose the combination of hyperparameters with the best AUPR on the validation set, and retrained the model with those hyperparameters to make predictions on the test set.
We trained our networks with the Adam optimizer <span class="citation" data-cites="c6d3lKFX">[<a href="#ref-c6d3lKFX" role="doc-biblioref">159</a>]</span>, with ReLU activation after the hidden layers and sigmoid activation to make predictions, and using binary cross-entropy as the loss function as implemented in the PyTorch <code>BCEWithLogitsLoss</code> function, through the <code>skorch</code> library which provides interoperability between PyTorch and scikit-learn.</p>
<h3 id="results-3">Results</h3>
<h4 id="using-diverse-data-modalities-to-predict-cancer-alterations">Using diverse data modalities to predict cancer alterations</h4>
<p>We collected five different data modalities from cancer samples in the TCGA Pan-Cancer Atlas, capturing five steps of cellular function that are perturbed by genetic alterations in cancer (Figure <a href="#fig:omics_overview">8</a>A).
These included gene expression (RNA-seq data), DNA methylation (27K and 450K Illumina BeadChip arrays), protein abundance (RPPA data), microRNA expression data, and patterns of somatic mutation (mutational signatures).
To link these diverse data modalities to changes in mutation status, we used elastic net logistic regression to predict the presence or absence of mutations in cancer genes, using these readouts as predictive features (Figure <a href="#fig:omics_overview">8</a>B).
We evaluated the resulting mutation status classifiers in a pan-cancer setting, preserving the proportions of each of the 33 cancer types in TCGA for eight train/test splits (4 folds x 2 replicates) in each of approximately 250 cancer genes (Figure <a href="#fig:omics_overview">8</a>C).</p>
<p>We sought to compare classifiers against a baseline where mutation labels are permuted (to identify genes whose mutation status correlates strongly with a functional signature in a given data type) and also to compare classifiers trained on true labels across different data types (to identify data types that are more or less predictive of mutations in a given gene).
To account for variation between dataset splits in making these comparisons, we treat classification metrics from the eight train/test splits as performance distributions, which we compare using <em>t</em>-tests.
We summarize performance across all genes in our cancer gene set using a similar approach to a volcano plot, in which each point is a gene.
In our summary plots, the x-axis shows the magnitude of the change in the classification metric between conditions, and the y-axis shows the <em>p</em>-value for the associated <em>t</em>-test (Figure <a href="#fig:omics_overview">8</a>C).</p>
<div id="fig:omics_overview" class="fignos">
<figure>
<img src="images/omics/figure_1.png" alt="Figure 8: A. Cancer mutations can perturb cellular function via a variety of cellular processes. Arrows represent major potential paths of information flow from a somatic mutation in DNA to its resulting cell phenotype; circular arrow represents the ability of certain mutations (e.g. in DNA damage repair genes) to alter somatic mutation patterns. Note that this does not reflect all possible relationships between cellular processes: for instance, changes in gene expression can lead to changes in somatic mutation rates. B. Predicting presence/absence of somatic alterations in cancer from diverse data modalities. In this study, we use functional readouts from TCGA as predictive features and the presence or absence of mutation in a given gene as labels. This reverses the primary direction of information flow shown in Panel A. C. Schematic of evaluation pipeline." />
<figcaption aria-hidden="true"><span>Figure 8:</span> <strong>A.</strong> Cancer mutations can perturb cellular function via a variety of cellular processes.
Arrows represent major potential paths of information flow from a somatic mutation in DNA to its resulting cell phenotype; circular arrow represents the ability of certain mutations (e.g. in DNA damage repair genes) to alter somatic mutation patterns.
Note that this does not reflect all possible relationships between cellular processes: for instance, changes in gene expression can lead to changes in somatic mutation rates.
<strong>B.</strong> Predicting presence/absence of somatic alterations in cancer from diverse data modalities.
In this study, we use functional readouts from TCGA as predictive features and the presence or absence of mutation in a given gene as labels.
This reverses the primary direction of information flow shown in Panel A.
<strong>C.</strong> Schematic of evaluation pipeline.</figcaption>
</figure>
</div>
<h4 id="selection-of-cancer-related-genes-improves-predictive-signal">Selection of cancer-related genes improves predictive signal</h4>
<p>As a baseline, we evaluated prediction of mutation status from gene expression data across several different gene sets.
Past work has evaluated mutation prediction for the top 50 most mutated genes in TCGA <span class="citation" data-cites="NsW0qxZF">[<a href="#ref-NsW0qxZF" role="doc-biblioref">51</a>]</span>, and we sought to extend this to a broader list of gene sets.
To evaluate whether using known cancer-related genes tends to improve prediction, we compiled a set of cancer-related genes (n=268) from Vogelstein et al. 2013 <span class="citation" data-cites="809OyWlC">[<a href="#ref-809OyWlC" role="doc-biblioref">106</a>]</span>, Bailey et al. 2018 <span class="citation" data-cites="A3VeCeaB">[<a href="#ref-A3VeCeaB" role="doc-biblioref">150</a>]</span>, and the COSMIC Cancer Gene Census <span class="citation" data-cites="DxovERyv">[<a href="#ref-DxovERyv" role="doc-biblioref">151</a>]</span>.
We compared performance on this curated gene set with performance on an equal number of genes sampled randomly after applying a mutation frequency threshold (n=268, see Methods for sampling details) and an equal number of the most mutated genes in TCGA (n=268).
For all gene sets, we used only the set of TCGA samples for which both gene expression and somatic mutation data exists, resulting in a total of 9,074 samples across all 33 cancer types.
This set of samples was further filtered for each target gene to cancer types containing at least 15 mutated samples and at least 5% of samples mutated for that cancer type.
As an alternate approach, we tried including/excluding entire genes using similar filters, and the results were consistent across filtering strategies (Additional File 1: Fig. S4).
We then evaluated the performance for each target gene in each of the three gene sets.</p>
<p>Overall, genes from the cancer-related gene set were more predictable than randomly chosen genes or those selected by total mutation count (Figure <a href="#fig:expression_gene_sets">9</a>A).
In total, for a significance threshold of <span class="math inline">\(\alpha = 0.001\)</span>, 120/268 genes (44.8%) in the cancer-related gene set are significantly predictable from gene expression data, compared to 14/268 genes (5.22%) in the random gene set and 80/268 genes (29.9%) in the most mutated gene set.
Of the 14 significantly predictable genes in the random gene set, 13 of them are also in the cancer-related gene set (highlighted with ‘X’ in Figure <a href="#fig:expression_gene_sets">9</a>B), and of the 80 significantly predictable genes in the most mutated gene set, 26 of them are also in the cancer-related gene set (highlighted in red in Figure <a href="#fig:expression_gene_sets">9</a>C).
These results suggest that selecting target genes for mutation prediction based on prior knowledge of their involvement in cancer pathways and processes, rather than randomly or based on mutation frequency alone, can improve predictive signal and identify more highly predictable mutations from gene expression data.</p>
<div id="fig:expression_gene_sets" class="fignos">
<figure>
<img src="images/omics/figure_2.png" style="width:90.0%" alt="Figure 9: A. Overall distribution of performance across three gene sets, using gene expression (RNA-seq) data to predict mutations. Each data point represents the mean cross-validated AUPR difference, compared with a baseline model trained on permuted mutation presence/absence labels, for one gene in the given gene set; notches show bootstrapped 95% confidence intervals. “random” = 268 random genes, “most mutated” = 268 most mutated genes, “cancer gene set” = 268 cancer related genes from curated gene sets. Significance stars indicate results of Bonferroni-corrected pairwise Wilcoxon tests: **: p &lt; 0.01, ***: p &lt; 0.001, ns: not statistically significant for a cutoff of p = 0.05. B, C, D. Volcano-like plots showing mutation presence/absence predictive performance for each gene in each of the three gene sets. The x-axis shows the difference in mean AUPR compared with a baseline model trained on permuted labels, and the y-axis shows p-values for a paired t-test comparing cross-validated AUPR values within folds. Points (genes) marked with an “X” are overlapping between the cancer gene set and either the random or most mutated gene set." />
<figcaption aria-hidden="true"><span>Figure 9:</span> <strong>A.</strong> Overall distribution of performance across three gene sets, using gene expression (RNA-seq) data to predict mutations.
Each data point represents the mean cross-validated AUPR difference, compared with a baseline model trained on permuted mutation presence/absence labels, for one gene in the given gene set; notches show bootstrapped 95% confidence intervals.
“random” = 268 random genes, “most mutated” = 268 most mutated genes, “cancer gene set” = 268 cancer related genes from curated gene sets.
Significance stars indicate results of Bonferroni-corrected pairwise Wilcoxon tests: **: <em>p</em> &lt; 0.01, ***: <em>p</em> &lt; 0.001, ns: not statistically significant for a cutoff of <em>p</em> = 0.05.
<strong>B, C, D.</strong> Volcano-like plots showing mutation presence/absence predictive performance for each gene in each of the three gene sets.
The <em>x</em>-axis shows the difference in mean AUPR compared with a baseline model trained on permuted labels, and the <em>y</em>-axis shows <em>p</em>-values for a paired <em>t</em>-test comparing cross-validated AUPR values within folds.
Points (genes) marked with an “X” are overlapping between the cancer gene set and either the random or most mutated gene set.</figcaption>
</figure>
</div>
<h4 id="gene-expression-predicts-cancer-mutation-status-more-effectively-than-dna-methylation">Gene expression predicts cancer mutation status more effectively than DNA methylation</h4>
<p>We compared gene expression with DNA methylation as downstream readouts of the effects of cancer alterations.
In these analyses, we considered both the 27K probe and 450K probe methylation datasets generated for the TCGA Pan-Cancer Atlas.
As target genes, we used the same combined cancer-related gene set described in the “Selection of cancer-related genes” section.
We used samples that had data for each of the data types being compared, including somatic mutation data to generate mutation labels.
This process retained 7,981 samples in the intersection of the expression, 27K methylation, 450K methylation, and mutation datasets, which we used for subsequent analyses.
The most frequent missing data types were somatic mutation data (1,114 samples) and 450K methylation data (1,072 samples) (Figure <a href="#fig:methylation">10</a>A).</p>
<p>For many genes, predictions are better than our baseline model where labels are permuted (values greater than 0 in the box plots), suggesting that there is considerable predictive signal in both expression and methylation datasets across the cancer-related gene set (Figure <a href="#fig:methylation">10</a>B).
On aggregate across all genes, predictive performance is best overall for gene expression.
Both before and after filtering for genes that exceed the significance threshold, gene expression with raw gene features provides a significant performance improvement relative to the 27K methylation and 450K methylation datasets (Figure <a href="#fig:methylation">10</a>B-C).
Results were similar with PCA-compressed gene expression features or raw CpG probes as predictors (Additional File 1: Fig. S5).</p>
<p>Considering each target gene in the cancer-related gene set individually, we observed that 113/272 genes significantly outperformed the permuted baseline using gene expression data, as compared to 62/272 genes for 27K methylation and 77/272 genes for 450K methylation (Figure <a href="#fig:methylation">10</a>D-F, more information about specific genes in Additional File 1: Fig. S2).
Some “well-predicted” genes that outperformed the permuted baseline tended to be similar between data types (Figure <a href="#fig:methylation">10</a>D-F; genes in the top right of each plot).
For example, <em>CIC</em> appears in the top right of all 3 plots, and <em>CCND1</em> appears in the top right of the gene expression and 450K methylation plots, suggesting that mutations in these genes have strong gene expression and DNA methylation signatures, and these signatures tend to be preserved across cancer types.</p>
<p>In addition to comparing mutation classifiers trained on different data types to the permuted baseline, we also compared classifiers trained on true labels directly to each other, for genes that performed significantly better than the baseline for both of the data types under consideration (Figure <a href="#fig:methylation">10</a>G-H).
We observed that 18/58 genes were significantly more predictable from expression data than 27K methylation data, and 21/69 genes were significantly more predictable from expression data than 450K methylation data.
In both cases, no genes were significantly more predictable using the methylation data types.
Still, we observed that some points were clustered around the origin, indicating that the data types appear to confer similar information about mutation status.
That is, in these cases, matching the gene being studied with the “correct” data modality seems to be unimportant: mutation status has a strong signature which can be extracted from both expression and DNA methylation data roughly equally.</p>
<p>We additionally compared pan-cancer survival prediction performance using principal components derived from each data type; in general, results were comparable across the three data types (Figure <a href="#fig:methylation">10</a>I).
All data types outperformed the covariate-only baseline (see <a href="#survival-prediction-using--omics-datasets">Methods</a>) for lower numbers of PC features included, although performance was similar to the baseline for higher numbers of PCs.
Confidence intervals between the best- and worst-performing data types overlap at most PC counts (with the exception of gene expression at 5,000 PC features), suggesting that similarly to mutation prediction, the three data types tend to have comparable effectiveness for pan-cancer survival prediction.</p>
<div id="fig:methylation" class="fignos">
<figure>
<img src="images/omics/figure_3.png" style="width:90.0%" alt="Figure 10: A. Count of overlapping samples between gene expression, 27K methylation, 450K methylation, and somatic mutation data used from TCGA. Only non-zero overlap counts are shown. Somatic mutation sample information is included because it is needed to generate the mutation presence/absence labels. B. Predictive performance for genes in the cancer-related gene set, using each of the three data types as predictors. The gene expression predictor uses the top 8000 gene features by mean absolute deviation, and the methylation predictors use the top 5000 principal components as predictive features. Significance stars indicate results of Bonferroni-corrected pairwise Wilcoxon tests: **: p &lt; 0.01, ***: p &lt; 0.001, ns: not statistically significant for a cutoff of p = 0.05. C. Predictive performance for genes where at least one of the considered data types predicts mutation labels significantly better than the permuted baseline. D-F. Predictive performance for each gene in the cancer-related gene set, for each data type, compared with a baseline model trained on permuted labels. G-H. Direct comparison of performance using gene expression and each methylation dataset, for genes that perform significantly better than the baseline for both data types. Points (genes) to the left of y=0 perform better using gene expression-derived features, and points to the right perform better using methylation-derived features. I. Pan-cancer survival prediction performance, quantified using c-index on the y-axis, for gene expression, 27K methylation, and 450K methylation. The x-axis shows results with varying numbers of principal components included for each data type. Models also included covariates for patient age, sample mutation burden, and sample cancer type; grey dotted line indicates mean performance for a covariate-only baseline model." />
<figcaption aria-hidden="true"><span>Figure 10:</span> <strong>A.</strong> Count of overlapping samples between gene expression, 27K methylation, 450K methylation, and somatic mutation data used from TCGA.
Only non-zero overlap counts are shown.
Somatic mutation sample information is included because it is needed to generate the mutation presence/absence labels.
<strong>B.</strong> Predictive performance for genes in the cancer-related gene set, using each of the three data types as predictors.
The gene expression predictor uses the top 8000 gene features by mean absolute deviation, and the methylation predictors use the top 5000 principal components as predictive features.
Significance stars indicate results of Bonferroni-corrected pairwise Wilcoxon tests: **: <em>p</em> &lt; 0.01, ***: <em>p</em> &lt; 0.001, ns: not statistically significant for a cutoff of <em>p</em> = 0.05.
<strong>C.</strong> Predictive performance for genes where at least one of the considered data types predicts mutation labels significantly better than the permuted baseline.
<strong>D-F.</strong> Predictive performance for each gene in the cancer-related gene set, for each data type, compared with a baseline model trained on permuted labels.
<strong>G-H.</strong> Direct comparison of performance using gene expression and each methylation dataset, for genes that perform significantly better than the baseline for both data types. Points (genes) to the left of y=0 perform better using gene expression-derived features, and points to the right perform better using methylation-derived features.
<strong>I.</strong> Pan-cancer survival prediction performance, quantified using c-index on the <em>y</em>-axis, for gene expression, 27K methylation, and 450K methylation. The <em>x</em>-axis shows results with varying numbers of principal components included for each data type. Models also included covariates for patient age, sample mutation burden, and sample cancer type; grey dotted line indicates mean performance for a covariate-only baseline model.</figcaption>
</figure>
</div>
<p>Focusing on several selected genes of interest, we observed that relative classifier performance varies by gene (Figure <a href="#fig:methylation_genes">11</a>).
Past work has indicated that mutations in <em>TP53</em> are highly predictable from gene expression data <span class="citation" data-cites="X6oQhIf8">[<a href="#ref-X6oQhIf8" role="doc-biblioref">128</a>]</span>, and we observed that the methylation datasets provided similar predictive performance (Figure <a href="#fig:methylation_genes">11</a>A).
Similarly, for <em>IDH1</em> both expression and methylation features result in similar performance, consistent with the previously observed role of IDH1 in regulating both DNA methylation and gene expression (Figure <a href="#fig:methylation_genes">11</a>D) <span class="citation" data-cites="KGJwKXkD">[<a href="#ref-KGJwKXkD" role="doc-biblioref">160</a>]</span>.
Mutations in <em>KRAS</em> and <em>ERBB2</em> (<em>HER2</em>) were most predictable from gene expression data, and in both cases the methylation datasets significantly outperformed the baseline as well (Figure <a href="#fig:methylation_genes">11</a>B and <a href="#fig:methylation_genes">11</a>E).
Gene expression signatures of <em>ERBB2</em> alterations are historically well-studied in breast cancer <span class="citation" data-cites="uBEARk2">[<a href="#ref-uBEARk2" role="doc-biblioref">161</a>]</span>, and samples with activating <em>ERBB2</em> mutations have recently been shown to share sensitivities to some small-molecule inhibitors across cancer types <span class="citation" data-cites="1c8GokzT">[<a href="#ref-1c8GokzT" role="doc-biblioref">162</a>]</span>.
These observations are consistent with the pan-cancer <em>ERBB2</em> mutant-associated expression signature that we observed in this study.
<em>NF1</em> mutations were also most predictable from gene expression data, although the gene expression-based <em>NF1</em> mutation classifier did not significantly outperform the baseline with permuted labels at a cutoff of <span class="math inline">\(\alpha = 0.001\)</span> (Figure <a href="#fig:methylation_genes">11</a>C).
<em>SETD2</em> is an example of a gene that is more predictable from the methylation datasets than from gene expression, although gene expression with raw gene features significantly outperformed the permuted baseline as well (Figure <a href="#fig:methylation_genes">11</a>F).
<em>SETD2</em> is widely mutated across cancer types and affects H3K36 histone methylation most directly, but SETD2-mediated changes in H3K36 methylation have been linked to dysregulation of diverse cellular processes including DNA methylation and RNA splicing <span class="citation" data-cites="rNm0rAz7 fdVKgtEF">[<a href="#ref-rNm0rAz7" role="doc-biblioref">137</a>,<a href="#ref-fdVKgtEF" role="doc-biblioref">163</a>]</span>.</p>
<div id="fig:methylation_genes" class="fignos">
<figure>
<img src="images/omics/figure_4.png" style="width:90.0%" alt="Figure 11: Performance across varying PCA dimensions for specific genes of interest. Dotted lines represent results for “raw” features (8,000 gene features for gene expression data and 8,000 CpG probes for both methylation datasets, selected by largest mean absolute deviation). Error bars and shaded regions show bootstrapped 95% confidence intervals. Stars in boxes show statistical testing results compared with permuted baseline model; each box refers to the model using the number of PCA components it is over (far right box = models with raw features). **: p &lt; 0.01, ***: p &lt; 0.001, no stars: not statistically significant for a cutoff of p = 0.05." />
<figcaption aria-hidden="true"><span>Figure 11:</span> Performance across varying PCA dimensions for specific genes of interest. Dotted lines represent results for “raw” features (8,000 gene features for gene expression data and 8,000 CpG probes for both methylation datasets, selected by largest mean absolute deviation). Error bars and shaded regions show bootstrapped 95% confidence intervals. Stars in boxes show statistical testing results compared with permuted baseline model; each box refers to the model using the number of PCA components it is over (far right box = models with raw features). **: <em>p</em> &lt; 0.01, ***: <em>p</em> &lt; 0.001, no stars: not statistically significant for a cutoff of <em>p</em> = 0.05.</figcaption>
</figure>
</div>
<h4 id="comparing-six-different-readouts-favors-expression-and-dna-methylation">Comparing six different readouts favors expression and DNA methylation</h4>
<p>Next, we expanded our comparison to all five functional data modalities (six total readouts, since there are two DNA methylation platforms) available in the TCGA Pan-Cancer Atlas.
As with previous experiments, we limited our comparison to the set of samples profiled for each readout, resulting in 5,226 samples with data for all readouts.
The data types with the most missing samples were RPPA data (2,215 samples that were missing RPPA data) and 450K methylation (630 samples that were missing 450K methylation data) (Figure <a href="#fig:all_data">12</a>A).
Summarized over all genes in the cancer-related gene set, we observed that gene expression tended to produce better predictions than the other data types (Figure <a href="#fig:all_data">12</a>B).
This remained true when we looked only at the set of genes having at least one significant predictor (i.e. “well-predicted” genes) (Figure <a href="#fig:all_data">12</a>C).</p>
<p>On the individual gene level, mutations in 33/217 genes were significantly predictable from RPPA data relative to the permuted baseline, compared to 25/217 genes from microRNA data and 2/217 genes from mutational signatures data (Figure <a href="#fig:all_data">12</a>D-F).
For the remaining data types on this smaller set of samples, 79/217 genes outperformed the baseline for gene expression data, 31/217 for 27k methylation, and 42/217 for 450k methylation.
Compared to the methylation experiments (Figure <a href="#fig:methylation">10</a>), we observed fewer “well-predicted” genes for the expression and methylation datasets here (likely due to the considerably smaller sample size) but relative performance was comparable (Additional File 1: Fig. S3).
Direct comparisons between each added data type and gene expression data showed that for most “well-predicted” genes, RPPA, microRNA and mutational signatures data generally provide similar or worse performance compared to gene expression (Figure <a href="#fig:all_data">12</a>G-I).</p>
<p>Performance using RPPA data (Figure <a href="#fig:all_data">12</a>G) is notable because of its drastically smaller dimensionality than the other data types (190 proteins, compared to thousands of features for the expression and methylation data types).
This suggests that each protein abundance measurement provides a high information content, although this is by design as the antibody probes used for the TCGA analysis were selected to cover established cancer-related pathways <span class="citation" data-cites="rBwmNJtL">[<a href="#ref-rBwmNJtL" role="doc-biblioref">164</a>]</span>.
Similarly, the scope of the features captured by the mutational signatures data we used is limited to single-base substitution signatures; a broader spectrum of possible signatures is described in previous work <span class="citation" data-cites="11yRODwNY 23jjG6I6">[<a href="#ref-11yRODwNY" role="doc-biblioref">140</a>,<a href="#ref-23jjG6I6" role="doc-biblioref">165</a>]</span> including doublet-substitution signatures, small indel signatures, and signatures of structural variation, but these were not readily available for the TCGA exome sequencing data.
The relatively poor predictive ability of mutational signatures likely stems from a combination of biological and technical factors, as there is no reason to expect that changes in somatic mutation patterns would be directly caused by most cancer driver mutations.
Two exceptions are <em>KMT2C</em> and <em>KMT2D</em> (Figure <a href="#fig:all_data">12</a>F), which may have a role in mediating DNA damage response <span class="citation" data-cites="HfPt51y9">[<a href="#ref-HfPt51y9" role="doc-biblioref">166</a>]</span>.</p>
<p>As in the expression/methylation comparison, we also compared pan-cancer survival prediction performance between all six readouts, using the top principal components derived from each data type to ensure comparable information content (Figure <a href="#fig:all_data">12</a>J).
All six readouts performed comparably for this smaller set of samples, with slightly better performance across PC feature dimensions for the 450K methylation array.
The covariate-only baseline predictor performed considerably worse than it did in the expression/methylation comparisons, with all -omics data types outperforming the baseline predictor at all PC numbers.</p>
<div id="fig:all_data" class="fignos">
<figure>
<img src="images/omics/figure_5.png" style="width:90.0%" alt="Figure 12: A. Overlap of TCGA samples between all data types used in mutation prediction comparisons. Only overlaps with more than 100 samples are shown. Somatic mutation sample information is included because it is needed to generate the mutation presence/absence labels. B. Overall distribution of performance per data type across 217 genes from the cancer-related gene set. Each data point represents mean cross-validated AUPR difference, compared with a baseline model trained on permuted labels, for one gene; notches show bootstrapped 95% confidence intervals. Significance stars indicate results of Bonferroni-corrected pairwise Wilcoxon tests: **: p &lt; 0.01, ***: p &lt; 0.001, ns: not statistically significant for a cutoff of p = 0.05. All pairwise tests were run, and corrected for, but only neighboring test results are shown. C. Overall performance distribution per data type for genes where the permuted baseline model is significantly outperformed for one or more data types, resulting in a total of 39 genes. D, E, F. Volcano-like plots showing predictive performance for each gene in the cancer-related gene set, in each of the added data types (RPPA, microRNA, mutational signatures). The x-axis shows the difference in mean AUPR compared with a baseline model trained on permuted labels, and the y-axis shows p-values for a paired t-test comparing cross-validated AUPR values within folds. G, H, I. Direct comparison of performance using gene expression and each added data type, showing only genes that perform significantly better than the baseline model for both data types. Points (genes) to the left of y=0 perform better using gene expression-derived features, and points to the right perform better using the added data type (RPPA, microRNA, and mutational signatures respectively). J. Pan-cancer survival prediction performance, quantified using c-index on the y-axis, for all data types. The x-axis shows results with varying numbers of principal components included for each data type. Models also included covariates for patient age, sample mutation burden, and sample cancer type; grey dotted line indicates mean performance for a covariate-only baseline model." />
<figcaption aria-hidden="true"><span>Figure 12:</span> <strong>A.</strong> Overlap of TCGA samples between all data types used in mutation prediction comparisons.
Only overlaps with more than 100 samples are shown.
Somatic mutation sample information is included because it is needed to generate the mutation presence/absence labels.
<strong>B.</strong> Overall distribution of performance per data type across 217 genes from the cancer-related gene set.
Each data point represents mean cross-validated AUPR difference, compared with a baseline model trained on permuted labels, for one gene; notches show bootstrapped 95% confidence intervals.
Significance stars indicate results of Bonferroni-corrected pairwise Wilcoxon tests: **: <em>p</em> &lt; 0.01, ***: <em>p</em> &lt; 0.001, ns: not statistically significant for a cutoff of <em>p</em> = 0.05.
All pairwise tests were run, and corrected for, but only neighboring test results are shown.
<strong>C.</strong> Overall performance distribution per data type for genes where the permuted baseline model is significantly outperformed for one or more data types, resulting in a total of 39 genes.
<strong>D, E, F.</strong> Volcano-like plots showing predictive performance for each gene in the cancer-related gene set, in each of the added data types (RPPA, microRNA, mutational signatures). The <em>x</em>-axis shows the difference in mean AUPR compared with a baseline model trained on permuted labels, and the <em>y</em>-axis shows <em>p</em>-values for a paired <em>t</em>-test comparing cross-validated AUPR values within folds.
<strong>G, H, I.</strong> Direct comparison of performance using gene expression and each added data type, showing only genes that perform significantly better than the baseline model for both data types. Points (genes) to the left of y=0 perform better using gene expression-derived features, and points to the right perform better using the added data type (RPPA, microRNA, and mutational signatures respectively).
<strong>J.</strong> Pan-cancer survival prediction performance, quantified using c-index on the <em>y</em>-axis, for all data types. The <em>x</em>-axis shows results with varying numbers of principal components included for each data type. Models also included covariates for patient age, sample mutation burden, and sample cancer type; grey dotted line indicates mean performance for a covariate-only baseline model.</figcaption>
</figure>
</div>
<p>When we constructed a heatmap depicting predictive performance for each gene across data types, we found that many genes tended to be well-predicted by more than one data type (Figure <a href="#fig:heatmap">13</a>).
Of the 86 genes that are well-predicted using at least one data type (grey circles in Figure <a href="#fig:heatmap">13</a>), 52/86 (60.5%) are well-predicted by multiple data types, meaning that multiple -omics readouts contain a detectable signature of presence/absence of a mutation in the corresponding gene.
Of the remaining 34 genes, 28/34 (82.4%) are well-predicted by gene expression alone.
This supports our observation that in a surprising number of cases, choosing the “correct” data modality is unimportant for driver genes with strong functional signatures, although gene expression may be the best “default” choice as it tends to be a strong predictor in the majority of cases.
Exceptions included <em>ERBB4</em>, <em>KMT2A</em>, <em>PIK3R1</em>, and <em>RPL22</em> (only well-predicted using RPPA data), <em>FAT4</em> (only well-predicted using microRNA data), and <em>KDM6A</em> (only well-predicted using 450K methylation data).</p>
<div id="fig:heatmap" class="fignos">
<figure>
<img src="images/omics/figure_6.png" style="width:90.0%" alt="Figure 13: Heatmap displaying predictive performance for mutations in each of the 217 genes from the cancer-related gene set, across all six TCGA data modalities. Each cell quantifies performance for a target gene, using predictive features derived from a particular data type. Grey shaded dots indicate that the given data type provides significantly better predictions than the permuted baseline for the given gene; black inner dots indicate the same and additionally that the given data type provides statistically equivalent performance to the data type with the best average performance (determined by pairwise t-tests across data types with FDR correction)." />
<figcaption aria-hidden="true"><span>Figure 13:</span> Heatmap displaying predictive performance for mutations in each of the 217 genes from the cancer-related gene set, across all six TCGA data modalities. Each cell quantifies performance for a target gene, using predictive features derived from a particular data type. Grey shaded dots indicate that the given data type provides significantly better predictions than the permuted baseline for the given gene; black inner dots indicate the same and additionally that the given data type provides statistically equivalent performance to the data type with the best average performance (determined by pairwise <em>t</em>-tests across data types with FDR correction).</figcaption>
</figure>
</div>
<h4 id="simple-multi-omics-integration-provides-little-performance-benefit">Simple multi-omics integration provides little performance benefit</h4>
<p>We also trained “multi-omics” classifiers to predict mutations in six well-studied and widely mutated driver genes across various cancer types: <em>EGFR</em>, <em>IDH1</em>, <em>KRAS</em>, <em>PIK3CA</em>, <em>SETD2</em>, and <em>TP53</em>.
Each of these genes is well-predicted from several data types in our earlier experiments (Figure <a href="#fig:heatmap">13</a>), consistent with having strong pan-cancer driver effects.
For the multi-omics classifiers, we considered all pairwise combinations of the top three performing individual data types (gene expression, 27K methylation, and 450K methylation), in addition to a model using all three data types.
We trained a classifier for multiple data types by concatenating features from the individual data types, then fitting the same elastic net logistic regression model as we used for the single-omics models.
Here, we show results using the top 5,000 principal components from each data type as predictive features, to ensure that feature count and scale is comparable among data types; results for raw features are shown in Additional File 1: Fig. S6.
We additionally ran the same experiments using a 3-layer fully-connected neural network for classification, with principal components as input, and results are shown in Additional File 1: Fig. S7.
In general, we found predictions using elastic net logistic regression to be more robust across cross-validation folds and hyperparameter choices than predictions using the neural network, although the neural network provided a slight performance improvement using multiple -omics types for some genes.</p>
<p>For each of the six target genes, we observed comparable performance between the best single-omics classifier (blue boxes in Figure <a href="#fig:multi_omics">14</a>A) and the best multi-omics classifier (orange boxes in Figure <a href="#fig:multi_omics">14</a>A).
Across all classifiers and data types, we found varied patterns based on the target gene.
For <em>IDH1</em> and <em>TP53</em> performance is relatively consistent regardless of data type(s), suggesting that baseline performance is high and there is little room for improvement as data is added (Figure <a href="#fig:multi_omics">14</a>C, G).
The <em>TP53</em> classifier using raw features showed a statistically significant improvement when multiple data types were integrated, although the difference in mean performance was relatively small (Additional File 1: Fig. S6, <em>p</em>=0.0078).
For <em>EGFR</em>, <em>KRAS</em>, and <em>PIK3CA</em>, combining gene expression with methylation data results in statistically equivalent or worse performance to gene expression alone; classifiers trained only on methylation data generally do not perform as well as those that integrate expression data (Figure <a href="#fig:multi_omics">14</a>B, D, E).
Previously, we saw that the best classifiers for <em>SETD2</em> used methylation data alone (Figure <a href="#fig:heatmap">13</a>).
When we added multiple data types to our <em>SETD2</em> classifier, we did observe an increase in performance (Figure <a href="#fig:multi_omics">14</a>F), although this improvement was not statistically significant in a paired-sample <em>t</em>-test for <span class="math inline">\(\alpha\)</span>=0.05 (<em>p</em>=0.078).
Overall, we observed that combining data types in a relatively simple manner, by concatenating features from each individual data type, provided little or no improvement in predictive ability over the best individual data type.
This supports our earlier observations of the redundancy of gene expression and methylation data as functional readouts, since our multi-omics classifiers are not in general able to extract gains in predictive performance as more data types are added for this set of cancer drivers.</p>
<div id="fig:multi_omics" class="fignos">
<figure>
<img src="images/omics/figure_7.png" style="width:90.0%" alt="Figure 14: A. Comparing the best-performing model (i.e. highest mean AUPR relative to permuted baseline) trained on a single data type against the best “multi-omics” model for each target gene. None of the differences between single-omics and multi-omics models were statistically significant using paired-sample Wilcoxon tests across cross-validation folds, for a threshold of 0.05. B-G. Classifier performance, relative to baseline with permuted labels, for mutation prediction models trained on various combinations of data types. Each panel shows performance for one of the six target genes; box plots show performance distribution over 8 evaluation sets (4 cross-validation folds x 2 replicates)." />
<figcaption aria-hidden="true"><span>Figure 14:</span> <strong>A.</strong> Comparing the best-performing model (i.e. highest mean AUPR relative to permuted baseline) trained on a single data type against the best “multi-omics” model for each target gene. None of the differences between single-omics and multi-omics models were statistically significant using paired-sample Wilcoxon tests across cross-validation folds, for a threshold of 0.05.
<strong>B-G.</strong> Classifier performance, relative to baseline with permuted labels, for mutation prediction models trained on various combinations of data types. Each panel shows performance for one of the six target genes; box plots show performance distribution over 8 evaluation sets (4 cross-validation folds x 2 replicates).</figcaption>
</figure>
</div>
<h3 id="discussion-1">Discussion</h3>
<p>We carried out a large-scale comparison of data types in the TCGA Pan-Cancer Atlas as functional readouts of genetic alterations in cancer, integrating results across cancer types and across driver genes.
Overall, we found that gene expression captures signatures of mutation state most effectively in general, relative to a baseline model, but we saw that for many genes other data types could be equally effective at predicting mutation presence or absence.
For pan-cancer survival prediction, we found that the functional readouts tended to be similarly effective, outperforming a simple baseline using age and sample mutation burden in most cases.
Our multi-omics modeling experiment indicated that the mutation state information captured by gene expression and DNA methylation is highly redundant, as added data types resulted in no gain or modest gains in classifier performance.</p>
<p>Comparing mutation status prediction using raw and PCA compressed expression and DNA methylation data, we observed that feature extraction using PCA provided no benefit compared to using raw gene or CpG probe features.
Other studies using DNA methylation array data have found that nonlinear dimension reduction methods, such as variational autoencoders and capsule networks, can be effective for extracting predictive features <span class="citation" data-cites="wOjMFUXP pubmed:34417465">[<a href="#ref-wOjMFUXP" role="doc-biblioref">167</a>,<a href="#ref-pubmed:34417465" role="doc-biblioref"><strong>pubmed:34417465?</strong></a>]</span>.
The latter approach is especially interesting because capsule networks and “capsule-like methods” can be constrained to extract features that align with known biology (i.e. that correspond to known disease pathways or CpG site annotations).
This can improve model interpretability as well as predictive performance.
Similar methods have been applied to extract biologically informed features from gene expression data (see, for instance, <span class="citation" data-cites="1HfJnooMx 7kIfUrlF">[<a href="#ref-1HfJnooMx" role="doc-biblioref">168</a>,<a href="#ref-7kIfUrlF" role="doc-biblioref">169</a>]</span>).
A more comprehensive study of dimension reduction methods in the context of mutation prediction, including the features selected by these methods and their biological relevance and interpretation, would be a beneficial area of future work.
In addition to methods for extracting features, another aspect of the study that could be explored further is methods for labeling samples as mutated or not more efficiently.
Although the mutation calls we used from MC3 represent the consensus of multiple algorithms aggregated through a standard pipeline, benchmarking other methods for identifying mutated samples could improve the utility of our method, such as calling mutations directly from RNA-seq data to avoid the need for paired samples <span class="citation" data-cites="eoXXrVpl gROOu6O4">[<a href="#ref-eoXXrVpl" role="doc-biblioref">170</a>,<a href="#ref-gROOu6O4" role="doc-biblioref">171</a>]</span>.</p>
<p>In contrast to many other studies demonstrating the benefits of integrating multiple -omics data types for various cancer-related prediction problems <span class="citation" data-cites="1AeRgHv3L tMmErZMX 4X1N4DJa shldZigM llajEDiU">[<a href="#ref-shldZigM" role="doc-biblioref">36</a>,<a href="#ref-1AeRgHv3L" role="doc-biblioref">172</a>,<a href="#ref-tMmErZMX" role="doc-biblioref">173</a>,<a href="#ref-4X1N4DJa" role="doc-biblioref">174</a>,<a href="#ref-llajEDiU" role="doc-biblioref">175</a>]</span>, we found that combining multiple data types to predict mutation status was generally not effective for this problem.
The method we used to integrate different data types by concatenating feature sets is sometimes referred to as “early” data integration (discussed in more detail in <span class="citation" data-cites="gIhNaGUg">[<a href="#ref-gIhNaGUg" role="doc-biblioref">176</a>]</span> and <span class="citation" data-cites="NfKILi8i">[<a href="#ref-NfKILi8i" role="doc-biblioref">177</a>]</span>).
It is possible that more sophisticated data integration methods, such as “intermediate” integration methods that learn a set of features jointly across datasets, would produce improved predictions.
We do not interpret our results as concrete evidence that multi-omics integration is not effective for this problem; rather, we see them as an indication that this is a challenging data integration problem for which further investigation is needed.
We also present this problem as a set of benchmark tasks on which multi-omics integration methods can be evaluated.
In addition to the methodological questions, the issue of data integration also has implications for the underlying biology: a more nuanced understanding of when different data readouts provide redundant information, and when they can contribute unique information about cancer pathology and development, could have many translational applications.</p>
<p>One limitation of the current study is that, for the mutation prediction problem, we only evaluated classifiers that were trained on pan-cancer data.
Considering every possible combination of target gene and TCGA cancer type (85 target genes x 33 cancer types x 6 data types) would have drastically increased the computational load and presented a large multiple testing burden.
Alternatively, choosing only a subset of gene/cancer type combinations to study would have biased our results toward known driver gene/cancer type relationships, which we aimed to avoid.
In future work it would be interesting to identify classifiers that perform well in a certain cancer type but not in the pan-cancer context and to compare these instances across different cancer types.
As a motivating example, other studies have shown that activating mutations in Ras isoforms (<em>HRAS</em>, <em>KRAS</em>, <em>NRAS</em>) tend to have similar effects to one another in thyroid cancer, producing similar gene expression signatures <span class="citation" data-cites="odNCMbto">[<a href="#ref-odNCMbto" role="doc-biblioref">55</a>]</span>.
In multiple myeloma, however, activating <em>KRAS</em> and <em>NRAS</em> mutations produce distinct expression signatures, necessitating separate classifiers <span class="citation" data-cites="1B2CloRMX">[<a href="#ref-1B2CloRMX" role="doc-biblioref">178</a>]</span>.
A high-throughput computational pipeline to identify cases where functional signatures of a particular cancer driver are either concordant or discordant between cancer types could identify opportunities for context-specific protein function prediction, improve biomarker identification, and suggest cases where drugs targeting specific alterations might produce discordant results in different cancer types.</p>
<p>As with any study relying on observational, cross-sectional data such as the TCGA Pan-Cancer Atlas, the conclusions that we can draw are limited by the data.
In particular, for any of our “well-predicted” genes (i.e. genes that, when mutated, have strong signatures in one or more data types), we cannot definitively distinguish correlation from causation.
To directly assess the effects of particular mutations on various data modalities, some studies use cell line data from sources such as the Cancer Cell Line Encyclopedia (CCLE) <span class="citation" data-cites="kcdvZmNN">[<a href="#ref-kcdvZmNN" role="doc-biblioref">9</a>]</span>.
While this approach could help to isolate the causal effect of a given mutation on a given cell line, cell lines are sometimes an imperfect match for the cancers they are derived from <span class="citation" data-cites="2AuD4Mwu">[<a href="#ref-2AuD4Mwu" role="doc-biblioref">179</a>]</span>.
We are also limited in that we cannot assign timing or clonal status to mutations, or fully characterize intratumor heterogeneity, with certainty from the bulk sequencing data generated by TCGA (although some features of tumor mutational processes over time can be estimated from bulk data, e.g. <span class="citation" data-cites="16yVVytBf">[<a href="#ref-16yVVytBf" role="doc-biblioref">180</a>]</span>).
As methods for generating large longitudinal datasets at single-cell resolution mature and scale, we will need to revise the way we think about cellular function and dysregulation in cancer cells, as dynamic and adaptive processes rather than a single representative snapshot of a tumor.</p>
<h3 id="conclusions-1">Conclusions</h3>
<p>Based on our results, for studies focused on the functional consequences of cancer mutations, we recommend that researchers cancers prioritize downstream readouts based on the gene or genes of interest (Figure <a href="#fig:heatmap">13</a>).
On balance, prediction of mutation status is best in general using gene expression data, and prediction of patient survival is similar for all data types in the study.
However, the finding that for many genes, multiple functional profiles contain much of the same information will be useful for some study designs, given varying cost and stability of different readouts.
In addition to gene expression, results using DNA methylation and RPPA measurements as predictive features were promising, especially considering the substantially lower dimensionality of the RPPA dataset compared to other data types.
It is important to note that the specific technologies chosen by TCGA, and the tradeoffs inherent in such a high-throughput study, could influence the comparison: it is possible that, for instance, another technology for measuring DNA methylation (such as bisulfite sequencing) or another technique for measuring protein abundance (such as mass spectrometry-based proteomics) could change performance for those data types.
Future technology advances, in both quality and quantity of data, are likely to improve our understanding of the full picture of functional consequences of mutations in cancer cells.</p>
<h3 id="declarations">Declarations</h3>
<h4 id="availability-of-data-and-materials">Availability of data and materials</h4>
<p>The datasets analyzed during this study were previously published as part of the TCGA Pan-Cancer Atlas project, and are publicly available from the NIH NCI Genomic Data Commons (GDC) <span class="citation" data-cites="KrfP0fMx">[<a href="#ref-KrfP0fMx" role="doc-biblioref">8</a>]</span>. The mutational signatures dataset was downloaded from the ICGC Data Portal <span class="citation" data-cites="pHDBQdQT">[<a href="#ref-pHDBQdQT" role="doc-biblioref">181</a>]</span>. Scripts used to download and preprocess the datasets for this study are available at <code>https://github.com/greenelab/mpmp/tree/master/00_download_data</code>.</p>
<p>All analyses were implemented in the Python programming language and are available at Zenodo <span class="citation" data-cites="J4w4e1MM">[<a href="#ref-J4w4e1MM" role="doc-biblioref">182</a>]</span> and in the following GitHub repository: <a href="https://github.com/greenelab/mpmp"><code>https://github.com/greenelab/mpmp</code></a> <span class="citation" data-cites="lVGBvUEc">[<a href="#ref-lVGBvUEc" role="doc-biblioref">183</a>]</span> under the open-source BSD 3-clause license.
Scripts to download large data files from GDC and other sources are located in the <code>00_download_data</code> directory.
Scripts to run experiments comparing data modalities used individually are located in the <code>02_classify_mutations</code> directory, scripts to run multi-omics experiments are located in the <code>05_classify_mutations_multimodal</code> directory, and scripts to run survival prediction experiments are located in the <code>06_predict_survival</code> directory.
The Python environment was managed using <code>conda</code>, and directions for setting up the environment can be found in the <code>README.md</code> file.
Most analyses were run on the HTC CPU cluster at the University of Pittsburgh, except the neural networks which were trained and evaluated on the PMACS LPC GPU cluster at the University of Pennsylvania; scripts for training classifiers both locally for a single gene and on a Slurm cluster to reproduce the analysis of many genes in parallel are provided in the linked GitHub repo.
This manuscript was written using Manubot <span class="citation" data-cites="YuJbg3zO">[<a href="#ref-YuJbg3zO" role="doc-biblioref">121</a>]</span> and is available on GitHub at <a href="https://github.com/greenelab/mpmp-manuscript"><code>https://github.com/greenelab/mpmp-manuscript</code></a> under the CC0-1.0 license <span class="citation" data-cites="3yukLr06">[<a href="#ref-3yukLr06" role="doc-biblioref">184</a>]</span> and at Zenodo <span class="citation" data-cites="30nThzUa">[<a href="#ref-30nThzUa" role="doc-biblioref">185</a>]</span>.</p>
<p>As a data resource, coefficients and hyperparameter choices for final models fit using all data types are available on Figshare: coefficients are available at <a href="https://doi.org/10.6084/m9.figshare.19576012"><code>https://doi.org/10.6084/m9.figshare.19576012</code></a> <span class="citation" data-cites="TIqbHsOH">[<a href="#ref-TIqbHsOH" role="doc-biblioref">186</a>]</span> and hyperparameters are at <a href="https://doi.org/10.6084/m9.figshare.19576048"><code>https://doi.org/10.6084/m9.figshare.19576048</code></a> <span class="citation" data-cites="EZnvMHZD">[<a href="#ref-EZnvMHZD" role="doc-biblioref">187</a>]</span>. File format/entries are described in the supplementary material in Additional File 1.</p>
<h4 id="acknowledgements">Acknowledgements</h4>
<p>We would like to thank Alexandra Lee, Ariel Hippen, Ben Heil, Milton Pividori, and Natalie Davidson for reviewing the software associated with this work and providing insightful feedback.
This research was supported in part by the University of Pittsburgh Center for Research Computing through the resources provided.
Figure 1 (the schematic of the background and evaluation pipeline) was created using BioRender.com.</p>
<h2 id="supplementary-material-for-widespread-redundancy-in--omics-profiles-of-cancer-mutation-states">Supplementary material for “Widespread redundancy in -omics profiles of cancer mutation states”</h2>
<p>A version of the main paper figures using the area under the receiver-operator curve (AUROC) metric rather than AUPR is available at <a href="https://doi.org/10.6084/m9.figshare.14919729"><code>https://doi.org/10.6084/m9.figshare.14919729</code></a>.</p>
<p>In a previous version of this paper, we ran our analysis only for the genes in the Vogelstein et al. 2013 gene set.
While there were some gene-to-gene differences in this set, we did not observe large differences between methylation and gene expression performances overall.
Scaling up the gene set by combining cancer gene sets from the literature as described in the methods/results sections affected the study results somewhat, as mutations in the added genes tend to be better predicted using gene expression than other data types.
During the revision, we explored the difference between the genes in this gene set and the genes in the “merged” cancer-related gene set but not in the Vogelstein genes.
GO analysis results for the Vogelstein genes are available at <a href="https://doi.org/10.6084/m9.figshare.19565890"><code>https://doi.org/10.6084/m9.figshare.19565890</code></a>, and results for the non-Vogelstein genes are available at <a href="https://doi.org/10.6084/m9.figshare.19565887"><code>https://doi.org/10.6084/m9.figshare.19565887</code></a>.
We noticed that the non-Vogelstein genes tend to be enriched for terms relating to transcription factors and transcriptional regulation.</p>
<p>As a data resource, coefficients and hyperparameter choices for final models fit using all data types are available on Figshare: coefficients are available at <a href="https://doi.org/10.6084/m9.figshare.19576012"><code>https://doi.org/10.6084/m9.figshare.19576012</code></a> and hyperparameters are at <a href="https://doi.org/10.6084/m9.figshare.19576048"><code>https://doi.org/10.6084/m9.figshare.19576048</code></a>.
Columns in the coefficients dataset correspond to target genes (gene symbols), and rows correspond either to PCA components (for 27K and 450K methylation), -omics features (for all other data types), or covariates (cancer type indicator variables or log(mutation burden)).
An ‘NA’ value in a cell indicates that feature was not used in the model for the corresponding gene (for an -omics feature this could mean it was not in the top 8000 features by MAD, for a cancer type feature this means that cancer type was not included in the training set based on our mutation filters).
A 0 value in a cell indicates that feature was included in model training, but it was not selected by the elastic net feature selection algorithm.
Columns in the hyperparameters dataset correspond to hyperparameters (alpha and l1_ratio for elastic net logistic regression) and rows correspond to target genes.
For the methylation data types, PCA results (score and loading matrices) corresponding to the coefficients data are also available at <a href="https://doi.org/10.6084/m9.figshare.19908034"><code>https://doi.org/10.6084/m9.figshare.19908034</code></a>.
These contain the top 5,000 principal components for each data type, which were used in the classifiers evaluated in the main paper.</p>
<p>Regarding the hyperparameters for the final models, recall that for the main figures in the paper, we evaluate each of our models using 2 replicates of 4-fold cross-validation.
For each of these folds (train/test splits), we further split the training set into train and validation sets to select hyperparameters, independently for each fold, and evaluate the models on the test set to get the results in the paper.
Because we are evaluating performance over multiple folds, it is not perfectly straightforward to get a single set of regression coefficients, since we have a (potentially different) set of coefficients for each cross-validation fold.
In order to synthesize these results into a single model for each gene in each data type, we selected one of the 8 sets of hyperparameters (from the 8 best models, 1 per CV fold) at random, with probability proportional to performance (AUPR) on the validation set used to select the hyperparameters, described above (so test set performance is not used here).
We then used the selected hyperparameters to train a single model on the entire dataset.</p>
<div id="fig:cancer_type_proportions" class="fignos">
<figure>
<img src="images/omics/supp_figure_8.png" style="width:80.0%" alt="Figure 15: Proportion of samples from each TCGA cancer type that are “dropped” as more data types are added to our analyses. We started with gene expression data, and for each added data type, we took the intersection of samples that were profiled for that data type and the previous data types, dropping all samples that were missing 1 or more data types. Overall, at each step, the proportions of “dropped” samples appear to be fairly evenly spread between cancer types, showing that in general we are not disproportionately losing one or several cancer types as more data modalities are added to our analyses." />
<figcaption aria-hidden="true"><span>Figure 15:</span> Proportion of samples from each TCGA cancer type that are “dropped” as more data types are added to our analyses. We started with gene expression data, and for each added data type, we took the intersection of samples that were profiled for that data type and the previous data types, dropping all samples that were missing 1 or more data types. Overall, at each step, the proportions of “dropped” samples appear to be fairly evenly spread between cancer types, showing that in general we are not disproportionately losing one or several cancer types as more data modalities are added to our analyses.</figcaption>
</figure>
</div>
<div id="fig:methylation_heatmap" class="fignos">
<figure>
<img src="images/omics/supp_figure_9.png" style="width:90.0%" alt="Figure 16: Heatmap displaying predictive performance for mutations in each of the 272 genes from the cancer-related gene set, across gene expression and the two DNA methylation arrays. Each cell quantifies performance for a target gene, using predictive features derived from a particular data type. Grey shaded dots indicate that the given data type provides significantly better predictions than the permuted baseline for the given gene; black inner dots indicate the same and additionally that the given data type provides statistically equivalent performance to the data type with the best average performance (determined by pairwise t-tests across data types with FDR correction)." />
<figcaption aria-hidden="true"><span>Figure 16:</span> Heatmap displaying predictive performance for mutations in each of the 272 genes from the cancer-related gene set, across gene expression and the two DNA methylation arrays. Each cell quantifies performance for a target gene, using predictive features derived from a particular data type. Grey shaded dots indicate that the given data type provides significantly better predictions than the permuted baseline for the given gene; black inner dots indicate the same and additionally that the given data type provides statistically equivalent performance to the data type with the best average performance (determined by pairwise <em>t</em>-tests across data types with FDR correction).</figcaption>
</figure>
</div>
<div id="fig:all_volcano_me" class="fignos">
<figure>
<img src="images/omics/supp_figure_10.png" style="width:90.0%" alt="Figure 17: Volcano-like plots showing predictive performance for each gene in the cancer-related gene set for expression and DNA methylation, on the sample set used for the “all data types” experiments. The first row shows performance relative to the permuted baseline, and the second row shows direct comparisons between data types for genes that outperformed the permuted baseline only for both data types. The x-axis shows the difference in mean AUPR compared with a baseline model trained on permuted labels, and the y-axis shows p-values for a paired t-test comparing cross-validated AUPR values within folds." />
<figcaption aria-hidden="true"><span>Figure 17:</span> Volcano-like plots showing predictive performance for each gene in the cancer-related gene set for expression and DNA methylation, on the sample set used for the “all data types” experiments. The first row shows performance relative to the permuted baseline, and the second row shows direct comparisons between data types for genes that outperformed the permuted baseline only for both data types. The <em>x</em>-axis shows the difference in mean AUPR compared with a baseline model trained on permuted labels, and the <em>y</em>-axis shows <em>p</em>-values for a paired <em>t</em>-test comparing cross-validated AUPR values within folds.</figcaption>
</figure>
</div>
<div id="fig:all_volcano_filter" class="fignos">
<figure>
<img src="images/omics/supp_figure_11.png" style="width:90.0%" alt="Figure 18: Volcano-like plots showing predictive performance for each gene in the cancer-related gene set for all data types, relative to the permuted baseline model, when genes are filtered based on the entire dataset rather than by cancer type. For this filtering approach, we included/excluded entire genes rather than individual cancer types: specifically, we trained a classifier for each gene where all cancer types combined had at least 5% mutated samples and at least 100 total mutated samples, resulting in 182 total classifiers. The x-axis shows the difference in mean AUPR compared with a baseline model trained on permuted labels, and the y-axis shows p-values for a paired t-test comparing cross-validated AUPR values within folds. Counts of genes making the significance threshold of 0.001: gene expression 81/182 (44.5%), 27K methylation 16/182 (8.8%), 450K methylation 1/182 (0.6%), RPPA 41/182 (22.5%), microRNA 25/182 (13.7%), mutational signatures 7/182 (3.9%)." />
<figcaption aria-hidden="true"><span>Figure 18:</span> Volcano-like plots showing predictive performance for each gene in the cancer-related gene set for all data types, relative to the permuted baseline model, when genes are filtered based on the entire dataset rather than by cancer type.
For this filtering approach, we included/excluded entire genes rather than individual cancer types: specifically, we trained a classifier for each gene where all cancer types combined had at least 5% mutated samples and at least 100 total mutated samples, resulting in 182 total classifiers.
The <em>x</em>-axis shows the difference in mean AUPR compared with a baseline model trained on permuted labels, and the <em>y</em>-axis shows <em>p</em>-values for a paired <em>t</em>-test comparing cross-validated AUPR values within folds.
Counts of genes making the significance threshold of 0.001: gene expression 81/182 (44.5%), 27K methylation 16/182 (8.8%), 450K methylation 1/182 (0.6%), RPPA 41/182 (22.5%), microRNA 25/182 (13.7%), mutational signatures 7/182 (3.9%).</figcaption>
</figure>
</div>
<div id="fig:me_compress_boxes" class="fignos">
<figure>
<img src="images/omics/supp_figure_12.png" style="width:90.0%" alt="Figure 19: Predictive performance for genes in the cancer-related gene set, using each of the three data types as predictors. The x-axis shows the number of PCA components used as features, “raw” = no PCA compression." />
<figcaption aria-hidden="true"><span>Figure 19:</span> Predictive performance for genes in the cancer-related gene set, using each of the three data types as predictors.
The <em>x</em>-axis shows the number of PCA components used as features, “raw” = no PCA compression.</figcaption>
</figure>
</div>
<div id="fig:multi_omics_raw_feats" class="fignos">
<figure>
<img src="images/omics/supp_figure_13.png" style="width:90.0%" alt="Figure 20: Top plot: comparing the best-performing model (i.e. highest mean AUPR relative to permuted baseline) trained on a single data type against the best “multi-omics” model for each target gene, using raw (not PCA compressed) features. For feature parity between data types, the top 20,000 features by mean absolute deviation were used for each data type. The difference between single-omics and multi-omics performance for TP53 was statistically significant (p=0.0078), but other differences between single-omics and multi-omics models were not statistically significant using paired-sample Wilcoxon tests across cross-validation folds, for a threshold of 0.05. Bottom plots: classifier performance, relative to baseline with permuted labels, for individual genes. Each panel shows performance for one of the six target genes; box plots show performance distribution over 8 evaluation sets (4 cross-validation folds x 2 replicates)." />
<figcaption aria-hidden="true"><span>Figure 20:</span> Top plot: comparing the best-performing model (i.e. highest mean AUPR relative to permuted baseline) trained on a single data type against the best “multi-omics” model for each target gene, using raw (not PCA compressed) features. For feature parity between data types, the top 20,000 features by mean absolute deviation were used for each data type.
The difference between single-omics and multi-omics performance for <em>TP53</em> was statistically significant (<em>p</em>=0.0078), but other differences between single-omics and multi-omics models were not statistically significant using paired-sample Wilcoxon tests across cross-validation folds, for a threshold of 0.05.
Bottom plots: classifier performance, relative to baseline with permuted labels, for individual genes. Each panel shows performance for one of the six target genes; box plots show performance distribution over 8 evaluation sets (4 cross-validation folds x 2 replicates).</figcaption>
</figure>
</div>
<div id="fig:multi_omics_mlp" class="fignos">
<figure>
<img src="images/omics/supp_figure_14.png" style="width:90.0%" alt="Figure 21: Top plot: comparing the best-performing model (i.e. highest mean AUPR relative to permuted baseline) trained on a single data type against the best “multi-omics” model for each target gene, using a 3-layer fully-connected neural network. The top 5,000 principal components were used as predictive features for each data type. The difference between single-omics and multi-omics performance for PIK3CA (p = 0.0156, in favor of multi-omics) and TP53 (p = 0.0391, in favor of single-omics) were statistically significant, but other differences between single-omics and multi-omics models were not statistically significant using paired-sample Wilcoxon tests across cross-validation folds, for a threshold of 0.05. Bottom plots: comparison of classifier performance between elastic net and fully-connected NN, relative to baseline with permuted labels, for individual genes. Each panel shows performance for one of the six target genes; box plots show performance distribution over 8 evaluation sets (4 cross-validation folds x 2 replicates)." />
<figcaption aria-hidden="true"><span>Figure 21:</span> Top plot: comparing the best-performing model (i.e. highest mean AUPR relative to permuted baseline) trained on a single data type against the best “multi-omics” model for each target gene, using a 3-layer fully-connected neural network. The top 5,000 principal components were used as predictive features for each data type.
The difference between single-omics and multi-omics performance for <em>PIK3CA</em> (<em>p</em> = 0.0156, in favor of multi-omics) and <em>TP53</em> (<em>p</em> = 0.0391, in favor of single-omics) were statistically significant, but other differences between single-omics and multi-omics models were not statistically significant using paired-sample Wilcoxon tests across cross-validation folds, for a threshold of 0.05.
Bottom plots: comparison of classifier performance between elastic net and fully-connected NN, relative to baseline with permuted labels, for individual genes. Each panel shows performance for one of the six target genes; box plots show performance distribution over 8 evaluation sets (4 cross-validation folds x 2 replicates).</figcaption>
</figure>
</div>
<h2 id="chapter-4-smaller-models-do-not-exhibit-superior-generalization-performance">Chapter 4: smaller models do not exhibit superior generalization performance</h2>
<p>This chapter has been posted as a preprint on bioRxiv (TODO) under the title “Smaller models do not exhibit superior generalization performance”.</p>
<p><strong>Contributions:</strong>
I designed and ran the experiments, created the figures, wrote the initial draft of the manuscript, and edited the manuscript. Casey S. Greene gave feedback and guidance on experiments, and edited the manuscript.</p>
<h3 id="abstract-3">Abstract</h3>
<p>Existing guidelines in statistical modeling for genomics hold that simpler models have advantages over more complex ones.
Potential advantages include cost, interpretability, and improved generalization across datasets or biological contexts.
In cancer transcriptomics, this manifests as a preference for small “gene signatures”, or groups of genes whose expression is used to define cancer subtypes or suggest therapeutic interventions.
To test the assumption that small gene signatures generalize better, we examined the generalization of mutation status prediction models across datasets (from cell lines to human tumors and vice-versa) and contexts (holding out entire cancer types from pan-cancer data).
We compared two simple procedures for model selection, one that exclusively relies on cross-validation performance and one that combines cross-validation performance with regularization strength.
We did not observe that more regularized signatures generalized better.
This result held across multiple problems and both linear models (LASSO logistic regression) and non-linear ones (neural networks).
When the goal of an analysis is to produce generalizable predictive models, we recommend choosing the ones that perform best on held-out data or in cross-validation, instead of those that are smaller or more regularized.</p>
<h3 id="introduction-2">Introduction</h3>
<p>Gene expression datasets are typically “wide”, with many gene features and relatively few samples.
These feature-rich datasets present obstacles in many aspects of machine learning, including overfitting and multicollinearity, and challenges in interpretation.
To facilitate the use of feature-rich gene expression data in machine learning models, feature selection and/or dimension reduction are commonly used to distill a more condensed data representation from the input space of all genes <span class="citation" data-cites="pt72xvSB 1Gjm0AvRK">[<a href="#ref-pt72xvSB" role="doc-biblioref">188</a>,<a href="#ref-1Gjm0AvRK" role="doc-biblioref">189</a>]</span>.
The intuition is that many gene expression features are likely irrelevant to the prediction problem, redundant, or contain no meaningful variation across samples, so transforming them or selecting a subset can generate a more reliable predictor.</p>
<p>In cancer transcriptomics, this preference for small, parsimonious sets of genes can be seen in the popularity of “gene signatures”.
These are groups of genes whose expression levels are used to define cancer subtypes or to predict prognosis or therapeutic response <span class="citation" data-cites="1z8H2EME 16Pv16qgL">[<a href="#ref-1z8H2EME" role="doc-biblioref">131</a>,<a href="#ref-16Pv16qgL" role="doc-biblioref">190</a>]</span>.
Many studies specify the size of the signature in the paper’s title or abstract, suggesting that the fewer genes in a gene signature, the better, e.g. <span class="citation" data-cites="1RajAavD dM8796ho OEMQX6hJ">[<a href="#ref-1RajAavD" role="doc-biblioref">191</a>,<a href="#ref-dM8796ho" role="doc-biblioref">192</a>,<a href="#ref-OEMQX6hJ" role="doc-biblioref">193</a>]</span>.
Clinically, there are many reasons why a smaller gene signature may be preferable, including cost (fewer genes may be less expensive to profile or validate, whereas a large signature likely requires a targeted array or NGS analysis <span class="citation" data-cites="fdtdInsC">[<a href="#ref-fdtdInsC" role="doc-biblioref">194</a>]</span>) and interpretability (it is easier to reason about the function and biological role of a smaller gene set than a large one since even disjoint gene signatures tend to converge on common biological pathways <span class="citation" data-cites="1HDl7WHTm Yto9fv8i">[<a href="#ref-1HDl7WHTm" role="doc-biblioref">195</a>,<a href="#ref-Yto9fv8i" role="doc-biblioref">196</a>]</span>).
There is also an underlying assumption that smaller gene signatures tend to be more robust: that for a new patient or in a new biological context, a smaller gene set or more parsimonious model will be more likely to maintain its predictive performance than a larger one.
This assumption has rarely been explicitly tested in genomics applications, but it is often included in guidelines or rules of thumb for statistical modeling or machine learning in biology, e.g. <span class="citation" data-cites="kAXq3qou wnVDbuuP HmaoBsQm">[<a href="#ref-kAXq3qou" role="doc-biblioref">197</a>,<a href="#ref-wnVDbuuP" role="doc-biblioref">198</a>,<a href="#ref-HmaoBsQm" role="doc-biblioref">199</a>]</span>, and it is related in spirit to information-theoretic model selection approaches such as the Akaike information criterion (AIC) and the Bayesian information criterion (BIC) <span class="citation" data-cites="13HbWPwpj PUJO2E4L">[<a href="#ref-13HbWPwpj" role="doc-biblioref">200</a>,<a href="#ref-PUJO2E4L" role="doc-biblioref">201</a>]</span>.</p>
<p>In this study, we sought to test the robustness assumption directly by evaluating model generalization across biological contexts, inspired by previous work on domain adaptation and transfer learning in cancer transcriptomics <span class="citation" data-cites="ZJNDbtzm QWTf9qrW L3dpDrOG">[<a href="#ref-ZJNDbtzm" role="doc-biblioref">202</a>,<a href="#ref-QWTf9qrW" role="doc-biblioref">203</a>,<a href="#ref-L3dpDrOG" role="doc-biblioref">204</a>]</span>.
We used two large, heterogeneous public cancer datasets: The Cancer Genome Atlas (TCGA) for human tumor sample data <span class="citation" data-cites="AK17eOgD">[<a href="#ref-AK17eOgD" role="doc-biblioref">120</a>]</span>, and the Cancer Cell Line Encyclopedia (CCLE) for human cell line data <span class="citation" data-cites="kcdvZmNN">[<a href="#ref-kcdvZmNN" role="doc-biblioref">9</a>]</span>.
These datasets contain overlapping -omics data types derived from distinct data sources, allowing us to quantify model generalization across data sources.
In addition, each dataset contains samples from a wide range of different cancer types/tissues of origin, allowing us to quantify model generalization across cancer types.
We trained both linear and non-linear models to predict mutation status (presence or absence) from RNA-seq gene expression for approximately 70 cancer driver genes, across varying levels of model simplicity and degrees of regularization, resulting in a variety of gene signature sizes.
We compared two simple procedures for model selection, one that combines cross-validation performance with model parsimony and one that only relies on cross-validation performance, for each classifier in each context.</p>
<p>Our results suggest that, in general, mutation status classification models that perform well in cross-validation within a biological context also generalize well across biological contexts.
There are some individual genes and some individual cancer types where more regularized well-performing models outperform the best-performing model.
However, we do not observe a systematic generalization advantage for smaller/more regularized models across all genes and cancer types.
These results provide evidence that good cross-validation performance within a biological context (data source or cancer type) is a sufficient proxy for robust performance across contexts.</p>
<h3 class="page_break_before" id="methods-2">Methods</h3>
<h4 id="mutation-data-download-and-preprocessing-1">Mutation data download and preprocessing</h4>
<p>To generate binary mutated/non-mutated gene labels for our machine learning model, we used mutation calls for TCGA samples from MC3 <span class="citation" data-cites="cDnF3a5Q">[<a href="#ref-cDnF3a5Q" role="doc-biblioref">107</a>]</span> and copy number threshold calls from GISTIC2.0 <span class="citation" data-cites="rDV1hhfF">[<a href="#ref-rDV1hhfF" role="doc-biblioref">108</a>]</span>.
MC3 mutation calls were downloaded from the Genomic Data Commons (GDC) of the National Cancer Institute, at <a href="https://gdc.cancer.gov/about-data/publications/pancanatlas" class="uri">https://gdc.cancer.gov/about-data/publications/pancanatlas</a>.
Thresholded copy number calls are from an older version of the GDC data and are available here: <a href="https://figshare.com/articles/dataset/TCGA_PanCanAtlas_Copy_Number_Data/6144122" class="uri">https://figshare.com/articles/dataset/TCGA_PanCanAtlas_Copy_Number_Data/6144122</a>.
We removed hypermutated samples, defined as two or more standard deviations above the mean non-silent somatic mutation count, from our dataset to reduce the number of false positives (i.e., non-driver mutations).
Any sample with either a non-silent somatic variant or a copy number variation (copy number gain in the target gene for oncogenes and copy number loss in the target gene for tumor suppressor genes) was included in the positive set; all remaining samples were considered negative for mutation in the target gene.</p>
<p>We followed a similar procedure to generate binary labels for cell lines from CCLE, using the data available on the DepMap download portal at <a href="https://depmap.org/portal/download/all/" class="uri">https://depmap.org/portal/download/all/</a>.
Mutation information was retrieved from the <code>OmicsSomaticMutations.csv</code> data file, and copy number inforamtion was retrieved from the <code>OmicsCNGene.csv</code> data file.
We thresholded the CNV log-ratios provided by CCLE into binary gain/loss calls using a lower threshold of log<sub>2</sub>(3/2) (i.e. cell lines with a log-ratio below this threshold were considered to have a full copy loss in the corresponding gene), and an upper threshold of log<sub>2</sub>(5/2) (i.e. cell lines with a log-ratio above this threshold were considered to have a full copy gain in the corresponding gene).
After applying the same hypermutation criteria that we used for TCGA, no cell lines in CCLE were identified as hypermutated.
After preprocesing, 1402 cell lines with mutation and copy number data remained.
We then combined non-silent point mutations and copy number gain/loss information into binary labels using the same criteria as for TCGA.</p>
<h4 id="gene-expression-data-download-and-preprocessing">Gene expression data download and preprocessing</h4>
<p>RNA sequencing data for TCGA was downloaded from GDC at the same link provided above for the Pan-Cancer Atlas.
We discarded non-protein-coding genes and genes that failed to map, and removed tumors that were measured from multiple sites.
After filtering to remove hypermutated samples and taking the intersection of samples with both mutation and gene expression data, 9074 TCGA samples remained.</p>
<p>RNA sequencing data for CCLE was downloaded from the DepMap download portal, linked above, in the <code>CCLE_expression.csv</code> data file.
After taking the intersection of CCLE cell lines with both mutation and gene expression data, 1402 cell lines remained.
For experiments making predictions across datasets (i.e., training models on TCGA and evaluating performance on CCLE, or vice-versa) we took the intersection of genes in both datasets, resulting in 16041 gene features.
For experiments where only TCGA data was used (i.e., evaluating models on held-out cancer types), we used all 16148 gene features present in TCGA after the filtering described above.</p>
<h4 id="cancer-gene-set-construction-1">Cancer gene set construction</h4>
<p>In order to study mutation status classification for a diverse set of cancer driver genes, we started with the set of 125 frequently altered genes from Vogelstein et al. <span class="citation" data-cites="MnWpJEWE">[<a href="#ref-MnWpJEWE" role="doc-biblioref">124</a>]</span> (all genes from Table S2A).
For each target gene, to ensure that the training dataset was reasonably balanced (i.e., that there would be enough mutated samples to train an effective classifier), we included only cancer types with at least 15 mutated samples and at least 5% mutated samples, which we refer to here as “valid” cancer types.
In some cases, this resulted in genes with no valid cancer types, which we dropped from the analysis.
Out of the 125 genes originally listed in the Vogelstein et al. cancer gene set, we retained 71 target genes for the TCGA to CCLE analysis, and 70 genes for the CCLE to TCGA analyses.
For these analyses, each gene needed at least one valid cancer type in TCGA and one valid cancer type in CCLE, to construct the train and test sets.
For the cancer type holdout analysis, we retained 56 target genes: in this case, each gene needed at least two valid cancer types in TCGA to be retained, one to train on and one to hold out.</p>
<h4 id="classifier-setup-and-cross-validation-design">Classifier setup and cross-validation design</h4>
<p>We trained logistic regression classifiers to predict whether or not a given sample had a mutational event in a given target gene using gene expression features as explanatory variables.
Our model was trained on gene expression data (X) to predict somatic mutation presence or absence (y) in a target gene.
To control for varying mutation burden per sample and to adjust for potential cancer type-specific expression patterns, we included one-hot encoded cancer type and log<sub>10</sub>(sample mutation count) in the model as covariates.
Since gene expression datasets tend to have many dimensions and comparatively few samples, we used a LASSO penalty to perform feature selection <span class="citation" data-cites="kX2zf6UE">[<a href="#ref-kX2zf6UE" role="doc-biblioref">110</a>]</span>.
LASSO logistic regression has the advantage of generating sparse models (some or most coefficients are 0), as well as having a single tunable hyperparameter which can be easily interpreted as an indicator of regularization strength/model simplicity.</p>
<p>LASSO (<span class="math inline">\(\l_1\)</span>-penalized) logistic regression finds the feature weights <span class="math inline">\(\hat{w} \in \mathbb{R}^{p}\)</span> solving the following optimization problem:</p>
<p><span class="math display">\[\hat{w} = \text{argmin}_{w} \ (C \cdot l(X, y; w)) + ||w||_1\]</span></p>
<p>where <span class="math inline">\(i \in \{1, \dots, n\}\)</span> denotes a sample in the dataset, <span class="math inline">\(X_i \in \mathbb{R}^{p}\)</span> denotes features (gene expression measurements) from the given sample, <span class="math inline">\(y_i \in \{0, 1\}\)</span> denotes the label (mutation presence/absence) for the given sample, and <span class="math inline">\(l(\cdot)\)</span> denotes the negative log-likelihood of the observed data given a particular choice of feature weights, i.e.</p>
<p><span class="math display">\[l(X, y; w) = -\sum_{i=1}^{n} y_i \log\left(\frac{1}{1 + e^{-w^{\top}X_i}}\right) + (1 - y_i) \log\left(1 - \frac{1}{1 + e^{-w^{\top}X_i}}\right)\]</span></p>
<p>Given weight values <span class="math inline">\(\hat{w}\)</span>, it is straightforward to predict the probability of a positive label (mutation in the target gene) <span class="math inline">\(P(y^{*} = 1 \mid X^{*}; \hat{w})\)</span> for a test sample <span class="math inline">\(X^{*}\)</span>:</p>
<p><span class="math display">\[P(y^{*} = 1 \mid X^{*}; \hat{w}) = \frac{1}{1 + e^{-\hat{w}^{\top}X^{*}}}\]</span></p>
<p>and the probability of no mutation in the target gene, <span class="math inline">\(P(y^{*} = 0 \mid X^{*}; \hat{w})\)</span>, is given by (1 - the above quantity).</p>
<p>This optimization problem leaves one hyperparameter to select: <span class="math inline">\(C\)</span>, which controls the inverse of the strength of the L1 penalty on the weight values (i.e. regularization strength scales with <span class="math inline">\(\frac{1}{C}\)</span>).
Although the LASSO optimization problem does not have a closed form solution, the loss function is convex, and iterative optimization algorithms are commonly used for finding reasonable solutions.
For fixed values of <span class="math inline">\(C\)</span>, we solved for <span class="math inline">\(\hat{w}\)</span> using <code>scikit-learn</code>’s <code>LogisticRegression</code> method <span class="citation" data-cites="43wsMmMv">[<a href="#ref-43wsMmMv" role="doc-biblioref">103</a>]</span>, which uses the coordinate descent optimization method implemented in <code>liblinear</code> <span class="citation" data-cites="Mm8xIDqq">[<a href="#ref-Mm8xIDqq" role="doc-biblioref">104</a>]</span>.
We selected this implementation rather than the <code>SGDClassifier</code> stochastic gradient descent implementation because coordinate descent/<code>liblinear</code> tends to generate sparser models and does not depend on a learning rate parameter, although after hyperparameter tuning performance is generally comparable between the implementations <span class="citation" data-cites="gOHkic9g">[<a href="#ref-gOHkic9g" role="doc-biblioref">205</a>]</span>.</p>
<p>To assess model selection across contexts (datasets and cancer types), we trained models using a variety of LASSO parameters on 75% of the training dataset, holding out 25% of the training dataset as the “cross-validation” set and also evaluating across contexts as the “test” set.
We trained models using <span class="math inline">\(C\)</span> values evenly spaced on a logarithmic scale between (10<sup>-3</sup>, 10<sup>7</sup>); i.e. the output of <code>numpy.logspace(-3, 7, 21)</code>.
This range was intended to give evenly distributed coverage across genes and cancer types that included “underfit” models (predicting only the mean or using very few features, poor performance on all datasets), “overfit” models (performing perfectly on training data but comparatively poorly on cross-validation and test data), and a wide variety of models in between that typically included the best fits to the cross-validation and test data.
To assess variability between train/CV splits, we used all 4 splits (25% holdout sets) x 2 random seeds for a total of 8 different training sets for each gene, using the same test set (i.e. all of the held-out context, either one cancer type or one dataset) in each case.</p>
<h4 id="best-model-vs.-smallest-good-model-analysis">“Best model” vs. “smallest good model” analysis</h4>
<p>For the “best” vs. “smallest good” model selection comparison, we started with 8 performance measurements (4 cross-validation folds x 2 random seeds) for each of 21 LASSO parameters.
We took the mean over these 8 measurements to get a single performance measurement for each model (LASSO parameter) on the holdout dataset, which has the same composition as the training set.
We used these per-parameter mean performance measurements to select the “best” model (LASSO parameter with the best performance on the holdout dataset), and the “smallest good” model (smallest LASSO parameter with performance in the top 25% of mean values on the holdout dataset, rounded up to the nearest integer).
For the distributions of differences shown in the Results, we took the difference in mean performance for the “best” and “smallest good” models for each gene, with positive differences indicating better performance for the “best” model and negative differences better performance for the “smallest good” model, for each gene.</p>
<h4 id="neural-network-setup-and-parameter-selection">Neural network setup and parameter selection</h4>
<p>As a tradeoff between computational cost and ability to represent non-linear decision boundaries, inspired by the architecture of the intermediate-complexity model described in <span class="citation" data-cites="b7RUXQvu">[<a href="#ref-b7RUXQvu" role="doc-biblioref">206</a>]</span>, we trained a three-layer fully connected neural network with ReLU nonlinearities <span class="citation" data-cites="Xk9rmxAA">[<a href="#ref-Xk9rmxAA" role="doc-biblioref">207</a>]</span> to predict mutation status.
For the experiments described in the main paper, we varied the size of the first hidden layer in the range {1, 2, 3, 4, 5, 10, 50, 100, 500, 1000}.
We fixed the size of the second hidden layer to be half of the size of the first hidden layer, rounded up to the nearest integer, and the size of the third hidden layer was the number of classes, 2 in our case.
Our models were trained for 100 epochs of mini-batch stochastic gradient descent in PyTorch <span class="citation" data-cites="iTP4h1rX">[<a href="#ref-iTP4h1rX" role="doc-biblioref">208</a>]</span>, using the Adam optimizer <span class="citation" data-cites="c6d3lKFX">[<a href="#ref-c6d3lKFX" role="doc-biblioref">159</a>]</span> and a fixed batch size of 50.
To select the remaining hyperparameters for each hidden layer size, we performed a random search over 10 combinations, with a single train/test split stratified by cancer type, using the following hyperparameter ranges: learning rate {0.1, 0.01, 0.001, 5e-4, 1e-4}, dropout proportion {0.1, 0.5, 0.75}, weight decay (L2 penalty) {0, 0.1, 1, 10, 100}.
We used the same train/cross-validation split strategy described above, generating 8 different performance measurements for each gene and hidden layer size, for the neural network experiments as well.</p>
<p>For the <em>EGFR</em> gene, we also ran experiments where we varied the dropout proportion and the weight decay hyperparameter as the regularization axis, and selected the remaining hyperparameters (including the hidden layer size) using a random search.
In these cases, we used a fixed range for dropout of {0.0, 0.05, 0.125, 0.25, 0.375, 0.5, 0.625, 0.75, 0.875, 0.95}, and a fixed range for weight decay of {0.0, 0.001, 0.005, 0.01, 0.05, 0.1, 0.2, 0.3, 0.4, 0.5, 0.75, 1.0, 10.0}.
All neural network analyses were performed on a Ubuntu 18.04 machine with a NVIDIA RTX 2060 GPU.</p>
<h3 class="page_break_before" id="results-4">Results</h3>
<h4 id="evaluating-model-generalization-using-public-cancer-data">Evaluating model generalization using public cancer data</h4>
<p>We collected data from the TCGA Pan-Cancer Atlas and the Cancer Cell Line Encyclopedia to predict the presence or absence of mutations in cancer genes, as a benchmark of cancer-related information content across cancer types and contexts.
We trained mutation status classifiers across approximately 70 genes involved in cancer development and progression from Vogelstein et al. 2013 <span class="citation" data-cites="809OyWlC">[<a href="#ref-809OyWlC" role="doc-biblioref">106</a>]</span>, using LASSO logistic regression with gene expression (RNA-seq) values as predictive features.
We fit each classifier across a variety of regularization parameters, resulting in models with a variety of different sparsity levels between the extremes of 0 nonzero features and all features included (Supplementary Figure <a href="#fig:average_sparsity">27</a>).
Inspired by the generalization experiments across tissues and model systems in <span class="citation" data-cites="ZJNDbtzm">[<a href="#ref-ZJNDbtzm" role="doc-biblioref">202</a>]</span>, we designed experiments to evaluate the generalization of mutation status classifiers across datasets (TCGA to CCLE and CCLE to TCGA) and across biological contexts (cancer types) within TCGA, relative to a within-dataset baseline (Figure <a href="#fig:generalization_overview">22</a>).</p>
<div id="fig:generalization_overview" class="fignos">
<figure>
<img src="images/generalization/figure_1.png" style="width:90.0%" alt="Figure 22: Schematic of experimental design. The colors of the “dots” in the training/model selection/model evaluation panels on the left correspond to train/CV/test curves in the following results figures." />
<figcaption aria-hidden="true"><span>Figure 22:</span> Schematic of experimental design. The colors of the “dots” in the training/model selection/model evaluation panels on the left correspond to train/CV/test curves in the following results figures.</figcaption>
</figure>
</div>
<h4 id="generalization-from-human-tumor-samples-to-cell-lines-is-more-effective-than-the-reverse">Generalization from human tumor samples to cell lines is more effective than the reverse</h4>
<p>To evaluate “cross-dataset” generalization, we trained mutation status classifiers on human tumor data from TCGA and evaluated them on cell line data from CCLE, as well as the reverse from CCLE to TCGA.
As an example, we examined <em>EGFR</em>, an oncogenic tyrosine kinase that is commonly mutated in diverse cancer types and cancer cell lines, including lung cancer, colorectal cancer, and glioblastoma <span class="citation" data-cites="qtYNZ2Q2 rzy1RWeA">[<a href="#ref-qtYNZ2Q2" role="doc-biblioref">209</a>,<a href="#ref-rzy1RWeA" role="doc-biblioref">210</a>]</span>.
For EGFR mutation status classifiers trained on TCGA and evaluated on CCLE, we saw that AUPR on cell lines was slightly worse than on held-out tumor samples, but comparable across regularization levels/LASSO parameters (Figure <a href="#fig:tcga_ccle_overall">23</a>A).
On the other hand, EGFR classifiers trained on CCLE and evaluated on TCGA performed considerably worse on human tumor samples as compared to held-out cell lines (Figure <a href="#fig:tcga_ccle_overall">23</a>B).</p>
<p>To explore these tendencies more generally, we compared performance across all genes in the Vogelstein et al. dataset, for both TCGA to CCLE and CCLE to TCGA generalization.
We measured the difference between performance on the holdout data within the training dataset and performance across datasets, with a positive difference indicating poor generalization (better holdout performance than test performance) and a 0 or negative difference indicating good generalization (comparable test performance to holdout performance).
For generalization from TCGA to CCLE, we observed that median AUPR differences were mostly centered around 0 for most genes, with some exceptions at the extremes (Figure <a href="#fig:tcga_ccle_overall">23</a>C; performance differences on the y-axis).
An example of a gene exhibiting poor generalization was <em>IDH1</em>, the leftmost gene in Figure <a href="#fig:tcga_ccle_overall">23</a>C, with good performance on held-out TCGA data and poor performance on CCLE data.
IDH-mutant glioma cell lines are poorly represented compared to IDH-mutant patient tumors, which may explain the difficulty of generalization to cell lines for <em>IDH1</em> mutation classifiers <span class="citation" data-cites="HQx0QuE4">[<a href="#ref-HQx0QuE4" role="doc-biblioref">211</a>]</span>.
For generalization from CCLE to TCGA, we observed a more pronounced upward shift toward better performance on CCLE and worse on TCGA, with most genes performing better on the CCLE holdout data and very few genes generalizing comparably to the TCGA samples (Figure <a href="#fig:tcga_ccle_overall">23</a>D).</p>
<div id="fig:tcga_ccle_overall" class="fignos">
<figure>
<img src="images/generalization/figure_2.png" style="width:80.0%" alt="Figure 23: A. EGFR mutation status prediction performance on training samples from TCGA (blue), held-out TCGA samples (orange), and CCLE samples (green), across varying LASSO parameters. B. EGFR mutation status prediction performance on training samples from CCLE (blue), held-out CCLE samples (orange), and TCGA samples (green). C. Difference in mutation status prediction performance for models trained on TCGA (holdout data) and evaluated on CCLE (test data), across 71 genes from Vogelstein et al. For each gene, the best model (LASSO parameter) was selected using holdout AUPR performance. Genes on x-axis are ordered by median AUPR difference across cross-validation splits, from highest to lowest. D. Difference in mutation status prediction performance for models trained on CCLE (holdout data) and evaluated on TCGA (test data), across 70 genes from Vogelstein et al." />
<figcaption aria-hidden="true"><span>Figure 23:</span> <strong>A.</strong> <em>EGFR</em> mutation status prediction performance on training samples from TCGA (blue), held-out TCGA samples (orange), and CCLE samples (green), across varying LASSO parameters.
<strong>B.</strong> <em>EGFR</em> mutation status prediction performance on training samples from CCLE (blue), held-out CCLE samples (orange), and TCGA samples (green).
<strong>C.</strong> Difference in mutation status prediction performance for models trained on TCGA (holdout data) and evaluated on CCLE (test data), across 71 genes from Vogelstein et al. For each gene, the best model (LASSO parameter) was selected using holdout AUPR performance. Genes on x-axis are ordered by median AUPR difference across cross-validation splits, from highest to lowest.
<strong>D.</strong> Difference in mutation status prediction performance for models trained on CCLE (holdout data) and evaluated on TCGA (test data), across 70 genes from Vogelstein et al.</figcaption>
</figure>
</div>
<h4 id="best-and-smallest-good-model-selection-strategies-perform-comparably">“Best” and “smallest good” model selection strategies perform comparably</h4>
<p>To address the question of whether more parsimonious models tend to generalize better or not, we designed two model selection schemes and compared them for the TCGA to CCLE and CCLE to TCGA mutation prediction problems (Figure <a href="#fig:tcga_ccle_smallest_best">24</a>A).
The “best” model selection scheme chooses the top-performing model/LASSO parameter on the holdout dataset from the same source as the training data and applies it to the test data from the other data source.
The intention of the “smallest good” model selection scheme is to balance parsimony with reasonable performance on the holdout data, since simply selecting the smallest possible model (generally, the dummy regressor/mean predictor) is not likely to generalize well.
To accomplish this, we first identify the top 25% of well-performing models on the holdout dataset; then, from this subset of models, we choose the smallest (i.e., highest LASSO parameter) to apply to the test data.
In both cases, we exclusively use the holdout data to select a model and only apply the model to out-of-dataset samples to evaluate generalization performance <em>after</em> model selection.</p>
<p>For TCGA to CCLE generalization, 31/71 genes (43.7%) had better performance for the “best” model, and 15/71 genes (21.1%) had better generalization performance with the “smallest good” model.
The other 25 genes had the same “best” and “smallest good” model (in other words, the “smallest good” model was also the best-performing overall, and the difference was 0) (Figure <a href="#fig:tcga_ccle_smallest_best">24</a>B).
For CCLE to TCGA generalization, 24/70 genes (34.3%) had better performance for the “best” model and 19/70 (27.1%) for the “smallest good,” with the other 27 having the same model fulfill both criteria (Figure <a href="#fig:tcga_ccle_smallest_best">24</a>C).
Overall, these results do not support the hypothesis that the most parsimonious model generalizes the best: for both generalization problems there are slightly more genes where the best-performing model on the holdout dataset is also the best-performing on the test set, although there are some genes where the “smallest good” approach works well.</p>
<p>We examined genes that fell into either category for TCGA to CCLE generalization (dotted lines on Figure <a href="#fig:tcga_ccle_smallest_best">24</a>B).
For <em>NF1</em>, the “best” model outperforms the “smallest good” model (Figure <a href="#fig:tcga_ccle_smallest_best">24</a>D).
Comparing holdout (orange) and cross-dataset (green) performance, both generally follow a similar trend, with the cross-dataset performance peaking when the holdout performance peaks at a regularization parameter of <span class="math inline">\(\alpha = 0.00316\)</span>.
<em>PIK3CA</em> is an example of the opposite, a gene where the “smallest good” model tends to outperform the “best” model (Figure <a href="#fig:tcga_ccle_smallest_best">24</a>E).
In this case, the peak for the cross-dataset performance occurs at a higher level of regularization (further left on the x-axis), at <span class="math inline">\(\alpha = 0.01\)</span>, than the peak for the holdout performance, at <span class="math inline">\(\alpha = 0.0316\)</span>.
This suggests that a <em>PIK3CA</em> mutation status classifier that is more parsimonious, but that has slightly worse performance, does tend to generalize better across datasets to CCLE.</p>
<div id="fig:tcga_ccle_smallest_best" class="fignos">
<figure>
<img src="images/generalization/figure_3.png" style="width:90.0%" alt="Figure 24: A. Schematic of “best” vs. “smallest good” model comparison experiments. B. Distribution of performance comparisons between “best” and “smallest good” model selection strategies, for TCGA -&gt; CCLE generalization. Positive x-axis values indicate better performance for the “best” model, negative values indicate better performance for the “smallest good” model. C. Distribution of performance comparisons between “best” and “smallest good” model selection strategies, for CCLE -&gt; TCGA generalization. D. NF1 mutation status prediction performance generalizing from TCGA (holdout, orange), to CCLE (green), with “best” and “smallest good” models labeled. E. PIK3CA mutation status prediction performance generalizing from TCGA (holdout, orange), to CCLE (green), with “best” and “smallest good” models labeled." />
<figcaption aria-hidden="true"><span>Figure 24:</span> <strong>A.</strong> Schematic of “best” vs. “smallest good” model comparison experiments.
<strong>B.</strong> Distribution of performance comparisons between “best” and “smallest good” model selection strategies, for TCGA -&gt; CCLE generalization. Positive x-axis values indicate better performance for the “best” model, negative values indicate better performance for the “smallest good” model.
<strong>C.</strong> Distribution of performance comparisons between “best” and “smallest good” model selection strategies, for CCLE -&gt; TCGA generalization.
<strong>D.</strong> <em>NF1</em> mutation status prediction performance generalizing from TCGA (holdout, orange), to CCLE (green), with “best” and “smallest good” models labeled.
<strong>E.</strong> <em>PIK3CA</em> mutation status prediction performance generalizing from TCGA (holdout, orange), to CCLE (green), with “best” and “smallest good” models labeled.</figcaption>
</figure>
</div>
<h4 id="generalization-across-cancer-types-yields-similar-results-to-generalization-across-datasets">Generalization across cancer types yields similar results to generalization across datasets</h4>
<p>To evaluate generalization across biological contexts within a dataset, we trained mutation prediction classifiers on all but one cancer type in TCGA, performed model selection on a holdout set stratified by cancer type, and held out the remaining cancer type as a test set.
We performed the same “best” vs. “smallest good” analysis that was previously described, across 294 gene/holdout cancer type combinations (Figure <a href="#fig:cancer_type_holdout">25</a>A).
We observed 133/294 gene/cancer type combinations (45.2%) that had better generalization performance with the “best” model, compared to 84/294 (28.6%) for the “smallest good” model.
The other 77 gene/cancer type combinations had the same “best” and “smallest good” model and thus no difference in performance.
This is consistent with our cross-dataset experiments, with slightly more instances where the “best” model on the stratified holdout data also generalizes the best, but no pronounced distributional shift in either direction.</p>
<p>We looked in more detail at two examples of gene/cancer type combinations, one on either side of the 0 point for cross-cancer type generalization.
For prediction of <em>SETD2</em> mutation status in papillary renal cell carcinoma, we observed the best cross-cancer type performance for relatively low levels of regularization/high x-axis values (Figure <a href="#fig:cancer_type_holdout">25</a>B).
For prediction of <em>CDKN2A</em> mutation status in low grade glioma, on the other hand, we observed the best cross-cancer generalization for a high level of regularization (<span class="math inline">\(\alpha = 0.01\)</span>), and generalization capability for the best parameter on the stratified holdout set (<span class="math inline">\(\alpha = 0.0316\)</span>) was lower (Figure <a href="#fig:cancer_type_holdout">25</a>C).</p>
<p>We aggregated results across genes for each cancer type, looking at performance in the held-out cancer type compared to performance on the stratified holdout set (Figure <a href="#fig:cancer_type_holdout">25</a>D).
Cancer types that were particularly difficult to generalize to (better performance on stratified data than cancer type holdout, or positive y-axis values) include testicular cancer (TGCT) and soft tissue sarcoma (SARC), which are notable because they are not carcinomas like the majority of cancer types included in TCGA, potentially making generalization harder.
We also aggregated results across cancer types for each gene, identifying a distinct set of genes where classifiers tend to generalize poorly no matter what cancer type is held out (Supplementary Figure <a href="#fig:average_perf_by_gene">28</a>).
Included in this set of genes with poor generalization performance are <em>HRAS</em>, <em>NRAS</em>, and <em>BRAF</em>, suggesting that a classifier that combines mutations in Ras pathway genes into a single “pathway mutation status” label (as described in <span class="citation" data-cites="17XHtmqPY">[<a href="#ref-17XHtmqPY" role="doc-biblioref">50</a>]</span>, or using more general computational approaches such as <span class="citation" data-cites="odNCMbto zbMcel6V">[<a href="#ref-odNCMbto" role="doc-biblioref">55</a>,<a href="#ref-zbMcel6V" role="doc-biblioref">58</a>]</span>) could be a better approach than separate classifiers for each gene.</p>
<p>In the cancer type aggregation plot (Figure <a href="#fig:cancer_type_holdout">25</a>D), thyroid carcinoma (THCA) stood out as a carcinoma that had poor performance when held out.
In our experiments, the only genes in which THCA is included as a held-out cancer type are <em>BRAF</em> and <em>NRAS</em>; generalization performance for both genes is below cross-validation performance, but slightly worse for <em>NRAS</em> than <em>BRAF</em> (Supplementary Figure <a href="#fig:thca_by_gene">29</a>).
Previous work suggests that <em>BRAF</em> mutation tends to have a different functional signature in THCA than other cancer types, and withholding THCA from the training set improved classifier performance, which could at least in part explain the difficulty of generalizing to THCA we observe <span class="citation" data-cites="17XHtmqPY">[<a href="#ref-17XHtmqPY" role="doc-biblioref">50</a>]</span>.</p>
<div id="fig:cancer_type_holdout" class="fignos">
<figure>
<img src="images/generalization/figure_4.png" style="width:90.0%" alt="Figure 25: A. Distribution of performance comparisons between “best” and “smallest good” model selection strategies, for generalization across TCGA cancer types. Each point is a gene/cancer type combination; positive x-axis values indicate better performance for the “best” model and negative values indicate better performance for the “smallest good” model. B. SETD2 mutation status prediction performance generalizing from other cancer types in TCGA (stratified holdout, orange) to papillary renal cell carcinoma (KIRP, green), with “best” and “smallest good” models labeled. C. CDKN2A mutation status prediction performance generalizing from other cancer types in TCGA (stratified holdout, orange) to low grade glioma (LGG, green), with “best” and “smallest good” models labeled. D. Distributions of performance difference between CV data (same cancer types as train data) and holdout data (cancer types not represented in train data), by held-out cancer type. Each point is a gene whose mutation status classifier was used to make predictions on out-of-dataset samples in the relevant cancer type." />
<figcaption aria-hidden="true"><span>Figure 25:</span> <strong>A.</strong> Distribution of performance comparisons between “best” and “smallest good” model selection strategies, for generalization across TCGA cancer types. Each point is a gene/cancer type combination; positive x-axis values indicate better performance for the “best” model and negative values indicate better performance for the “smallest good” model.
<strong>B.</strong> <em>SETD2</em> mutation status prediction performance generalizing from other cancer types in TCGA (stratified holdout, orange) to papillary renal cell carcinoma (KIRP, green), with “best” and “smallest good” models labeled.
<strong>C.</strong> <em>CDKN2A</em> mutation status prediction performance generalizing from other cancer types in TCGA (stratified holdout, orange) to low grade glioma (LGG, green), with “best” and “smallest good” models labeled.
<strong>D.</strong> Distributions of performance difference between CV data (same cancer types as train data) and holdout data (cancer types not represented in train data), by held-out cancer type. Each point is a gene whose mutation status classifier was used to make predictions on out-of-dataset samples in the relevant cancer type.</figcaption>
</figure>
</div>
<h4 id="small-neural-network-hidden-layer-sizes-tend-to-generalize-poorly">Small neural network hidden layer sizes tend to generalize poorly</h4>
<p>To test whether or not findings generalize to non-linear models, we trained a 3-layer neural network to predict mutation status from gene expression for generalization from TCGA to CCLE, and we varied the size of the first hidden layer to control regularization/model complexity.
We fixed the size of the second hidden layer to be half the size of the first layer, rounded up to the nearest integer; further details in Methods.
For <em>EGFR</em> mutation status prediction, we saw that performance for small hidden layer sizes was noisy, but generally lower than for higher hidden layer sizes (Figure <a href="#fig:tcga_ccle_nn">26</a>A).
On average, over all 71 genes from Vogelstein et al., performance on both held-out TCGA data and CCLE data tends to increase until a hidden layer size of 10-50, then flatten (Figure <a href="#fig:tcga_ccle_nn">26</a>B).
To explore additional approaches to neural network regularization, we also tried varying dropout and weight decay for <em>EGFR</em> and <em>KRAS</em> mutation status classification while holding the hidden layer size constant.
Results followed a similar trend, with generalization performance generally tracking performance on holdout data (Supplementary Figure <a href="#fig:nn_dropout_wd">30</a>).</p>
<p>In order to measure which hidden layer sizes tended to perform relatively well or poorly, across different mutated cancer genes with different effect sizes, we ranked the range of hidden layer sizes by their generalization performance on CCLE (with low ranks representing good performance, and high ranks representing poor performance; Figure <a href="#fig:tcga_ccle_nn">26</a>C).
For each hidden layer size, we then visualized the distribution of ranks above and below the median rank of 5.5/10; a high proportion of ranks above the median (True, or blue bar) signifies poor overall performance for that hidden layer size, and a high proportion of ranks below the median (False, or orange bar) signifies good performance.
We saw that small hidden layer sizes tended to generalize poorly (&lt;5, but most pronounced for 1 and 2), and intermediate hidden layer sizes tended to generalize well (10-100, and sometimes 500/1000).
This suggests that some degree of parsimony/simplicity could be useful, but very simple models do not tend to generalize well.
We also performed the same “best”/“smallest good” analysis as with the linear models, using hidden layer size as the regularization axis instead of LASSO regularization strength.
We observed a distribution centered around 0, suggesting that the “best” and “smallest good” models tend to generalize similarly (Figure <a href="#fig:tcga_ccle_nn">26</a>D).
28/71 genes (45.2%) had better generalization performance with the “best” model, compared to 21/71 (28.6%) for the “smallest good” model and 22 with the same “best” and “smallest good” model.</p>
<div id="fig:tcga_ccle_nn" class="fignos">
<figure>
<img src="images/generalization/figure_5.png" style="width:90.0%" alt="Figure 26: A. EGFR mutation status prediction performance on training samples from TCGA (blue), held-out TCGA samples (orange), and CCLE samples (green), across varying neural network hidden layer sizes. B. Mutation status prediction performance summarized across all genes from Vogelstein et al. on training samples from TCGA (blue), held-out TCGA samples (orange), and CCLE samples (green), across varying neural network hidden layer sizes. C. Distribution of ranked performance values above/below the median rank for each gene, for each of the hidden layer sizes evaluated. Lower ranks indicate better performance and higher ranks indicate worse performance, relative to other hidden layer sizes. D. Distribution of performance comparisons between “best” and “smallest good” model selection strategies, for TCGA -&gt; CCLE generalization with neural network hidden layer size as the regularization axis. Positive x-axis values indicate better performance for the “best” model, negative values indicate better performance for the “smallest good” model." />
<figcaption aria-hidden="true"><span>Figure 26:</span> <strong>A.</strong> <em>EGFR</em> mutation status prediction performance on training samples from TCGA (blue), held-out TCGA samples (orange), and CCLE samples (green), across varying neural network hidden layer sizes.
<strong>B.</strong> Mutation status prediction performance summarized across all genes from Vogelstein et al. on training samples from TCGA (blue), held-out TCGA samples (orange), and CCLE samples (green), across varying neural network hidden layer sizes.
<strong>C.</strong> Distribution of ranked performance values above/below the median rank for each gene, for each of the hidden layer sizes evaluated. Lower ranks indicate better performance and higher ranks indicate worse performance, relative to other hidden layer sizes.
<strong>D.</strong> Distribution of performance comparisons between “best” and “smallest good” model selection strategies, for TCGA -&gt; CCLE generalization with neural network hidden layer size as the regularization axis. Positive x-axis values indicate better performance for the “best” model, negative values indicate better performance for the “smallest good” model.</figcaption>
</figure>
</div>
<h3 id="discussion-2">Discussion</h3>
<p>Using public cancer genomics and transcriptomics data from TCGA and CCLE, we studied generalization of mutation status classifiers for a wide variety of cancer driver genes.
We designed experiments to evaluate generalization across biological contexts by holding out cancer types in TCGA, and to evaluate generalization across datasets by training models on TCGA and evaluating them on CCLE, and vice-versa.
We found that, in general, smaller or more parsimonious models do not tend to generalize more effectively across cancer types or across datasets, and in the absence of prior knowledge about a prediction problem, simply choosing the model that performs the best on a holdout dataset is at least as effective for selecting models that generalize.</p>
<p>Our results were similar in both linear models (LASSO logistic regression) and non-linear deep neural networks when using hidden layer size as the regularization parameter of interest.
In our non-linear model experiments, we did not observe better generalization across datasets for fully connected neural networks with fewer hidden layer nodes, and our preliminary results indicated a similar trend for dropout and weight decay.
Compared to linear models, it is less clear how to define a “small” or “parsimonious” neural network model since there are many regularization techniques that one may use to control complexity.
Rather than simply removing nodes and keeping the network fully connected, another approach to parsimony could be to select an inductive bias to guide the size reduction of the network.
Existing examples include network structures guided by protein-protein interaction networks or function/pathway ontologies <span class="citation" data-cites="1HfJnooMx TIQTmEOG YOm0bqVR Mflfx7GP">[<a href="#ref-TIQTmEOG" role="doc-biblioref">91</a>,<a href="#ref-1HfJnooMx" role="doc-biblioref">168</a>,<a href="#ref-YOm0bqVR" role="doc-biblioref">212</a>,<a href="#ref-Mflfx7GP" role="doc-biblioref">213</a>]</span>.
It is possible that a smaller neural network with a structure that corresponds more appropriately to the prediction problem would achieve better generalization results, although choosing an apt network structure or data source can be a challenging aspect of such efforts.</p>
<p>For generalization from CCLE to TCGA, we observed that performance was generally worse on human tumor samples from TCGA than for held-out cell lines.
This could, at least in part, be a function of sample size: the number of cell lines in CCLE is approximately an order of magnitude smaller than the number of tumor samples in TCGA (~10,000 samples in TCGA vs. ~1,500 cell lines in CCLE, although the exact number of samples used to train and evaluate our classifiers varies by gene, see Methods for further detail).
There are also plausible biological and technical explanations for the difficulty of generalizing to human tumor samples.
This result could reflect the imperfect and limited nature of cancer cell lines as a model system for human tumors, which previous studies have pointed out <span class="citation" data-cites="1HnyQJPcl vYqd0FZY 18vM3pLD6">[<a href="#ref-1HnyQJPcl" role="doc-biblioref">214</a>,<a href="#ref-vYqd0FZY" role="doc-biblioref">215</a>,<a href="#ref-18vM3pLD6" role="doc-biblioref">216</a>]</span>.
In addition, the CCLE data is collected and processed uniformly, as described in <span class="citation" data-cites="kcdvZmNN">[<a href="#ref-kcdvZmNN" role="doc-biblioref">9</a>]</span>, while the TCGA data is processed by a uniform pipeline but collected from a wide variety of different cancer centers around the US <span class="citation" data-cites="AK17eOgD">[<a href="#ref-AK17eOgD" role="doc-biblioref">120</a>]</span>.</p>
<p>When we ranked cancer types in order of their generalization difficulty aggregated across genes, we noticed a slight tendency toward non-carcinoma cancer types (TGCT, SARC, SKCM) being difficult to generalize to.
It has been pointed out in other biological data types that holding out entire contexts or domains is necesssary for a full picture of generalization performance <span class="citation" data-cites="Inb9g7Za 13prIHiSM">[<a href="#ref-Inb9g7Za" role="doc-biblioref">217</a>,<a href="#ref-13prIHiSM" role="doc-biblioref">218</a>]</span>, which our results corroborate.
This highlights a potential weakness of using TCGA’s carcinoma-dominant pan-cancer data as a training set for a broad range of tasks, for instance in foundation models which are becoming feasible for some genomics applications <span class="citation" data-cites="NMsylj71 r5y0HbhJ Uetau8wh">[<a href="#ref-NMsylj71" role="doc-biblioref">13</a>,<a href="#ref-r5y0HbhJ" role="doc-biblioref">14</a>,<a href="#ref-Uetau8wh" role="doc-biblioref">15</a>]</span>.
One caveat of our analysis is that each cancer type is included in the training data or held out for a different subset of genes, so it is difficult to detangle gene-specific effects (some mutations have less distinguishable functional effects on gene expression than others) from cancer type-specific effects (some cancer types are less similar to each other than others) on prediction performance using our experimental design.</p>
<h3 id="conclusion">Conclusion</h3>
<p>Without directly evaluating model generalization, it is tempting to assume that simpler models will generalize better than more complex ones, and previous studies and sets of guidelines suggest this rule of thumb <span class="citation" data-cites="cU8r76Rn kAXq3qou wnVDbuuP HmaoBsQm">[<a href="#ref-kAXq3qou" role="doc-biblioref">197</a>,<a href="#ref-wnVDbuuP" role="doc-biblioref">198</a>,<a href="#ref-HmaoBsQm" role="doc-biblioref">199</a>,<a href="#ref-cU8r76Rn" role="doc-biblioref">219</a>]</span>.
However, we do not observe strong evidence that simpler models inherently generalize more effectively than more complex ones.
There may be other reasons to train small models or to look for the best model of a certain size/sparsity, such as biomarker interpretability or assay cost.
Our results underscore the importance of defining clear goals for each analysis.
If the goal is to achieve generalization across contexts or datasets, whenever possible we recommend directly evaluating generalization.
When it is not feasible, we recommend choosing the model that performs the best on unseen data via cross-validation or a holdout dataset.</p>
<h3 id="data-and-code-availability-1">Data and code availability</h3>
<p>The data from TCGA analyzed during this study were previously published as part of the TCGA Pan-Cancer Atlas project <span class="citation" data-cites="AK17eOgD">[<a href="#ref-AK17eOgD" role="doc-biblioref">120</a>]</span>, and are available from the NIH NCI Genomic Data Commons (GDC).
The data from CCLE analyzed during this study were previously published <span class="citation" data-cites="kcdvZmNN">[<a href="#ref-kcdvZmNN" role="doc-biblioref">9</a>]</span>, and are available from the Broad Institute’s DepMap Portal.
Raw classification results, performance figures for all genes in the Vogelstein et al. 2013 dataset, and parameter selection results and performance comparisons for each individual gene in the “best vs. smallest good” analyses are available on Figshare at <a href="https://doi.org/10.6084/m9.figshare.23826450" class="uri">https://doi.org/10.6084/m9.figshare.23826450</a>, under a CC0 license.
The scripts used to download and preprocess the datasets for this study are available at <a href="https://github.com/greenelab/pancancer-evaluation/tree/master/00_process_data" class="uri">https://github.com/greenelab/pancancer-evaluation/tree/master/00_process_data</a>.
Scripts for TCGA &lt;-&gt; CCLE comparisons (Figures 2 and 3) and neural network experiments (Figure 5) are available in the <a href="https://github.com/greenelab/pancancer-evaluation/tree/master/08_cell_line_prediction" class="uri">https://github.com/greenelab/pancancer-evaluation/tree/master/08_cell_line_prediction</a> directory.
Scripts for TCGA cancer type comparisons (Figure 4) are available in the <a href="https://github.com/greenelab/pancancer-evaluation/tree/master/02_cancer_type_classification" class="uri">https://github.com/greenelab/pancancer-evaluation/tree/master/02_cancer_type_classification</a> directory.
All scripts are available under the open-source BSD 3-clause license.</p>
<p>This manuscript was written using Manubot <span class="citation" data-cites="YuJbg3zO">[<a href="#ref-YuJbg3zO" role="doc-biblioref">121</a>]</span> and is available on GitHub at <a href="https://github.com/greenelab/generalization-manuscript" class="uri">https://github.com/greenelab/generalization-manuscript</a> under the CC0-1.0 license.
This research was supported in part by the University of Pittsburgh Center for Research Computing through the resources provided. Specifically, this work used the HTC cluster, which is supported by NIH award number S10OD028483.</p>
<h3 id="supplementary-material-1">Supplementary Material</h3>
<div id="fig:average_sparsity" class="fignos">
<figure>
<img src="images/generalization/supp_figure_1.png" style="width:100.0%" alt="Figure 27: Number of nonzero coefficients (model sparsity) across varying regularization parameters, for 71 genes (TCGA to CCLE prediction, top) and 70 genes (CCLE to TCGA prediction, bottom) in the Vogelstein et al. dataset." />
<figcaption aria-hidden="true"><span>Figure 27:</span> Number of nonzero coefficients (model sparsity) across varying regularization parameters, for 71 genes (TCGA to CCLE prediction, top) and 70 genes (CCLE to TCGA prediction, bottom) in the Vogelstein et al. dataset.</figcaption>
</figure>
</div>
<div id="fig:average_perf_by_gene" class="fignos">
<figure>
<img src="images/generalization/supp_figure_2.png" class="page_break_before" style="width:100.0%" alt="Figure 28: Distributions of performance difference between cross-validation data (same cancer types as training data) and holdout data (cancer types not represented in data), grouped by held-out gene. Each point shows performance for a single train/validation split for one cancer type that was held out, using a classifier trained to predict mutations in the given gene." />
<figcaption aria-hidden="true"><span>Figure 28:</span> Distributions of performance difference between cross-validation data (same cancer types as training data) and holdout data (cancer types not represented in data), grouped by held-out gene. Each point shows performance for a single train/validation split for one cancer type that was held out, using a classifier trained to predict mutations in the given gene.</figcaption>
</figure>
</div>
<div id="fig:thca_by_gene" class="fignos">
<figure>
<img src="images/generalization/supp_figure_3.png" class="page_break_before" style="width:100.0%" alt="Figure 29: Top row: Distribution of performance differences when thyroid cancer (THCA) data is held out from training setacross seeds/folds, grouped by gene. Bottom row: Distributions of performance differences for genes where THCA is included in training/holdout sets, relative to other cancer types that are included." />
<figcaption aria-hidden="true"><span>Figure 29:</span> Top row: Distribution of performance differences when thyroid cancer (THCA) data is held out from training setacross seeds/folds, grouped by gene. Bottom row: Distributions of performance differences for genes where THCA is included in training/holdout sets, relative to other cancer types that are included.</figcaption>
</figure>
</div>
<div id="fig:nn_dropout_wd" class="fignos">
<figure>
<img src="images/generalization/supp_figure_4.png" class="page_break_before" style="width:100.0%" alt="Figure 30: Performance vs. dropout parameter (first column) and weight decay strength (second column), for EGFR mutation prediction (first row) and KRAS mutation prediction (second row) using a 3-layer fully connected neural network trained on TCGA (blue/orange) and evaluated on CCLE (green)." />
<figcaption aria-hidden="true"><span>Figure 30:</span> Performance vs. dropout parameter (first column) and weight decay strength (second column), for EGFR mutation prediction (first row) and KRAS mutation prediction (second row) using a 3-layer fully connected neural network trained on TCGA (blue/orange) and evaluated on CCLE (green).</figcaption>
</figure>
</div>
<h2 id="chapter-5-future-directions-and-conclusion">Chapter 5: future directions and conclusion</h2>
<p><strong>Contributions:</strong>
I wrote this chapter for the purpose of this dissertation.</p>
<p>In this dissertation, we have identified complexities of the structure of large cancer -omics datasets that make building predictive models less straightforward than on more uniform datasets, such as those often considered in large-scale machine learning applications to text and images.
In chapter 2, our results indicated that model selection depends heavily on which optimizer is used to fit the model, even when relying on a fairly simple linear modeling strategy, logistic regression.
More specifically, we concluded that stochastic gradient descent is less likely to overfit for lower levels of explicit regularization (smaller LASSO parameters) because it finds a slightly less good loss function optimum, suggesting a form of “implicit regularization” that coordinate descent does not appear to possess.
We showed that this difference in optimizer behavior tends to hold across many different genes/classifiers, emphasizing that optimizer choice is important to carefully consider and document in the design of machine learning studies in transcriptomics.</p>
<p>In chapter 3, we compared -omics types as predictors of mutational state, finding that for a set of approximately 250 driver genes, a variety of -omics types could serve as effective predictors, and most driver genes perturbed most functional readouts.
We also saw that although different -omics types in TCGA may have slightly variable information content related to cancer function, much of it is redundant in the sense that combining -omics layers generally does not result in a large performance boost.
We selected 6 frequently mutated pan-cancer driver genes, and for each of the 6 we saw that the best-performing classifier using a single -omics type (gene expression, 27K methylation, or 450K methylation) outperformed classifiers using all possible combinations of the 3 -omics datasets.
In the context of the existing wide variety of multi-omics integration methods, reviewed in detail in <span class="citation" data-cites="NfKILi8i 10bvAmkln">[<a href="#ref-NfKILi8i" role="doc-biblioref">177</a>,<a href="#ref-10bvAmkln" role="doc-biblioref">220</a>]</span> and others, this result is somewhat surprising.</p>
<p>Our experiments in Chapter 4 showed that generalization of mutation status classifiers across cancer types and model systems can vary highly based on the held-out biological context.
In contrast to existing guidelines, our results show that smaller or simpler models do not in general tend to be more robust across contexts (datasets or cancer types) than models that optimize cross-validation performance within a biological context.
We saw similar results for linear models regularized using a LASSO penalty, and for neural networks regularized in a variety of ways, including limiting the size of hidden layers and using different dropout proportions.
This has implications in cases where one trains a model on a single transcriptomic dataset with the intent of applying it to other datasets or applying it across a range of problems, suggesting that optimizing cross-validation performance on a held-out subset of the training data is sufficient.</p>
<p>Taken together, these results have helped to better characterize the complexities of machine learning applied to large cancer datasets, particularly in transcriptomics but with some conclusions about other -omics types in Chapter 3.
The following sections outline areas where future work could expand upon our results, with applications to development of methods and datasets, as well as downstream biological/clinical applications of the ideas described here.</p>
<h3 id="multi-omics-integration">Multi-omics integration</h3>
<p>In chapter 2, we saw the somewhat counterintuitive result that when gene expression and DNA methylation readouts were combined, they failed to predict mutations in 6 key pan-cancer driver genes better than expression or methylation data alone.
We also noted in the discussion section of chapter 2 that the multi-omics integration method we tried was a fairly simple one: we combined the feature sets of each -omics type and fit a model directly to this larger dataset, sometimes referred to as “early” integration.
Multi-omics integration is a particularly salient problem in single-cell analysis, where data are generally sparse and often unpaired, or measured on different cells (reviewed in <span class="citation" data-cites="SLbB6cV8 9lrRlyX2 13Db93XWH">[<a href="#ref-SLbB6cV8" role="doc-biblioref">221</a>,<a href="#ref-9lrRlyX2" role="doc-biblioref">222</a>,<a href="#ref-13Db93XWH" role="doc-biblioref">223</a>]</span> and elsewhere).
However, many existing methods are not designed to incorporate axes of variation other than those typically included in single-cell studies, such as mutation status or cancer type, although some incorporate supervision or prior knowledge (e.g. <span class="citation" data-cites="g7bzDkos bcArgYmm hzkzGLGE">[<a href="#ref-g7bzDkos" role="doc-biblioref">224</a>,<a href="#ref-bcArgYmm" role="doc-biblioref">225</a>,<a href="#ref-hzkzGLGE" role="doc-biblioref">226</a>]</span>).
More specific characterization of the conditions under which integrating multiple -omics types provides additional information relative to some labeling, as well as identifying conditions when -omics types are functionally redundant, would have important implications for study design in both bulk and single-cell contexts.</p>
<h3 id="cancer-model-systems-and-data">Cancer model systems and data</h3>
<p>We showed in chapter 4 that when mutation prediction models are trained on cell line data, performance on tumor samples is generally worse than performance on held out cell lines.
We also saw that the converse is not true: models trained on tumor samples tend to generalize relatively well to cell lines.
This is likely due at least in part to biological differences between cancer cell lines and the tumors they aim to model.
Other studies have also observed that cell line gene expression is less representative of tumor samples than PDX mouse models <span class="citation" data-cites="lhYg2Yc3">[<a href="#ref-lhYg2Yc3" role="doc-biblioref">118</a>]</span> and genetically engineered mouse models <span class="citation" data-cites="1BYOCoFwl">[<a href="#ref-1BYOCoFwl" role="doc-biblioref">227</a>]</span>, although this can vary by cancer type.
In chapter 1, we noted the general reliance of existing drug response prediction efforts on public cell line datasets such as CCLE and GDSC, most likely due to the size (both in terms of sample size and number of -omics data types profiled) and accessibility of the datasets.
Together with our results, this underscores the need for larger and more representative compendia of model system data that may be biologically closer to human tumors, including mouse models, tumoroids, and others.</p>
<p>Another desired generalization property that we did not explore in this dissertation is between adult and pediatric cancers.
There are fundamental differences between adult and pediatric cancer that make transferring therapies developed on adult cancers to their pediatric equivalents a non-trivial problem.
In particular, pediatric cancers tend to have a much lower overall mutational burden, and the genes that drive pediatric cancers have limited overlap with their adult counterparts <span class="citation" data-cites="10DTHeWmL oRtHYf3u">[<a href="#ref-10DTHeWmL" role="doc-biblioref">228</a>,<a href="#ref-oRtHYf3u" role="doc-biblioref">229</a>]</span>.
With the increasing availability of public pediatric cancer -omics datasets including the Therapeutically Applicable Research to Generate Effective Treatments (TARGET) project <span class="citation" data-cites="142nJ47j4">[<a href="#ref-142nJ47j4" role="doc-biblioref">230</a>]</span>, the Open Pediatric Brain Tumor Atlas (OpenPBTA) <span class="citation" data-cites="5VXMHJ7N">[<a href="#ref-5VXMHJ7N" role="doc-biblioref">231</a>]</span>, and Pediatric DepMap <span class="citation" data-cites="HXL9NZ9Z">[<a href="#ref-HXL9NZ9Z" role="doc-biblioref">21</a>]</span>, developing models that are either specific to pediatric cancer or that integrate both adult and pediatric cancer data is starting to become a possibility.
Previous work has shown that a machine learning classifier to predict <em>TP53</em> inactivation trained on data from TCGA generalized effectively to pediatric PDX tumors, but a similarly applied Ras activation classifier failed to generalize effectively <span class="citation" data-cites="SDwYl8uA">[<a href="#ref-SDwYl8uA" role="doc-biblioref">232</a>]</span>.
In the future, it would be useful to more broadly characterize cases where classifiers generalize effectively or fail to generalize in pediatric datasets, and to develop methods to incorporate abundant adult tumor data into models that can derive insight from pediatric samples that are more limited in quantity.</p>
<h3 id="validation-and-clinical-application-of-mutation-status-classifiers">Validation and clinical application of mutation status classifiers</h3>
<p>Our “leave-one-cancer-type-out” experiments described in chapter 4 suggest that cancer types in TCGA can vary highly in how easy or hard they are to generalize to, or how similar or different they are from the rest of the dataset.
This emphasizes the need for carefully choosing cross-validation holdout sets, since cross-validation sets that are stratified by cancer type, which is more typical for TCGA or CCLE analyses, may give an overly optimistic view of performance as compared to evaluation on an entirely held out cancer type.
A similar experimental structure to ours for evaluating drug response prediction in held-out tissues was also described in <span class="citation" data-cites="ZJNDbtzm">[<a href="#ref-ZJNDbtzm" role="doc-biblioref">202</a>]</span>, and related work has detailed problems in genomic prediction where generalization to new cell types may be desirable <span class="citation" data-cites="Inb9g7Za">[<a href="#ref-Inb9g7Za" role="doc-biblioref">217</a>]</span>.</p>
<p>Differentiating similar from divergent behavior of driver mutations in the same gene across tissues/cancer types could have direct applications in basket clinical trial design, where the basic idea is that patients are included in the trial across multiple cancer types based on the presence or absence of a targetable mutation or molecular alteration.
A previous study explored the idea of selecting cancer types in a more principled way based on proteomics data, identifying several mutations with convergent effects across cancers including <em>ERBB2</em> amplification and <em>KRAS</em>/<em>NRAS</em> mutation <span class="citation" data-cites="pRTkOXGI">[<a href="#ref-pRTkOXGI" role="doc-biblioref">143</a>]</span>,
The experimental setup we described in chapters 3 and 4 could be used in a complementary manner to integrate more data types and provide improved statistical power across a variety of genes and cancer types.</p>
<p>Moreover, the success or failure of a mutation status classifier in identifying samples that may be susceptible to a targeted therapy is likely to be highly dependent on the definition of “mutated” that is settled on to train the classifier.
In the work described in this dissertation, we use a somewhat coarse-grained definition, where a sample “has a mutation” in a gene if there is either any non-silent point mutation in that gene, or a copy number variation in the correct direction for the gene’s function in cancer (gain for oncogenes or loss for tumor suppressors).
However, there are biological reasons to believe that both more flexibility in positive class definition (e.g. epigenetic modifications to genes, such as silencing via DNA methylation, or pathway cross-talk wherein mutations in different genes in the same signaling pathway could have the same functional effect) and/or more stringency in positive class definition (e.g. not all non-silent mutations in the same gene are necessarily functional or have a convergent function, as explored in <span class="citation" data-cites="7gROlS3">[<a href="#ref-7gROlS3" role="doc-biblioref">57</a>]</span>) could lead to more performant and robust classifiers in certain cases.</p>
<p>The iterative process implemented in the LURE algorithm <span class="citation" data-cites="odNCMbto">[<a href="#ref-odNCMbto" role="doc-biblioref">55</a>]</span> is one approach to identifying different molecular alterations that have convergent functions in cancer, but the analysis in the study describing LURE is limited to individual cancer types and known driver mutations.
A more flexible algorithm could simultaneously identify patients whose tumors have high pan-cancer functional similarity to a known driver alteration, and cancer types or even whole datasets/patient cohorts to include in model training in which this functional similarity is most pronounced, as a tool for rational design of basket clinical trials.
Validation of such an algorithm to the standards of clinical use would be a challenge, and would likely depend at least initially on retrospective information about driver events in certain genes, in contrast to its desired prospective use in selecting clinical trial cohorts.</p>
<h3 id="final-remarks">Final remarks</h3>
<p>Broadly, the fields of cancer genomics and computational oncology are rapidly evolving, both due to increases in the size and composition of public datasets, and due to the availability of more diverse types and sources of data.
This has resulted in new opportunities in methods development and new leads for clinical applications, but it also presents challenges due to the heterogeneity of these datasets.
We enumerated several of these challenges in this thesis, and described proposed solutions and insights, but there are sure to be many more hurdles in the future that we have not anticipated here.
Examples may include age-specific, sex-specific, and ethnic/racial group-specific effects of driver mutations or therapeutics, different sequencing or data generation methods, and different preprocessing or standardization approaches, all of which can affect modeling downstream.
Although on the surface it may seem like there are many potential pitfalls for applications of machine learning to large cancer datasets, we hope our work here shows that data heterogeneity can also present exciting opportunities for methodological creativity and biological insight in future research.
Overall, the convergence of large, public cancer genomics data and methods designed for the specific characteristics of those data should be a cause for optimism.</p>
<h2 class="page_break_before" id="references">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs" class="references csl-bib-body" role="doc-bibliography">
<div id="ref-fsabqks" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline"><strong>The European Society for Medical Oncology (ESMO) Precision Medicine Glossary</strong> <div class="csl-block">LR Yates, J Seoane, C Le Tourneau, LL Siu, R Marais, S Michiels, JC Soria, P Campbell, N Normanno, A Scarpa, … F Andre</div> <em>Annals of Oncology</em> (2018-01) <a href="https://doi.org/gcjs62">https://doi.org/gcjs62</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/annonc/mdx707">10.1093/annonc/mdx707</a></div></div>
</div>
<div id="ref-t9NvuXx2" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline"><strong>Gastric Cancer: Trastuzumab Trial Results Spur Search for Other Targets</strong> <div class="csl-block">K Hede</div> <em>JNCI Journal of the National Cancer Institute</em> (2009-09-15) <a href="https://doi.org/b7djgm">https://doi.org/b7djgm</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/jnci/djp341">10.1093/jnci/djp341</a></div></div>
</div>
<div id="ref-qSH6r3Uo" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline"><strong>BTK inhibitors in the treatment of hematological malignancies and inflammatory diseases: mechanisms and clinical studies</strong> <div class="csl-block">Aqu Alu, Hong Lei, Xuejiao Han, Yuquan Wei, Xiawei Wei</div> <em>Journal of Hematology &amp;amp; Oncology</em> (2022-10-01) <a href="https://doi.org/gsnqv2">https://doi.org/gsnqv2</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13045-022-01353-w">10.1186/s13045-022-01353-w</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36183125">36183125</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9526392">PMC9526392</a></div></div>
</div>
<div id="ref-CkczGwEJ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline"><strong>Emerging strategies to overcome resistance to third-generation EGFR inhibitors</strong> <div class="csl-block">Kunyu Shi, Guan Wang, Junping Pei, Jifa Zhang, Jiaxing Wang, Liang Ouyang, Yuxi Wang, Weimin Li</div> <em>Journal of Hematology &amp;amp; Oncology</em> (2022-07-15) <a href="https://doi.org/grr735">https://doi.org/grr735</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13045-022-01311-6">10.1186/s13045-022-01311-6</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/35840984">35840984</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9287895">PMC9287895</a></div></div>
</div>
<div id="ref-c9AdwXLE" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline"><strong>A decade of clinical development of PARP inhibitors in perspective</strong> <div class="csl-block">J Mateo, CJ Lord, V Serra, A Tutt, J Balmaña, M Castroviejo-Bermejo, C Cruz, A Oaknin, SB Kaye, JS de Bono</div> <em>Annals of Oncology</em> (2019-09) <a href="https://doi.org/gk6cjp">https://doi.org/gk6cjp</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/annonc/mdz192">10.1093/annonc/mdz192</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31218365">31218365</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6771225">PMC6771225</a></div></div>
</div>
<div id="ref-11ZPgRuLk" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline"><strong>Oncogenic Signaling Pathways in The Cancer Genome Atlas</strong> <div class="csl-block">Francisco Sanchez-Vega, Marco Mina, Joshua Armenia, Walid K Chatila, Augustin Luna, Konnor C La, Sofia Dimitriadoy, David L Liu, Havish S Kantheti, Sadegh Saghafinia, … Armaz Mariamidze</div> <em>Cell</em> (2018-04) <a href="https://doi.org/gc7r9b">https://doi.org/gc7r9b</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cell.2018.03.035">10.1016/j.cell.2018.03.035</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29625050">29625050</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6070353">PMC6070353</a></div></div>
</div>
<div id="ref-5N9Iz2gd" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline"><strong>A modular master regulator landscape controls cancer transcriptional identity</strong> <div class="csl-block">Evan O Paull, Alvaro Aytes, Sunny J Jones, Prem S Subramaniam, Federico M Giorgi, Eugene F Douglass, Somnath Tagore, Brennan Chu, Alessandro Vasciaveo, Siyuan Zheng, … Andrea Califano</div> <em>Cell</em> (2021-01) <a href="https://doi.org/gh2s7r">https://doi.org/gh2s7r</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cell.2020.11.045">10.1016/j.cell.2020.11.045</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33434495">33434495</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8103356">PMC8103356</a></div></div>
</div>
<div id="ref-KrfP0fMx" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline"><strong>TCGA Pan-Cancer Atlas</strong> <div class="csl-block">The Cancer Genome Atlas (TCGA) Research Network</div> <em>Webpage</em> (2022) <a href="https://gdc.cancer.gov/about-data/publications/pancanatlas">https://gdc.cancer.gov/about-data/publications/pancanatlas</a></div>
</div>
<div id="ref-kcdvZmNN" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline"><strong>Next-generation characterization of the Cancer Cell Line Encyclopedia</strong> <div class="csl-block">Mahmoud Ghandi, Franklin W Huang, Judit Jané-Valbuena, Gregory V Kryukov, Christopher C Lo, ERobert McDonald III, Jordi Barretina, Ellen T Gelfand, Craig M Bielski, Haoxin Li, … William R Sellers</div> <em>Nature</em> (2019-05) <a href="https://doi.org/gf2m3h">https://doi.org/gf2m3h</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41586-019-1186-3">10.1038/s41586-019-1186-3</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31068700">31068700</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6697103">PMC6697103</a></div></div>
</div>
<div id="ref-lJFuND9F" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline"><strong>Genomics of Drug Sensitivity in Cancer (GDSC): a resource for therapeutic biomarker discovery in cancer cells</strong> <div class="csl-block">Wanjuan Yang, Jorge Soares, Patricia Greninger, Elena J Edelman, Howard Lightfoot, Simon Forbes, Nidhi Bindal, Dave Beare, James A Smith, IRichard Thompson, … Mathew J Garnett</div> <em>Nucleic Acids Research</em> (2012-11-22) <a href="https://doi.org/f4jrdn">https://doi.org/f4jrdn</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gks1111">10.1093/nar/gks1111</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23180760">23180760</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3531057">PMC3531057</a></div></div>
</div>
<div id="ref-m16z46g3" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline"><strong>Can Foundation Models Wrangle Your Data?</strong> <div class="csl-block">Avanika Narayan, Ines Chami, Laurel Orr, Simran Arora, Christopher Ré</div> <em>arXiv</em> (2022-12-27) <a href="https://arxiv.org/abs/2205.09911">https://arxiv.org/abs/2205.09911</a></div>
</div>
<div id="ref-9Swcx8ai" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline"><strong>On the Opportunities and Risks of Foundation Models</strong> <div class="csl-block">Rishi Bommasani, Drew A Hudson, Ehsan Adeli, Russ Altman, Simran Arora, Sydney von Arx, Michael S Bernstein, Jeannette Bohg, Antoine Bosselut, Emma Brunskill, … Percy Liang</div> <em>arXiv</em> (2022-07-14) <a href="https://arxiv.org/abs/2108.07258">https://arxiv.org/abs/2108.07258</a></div>
</div>
<div id="ref-NMsylj71" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline"><strong>HyenaDNA: Long-Range Genomic Sequence Modeling at Single Nucleotide Resolution</strong> <div class="csl-block">Eric Nguyen, Michael Poli, Marjan Faizi, Armin Thomas, Callum Birch-Sykes, Michael Wornow, Aman Patel, Clayton Rabideau, Stefano Massaroli, Yoshua Bengio, … Chris Ré</div> <em>arXiv</em> (2023-06-29) <a href="https://arxiv.org/abs/2306.15794">https://arxiv.org/abs/2306.15794</a></div>
</div>
<div id="ref-r5y0HbhJ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">14. </div><div class="csl-right-inline"><strong>scGPT: Towards Building a Foundation Model for Single-Cell Multi-omics Using Generative AI</strong> <div class="csl-block">Haotian Cui, Chloe Wang, Hassaan Maan, Kuan Pang, Fengning Luo, Bo Wang</div> <em>Cold Spring Harbor Laboratory</em> (2023-05-01) <a href="https://doi.org/gshk6p">https://doi.org/gshk6p</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2023.04.30.538439">10.1101/2023.04.30.538439</a></div></div>
</div>
<div id="ref-Uetau8wh" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">15. </div><div class="csl-right-inline"><strong>Large Scale Foundation Model on Single-cell Transcriptomics</strong> <div class="csl-block">Minsheng Hao, Jing Gong, Xin Zeng, Chiming Liu, Yucheng Guo, Xingyi Cheng, Taifeng Wang, Jianzhu Ma, Le Song, Xuegong Zhang</div> <em>Cold Spring Harbor Laboratory</em> (2023-05-31) <a href="https://doi.org/gshk6q">https://doi.org/gshk6q</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2023.05.29.542705">10.1101/2023.05.29.542705</a></div></div>
</div>
<div id="ref-ZB5QpmE" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">16. </div><div class="csl-right-inline"><strong>An Integrated TCGA Pan-Cancer Clinical Data Resource to Drive High-Quality Survival Outcome Analytics</strong> <div class="csl-block">Jianfang Liu, Tara Lichtenberg, Katherine A Hoadley, Laila M Poisson, Alexander J Lazar, Andrew D Cherniack, Albert J Kovatich, Christopher C Benz, Douglas A Levine, Adrian V Lee, … Armaz Mariamidze</div> <em>Cell</em> (2018-04) <a href="https://doi.org/gc93hc">https://doi.org/gc93hc</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cell.2018.02.052">10.1016/j.cell.2018.02.052</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29625055">29625055</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6066282">PMC6066282</a></div></div>
</div>
<div id="ref-IKDYyuvh" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">17. </div><div class="csl-right-inline"><strong>The International Cancer Genome Consortium Data Portal</strong> <div class="csl-block">Junjun Zhang, Rosita Bajari, Dusan Andric, Francois Gerthoffert, Alexandru Lepsa, Hardeep Nahal-Bose, Lincoln D Stein, Vincent Ferretti</div> <em>Nature Biotechnology</em> (2019-03-15) <a href="https://doi.org/gkn7xh">https://doi.org/gkn7xh</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41587-019-0055-9">10.1038/s41587-019-0055-9</a></div></div>
</div>
<div id="ref-12f5xcS10" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">18. </div><div class="csl-right-inline"><strong>Pan-cancer analysis of whole genomes</strong><div class="csl-block">, Lauri A Aaltonen, Federico Abascal, Adam Abeshouse, Hiroyuki Aburatani, David J Adams, Nishant Agrawal, Keun Soo Ahn, Sung-Min Ahn, Hiroshi Aikata, … Christian von Mering</div> <em>Nature</em> (2020-02-05) <a href="https://doi.org/ggkfng">https://doi.org/ggkfng</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41586-020-1969-6">10.1038/s41586-020-1969-6</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32025007">32025007</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7025898">PMC7025898</a></div></div>
</div>
<div id="ref-8oOo13r" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">19. </div><div class="csl-right-inline"><strong>AACR Project GENIE: Powering Precision Medicine through an International Consortium</strong><div class="csl-block">, Fabrice André, Monica Arnedos, Alexander S Baras, José Baselga, Philippe L Bedard, Michael F Berger, Mariska Bierkens, Fabien Calvo, Ethan Cerami, … </div> <em>Cancer Discovery</em> (2017-08-01) <a href="https://doi.org/ggnnrn">https://doi.org/ggnnrn</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1158/2159-8290.cd-17-0151">10.1158/2159-8290.cd-17-0151</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28572459">28572459</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5611790">PMC5611790</a></div></div>
</div>
<div id="ref-RnnKuIPr" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">20. </div><div class="csl-right-inline"><strong>Integrated cross-study datasets of genetic dependencies in cancer</strong> <div class="csl-block">Clare Pacini, Joshua M Dempster, Isabella Boyle, Emanuel Gonçalves, Hanna Najgebauer, Emre Karakoc, Dieudonne van der Meer, Andrew Barthorpe, Howard Lightfoot, Patricia Jaaks, … Francesco Iorio</div> <em>Nature Communications</em> (2021-03-12) <a href="https://doi.org/gnpt56">https://doi.org/gnpt56</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41467-021-21898-7">10.1038/s41467-021-21898-7</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33712601">33712601</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7955067">PMC7955067</a></div></div>
</div>
<div id="ref-HXL9NZ9Z" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">21. </div><div class="csl-right-inline"><strong>A first-generation pediatric cancer dependency map</strong> <div class="csl-block">Neekesh V Dharia, Guillaume Kugener, Lillian M Guenther, Clare F Malone, Adam D Durbin, Andrew L Hong, Thomas P Howard, Pratiti Bandopadhayay, Caroline S Wechsler, Iris Fung, … Kimberly Stegmaier</div> <em>Nature Genetics</em> (2021-03-22) <a href="https://doi.org/f37p">https://doi.org/f37p</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41588-021-00819-w">10.1038/s41588-021-00819-w</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33753930">33753930</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8049517">PMC8049517</a></div></div>
</div>
<div id="ref-F7lIlh2N" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">22. </div><div class="csl-right-inline"><strong>A Next Generation Connectivity Map: L1000 Platform and the First 1,000,000 Profiles</strong> <div class="csl-block">Aravind Subramanian, Rajiv Narayan, Steven M Corsello, David D Peck, Ted E Natoli, Xiaodong Lu, Joshua Gould, John F Davis, Andrew A Tubelli, Jacob K Asiedu, … Todd R Golub</div> <em>Cell</em> (2017-11) <a href="https://doi.org/cgwt">https://doi.org/cgwt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cell.2017.10.049">10.1016/j.cell.2017.10.049</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29195078">29195078</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5990023">PMC5990023</a></div></div>
</div>
<div id="ref-ZRWXpNGs" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">23. </div><div class="csl-right-inline"><strong>Discovering the anticancer potential of non-oncology drugs by systematic viability profiling</strong> <div class="csl-block">Steven M Corsello, Rohith T Nagari, Ryan D Spangler, Jordan Rossen, Mustafa Kocak, Jordan G Bryan, Ranad Humeidi, David Peck, Xiaoyun Wu, Andrew A Tang, … Todd R Golub</div> <em>Nature Cancer</em> (2020-01-20) <a href="https://doi.org/ggr7x4">https://doi.org/ggr7x4</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s43018-019-0018-6">10.1038/s43018-019-0018-6</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32613204">32613204</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7328899">PMC7328899</a></div></div>
</div>
<div id="ref-BFfDQGl5" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">24. </div><div class="csl-right-inline"><strong>High-throughput screening using patient-derived tumor xenografts to predict clinical trial drug response</strong> <div class="csl-block">Hui Gao, Joshua M Korn, Stéphane Ferretti, John E Monahan, Youzhen Wang, Mallika Singh, Chao Zhang, Christian Schnell, Guizhi Yang, Yun Zhang, … William R Sellers</div> <em>Nature Medicine</em> (2015-10-19) <a href="https://doi.org/f7xmb3">https://doi.org/f7xmb3</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nm.3954">10.1038/nm.3954</a></div></div>
</div>
<div id="ref-Ct0A32uh" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">25. </div><div class="csl-right-inline"><strong>Comprehensive characterization of 536 patient-derived xenograft models prioritizes candidates for targeted treatment</strong> <div class="csl-block">Hua Sun, Song Cao, RJay Mashl, Chia-Kuei Mo, Simone Zaccaria, Michael C Wendl, Sherri R Davies, Matthew H Bailey, Tina M Primeau, Jeremy Hoog, … </div> <em>Nature Communications</em> (2021-08-24) <a href="https://doi.org/gnzdnk">https://doi.org/gnzdnk</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41467-021-25177-3">10.1038/s41467-021-25177-3</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34429404">34429404</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8384880">PMC8384880</a></div></div>
</div>
<div id="ref-yoGhmtgc" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">26. </div><div class="csl-right-inline"><strong>NCI Patient-Derived Models Repository</strong> <div class="csl-block">National Cancer Institute (NCI), National Institutes of Health</div> <em>Webpage</em> (2023) <a href="https://pdmr.cancer.gov/">https://pdmr.cancer.gov/</a></div>
</div>
<div id="ref-MHNiNgKE" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">27. </div><div class="csl-right-inline"><strong>Convolutional neural network models for cancer type prediction based on gene expression</strong> <div class="csl-block">Milad Mostavi, Yu-Chiao Chiu, Yufei Huang, Yidong Chen</div> <em>BMC Medical Genomics</em> (2020-04) <a href="https://doi.org/gqftgb">https://doi.org/gqftgb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12920-020-0677-2">10.1186/s12920-020-0677-2</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32241303">32241303</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7119277">PMC7119277</a></div></div>
</div>
<div id="ref-16isU2R76" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">28. </div><div class="csl-right-inline"><strong>A Neural Network Framework for Predicting the Tissue-of-Origin of 15 Common Cancer Types Based on RNA-Seq Data</strong> <div class="csl-block">Binsheng He, Yanxiang Zhang, Zhen Zhou, Bo Wang, Yuebin Liang, Jidong Lang, Huixin Lin, Pingping Bing, Lan Yu, Dejun Sun, … Geng Tian</div> <em>Frontiers in Bioengineering and Biotechnology</em> (2020-08-05) <a href="https://doi.org/ghdqmt">https://doi.org/ghdqmt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3389/fbioe.2020.00737">10.3389/fbioe.2020.00737</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32850691">32850691</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7419649">PMC7419649</a></div></div>
</div>
<div id="ref-NgF0DB31" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">29. </div><div class="csl-right-inline"><strong>Gene-Expression-Based Cancer Subtypes Prediction Through Feature Selection and Transductive SVM</strong> <div class="csl-block">U Maulik, A Mukhopadhyay, D Chakraborty</div> <em>IEEE Transactions on Biomedical Engineering</em> (2013-04) <a href="https://doi.org/f4swmp">https://doi.org/f4swmp</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1109/tbme.2012.2225622">10.1109/tbme.2012.2225622</a></div></div>
</div>
<div id="ref-3oNamamF" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">30. </div><div class="csl-right-inline"><strong>Multi-label transcriptional classification of colorectal cancer reflects tumor cell population heterogeneity</strong> <div class="csl-block">Silvia Cascianelli, Chiara Barbera, Alexandra Ambra Ulla, Elena Grassi, Barbara Lupo, Diego Pasini, Andrea Bertotti, Livio Trusolino, Enzo Medico, Claudio Isella, Marco Masseroli</div> <em>Genome Medicine</em> (2023-05-15) <a href="https://doi.org/gsn623">https://doi.org/gsn623</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13073-023-01176-5">10.1186/s13073-023-01176-5</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37189167">37189167</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10184353">PMC10184353</a></div></div>
</div>
<div id="ref-hocatXNe" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">31. </div><div class="csl-right-inline"><strong>Developmental Deconvolution for Classification of Cancer Origin</strong> <div class="csl-block">Enrico Moiso, Alexander Farahani, Hetal D Marble, Austin Hendricks, Samuel Mildrum, Stuart Levine, Jochen K Lennerz, Salil Garg</div> <em>Cancer Discovery</em> (2022-08-30) <a href="https://doi.org/gqq9x7">https://doi.org/gqq9x7</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1158/2159-8290.cd-21-1443">10.1158/2159-8290.cd-21-1443</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36041084">36041084</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9627133">PMC9627133</a></div></div>
</div>
<div id="ref-jwzDipUG" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">32. </div><div class="csl-right-inline"><strong>Machine learning for genetics-based classification and treatment response prediction in cancer of unknown primary</strong> <div class="csl-block">Intae Moon, Jaclyn LoPiccolo, Sylvan C Baca, Lynette M Sholl, Kenneth L Kehl, Michael J Hassett, David Liu, Deborah Schrag, Alexander Gusev</div> <em>Nature Medicine</em> (2023-08) <a href="https://doi.org/gsn62s">https://doi.org/gsn62s</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41591-023-02482-6">10.1038/s41591-023-02482-6</a></div></div>
</div>
<div id="ref-RjkBa6QW" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">33. </div><div class="csl-right-inline"><strong>A deep learning system accurately classifies primary and metastatic cancers using passenger mutation patterns</strong> <div class="csl-block">Wei Jiao, Gurnit Atwal, Paz Polak, Rosa Karlic, Edwin Cuppen, Fatima Al-Shahrour, Gurnit Atwal, Peter J Bailey, Andrew V Biankin, Paul C Boutros, … </div> <em>Nature Communications</em> (2020-02-05) <a href="https://doi.org/ggkfnh">https://doi.org/ggkfnh</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41467-019-13825-8">10.1038/s41467-019-13825-8</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32024849">32024849</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7002586">PMC7002586</a></div></div>
</div>
<div id="ref-T2RxGl4r" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">34. </div><div class="csl-right-inline"><strong>CancerSiamese: one-shot learning for predicting primary and metastatic tumor types unseen during model training</strong> <div class="csl-block">Milad Mostav, Yu-Chiao Chiu, Yidong Chen, Yufei Huang</div> <em>Cold Spring Harbor Laboratory</em> (2020-09-09) <a href="https://doi.org/gsn62x">https://doi.org/gsn62x</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2020.09.07.286583">10.1101/2020.09.07.286583</a></div></div>
</div>
<div id="ref-1BSWdoRe0" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">35. </div><div class="csl-right-inline"><strong>PLUS: Predicting cancer metastasis potential based on positive and unlabeled learning</strong> <div class="csl-block">Junyi Zhou, Xiaoyu Lu, Wennan Chang, Changlin Wan, Xiongbin Lu, Chi Zhang, Sha Cao</div> <em>PLOS Computational Biology</em> (2022-03-29) <a href="https://doi.org/gsn624">https://doi.org/gsn624</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pcbi.1009956">10.1371/journal.pcbi.1009956</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/35349572">35349572</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8992993">PMC8992993</a></div></div>
</div>
<div id="ref-shldZigM" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">36. </div><div class="csl-right-inline"><strong>MOLI: multi-omics late integration with deep neural networks for drug response prediction</strong> <div class="csl-block">Hossein Sharifi-Noghabi, Olga Zolotareva, Colin C Collins, Martin Ester</div> <em>Bioinformatics</em> (2019-07) <a href="https://doi.org/fxbz">https://doi.org/fxbz</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btz318">10.1093/bioinformatics/btz318</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31510700">31510700</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6612815">PMC6612815</a></div></div>
</div>
<div id="ref-WRbRNpls" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">37. </div><div class="csl-right-inline"><strong>Dr.VAE: improving drug response prediction via modeling of drug perturbation effects</strong> <div class="csl-block">Ladislav Rampášek, Daniel Hidru, Petr Smirnov, Benjamin Haibe-Kains, Anna Goldenberg</div> <em>Bioinformatics</em> (2019-03-08) <a href="https://doi.org/gsn625">https://doi.org/gsn625</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btz158">10.1093/bioinformatics/btz158</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30850846">30850846</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6761940">PMC6761940</a></div></div>
</div>
<div id="ref-D5XNExDI" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">38. </div><div class="csl-right-inline"><strong>Predicting patient treatment response and resistance via single-cell transcriptomics of their tumors</strong> <div class="csl-block">Sanju Sinha, Rahulsimham Vegesna, Saugato Rahman Dhruba, Wei Wu, DLucas Kerr, Oleg V Stroganov, Ivan Grishagin, Kenneth D Aldape, Collin M Blakely, Peng Jiang, … Eytan Ruppin</div> <em>Cold Spring Harbor Laboratory</em> (2022-01-12) <a href="https://doi.org/gsn62z">https://doi.org/gsn62z</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2022.01.11.475728">10.1101/2022.01.11.475728</a></div></div>
</div>
<div id="ref-AzsCu0Vu" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">39. </div><div class="csl-right-inline"><strong>Drug sensitivity prediction from cell line-based pharmacogenomics data: guidelines for developing machine learning models</strong> <div class="csl-block">Hossein Sharifi-Noghabi, Soheil Jahangiri-Tazehkand, Petr Smirnov, Casey Hon, Anthony Mammoliti, Sisira Kadambat Nair, Arvind Singh Mer, Martin Ester, Benjamin Haibe-Kains</div> <em>Briefings in Bioinformatics</em> (2021-08-11) <a href="https://doi.org/gsn62w">https://doi.org/gsn62w</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bib/bbab294">10.1093/bib/bbab294</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34382071">34382071</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8575017">PMC8575017</a></div></div>
</div>
<div id="ref-17Pw4cclY" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">40. </div><div class="csl-right-inline"><strong>Deep transfer learning of cancer drug responses by integrating bulk and single-cell RNA-seq data</strong> <div class="csl-block">Junyi Chen, Xiaoying Wang, Anjun Ma, Qi-En Wang, Bingqiang Liu, Lang Li, Dong Xu, Qin Ma</div> <em>Nature Communications</em> (2022-10-30) <a href="https://doi.org/gsn62r">https://doi.org/gsn62r</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41467-022-34277-7">10.1038/s41467-022-34277-7</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36310235">36310235</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9618578">PMC9618578</a></div></div>
</div>
<div id="ref-1FBomJOo7" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">41. </div><div class="csl-right-inline"><strong>A performance evaluation of drug response prediction models for individual drugs</strong> <div class="csl-block">Aron Park, Yeeun Lee, Seungyoon Nam</div> <em>Scientific Reports</em> (2023-07-24) <a href="https://doi.org/gsn62t">https://doi.org/gsn62t</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41598-023-39179-2">10.1038/s41598-023-39179-2</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37488424">37488424</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10366128">PMC10366128</a></div></div>
</div>
<div id="ref-11fRJjEgO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">42. </div><div class="csl-right-inline"><strong>Improved prediction of immune checkpoint blockade efficacy across multiple cancer types</strong> <div class="csl-block">Diego Chowell, Seong-Keun Yoo, Cristina Valero, Alessandro Pastore, Chirag Krishna, Mark Lee, Douglas Hoen, Hongyu Shi, Daniel W Kelly, Neal Patel, … Timothy A Chan</div> <em>Nature Biotechnology</em> (2021-11-01) <a href="https://doi.org/gnbhsf">https://doi.org/gnbhsf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41587-021-01070-8">10.1038/s41587-021-01070-8</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34725502">34725502</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9363980">PMC9363980</a></div></div>
</div>
<div id="ref-4jSZ7PTT" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">43. </div><div class="csl-right-inline"><strong>Tumor monocyte content predicts immunochemotherapy outcomes in esophageal adenocarcinoma</strong> <div class="csl-block">Thomas M Carroll, Joseph A Chadwick, Richard P Owen, Michael J White, Joseph Kaplinsky, Iliana Peneva, Anna Frangou, Phil F Xie, Jaeho Chang, Andrew Roth, … Phil F Xie</div> <em>Cancer Cell</em> (2023-07) <a href="https://doi.org/gsf65j">https://doi.org/gsf65j</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.ccell.2023.06.006">10.1016/j.ccell.2023.06.006</a></div></div>
</div>
<div id="ref-lw3yfGOX" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">44. </div><div class="csl-right-inline"><strong>Is tumor mutational burden predictive of response to immunotherapy?</strong> <div class="csl-block">Carino Gurjao, Dina Tsukrov, Maxim Imakaev, Lovelace J Luquette, Leonid A Mirny</div> <em>Cold Spring Harbor Laboratory</em> (2020-09-04) <a href="https://doi.org/jjcw">https://doi.org/jjcw</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2020.09.03.260265">10.1101/2020.09.03.260265</a></div></div>
</div>
<div id="ref-1DjBaykxF" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">45. </div><div class="csl-right-inline"><strong>Assessment of performance of survival prediction models for cancer prognosis</strong> <div class="csl-block">Hung-Chia Chen, Ralph L Kodell, Kuang Fu Cheng, James J Chen</div> <em>BMC Medical Research Methodology</em> (2012-07-23) <a href="https://doi.org/gbbf2p">https://doi.org/gbbf2p</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/1471-2288-12-102">10.1186/1471-2288-12-102</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22824262">22824262</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3410808">PMC3410808</a></div></div>
</div>
<div id="ref-2ioatXdg" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">46. </div><div class="csl-right-inline"><strong>Combining multidimensional genomic measurements for predicting cancer prognosis: observations from TCGA</strong> <div class="csl-block">Q Zhao, X Shi, Y Xie, J Huang, B Shia, S Ma</div> <em>Briefings in Bioinformatics</em> (2014-03-13) <a href="https://doi.org/gpf5ds">https://doi.org/gpf5ds</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bib/bbu003">10.1093/bib/bbu003</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24632304">24632304</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4375393">PMC4375393</a></div></div>
</div>
<div id="ref-1AopCEsxs" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">47. </div><div class="csl-right-inline"><strong>Pan-cancer evaluation of gene expression and somatic alteration data for cancer prognosis prediction</strong> <div class="csl-block">Xingyu Zheng, Christopher I Amos, HRobert Frost</div> <em>BMC Cancer</em> (2021-09-25) <a href="https://doi.org/gsn63z">https://doi.org/gsn63z</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12885-021-08796-3">10.1186/s12885-021-08796-3</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34563154">34563154</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8467202">PMC8467202</a></div></div>
</div>
<div id="ref-OWaqmgwK" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">48. </div><div class="csl-right-inline"><strong>Machine learning applications in cancer prognosis and prediction</strong> <div class="csl-block">Konstantina Kourou, Themis P Exarchos, Konstantinos P Exarchos, Michalis V Karamouzis, Dimitrios I Fotiadis</div> <em>Computational and Structural Biotechnology Journal</em> (2015) <a href="https://doi.org/gcpfs9">https://doi.org/gcpfs9</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.csbj.2014.11.005">10.1016/j.csbj.2014.11.005</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25750696">25750696</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4348437">PMC4348437</a></div></div>
</div>
<div id="ref-zPJFCX9h" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">49. </div><div class="csl-right-inline"><strong>Modeling RAS Phenotype in Colorectal Cancer Uncovers Novel Molecular Traits of RAS Dependency and Improves Prediction of Response to Targeted Agents in Patients</strong> <div class="csl-block">Justin Guinney, Charles Ferté, Jonathan Dry, Robert McEwen, Gilles Manceau, KJ Kao, Kai-Ming Chang, Claus Bendtsen, Kevin Hudson, Erich Huang, … Pierre Laurent-Puig</div> <em>Clinical Cancer Research</em> (2014-01-01) <a href="https://doi.org/f5njhn">https://doi.org/f5njhn</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1158/1078-0432.ccr-13-1943">10.1158/1078-0432.ccr-13-1943</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24170544">24170544</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4141655">PMC4141655</a></div></div>
</div>
<div id="ref-17XHtmqPY" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">50. </div><div class="csl-right-inline"><strong>Machine Learning Detects Pan-cancer Ras Pathway Activation in The Cancer Genome Atlas</strong> <div class="csl-block">Gregory P Way, Francisco Sanchez-Vega, Konnor La, Joshua Armenia, Walid K Chatila, Augustin Luna, Chris Sander, Andrew D Cherniack, Marco Mina, Giovanni Ciriello, … Armaz Mariamidze</div> <em>Cell Reports</em> (2018-04) <a href="https://doi.org/gfspsb">https://doi.org/gfspsb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.celrep.2018.03.046">10.1016/j.celrep.2018.03.046</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29617658">29617658</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5918694">PMC5918694</a></div></div>
</div>
<div id="ref-NsW0qxZF" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">51. </div><div class="csl-right-inline"><strong>Compressing gene expression data using multiple latent space dimensionalities learns complementary biological representations</strong> <div class="csl-block">Gregory P Way, Michael Zietz, Vincent Rubinetti, Daniel S Himmelstein, Casey S Greene</div> <em>Genome Biology</em> (2020-05-11) <a href="https://doi.org/gg2mjh">https://doi.org/gg2mjh</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-020-02021-3">10.1186/s13059-020-02021-3</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32393369">32393369</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7212571">PMC7212571</a></div></div>
</div>
<div id="ref-MUt3ESlf" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">52. </div><div class="csl-right-inline"><strong>Prediction of PIK3CA mutations from cancer gene expression data</strong> <div class="csl-block">Jun Kang, Ahwon Lee, Youn Soo Lee</div> <em>PLOS ONE</em> (2020-11-09) <a href="https://doi.org/gjmd3s">https://doi.org/gjmd3s</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pone.0241514">10.1371/journal.pone.0241514</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33166334">33166334</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7652327">PMC7652327</a></div></div>
</div>
<div id="ref-j19ozmy" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">53. </div><div class="csl-right-inline"><strong>BRCAness revisited</strong> <div class="csl-block">Christopher J Lord, Alan Ashworth</div> <em>Nature Reviews Cancer</em> (2016-01-18) <a href="https://doi.org/gd6h8d">https://doi.org/gd6h8d</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nrc.2015.21">10.1038/nrc.2015.21</a></div></div>
</div>
<div id="ref-19oMq1OuY" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">54. </div><div class="csl-right-inline"><strong>Ph-like acute lymphoblastic leukemia</strong> <div class="csl-block">Thai Hoa Tran, Mignon L Loh</div> <em>Hematology</em> (2016-12-02) <a href="https://doi.org/gsn626">https://doi.org/gsn626</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1182/asheducation-2016.1.561">10.1182/asheducation-2016.1.561</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27913529">27913529</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6142516">PMC6142516</a></div></div>
</div>
<div id="ref-odNCMbto" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">55. </div><div class="csl-right-inline"><strong>Using Transcriptional Signatures to Find Cancer Drivers with LURE</strong> <div class="csl-block">David Haan, Ruikang Tao, Verena Friedl, Ioannis N Anastopoulos, Christopher K Wong, Alana S Weinstein, Joshua M Stuart</div> <em>Biocomputing 2020</em> (2019-11-27) <a href="https://doi.org/gjmd4t">https://doi.org/gjmd4t</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1142/9789811215636_0031">10.1142/9789811215636_0031</a></div></div>
</div>
<div id="ref-sngWnSGq" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">56. </div><div class="csl-right-inline"><strong>Identify non-mutational p53 functional deficiency in human cancers</strong> <div class="csl-block">Qianpeng Li, Yang Zhang, Sicheng Luo, Zhang Zhang, Ann L Oberg, David E Kozono, Hua Lu, Jann N Sarkaria, Lina Ma, Liguo Wang</div> <em>Cold Spring Harbor Laboratory</em> (2022-07-31) <a href="https://doi.org/gsn622">https://doi.org/gsn622</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2022.07.28.501874">10.1101/2022.07.28.501874</a></div></div>
</div>
<div id="ref-7gROlS3" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">57. </div><div class="csl-right-inline"><strong>Systematic interrogation of mutation groupings reveals divergent downstream expression programs within key cancer genes</strong> <div class="csl-block">Michal R Grzadkowski, Hannah D Holly, Julia Somers, Emek Demir</div> <em>BMC Bioinformatics</em> (2021-05-06) <a href="https://doi.org/gjwvwj">https://doi.org/gjwvwj</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12859-021-04147-y">10.1186/s12859-021-04147-y</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33957863">33957863</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8101181">PMC8101181</a></div></div>
</div>
<div id="ref-zbMcel6V" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">58. </div><div class="csl-right-inline"><strong>Identification of phenocopies improves prediction of targeted therapy response over DNA mutations alone</strong> <div class="csl-block">Hamza Bakhtiar, Kyle T Helzer, Yeonhee Park, Yi Chen, Nicholas R Rydzewski, Matthew L Bootsma, Yue Shi, Paul M Harari, Marina Sharifi, Martin Sjöström, … Shuang G Zhao</div> <em>npj Genomic Medicine</em> (2022-10-17) <a href="https://doi.org/gsjxt6">https://doi.org/gsjxt6</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41525-022-00328-7">10.1038/s41525-022-00328-7</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36253482">36253482</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9576758">PMC9576758</a></div></div>
</div>
<div id="ref-19JSolFTy" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">59. </div><div class="csl-right-inline"><strong>HRDetect is a predictor of BRCA1 and BRCA2 deficiency based on mutational signatures</strong> <div class="csl-block">Helen Davies, Dominik Glodzik, Sandro Morganella, Lucy R Yates, Johan Staaf, Xueqing Zou, Manasa Ramakrishna, Sancha Martin, Sandrine Boyault, Anieta M Sieuwerts, … Serena Nik-Zainal</div> <em>Nature Medicine</em> (2017-03-13) <a href="https://doi.org/b37d">https://doi.org/b37d</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nm.4292">10.1038/nm.4292</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28288110">28288110</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5833945">PMC5833945</a></div></div>
</div>
<div id="ref-1EZ2xyPvl" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">60. </div><div class="csl-right-inline"><strong>The clinical utility of integrative genomics in childhood cancer extends beyond targetable mutations</strong> <div class="csl-block">Anita Villani, Scott Davidson, Nisha Kanwar, Winnie W Lo, Yisu Li, Sarah Cohen-Gogo, Fabio Fuligni, Lisa-Monique Edward, Nicholas Light, Mehdi Layeghifard, … Adam Shlien</div> <em>Nature Cancer</em> (2022-12-30) <a href="https://doi.org/gsn62v">https://doi.org/gsn62v</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s43018-022-00474-y">10.1038/s43018-022-00474-y</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36585449">36585449</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9970873">PMC9970873</a></div></div>
</div>
<div id="ref-1Echl6MYV" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">61. </div><div class="csl-right-inline"><strong>Multi-scale characterisation of homologous recombination deficiency in breast cancer</strong> <div class="csl-block">Daniel H Jacobson, Shi Pan, Jasmin Fisher, Maria Secrier</div> <em>Cold Spring Harbor Laboratory</em> (2023-08-23) <a href="https://doi.org/gsqddh">https://doi.org/gsqddh</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2023.08.23.554414">10.1101/2023.08.23.554414</a></div></div>
</div>
<div id="ref-oqXUIdaZ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">62. </div><div class="csl-right-inline"><strong>Synthetic lethality-mediated precision oncology via the tumor transcriptome</strong> <div class="csl-block">Joo Sang Lee, Nishanth Ulhas Nair, Gal Dinstag, Lesley Chapman, Youngmin Chung, Kun Wang, Sanju Sinha, Hongui Cha, Dasol Kim, Alexander V Schperberg, … Eytan Ruppin</div> <em>Cell</em> (2021-04) <a href="https://doi.org/gjtcvt">https://doi.org/gjtcvt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cell.2021.03.030">10.1016/j.cell.2021.03.030</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33857424">33857424</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9310669">PMC9310669</a></div></div>
</div>
<div id="ref-1FF1hUyo8" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">63. </div><div class="csl-right-inline"><strong>Molecular Profiling and Targeted Therapy for Advanced Thoracic Malignancies: A Biomarker-Derived, Multiarm, Multihistology Phase II Basket Trial</strong> <div class="csl-block">Ariel Lopez-Chavez, Anish Thomas, Arun Rajan, Mark Raffeld, Betsy Morrow, Ronan Kelly, Corey Allan Carter, Udayan Guha, Keith Killian, Christopher C Lau, … Giuseppe Giaccone</div> <em>Journal of Clinical Oncology</em> (2015-03-20) <a href="https://doi.org/f7f87r">https://doi.org/f7f87r</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1200/jco.2014.58.2007">10.1200/jco.2014.58.2007</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25667274">25667274</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4356709">PMC4356709</a></div></div>
</div>
<div id="ref-1B2Ue9gnv" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">64. </div><div class="csl-right-inline"><strong>Machine Learning in Genomic Medicine: A Review of Computational Problems and Data Sets</strong> <div class="csl-block">Michael KK Leung, Andrew Delong, Babak Alipanahi, Brendan J Frey</div> <em>Proceedings of the IEEE</em> (2016-01) <a href="https://doi.org/f75grb">https://doi.org/f75grb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1109/jproc.2015.2494198">10.1109/jproc.2015.2494198</a></div></div>
</div>
<div id="ref-JPGgWnoo" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">65. </div><div class="csl-right-inline"><strong>Diet Networks: Thin Parameters for Fat Genomics</strong> <div class="csl-block">Adriana Romero, Pierre Luc Carrier, Akram Erraqabi, Tristan Sylvain, Alex Auvolat, Etienne Dejoie, Marc-André Legault, Marie-Pierre Dubé, Julie G Hussin, Yoshua Bengio</div> <em>arXiv</em> (2017-03-21) <a href="https://arxiv.org/abs/1611.09340">https://arxiv.org/abs/1611.09340</a></div>
</div>
<div id="ref-lt4BNUoG" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">66. </div><div class="csl-right-inline"><strong>ImageNet: A large-scale hierarchical image database</strong> <div class="csl-block">Jia Deng, Wei Dong, Richard Socher, Li-Jia Li, Kai Li, Li Fei-Fei</div> <em>2009 IEEE Conference on Computer Vision and Pattern Recognition</em> (2009-06) <a href="https://doi.org/cvc7xp">https://doi.org/cvc7xp</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1109/cvpr.2009.5206848">10.1109/cvpr.2009.5206848</a></div></div>
</div>
<div id="ref-yJtbkmny" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">67. </div><div class="csl-right-inline"><strong>Learning Word Vectors for Sentiment Analysis</strong> <div class="csl-block">Andrew L Maas, Raymond E Daly, Peter T Pham, Dan Huang, Andrew Y Ng, Christopher Potts</div> <em>Proceedings of the 49th Annual Meeting of the Association for Computational Linguistics: Human Language Technologies</em> (2011-06) <a href="https://aclanthology.org/P11-1015">https://aclanthology.org/P11-1015</a></div>
</div>
<div id="ref-Ki2ij7zE" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">68. </div><div class="csl-right-inline"><strong>Pathway-level information extractor (PLIER) for gene expression data</strong> <div class="csl-block">Weiguang Mao, Elena Zaslavsky, Boris M Hartmann, Stuart C Sealfon, Maria Chikina</div> <em>Nature Methods</em> (2019-06-27) <a href="https://doi.org/gf75g6">https://doi.org/gf75g6</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41592-019-0456-1">10.1038/s41592-019-0456-1</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31249421">31249421</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7262669">PMC7262669</a></div></div>
</div>
<div id="ref-15p5LWIVP" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">69. </div><div class="csl-right-inline"><strong>Gene set enrichment analysis: A knowledge-based approach for interpreting genome-wide expression profiles</strong> <div class="csl-block">Aravind Subramanian, Pablo Tamayo, Vamsi K Mootha, Sayan Mukherjee, Benjamin L Ebert, Michael A Gillette, Amanda Paulovich, Scott L Pomeroy, Todd R Golub, Eric S Lander, Jill P Mesirov</div> <em>Proceedings of the National Academy of Sciences</em> (2005-09-30) <a href="https://doi.org/d4qbh8">https://doi.org/d4qbh8</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1073/pnas.0506580102">10.1073/pnas.0506580102</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/16199517">16199517</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1239896">PMC1239896</a></div></div>
</div>
<div id="ref-14rnBunuZ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">70. </div><div class="csl-right-inline"><strong>MultiPLIER: A Transfer Learning Framework for Transcriptomics Reveals Systemic Features of Rare Disease</strong> <div class="csl-block">Jaclyn N Taroni, Peter C Grayson, Qiwen Hu, Sean Eddy, Matthias Kretzler, Peter A Merkel, Casey S Greene</div> <em>Cell Systems</em> (2019-05) <a href="https://doi.org/gf75g5">https://doi.org/gf75g5</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cels.2019.04.003">10.1016/j.cels.2019.04.003</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31121115">31121115</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6538307">PMC6538307</a></div></div>
</div>
<div id="ref-6SPTvFXq" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">71. </div><div class="csl-right-inline"><strong>Reproducible RNA-seq analysis using recount2</strong> <div class="csl-block">Leonardo Collado-Torres, Abhinav Nellore, Kai Kammers, Shannon E Ellis, Margaret A Taub, Kasper D Hansen, Andrew E Jaffe, Ben Langmead, Jeffrey T Leek</div> <em>Nature Biotechnology</em> (2017-04) <a href="https://doi.org/gf75hp">https://doi.org/gf75hp</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nbt.3838">10.1038/nbt.3838</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28398307">28398307</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6742427">PMC6742427</a></div></div>
</div>
<div id="ref-1Bb8CyeTY" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">72. </div><div class="csl-right-inline"><strong>PASNet: pathway-associated sparse deep neural network for prognosis prediction from high-throughput data</strong> <div class="csl-block">Jie Hao, Youngsoon Kim, Tae-Kyung Kim, Mingon Kang</div> <em>BMC Bioinformatics</em> (2018-12) <a href="https://doi.org/gf75g9">https://doi.org/gf75g9</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12859-018-2500-z">10.1186/s12859-018-2500-z</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30558539">30558539</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6296065">PMC6296065</a></div></div>
</div>
<div id="ref-c2Oc0eIx" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">73. </div><div class="csl-right-inline"><strong>Network-based stratification of tumor mutations</strong> <div class="csl-block">Matan Hofree, John P Shen, Hannah Carter, Andrew Gross, Trey Ideker</div> <em>Nature Methods</em> (2013-09-15) <a href="https://doi.org/gbd257">https://doi.org/gbd257</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nmeth.2651">10.1038/nmeth.2651</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24037242">24037242</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3866081">PMC3866081</a></div></div>
</div>
<div id="ref-17fvHtbrH" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">74. </div><div class="csl-right-inline"><strong>netNMF-sc: Leveraging gene-gene interactions for imputation and dimensionality reduction in single-cell expression analysis</strong> <div class="csl-block">Rebecca Elyanow, Bianca Dumitrascu, Barbara E Engelhardt, Benjamin J Raphael</div> <em>Cold Spring Harbor Laboratory</em> (2019-02-08) <a href="https://doi.org/gf386x">https://doi.org/gf386x</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/544346">10.1101/544346</a></div></div>
</div>
<div id="ref-3VYPTgXw" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">75. </div><div class="csl-right-inline"><strong>COEXPEDIA: exploring biomedical hypotheses via co-expressions associated with medical subject headings (MeSH)</strong> <div class="csl-block">Sunmo Yang, Chan Yeong Kim, Sohyun Hwang, Eiru Kim, Hyojin Kim, Hongseok Shim, Insuk Lee</div> <em>Nucleic Acids Research</em> (2016-09-26) <a href="https://doi.org/f9v38k">https://doi.org/f9v38k</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gkw868">10.1093/nar/gkw868</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27679477">27679477</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5210615">PMC5210615</a></div></div>
</div>
<div id="ref-LLInUBEI" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">76. </div><div class="csl-right-inline"><strong>Improving performance of deep learning models with axiomatic attribution priors and expected gradients</strong> <div class="csl-block">Gabriel Erion, Joseph D Janizek, Pascal Sturmfels, Scott Lundberg, Su-In Lee</div> <em>arXiv</em> (2020-11-12) <a href="https://arxiv.org/abs/1906.10670">https://arxiv.org/abs/1906.10670</a></div>
</div>
<div id="ref-1BbuXJuIl" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">77. </div><div class="csl-right-inline"><strong>A novel network regularized matrix decomposition method to detect mutated cancer genes in tumour samples with inter-patient heterogeneity</strong> <div class="csl-block">Jianing Xi, Ao Li, Minghui Wang</div> <em>Scientific Reports</em> (2017-06-06) <a href="https://doi.org/gcq9j7">https://doi.org/gcq9j7</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41598-017-03141-w">10.1038/s41598-017-03141-w</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28588243">28588243</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5460199">PMC5460199</a></div></div>
</div>
<div id="ref-12cJO5Pse" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">78. </div><div class="csl-right-inline"><strong>PIMKL: Pathway-Induced Multiple Kernel Learning</strong> <div class="csl-block">Matteo Manica, Joris Cadow, Roland Mathis, María Rodríguez Martínez</div> <em>npj Systems Biology and Applications</em> (2019-03-05) <a href="https://doi.org/gf8ck6">https://doi.org/gf8ck6</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41540-019-0086-3">10.1038/s41540-019-0086-3</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30854223">30854223</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6401099">PMC6401099</a></div></div>
</div>
<div id="ref-g8OoyIPj" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">79. </div><div class="csl-right-inline"><strong>Robust phenotype prediction from gene expression data using differential shrinkage of co-regulated genes</strong> <div class="csl-block">Kourosh Zarringhalam, David Degras, Christoph Brockel, Daniel Ziemek</div> <em>Scientific Reports</em> (2018-01-19) <a href="https://doi.org/gcwzdn">https://doi.org/gcwzdn</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41598-018-19635-0">10.1038/s41598-018-19635-0</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29352257">29352257</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5775343">PMC5775343</a></div></div>
</div>
<div id="ref-q9Fhy8eq" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">80. </div><div class="csl-right-inline"><strong>STRING v10: protein–protein interaction networks, integrated over the tree of life</strong> <div class="csl-block">Damian Szklarczyk, Andrea Franceschini, Stefan Wyder, Kristoffer Forslund, Davide Heller, Jaime Huerta-Cepas, Milan Simonovic, Alexander Roth, Alberto Santos, Kalliopi P Tsafou, … Christian von Mering</div> <em>Nucleic Acids Research</em> (2014-10-28) <a href="https://doi.org/f64rfn">https://doi.org/f64rfn</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gku1003">10.1093/nar/gku1003</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25352553">25352553</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4383874">PMC4383874</a></div></div>
</div>
<div id="ref-19wuAzYvo" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">81. </div><div class="csl-right-inline"><strong>A biological network-based regularized artificial neural network model for robust phenotype prediction from gene expression data</strong> <div class="csl-block">Tianyu Kang, Wei Ding, Luoyan Zhang, Daniel Ziemek, Kourosh Zarringhalam</div> <em>BMC Bioinformatics</em> (2017-12) <a href="https://doi.org/gf8cm6">https://doi.org/gf8cm6</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12859-017-1984-2">10.1186/s12859-017-1984-2</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29258445">29258445</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5735940">PMC5735940</a></div></div>
</div>
<div id="ref-owp8L957" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">82. </div><div class="csl-right-inline"><strong>Using neural networks for reducing the dimensions of single-cell RNA-Seq data</strong> <div class="csl-block">Chieh Lin, Siddhartha Jain, Hannah Kim, Ziv Bar-Joseph</div> <em>Nucleic Acids Research</em> (2017-07-31) <a href="https://doi.org/gcnzb7">https://doi.org/gcnzb7</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gkx681">10.1093/nar/gkx681</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28973464">28973464</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5737331">PMC5737331</a></div></div>
</div>
<div id="ref-e4tSAJkK" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">83. </div><div class="csl-right-inline"><strong>Genetic Neural Networks: an artificial neural network architecture for capturing gene expression relationships</strong> <div class="csl-block">Ameen Eetemadi, Ilias Tagkopoulos</div> <em>Bioinformatics</em> (2018-11-19) <a href="https://doi.org/gfnks6">https://doi.org/gfnks6</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/bty945">10.1093/bioinformatics/bty945</a></div></div>
</div>
<div id="ref-PuUYD4zV" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">84. </div><div class="csl-right-inline"><strong>Nonparametric pathway-based regression models for analysis of genomic data</strong> <div class="csl-block">Z Wei, H Li</div> <em>Biostatistics</em> (2006-06-13) <a href="https://doi.org/fdmgqm">https://doi.org/fdmgqm</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/biostatistics/kxl007">10.1093/biostatistics/kxl007</a></div></div>
</div>
<div id="ref-MPgqEKuv" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">85. </div><div class="csl-right-inline"><strong>Network-constrained regularization and variable selection for analysis of genomic data</strong> <div class="csl-block">Caiyan Li, Hongzhe Li</div> <em>Bioinformatics</em> (2008-03-01) <a href="https://doi.org/fk8n4b">https://doi.org/fk8n4b</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btn081">10.1093/bioinformatics/btn081</a></div></div>
</div>
<div id="ref-1AuHIuXiR" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">86. </div><div class="csl-right-inline"><strong>&lt;i&gt;netReg&lt;/i&gt;: network-regularized linear models for biological association studies</strong> <div class="csl-block">Simon Dirmeier, Christiane Fuchs, Nikola S Mueller, Fabian J Theis</div> <em>Bioinformatics</em> (2017-10-25) <a href="https://doi.org/gcg9xq">https://doi.org/gcg9xq</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btx677">10.1093/bioinformatics/btx677</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29077797">29077797</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6030897">PMC6030897</a></div></div>
</div>
<div id="ref-9SBhyy2x" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">87. </div><div class="csl-right-inline"><strong>Graph-regularized dual Lasso for robust eQTL mapping</strong> <div class="csl-block">Wei Cheng, Xiang Zhang, Zhishan Guo, Yu Shi, Wei Wang</div> <em>Bioinformatics</em> (2014-06-11) <a href="https://doi.org/f58j6m">https://doi.org/f58j6m</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btu293">10.1093/bioinformatics/btu293</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24931977">24931977</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4058913">PMC4058913</a></div></div>
</div>
<div id="ref-m7CdzEUG" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">88. </div><div class="csl-right-inline"><strong>Integrative analysis of genetical genomics data incorporating network structures</strong> <div class="csl-block">Bin Gao, Xu Liu, Hongzhe Li, Yuehua Cui</div> <em>Biometrics</em> (2019-04-29) <a href="https://doi.org/gf8f9q">https://doi.org/gf8f9q</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1111/biom.13072">10.1111/biom.13072</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31009063">31009063</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6810723">PMC6810723</a></div></div>
</div>
<div id="ref-eH3LaU5K" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">89. </div><div class="csl-right-inline"><strong>The Gene Ontology Resource: 20 years and still GOing strong</strong> <em>Nucleic Acids Research</em> (2018-11-05) <a href="https://doi.org/gf63mb">https://doi.org/gf63mb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gky1055">10.1093/nar/gky1055</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30395331">30395331</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6323945">PMC6323945</a></div></div>
</div>
<div id="ref-qQP20moO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">90. </div><div class="csl-right-inline"><strong>Using deep learning to model the hierarchical structure and function of a cell</strong> <div class="csl-block">Jianzhu Ma, Michael Ku Yu, Samson Fong, Keiichiro Ono, Eric Sage, Barry Demchak, Roded Sharan, Trey Ideker</div> <em>Nature Methods</em> (2018-03-05) <a href="https://doi.org/gc46jp">https://doi.org/gc46jp</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nmeth.4627">10.1038/nmeth.4627</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29505029">29505029</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5882547">PMC5882547</a></div></div>
</div>
<div id="ref-TIQTmEOG" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">91. </div><div class="csl-right-inline"><strong>DeepGO: predicting protein functions from sequence and interactions using a deep ontology-aware classifier</strong> <div class="csl-block">Maxat Kulmanov, Mohammed Asif Khan, Robert Hoehndorf</div> <em>Bioinformatics</em> (2017-10-03) <a href="https://doi.org/gc3nb8">https://doi.org/gc3nb8</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btx624">10.1093/bioinformatics/btx624</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29028931">29028931</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5860606">PMC5860606</a></div></div>
</div>
<div id="ref-Cf5duPBD" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">92. </div><div class="csl-right-inline"><strong>DeepGOPlus: improved protein function prediction from sequence</strong> <div class="csl-block">Maxat Kulmanov, Robert Hoehndorf</div> <em>Bioinformatics</em> (2019-07-27) <a href="https://doi.org/gf84d8">https://doi.org/gf84d8</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btz595">10.1093/bioinformatics/btz595</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31350877">31350877</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9883727">PMC9883727</a></div></div>
</div>
<div id="ref-uQ5z1fAc" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">93. </div><div class="csl-right-inline"><strong>Predictive Modeling of Microbiome Data Using a Phylogeny-Regularized Generalized Linear Mixed Model</strong> <div class="csl-block">Jian Xiao, Li Chen, Stephen Johnson, Yue Yu, Xianyang Zhang, Jun Chen</div> <em>Frontiers in Microbiology</em> (2018-06-27) <a href="https://doi.org/gdtz4z">https://doi.org/gdtz4z</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3389/fmicb.2018.01391">10.3389/fmicb.2018.01391</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29997602">29997602</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6030386">PMC6030386</a></div></div>
</div>
<div id="ref-VvllhwW1" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">94. </div><div class="csl-right-inline"><strong>A Phylogeny-Regularized Sparse Regression Model for Predictive Modeling of Microbial Community Data</strong> <div class="csl-block">Jian Xiao, Li Chen, Yue Yu, Xianyang Zhang, Jun Chen</div> <em>Frontiers in Microbiology</em> (2018-12-19) <a href="https://doi.org/gf8qcc">https://doi.org/gf8qcc</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3389/fmicb.2018.03112">10.3389/fmicb.2018.03112</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30619188">30619188</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6305753">PMC6305753</a></div></div>
</div>
<div id="ref-VzWLIpJn" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">95. </div><div class="csl-right-inline"><strong>Tumor Copy Number Deconvolution Integrating Bulk and Single-Cell Sequencing Data</strong> <div class="csl-block">Haoyun Lei, Bochuan Lyu, EMichael Gertz, Alejandro A Schäffer, Xulian Shi, Kui Wu, Guibo Li, Liqin Xu, Yong Hou, Michael Dean, Russell Schwartz</div> <em>Lecture Notes in Computer Science</em> (2019) <a href="https://doi.org/gf8qck">https://doi.org/gf8qck</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/978-3-030-17083-7_11">10.1007/978-3-030-17083-7_11</a></div></div>
</div>
<div id="ref-1Dk2FpWk6" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">96. </div><div class="csl-right-inline"><strong>CYCLOPS reveals human transcriptional rhythms in health and disease</strong> <div class="csl-block">Ron C Anafi, Lauren J Francey, John B Hogenesch, Junhyong Kim</div> <em>Proceedings of the National Academy of Sciences</em> (2017-04-24) <a href="https://doi.org/f9796k">https://doi.org/f9796k</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1073/pnas.1619320114">10.1073/pnas.1619320114</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28439010">28439010</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5441789">PMC5441789</a></div></div>
</div>
<div id="ref-15jrSlkXD" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">97. </div><div class="csl-right-inline"><strong>Circular Nodes in Neural Networks</strong> <div class="csl-block">Michael J Kirby, Rick Miranda</div> <em>Neural Computation</em> (1996-02-15) <a href="https://doi.org/ffcww8">https://doi.org/ffcww8</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1162/neco.1996.8.2.390">10.1162/neco.1996.8.2.390</a></div></div>
</div>
<div id="ref-kKiwlzZq" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">98. </div><div class="csl-right-inline"><strong>Network-based Biased Tree Ensembles (NetBiTE) for Drug Sensitivity Prediction and Drug Sensitivity Biomarker Identification in Cancer</strong> <div class="csl-block">Ali Oskooei, Matteo Manica, Roland Mathis, Maria Rodriguez Martinez</div> <em>arXiv</em> (2019-04-29) <a href="https://arxiv.org/abs/1808.06603">https://arxiv.org/abs/1808.06603</a></div>
</div>
<div id="ref-1CdDn310M" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">99. </div><div class="csl-right-inline"><strong>The ability to classify patients based on gene-expression data varies by algorithm and performance metric</strong> <div class="csl-block">Stephen R Piccolo, Avery Mecham, Nathan P Golightly, Jérémie L Johnson, Dustin B Miller</div> <em>PLOS Computational Biology</em> (2022-03-11) <a href="https://doi.org/gr43qd">https://doi.org/gr43qd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pcbi.1009926">10.1371/journal.pcbi.1009926</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/35275931">35275931</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8942277">PMC8942277</a></div></div>
</div>
<div id="ref-bOvwnoRq" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">100. </div><div class="csl-right-inline"><strong>Supervised learning is an accurate method for network-based gene classification</strong> <div class="csl-block">Renming Liu, Christopher A Mancuso, Anna Yannakopoulos, Kayla A Johnson, Arjun Krishnan</div> <em>Bioinformatics</em> (2020-04-14) <a href="https://doi.org/gmvnfc">https://doi.org/gmvnfc</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btaa150">10.1093/bioinformatics/btaa150</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32129827">32129827</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7267831">PMC7267831</a></div></div>
</div>
<div id="ref-lnK82Ey6" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">101. </div><div class="csl-right-inline"><strong>Supervised Risk Predictor of Breast Cancer Based on Intrinsic Subtypes</strong> <div class="csl-block">Joel S Parker, Michael Mullins, Maggie CU Cheang, Samuel Leung, David Voduc, Tammi Vickery, Sherri Davies, Christiane Fauron, Xiaping He, Zhiyuan Hu, … Philip S Bernard</div> <em>Journal of Clinical Oncology</em> (2009-03-10) <a href="https://doi.org/c2688w">https://doi.org/c2688w</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1200/jco.2008.18.1370">10.1200/jco.2008.18.1370</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/19204204">19204204</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2667820">PMC2667820</a></div></div>
</div>
<div id="ref-OW5P00JC" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">102. </div><div class="csl-right-inline"><strong>Prediction of adjuvant chemotherapy benefit in endocrine responsive, early breast cancer using multigene assays</strong> <div class="csl-block">Kathy S Albain, Soonmyung Paik, Laura van't Veer</div> <em>The Breast</em> (2009-10) <a href="https://doi.org/bp4rtw">https://doi.org/bp4rtw</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/s0960-9776(09)70290-5">10.1016/s0960-9776(09)70290-5</a></div></div>
</div>
<div id="ref-43wsMmMv" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">103. </div><div class="csl-right-inline"><strong>Scikit-learn: Machine Learning in Python</strong> <div class="csl-block">Fabian Pedregosa, Gaël Varoquaux, Alexandre Gramfort, Vincent Michel, Bertrand Thirion, Olivier Grisel, Mathieu Blondel, Peter Prettenhofer, Ron Weiss, Vincent Dubourg, … Édouard Duchesnay</div> <em>Journal of Machine Learning Research</em> (2011) <a href="http://jmlr.org/papers/v12/pedregosa11a.html">http://jmlr.org/papers/v12/pedregosa11a.html</a></div>
</div>
<div id="ref-Mm8xIDqq" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">104. </div><div class="csl-right-inline"><strong>LIBLINEAR: A Library for Large Linear Classification</strong> <div class="csl-block">Rong-En Fan, Kai-Wei Chang, Cho-Jui Hsieh, Xiang-Rui Wang, Chih-Jen Lin</div> <em>Journal of Machine Learning Research</em> (2008) <a href="http://jmlr.org/papers/v9/fan08a.html">http://jmlr.org/papers/v9/fan08a.html</a></div>
</div>
<div id="ref-x24AA6cY" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">105. </div><div class="csl-right-inline"><strong>Online Learning and Stochastic Approximations</strong> <div class="csl-block">Leon Bottou</div> (1998) <a href="https://wiki.eecs.yorku.ca/course_archive/2012-13/F/6328/_media/bottou-onlinelearning-98.pdf">https://wiki.eecs.yorku.ca/course_archive/2012-13/F/6328/_media/bottou-onlinelearning-98.pdf</a></div>
</div>
<div id="ref-809OyWlC" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">106. </div><div class="csl-right-inline"><strong>Cancer Genome Landscapes</strong> <div class="csl-block">B Vogelstein, N Papadopoulos, VE Velculescu, S Zhou, LA Diaz, KW Kinzler</div> <em>Science</em> (2013-03-28) <a href="https://doi.org/6rg">https://doi.org/6rg</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1126/science.1235122">10.1126/science.1235122</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23539594">23539594</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3749880">PMC3749880</a></div></div>
</div>
<div id="ref-cDnF3a5Q" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">107. </div><div class="csl-right-inline"><strong>Scalable Open Science Approach for Mutation Calling of Tumor Exomes Using Multiple Genomic Pipelines</strong> <div class="csl-block">Kyle Ellrott, Matthew H Bailey, Gordon Saksena, Kyle R Covington, Cyriac Kandoth, Chip Stewart, Julian Hess, Singer Ma, Kami E Chiotti, Michael McLellan, … Armaz Mariamidze</div> <em>Cell Systems</em> (2018-03) <a href="https://doi.org/gf9twn">https://doi.org/gf9twn</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cels.2018.03.002">10.1016/j.cels.2018.03.002</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29596782">29596782</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6075717">PMC6075717</a></div></div>
</div>
<div id="ref-rDV1hhfF" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">108. </div><div class="csl-right-inline"><strong>GISTIC2.0 facilitates sensitive and confident localization of the targets of focal somatic copy-number alteration in human cancers</strong> <div class="csl-block">Craig H Mermel, Steven E Schumacher, Barbara Hill, Matthew L Meyerson, Rameen Beroukhim, Gad Getz</div> <em>Genome Biology</em> (2011-04) <a href="https://doi.org/dzhjqh">https://doi.org/dzhjqh</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/gb-2011-12-4-r41">10.1186/gb-2011-12-4-r41</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21527027">21527027</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3218867">PMC3218867</a></div></div>
</div>
<div id="ref-cAxtCrBP" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">109. </div><div class="csl-right-inline"><strong>Widespread redundancy in -omics profiles of cancer mutation states</strong> <div class="csl-block">Jake Crawford, Brock C Christensen, Maria Chikina, Casey S Greene</div> <em>Genome Biology</em> (2022-06-27) <a href="https://doi.org/gqfqnm">https://doi.org/gqfqnm</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-022-02705-y">10.1186/s13059-022-02705-y</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/35761387">35761387</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9238138">PMC9238138</a></div></div>
</div>
<div id="ref-kX2zf6UE" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">110. </div><div class="csl-right-inline"><strong>Regression Shrinkage and Selection Via the Lasso</strong> <div class="csl-block">Robert Tibshirani</div> <em>Journal of the Royal Statistical Society: Series B (Methodological)</em> (1996-01) <a href="https://doi.org/gfn45m">https://doi.org/gfn45m</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1111/j.2517-6161.1996.tb02080.x">10.1111/j.2517-6161.1996.tb02080.x</a></div></div>
</div>
<div id="ref-kwgYU3iP" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">111. </div><div class="csl-right-inline"><strong>The Benefits of Implicit Regularization from SGD in Least Squares Problems</strong> <div class="csl-block">Difan Zou, Jingfeng Wu, Vladimir Braverman, Quanquan Gu, Dean P Foster, Sham M Kakade</div> <em>arXiv</em> (2022-07-12) <a href="https://arxiv.org/abs/2108.04552">https://arxiv.org/abs/2108.04552</a></div>
</div>
<div id="ref-DIONdBGr" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">112. </div><div class="csl-right-inline"><strong>Can Implicit Bias Explain Generalization? Stochastic Convex Optimization as a Case Study</strong> <div class="csl-block">Assaf Dauber, Meir Feder, Tomer Koren, Roi Livni</div> <em>arXiv</em> (2020-12-23) <a href="https://arxiv.org/abs/2003.06152">https://arxiv.org/abs/2003.06152</a></div>
</div>
<div id="ref-kajx4KkT" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">113. </div><div class="csl-right-inline"><strong>Benign Overfitting of Constant-Stepsize SGD for Linear Regression</strong> <div class="csl-block">Difan Zou, Jingfeng Wu, Vladimir Braverman, Quanquan Gu, Sham Kakade</div> <em>Proceedings of Thirty Fourth Conference on Learning Theory</em> (2021-07-21) <a href="https://proceedings.mlr.press/v134/zou21a.html">https://proceedings.mlr.press/v134/zou21a.html</a></div>
</div>
<div id="ref-1HdGWY10w" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">114. </div><div class="csl-right-inline"><strong>Understanding deep learning (still) requires rethinking generalization</strong> <div class="csl-block">Chiyuan Zhang, Samy Bengio, Moritz Hardt, Benjamin Recht, Oriol Vinyals</div> <em>Communications of the ACM</em> (2021-02-22) <a href="https://doi.org/gh57fd">https://doi.org/gh57fd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1145/3446776">10.1145/3446776</a></div></div>
</div>
<div id="ref-jhZXoLtp" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">115. </div><div class="csl-right-inline"><strong>Benign overfitting in linear regression</strong> <div class="csl-block">Peter L Bartlett, Philip M Long, Gábor Lugosi, Alexander Tsigler</div> <em>Proceedings of the National Academy of Sciences</em> (2020-04-24) <a href="https://doi.org/gjgsxq">https://doi.org/gjgsxq</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1073/pnas.1907378117">10.1073/pnas.1907378117</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32332161">32332161</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7720150">PMC7720150</a></div></div>
</div>
<div id="ref-1Cb94HrPD" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">116. </div><div class="csl-right-inline"><strong>Understanding deep learning requires rethinking generalization</strong> <div class="csl-block">Chiyuan Zhang, Samy Bengio, Moritz Hardt, Benjamin Recht, Oriol Vinyals</div> <em>arXiv</em> (2017-02-28) <a href="https://arxiv.org/abs/1611.03530">https://arxiv.org/abs/1611.03530</a></div>
</div>
<div id="ref-10aVUDFmA" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">117. </div><div class="csl-right-inline"><strong>Regularization Paths for Generalized Linear Models via Coordinate Descent</strong> <div class="csl-block">Jerome Friedman, Trevor Hastie, Robert Tibshirani</div> <em>Journal of Statistical Software</em> (2010) <a href="https://doi.org/bb3d">https://doi.org/bb3d</a> <div class="csl-block">DOI: <a href="https://doi.org/10.18637/jss.v033.i01">10.18637/jss.v033.i01</a></div></div>
</div>
<div id="ref-lhYg2Yc3" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">118. </div><div class="csl-right-inline"><strong>Evaluating cancer cell line and patient-derived xenograft recapitulation of tumor and non-diseased tissue gene expression profiles&lt;i&gt;in silico&lt;/i&gt;</strong> <div class="csl-block">Avery S Williams, Elizabeth J Wilk, Jennifer L Fisher, Brittany N Lasseigne</div> <em>Cold Spring Harbor Laboratory</em> (2023-04-13) <a href="https://doi.org/gr6jr4">https://doi.org/gr6jr4</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2023.04.11.536431">10.1101/2023.04.11.536431</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37090499">37090499</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10120639">PMC10120639</a></div></div>
</div>
<div id="ref-1CcaMqQZg" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">119. </div><div class="csl-right-inline"><strong>Gene expression has more power for predicting &lt;i&gt;in vitro&lt;/i&gt; cancer cell vulnerabilities than genomics</strong> <div class="csl-block">Joshua M Dempster, John M Krill-Burger, James M McFarland, Allison Warren, Jesse S Boehm, Francisca Vazquez, William C Hahn, Todd R Golub, Aviad Tsherniak</div> <em>Cold Spring Harbor Laboratory</em> (2020-02-24) <a href="https://doi.org/ghczbr">https://doi.org/ghczbr</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2020.02.21.959627">10.1101/2020.02.21.959627</a></div></div>
</div>
<div id="ref-AK17eOgD" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">120. </div><div class="csl-right-inline"><strong>The Cancer Genome Atlas Pan-Cancer analysis project</strong> <div class="csl-block">John N Weinstein, Eric A Collisson, Gordon B Mills, Kenna RMills Shaw, Brad A Ozenberger, Kyle Ellrott, Ilya Shmulevich, Chris Sander, Joshua M Stuart</div> <em>Nature Genetics</em> (2013-09-26) <a href="https://doi.org/f3nt5c">https://doi.org/f3nt5c</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/ng.2764">10.1038/ng.2764</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24071849">24071849</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3919969">PMC3919969</a></div></div>
</div>
<div id="ref-YuJbg3zO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">121. </div><div class="csl-right-inline"><strong>Open collaborative writing with Manubot</strong> <div class="csl-block">Daniel S Himmelstein, Vincent Rubinetti, David R Slochower, Dongbo Hu, Venkat S Malladi, Casey S Greene, Anthony Gitter</div> <em>PLOS Computational Biology</em> (2019-06-24) <a href="https://doi.org/c7np">https://doi.org/c7np</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pcbi.1007128">10.1371/journal.pcbi.1007128</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31233491">31233491</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6611653">PMC6611653</a></div></div>
</div>
<div id="ref-1CHUpIalb" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">122. </div><div class="csl-right-inline"><strong>Systematic identification of mutations and copy number alterations associated with cancer patient prognosis</strong> <div class="csl-block">Joan C Smith, Jason M Sheltzer</div> <em>eLife</em> (2018-12-11) <a href="https://doi.org/gf4zgg">https://doi.org/gf4zgg</a> <div class="csl-block">DOI: <a href="https://doi.org/10.7554/elife.39217">10.7554/elife.39217</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30526857">30526857</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6289580">PMC6289580</a></div></div>
</div>
<div id="ref-Lm19TMrZ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">123. </div><div class="csl-right-inline"><strong>Challenges in identifying cancer genes by analysis of exome sequencing data</strong> <div class="csl-block">Matan Hofree, Hannah Carter, Jason F Kreisberg, Sourav Bandyopadhyay, Paul S Mischel, Stephen Friend, Trey Ideker</div> <em>Nature Communications</em> (2016-07-15) <a href="https://doi.org/f8x7t3">https://doi.org/f8x7t3</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/ncomms12096">10.1038/ncomms12096</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27417679">27417679</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4947162">PMC4947162</a></div></div>
</div>
<div id="ref-MnWpJEWE" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">124. </div><div class="csl-right-inline"><strong>Evaluating the evaluation of cancer driver genes</strong> <div class="csl-block">Collin J Tokheim, Nickolas Papadopoulos, Kenneth W Kinzler, Bert Vogelstein, Rachel Karchin</div> <em>Proceedings of the National Academy of Sciences</em> (2016-11-22) <a href="https://doi.org/f9d77w">https://doi.org/f9d77w</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1073/pnas.1616440113">10.1073/pnas.1616440113</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27911828">27911828</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5167163">PMC5167163</a></div></div>
</div>
<div id="ref-YKm7IBTV" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">125. </div><div class="csl-right-inline"><strong>Detailed modeling of positive selection improves detection of cancer driver genes</strong> <div class="csl-block">Siming Zhao, Jun Liu, Pranav Nanga, Yuwen Liu, AErcument Cicek, Nicholas Knoblauch, Chuan He, Matthew Stephens, Xin He</div> <em>Nature Communications</em> (2019-07-30) <a href="https://doi.org/gjmhnn">https://doi.org/gjmhnn</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41467-019-11284-9">10.1038/s41467-019-11284-9</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31363082">31363082</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6667447">PMC6667447</a></div></div>
</div>
<div id="ref-17HuO3ZJW" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">126. </div><div class="csl-right-inline"><strong>Review: Precision medicine and driver mutations: Computational methods, functional assays and conformational principles for interpreting cancer drivers</strong> <div class="csl-block">Ruth Nussinov, Hyunbum Jang, Chung-Jung Tsai, Feixiong Cheng</div> <em>PLOS Computational Biology</em> (2019-03-28) <a href="https://doi.org/gg8jhm">https://doi.org/gg8jhm</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pcbi.1006658">10.1371/journal.pcbi.1006658</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30921324">30921324</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6438456">PMC6438456</a></div></div>
</div>
<div id="ref-Z5JvAfru" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">127. </div><div class="csl-right-inline"><strong>Identification of pan-cancer Ras pathway activation with deep learning</strong> <div class="csl-block">Xiangtao Li, Shaochuan Li, Yunhe Wang, Shixiong Zhang, Ka-Chun Wong</div> <em>Briefings in Bioinformatics</em> (2020-10-30) <a href="https://doi.org/gjmd3p">https://doi.org/gjmd3p</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bib/bbaa258">10.1093/bib/bbaa258</a></div></div>
</div>
<div id="ref-X6oQhIf8" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">128. </div><div class="csl-right-inline"><strong>Genomic and Molecular Landscape of DNA Damage Repair Deficiency across The Cancer Genome Atlas</strong> <div class="csl-block">Theo A Knijnenburg, Linghua Wang, Michael T Zimmermann, Nyasha Chambwe, Galen F Gao, Andrew D Cherniack, Huihui Fan, Hui Shen, Gregory P Way, Casey S Greene, … Armaz Mariamidze</div> <em>Cell Reports</em> (2018-04) <a href="https://doi.org/gfspsc">https://doi.org/gfspsc</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.celrep.2018.03.076">10.1016/j.celrep.2018.03.076</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29617664">29617664</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5961503">PMC5961503</a></div></div>
</div>
<div id="ref-ekDcxjcr" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">129. </div><div class="csl-right-inline"><strong>Systematic interrogation of mutation groupings reveals divergent downstream expression programs within key cancer genes</strong> <div class="csl-block">Michal R Grzadkowski, Hannah Manning, Julia Somers, Emek Demir</div> <em>Cold Spring Harbor Laboratory</em> (2020-06-03) <a href="https://doi.org/gjmd7t">https://doi.org/gjmd7t</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2020.06.02.128850">10.1101/2020.06.02.128850</a></div></div>
</div>
<div id="ref-pLo10xih" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">130. </div><div class="csl-right-inline"><strong>Reverse regression increases power for detecting trans-eQTLs</strong> <div class="csl-block">Saikat Banerjee, Franco L Simonetti, Kira E Detrois, Anubhav Kaphle, Raktim Mitra, Rahul Nagial, Johannes Söding</div> <em>Cold Spring Harbor Laboratory</em> (2020-05-09) <a href="https://doi.org/gjmhd8">https://doi.org/gjmhd8</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2020.05.07.083386">10.1101/2020.05.07.083386</a></div></div>
</div>
<div id="ref-1z8H2EME" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">131. </div><div class="csl-right-inline"><strong>Cancer transcriptome profiling at the juncture of clinical translation</strong> <div class="csl-block">Marcin Cieślik, Arul M Chinnaiyan</div> <em>Nature Reviews Genetics</em> (2017-12-27) <a href="https://doi.org/gcsmnr">https://doi.org/gcsmnr</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nrg.2017.96">10.1038/nrg.2017.96</a></div></div>
</div>
<div id="ref-1Cxnszz2L" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">132. </div><div class="csl-right-inline"><strong>Identification of a CpG Island Methylator Phenotype that Defines a Distinct Subgroup of Glioma</strong> <div class="csl-block">Houtan Noushmehr, Daniel J Weisenberger, Kristin Diefes, Heidi S Phillips, Kanan Pujara, Benjamin P Berman, Fei Pan, Christopher E Pelloski, Erik P Sulman, Krishna P Bhat, … Kenneth Aldape</div> <em>Cancer Cell</em> (2010-05) <a href="https://doi.org/dbtmsd">https://doi.org/dbtmsd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.ccr.2010.03.017">10.1016/j.ccr.2010.03.017</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/20399149">20399149</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2872684">PMC2872684</a></div></div>
</div>
<div id="ref-WN82rY7P" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">133. </div><div class="csl-right-inline"><strong>DNA Methylation, Isocitrate Dehydrogenase Mutation, and Survival in Glioma</strong> <div class="csl-block">Brock C Christensen, Ashley A Smith, Shichun Zheng, Devin C Koestler, EAndres Houseman, Carmen J Marsit, Joseph L Wiemels, Heather H Nelson, Margaret R Karagas, Margaret R Wrensch, … John K Wiencke</div> <em>JNCI: Journal of the National Cancer Institute</em> (2011-01-19) <a href="https://doi.org/bbjqf9">https://doi.org/bbjqf9</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/jnci/djq497">10.1093/jnci/djq497</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21163902">21163902</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3022619">PMC3022619</a></div></div>
</div>
<div id="ref-Sp8RO1w7" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">134. </div><div class="csl-right-inline"><strong>&lt;i&gt;IDH1&lt;/i&gt;and&lt;i&gt;IDH2&lt;/i&gt;Mutations in Gliomas</strong> <div class="csl-block">Hai Yan, DWilliams Parsons, Genglin Jin, Roger McLendon, BAhmed Rasheed, Weishi Yuan, Ivan Kos, Ines Batinic-Haberle, Siân Jones, Gregory J Riggins, … Darell D Bigner</div> <em>New England Journal of Medicine</em> (2009-02-19) <a href="https://doi.org/btz6db">https://doi.org/btz6db</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1056/nejmoa0808710">10.1056/nejmoa0808710</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/19228619">19228619</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2820383">PMC2820383</a></div></div>
</div>
<div id="ref-FYj8JHBX" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">135. </div><div class="csl-right-inline"><strong>IDH mutation impairs histone demethylation and results in a block to cell differentiation</strong> <div class="csl-block">Chao Lu, Patrick S Ward, Gurpreet S Kapoor, Dan Rohle, Sevin Turcan, Omar Abdel-Wahab, Christopher R Edwards, Raya Khanin, Maria E Figueroa, Ari Melnick, … Craig B Thompson</div> <em>Nature</em> (2012-02-15) <a href="https://doi.org/f4msnt">https://doi.org/f4msnt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nature10860">10.1038/nature10860</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22343901">22343901</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3478770">PMC3478770</a></div></div>
</div>
<div id="ref-19sDhpROP" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">136. </div><div class="csl-right-inline"><strong>Connections between TET proteins and aberrant DNA modification in cancer</strong> <div class="csl-block">Yun Huang, Anjana Rao</div> <em>Trends in Genetics</em> (2014-10) <a href="https://doi.org/f6jm7v">https://doi.org/f6jm7v</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.tig.2014.07.005">10.1016/j.tig.2014.07.005</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25132561">25132561</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4337960">PMC4337960</a></div></div>
</div>
<div id="ref-rNm0rAz7" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">137. </div><div class="csl-right-inline"><strong>SETting the Stage for Cancer Development: SETD2 and the Consequences of Lost Methylation</strong> <div class="csl-block">Catherine C Fahey, Ian J Davis</div> <em>Cold Spring Harbor Perspectives in Medicine</em> (2017-02-03) <a href="https://doi.org/gjmfvq">https://doi.org/gjmfvq</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/cshperspect.a026468">10.1101/cshperspect.a026468</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28159833">28159833</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5411680">PMC5411680</a></div></div>
</div>
<div id="ref-18LSSJo3W" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">138. </div><div class="csl-right-inline"><strong>Mechanisms underlying mutational signatures in human cancers</strong> <div class="csl-block">Thomas Helleday, Saeed Eshtad, Serena Nik-Zainal</div> <em>Nature Reviews Genetics</em> (2014-07-01) <a href="https://doi.org/f25gnp">https://doi.org/f25gnp</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nrg3729">10.1038/nrg3729</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24981601">24981601</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6044419">PMC6044419</a></div></div>
</div>
<div id="ref-pWXLXywn" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">139. </div><div class="csl-right-inline"><strong>Quantitative Proteomics of the Cancer Cell Line Encyclopedia</strong> <div class="csl-block">David P Nusinow, John Szpyt, Mahmoud Ghandi, Christopher M Rose, ERobert McDonald III, Marian Kalocsay, Judit Jané-Valbuena, Ellen Gelfand, Devin K Schweppe, Mark Jedrychowski, … Steven P Gygi</div> <em>Cell</em> (2020-01) <a href="https://doi.org/ggxbh5">https://doi.org/ggxbh5</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cell.2019.12.023">10.1016/j.cell.2019.12.023</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31978347">31978347</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7339254">PMC7339254</a></div></div>
</div>
<div id="ref-11yRODwNY" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">140. </div><div class="csl-right-inline"><strong>The repertoire of mutational signatures in human cancer</strong> <div class="csl-block">Ludmil B Alexandrov, Jaegil Kim, Nicholas J Haradhvala, Mi Ni Huang, Alvin Wei Tian Ng, Yang Wu, Arnoud Boot, Kyle R Covington, Dmitry A Gordenin, Erik N Bergstrom, … </div> <em>Nature</em> (2020-02-05) <a href="https://doi.org/ggkfnv">https://doi.org/ggkfnv</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41586-020-1943-3">10.1038/s41586-020-1943-3</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32025018">32025018</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7054213">PMC7054213</a></div></div>
</div>
<div id="ref-UTtkLysq" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">141. </div><div class="csl-right-inline"><strong>Significant associations between driver gene mutations and DNA methylation alterations across many cancer types</strong> <div class="csl-block">Yun-Ching Chen, Valer Gotea, Gennady Margolin, Laura Elnitski</div> <em>PLOS Computational Biology</em> (2017-11-10) <a href="https://doi.org/gchz8h">https://doi.org/gchz8h</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pcbi.1005840">10.1371/journal.pcbi.1005840</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29125844">29125844</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5709060">PMC5709060</a></div></div>
</div>
<div id="ref-YBXY18s4" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">142. </div><div class="csl-right-inline"><strong>A pan-cancer analysis of driver gene mutations, DNA methylation and gene expressions reveals that chromatin remodeling is a major mechanism inducing global changes in cancer epigenomes</strong> <div class="csl-block">Ahrim Youn, Kyung In Kim, Raul Rabadan, Benjamin Tycko, Yufeng Shen, Shuang Wang</div> <em>BMC Medical Genomics</em> (2018-11-06) <a href="https://doi.org/gjmhfb">https://doi.org/gjmhfb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12920-018-0425-z">10.1186/s12920-018-0425-z</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30400878">30400878</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6218985">PMC6218985</a></div></div>
</div>
<div id="ref-pRTkOXGI" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">143. </div><div class="csl-right-inline"><strong>Computational analysis reveals histotype-dependent molecular profile and actionable mutation effects across cancers</strong> <div class="csl-block">Daniel Heim, Grégoire Montavon, Peter Hufnagl, Klaus-Robert Müller, Frederick Klauschen</div> <em>Genome Medicine</em> (2018-11-15) <a href="https://doi.org/gjmhfc">https://doi.org/gjmhfc</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13073-018-0591-9">10.1186/s13073-018-0591-9</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30442178">30442178</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6238410">PMC6238410</a></div></div>
</div>
<div id="ref-ZuUqcptb" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">144. </div><div class="csl-right-inline"><strong>CNAmet: an R package for integrating copy number, methylation and expression data</strong> <div class="csl-block">Riku Louhimo, Sampsa Hautaniemi</div> <em>Bioinformatics</em> (2011-01-12) <a href="https://doi.org/fbq4p2">https://doi.org/fbq4p2</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btr019">10.1093/bioinformatics/btr019</a></div></div>
</div>
<div id="ref-lQ8oalgu" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">145. </div><div class="csl-right-inline"><strong>Impacts of somatic mutations on gene expression: an association perspective</strong> <div class="csl-block">Peilin Jia, Zhongming Zhao</div> <em>Briefings in Bioinformatics</em> (2016-04-28) <a href="https://doi.org/gjnd5b">https://doi.org/gjnd5b</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bib/bbw037">10.1093/bib/bbw037</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27127206">27127206</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5862283">PMC5862283</a></div></div>
</div>
<div id="ref-StwDSCo7" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">146. </div><div class="csl-right-inline"><strong>Inference of patient-specific pathway activities from multi-dimensional cancer genomics data using PARADIGM</strong> <div class="csl-block">Charles J Vaske, Stephen C Benz, JZachary Sanborn, Dent Earl, Christopher Szeto, Jingchun Zhu, David Haussler, Joshua M Stuart</div> <em>Bioinformatics</em> (2010-06-01) <a href="https://doi.org/bcvgjf">https://doi.org/bcvgjf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btq182">10.1093/bioinformatics/btq182</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/20529912">20529912</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2881367">PMC2881367</a></div></div>
</div>
<div id="ref-1BMNB7BTo" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">147. </div><div class="csl-right-inline"><strong>Systematic analysis of somatic mutations impacting gene expression in 12 tumour types</strong> <div class="csl-block">Jiarui Ding, Melissa K McConechy, Hugo M Horlings, Gavin Ha, Fong Chun Chan, Tyler Funnell, Sarah C Mullaly, Jüri Reimand, Ali Bashashati, Gary D Bader, … Sohrab P Shah</div> <em>Nature Communications</em> (2015-10-05) <a href="https://doi.org/f7z86p">https://doi.org/f7z86p</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/ncomms9554">10.1038/ncomms9554</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26436532">26436532</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4600750">PMC4600750</a></div></div>
</div>
<div id="ref-lF0Fr9LL" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">148. </div><div class="csl-right-inline"><strong>Scikit-learn: Machine Learning in Python</strong> <div class="csl-block">Fabian Pedregosa, Gaël Varoquaux, Alexandre Gramfort, Vincent Michel, Bertrand Thirion, Olivier Grisel, Mathieu Blondel, Peter Prettenhofer, Ron Weiss, Vincent Dubourg, … Édouard Duchesnay</div> <em>Journal of Machine Learning Research</em> (2011) <a href="http://jmlr.org/papers/v12/pedregosa11a.html">http://jmlr.org/papers/v12/pedregosa11a.html</a></div>
</div>
<div id="ref-Zg2jmD6p" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">149. </div><div class="csl-right-inline"><strong>A beta-mixture quantile normalization method for correcting probe design bias in Illumina Infinium 450 k DNA methylation data</strong> <div class="csl-block">Andrew E Teschendorff, Francesco Marabita, Matthias Lechner, Thomas Bartlett, Jesper Tegner, David Gomez-Cabrero, Stephan Beck</div> <em>Bioinformatics</em> (2012-11-21) <a href="https://doi.org/f25mvt">https://doi.org/f25mvt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/bts680">10.1093/bioinformatics/bts680</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23175756">23175756</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3546795">PMC3546795</a></div></div>
</div>
<div id="ref-A3VeCeaB" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">150. </div><div class="csl-right-inline"><strong>Comprehensive Characterization of Cancer Driver Genes and Mutations</strong> <div class="csl-block">Matthew H Bailey, Collin Tokheim, Eduard Porta-Pardo, Sohini Sengupta, Denis Bertrand, Amila Weerasinghe, Antonio Colaprico, Michael C Wendl, Jaegil Kim, Brendan Reardon, … Armaz Mariamidze</div> <em>Cell</em> (2018-04) <a href="https://doi.org/gc7r88">https://doi.org/gc7r88</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cell.2018.02.060">10.1016/j.cell.2018.02.060</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29625053">29625053</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6029450">PMC6029450</a></div></div>
</div>
<div id="ref-DxovERyv" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">151. </div><div class="csl-right-inline"><strong>The COSMIC Cancer Gene Census: describing genetic dysfunction across all human cancers</strong> <div class="csl-block">Zbyslaw Sondka, Sally Bamford, Charlotte G Cole, Sari A Ward, Ian Dunham, Simon A Forbes</div> <em>Nature Reviews Cancer</em> (2018-10-06) <a href="https://doi.org/gfb3z4">https://doi.org/gfb3z4</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41568-018-0060-1">10.1038/s41568-018-0060-1</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30293088">30293088</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6450507">PMC6450507</a></div></div>
</div>
<div id="ref-JZNkB8d7" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">152. </div><div class="csl-right-inline"><strong>Regularization and Variable Selection Via the Elastic Net</strong> <div class="csl-block">Hui Zou, Trevor Hastie</div> <em>Journal of the Royal Statistical Society Series B: Statistical Methodology</em> (2005-03-09) <a href="https://doi.org/b8cwwr">https://doi.org/b8cwwr</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1111/j.1467-9868.2005.00503.x">10.1111/j.1467-9868.2005.00503.x</a></div></div>
</div>
<div id="ref-WTSzUd5Y" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">153. </div><div class="csl-right-inline"><strong>An introduction to ROC analysis</strong> <div class="csl-block">Tom Fawcett</div> <em>Pattern Recognition Letters</em> (2006-06) <a href="https://doi.org/bpsghb">https://doi.org/bpsghb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.patrec.2005.10.010">10.1016/j.patrec.2005.10.010</a></div></div>
</div>
<div id="ref-pazxvNcz" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">154. </div><div class="csl-right-inline"><strong>A critical investigation of recall and precision as measures of retrieval system performance</strong> <div class="csl-block">Vijay Raghavan, Peter Bollmann, Gwang S Jung</div> <em>ACM Transactions on Information Systems</em> (1989-07) <a href="https://doi.org/bg4tps">https://doi.org/bg4tps</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1145/65943.65945">10.1145/65943.65945</a></div></div>
</div>
<div id="ref-LB4mDas6" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">155. </div><div class="csl-right-inline"><strong>The Precision-Recall Plot Is More Informative than the ROC Plot When Evaluating Binary Classifiers on Imbalanced Datasets</strong> <div class="csl-block">Takaya Saito, Marc Rehmsmeier</div> <em>PLOS ONE</em> (2015-03-04) <a href="https://doi.org/f69237">https://doi.org/f69237</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pone.0118432">10.1371/journal.pone.0118432</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25738806">25738806</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4349800">PMC4349800</a></div></div>
</div>
<div id="ref-lWf6zHNO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">156. </div><div class="csl-right-inline"><strong>The MCC-F1 curve: a performance evaluation technique for binary classification</strong> <div class="csl-block">Chang Cao, Davide Chicco, Michael M Hoffman</div> <em>arXiv</em> (2020-06-23) <a href="https://arxiv.org/abs/2006.11278">https://arxiv.org/abs/2006.11278</a></div>
</div>
<div id="ref-CZHe0A4l" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">157. </div><div class="csl-right-inline"><strong>Genome-wide identification and analysis of prognostic features in human cancers</strong> <div class="csl-block">Joan C Smith, Jason M Sheltzer</div> <em>Cold Spring Harbor Laboratory</em> (2021-06-01) <a href="https://doi.org/gmqfqt">https://doi.org/gmqfqt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2021.06.01.446243">10.1101/2021.06.01.446243</a></div></div>
</div>
<div id="ref-wiw7Q7M7" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">158. </div><div class="csl-right-inline"><strong>scikit-survival: A Library for Time-to-Event Analysis Built on Top of scikit-learn</strong> <div class="csl-block">Sebastian Pölsterl</div> <em>Journal of Machine Learning Research</em> (2020) <a href="http://jmlr.org/papers/v21/20-729.html">http://jmlr.org/papers/v21/20-729.html</a></div>
</div>
<div id="ref-c6d3lKFX" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">159. </div><div class="csl-right-inline"><strong>Adam: A Method for Stochastic Optimization</strong> <div class="csl-block">Diederik P Kingma, Jimmy Ba</div> <em>arXiv</em> (2017-01-31) <a href="https://arxiv.org/abs/1412.6980">https://arxiv.org/abs/1412.6980</a></div>
</div>
<div id="ref-KGJwKXkD" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">160. </div><div class="csl-right-inline"><strong>Integrated genomic characterization of IDH1-mutant glioma malignant progression</strong> <div class="csl-block">Hanwen Bai, Akdes Serin Harmancı, EZeynep Erson-Omay, Jie Li, Süleyman Coşkun, Matthias Simon, Boris Krischek, Koray Özduman, SBülent Omay, Eric A Sorensen, … Murat Günel</div> <em>Nature Genetics</em> (2015-11-30) <a href="https://doi.org/f895hx">https://doi.org/f895hx</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/ng.3457">10.1038/ng.3457</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26618343">26618343</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4829945">PMC4829945</a></div></div>
</div>
<div id="ref-uBEARk2" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">161. </div><div class="csl-right-inline"><strong>Identification and validation of an ERBB2 gene expression signature in breast cancers</strong> <div class="csl-block">François Bertucci, Nathalie Borie, Christophe Ginestier, Agnès Groulet, Emmanuelle Charafe-Jauffret, José Adélaïde, Jeannine Geneix, Loïc Bachelart, Pascal Finetti, Alane Koki, … Daniel Birnbaum</div> <em>Oncogene</em> (2004-01-26) <a href="https://doi.org/dq9kvk">https://doi.org/dq9kvk</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/sj.onc.1207361">10.1038/sj.onc.1207361</a></div></div>
</div>
<div id="ref-1c8GokzT" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">162. </div><div class="csl-right-inline"><strong>Pan-Cancer Landscape and Analysis of ERBB2 Mutations Identifies Poziotinib as a Clinically Active Inhibitor and Enhancer of T-DM1 Activity</strong> <div class="csl-block">Jacqulyne P Robichaux, Yasir Y Elamin, RSK Vijayan, Monique B Nilsson, Lemei Hu, Junqin He, Fahao Zhang, Marlese Pisegna, Alissa Poteete, Huiying Sun, … John V Heymach</div> <em>Cancer Cell</em> (2019-10) <a href="https://doi.org/ggcv27">https://doi.org/ggcv27</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.ccell.2019.09.001">10.1016/j.ccell.2019.09.001</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31588020">31588020</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6944069">PMC6944069</a></div></div>
</div>
<div id="ref-fdVKgtEF" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">163. </div><div class="csl-right-inline"><strong>Shaping the cellular landscape with Set2/SETD2 methylation</strong> <div class="csl-block">Stephen L McDaniel, Brian D Strahl</div> <em>Cellular and Molecular Life Sciences</em> (2017-04-06) <a href="https://doi.org/gbrd9b">https://doi.org/gbrd9b</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/s00018-017-2517-x">10.1007/s00018-017-2517-x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28386724">28386724</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5545052">PMC5545052</a></div></div>
</div>
<div id="ref-rBwmNJtL" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">164. </div><div class="csl-right-inline"><strong>TCPA: a resource for cancer functional proteomics data</strong> <div class="csl-block">Jun Li, Yiling Lu, Rehan Akbani, Zhenlin Ju, Paul L Roebuck, Wenbin Liu, Ji-Yeon Yang, Bradley M Broom, Roeland GW Verhaak, David W Kane, … Han Liang</div> <em>Nature Methods</em> (2013-09-15) <a href="https://doi.org/gffkjm">https://doi.org/gffkjm</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nmeth.2650">10.1038/nmeth.2650</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24037243">24037243</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4076789">PMC4076789</a></div></div>
</div>
<div id="ref-23jjG6I6" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">165. </div><div class="csl-right-inline"><strong>Patterns of somatic structural variation in human cancer genomes</strong> <div class="csl-block">Yilong Li, Nicola D Roberts, Jeremiah A Wala, Ofer Shapira, Steven E Schumacher, Kiran Kumar, Ekta Khurana, Sebastian Waszak, Jan O Korbel, James E Haber, … </div> <em>Nature</em> (2020-02-05) <a href="https://doi.org/ggkfnw">https://doi.org/ggkfnw</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41586-019-1913-9">10.1038/s41586-019-1913-9</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32025012">32025012</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7025897">PMC7025897</a></div></div>
</div>
<div id="ref-HfPt51y9" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">166. </div><div class="csl-right-inline"><strong>Recruitment of KMT2C/MLL3 to DNA Damage Sites Mediates DNA Damage Responses and Regulates PARP Inhibitor Sensitivity in Cancer</strong> <div class="csl-block">Antao Chang, Liang Liu, Justin M Ashby, Dan Wu, Yanan Chen, Stacey S O'Neill, Shan Huang, Juan Wang, Guanwen Wang, Dongmei Cheng, … Peiqing Sun</div> <em>Cancer Research</em> (2021-04-14) <a href="https://doi.org/gpt2z4">https://doi.org/gpt2z4</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1158/0008-5472.can-21-0688">10.1158/0008-5472.can-21-0688</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33853832">33853832</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8260460">PMC8260460</a></div></div>
</div>
<div id="ref-wOjMFUXP" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">167. </div><div class="csl-right-inline"><strong>MethylNet: an automated and modular deep learning approach for DNA methylation analysis</strong> <div class="csl-block">Joshua J Levy, Alexander J Titus, Curtis L Petersen, Youdinghuan Chen, Lucas A Salas, Brock C Christensen</div> <em>BMC Bioinformatics</em> (2020-03-17) <a href="https://doi.org/ggxvrz">https://doi.org/ggxvrz</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12859-020-3443-8">10.1186/s12859-020-3443-8</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32183722">32183722</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7076991">PMC7076991</a></div></div>
</div>
<div id="ref-1HfJnooMx" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">168. </div><div class="csl-right-inline"><strong>Predicting Drug Response and Synergy Using a Deep Learning Model of Human Cancer Cells</strong> <div class="csl-block">Brent M Kuenzi, Jisoo Park, Samson H Fong, Kyle S Sanchez, John Lee, Jason F Kreisberg, Jianzhu Ma, Trey Ideker</div> <em>Cancer Cell</em> (2020-11) <a href="https://doi.org/gh7z2n">https://doi.org/gh7z2n</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.ccell.2020.09.014">10.1016/j.ccell.2020.09.014</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33096023">33096023</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7737474">PMC7737474</a></div></div>
</div>
<div id="ref-7kIfUrlF" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">169. </div><div class="csl-right-inline"><strong>Using Biological Constraints to Improve Prediction in Precision Oncology</strong> <div class="csl-block">Mohamed Omar, Wikum Dinalankara, Lotte Mulder, Tendai Coady, Claudio Zanettini, Eddie Luidy Imada, Laurent Younes, Donald Geman, Luigi Marchionni</div> <em>Cold Spring Harbor Laboratory</em> (2021-05-27) <a href="https://doi.org/gkcc43">https://doi.org/gkcc43</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2021.05.25.445604">10.1101/2021.05.25.445604</a></div></div>
</div>
<div id="ref-eoXXrVpl" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">170. </div><div class="csl-right-inline"><strong>Reliable Identification of Genomic Variants from RNA-Seq Data</strong> <div class="csl-block">Robert Piskol, Gokul Ramaswami, Jin Billy Li</div> <em>The American Journal of Human Genetics</em> (2013-10) <a href="https://doi.org/f5fk6v">https://doi.org/f5fk6v</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.ajhg.2013.08.008">10.1016/j.ajhg.2013.08.008</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24075185">24075185</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3791257">PMC3791257</a></div></div>
</div>
<div id="ref-gROOu6O4" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">171. </div><div class="csl-right-inline"><strong>Detection and benchmarking of somatic mutations in cancer genomes using RNA-seq data</strong> <div class="csl-block">Alexandre Coudray, Anna M Battenhouse, Philipp Bucher, Vishwanath R Iyer</div> <em>PeerJ</em> (2018-07-31) <a href="https://doi.org/gd2f3q">https://doi.org/gd2f3q</a> <div class="csl-block">DOI: <a href="https://doi.org/10.7717/peerj.5362">10.7717/peerj.5362</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30083469">30083469</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6074801">PMC6074801</a></div></div>
</div>
<div id="ref-1AeRgHv3L" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">172. </div><div class="csl-right-inline"><strong>Principles and methods of integrative genomic analyses in cancer</strong> <div class="csl-block">Vessela N Kristensen, Ole Christian Lingjærde, Hege G Russnes, Hans Kristian M Vollan, Arnoldo Frigessi, Anne-Lise Børresen-Dale</div> <em>Nature Reviews Cancer</em> (2014-04-24) <a href="https://doi.org/gf66jv">https://doi.org/gf66jv</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nrc3721">10.1038/nrc3721</a></div></div>
</div>
<div id="ref-tMmErZMX" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">173. </div><div class="csl-right-inline"><strong>Predicting censored survival data based on the interactions between meta-dimensional omics data in breast cancer</strong> <div class="csl-block">Dokyoon Kim, Ruowang Li, Scott M Dudek, Marylyn D Ritchie</div> <em>Journal of Biomedical Informatics</em> (2015-08) <a href="https://doi.org/f7n49h">https://doi.org/f7n49h</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.jbi.2015.05.019">10.1016/j.jbi.2015.05.019</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26048077">26048077</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4550096">PMC4550096</a></div></div>
</div>
<div id="ref-4X1N4DJa" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">174. </div><div class="csl-right-inline"><strong>Combining DNA methylation and RNA sequencing data of cancer for supervised knowledge extraction</strong> <div class="csl-block">Eleonora Cappelli, Giovanni Felici, Emanuel Weitschek</div> <em>BioData Mining</em> (2018-10-25) <a href="https://doi.org/gkcdbm">https://doi.org/gkcdbm</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13040-018-0184-6">10.1186/s13040-018-0184-6</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30386434">30386434</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6203208">PMC6203208</a></div></div>
</div>
<div id="ref-llajEDiU" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">175. </div><div class="csl-right-inline"><strong>Accurate cancer phenotype prediction with AKLIMATE, a stacked kernel learner integrating multimodal genomic data and pathway knowledge</strong> <div class="csl-block">Vladislav Uzunangelov, Christopher K Wong, Joshua M Stuart</div> <em>PLOS Computational Biology</em> (2021-04-16) <a href="https://doi.org/gkcdbn">https://doi.org/gkcdbn</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pcbi.1008878">10.1371/journal.pcbi.1008878</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33861732">33861732</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8081343">PMC8081343</a></div></div>
</div>
<div id="ref-gIhNaGUg" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">176. </div><div class="csl-right-inline"><strong>Methods for biological data integration: perspectives and challenges</strong> <div class="csl-block">Vladimir Gligorijević, Nataša Pržulj</div> <em>Journal of The Royal Society Interface</em> (2015-11) <a href="https://doi.org/bdzp">https://doi.org/bdzp</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1098/rsif.2015.0571">10.1098/rsif.2015.0571</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26490630">26490630</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4685837">PMC4685837</a></div></div>
</div>
<div id="ref-NfKILi8i" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">177. </div><div class="csl-right-inline"><strong>Machine learning for integrating data in biology and medicine: Principles, practice, and opportunities</strong> <div class="csl-block">Marinka Zitnik, Francis Nguyen, Bo Wang, Jure Leskovec, Anna Goldenberg, Michael M Hoffman</div> <em>Information Fusion</em> (2019-10) <a href="https://doi.org/gf7rj8">https://doi.org/gf7rj8</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.inffus.2018.09.012">10.1016/j.inffus.2018.09.012</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30467459">30467459</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6242341">PMC6242341</a></div></div>
</div>
<div id="ref-1B2CloRMX" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">178. </div><div class="csl-right-inline"><strong>Integrated phosphoproteomics and transcriptional classifiers reveal hidden RAS signaling dynamics in multiple myeloma</strong> <div class="csl-block">Yu-Hsiu T Lin, Gregory P Way, Benjamin G Barwick, Margarette C Mariano, Makeba Marcoulis, Ian D Ferguson, Christoph Driessen, Lawrence H Boise, Casey S Greene, Arun P Wiita</div> <em>Blood Advances</em> (2019-10-29) <a href="https://doi.org/gg7m56">https://doi.org/gg7m56</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1182/bloodadvances.2019000303">10.1182/bloodadvances.2019000303</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31698452">31698452</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6855123">PMC6855123</a></div></div>
</div>
<div id="ref-2AuD4Mwu" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">179. </div><div class="csl-right-inline"><strong>Comprehensive transcriptomic analysis of cell lines as models of primary tumors across 22 tumor types</strong> <div class="csl-block">K Yu, B Chen, D Aran, J Charalel, C Yau, DM Wolf, LJ van ‘t Veer, AJ Butte, T Goldstein, M Sirota</div> <em>Nature Communications</em> (2019-08-08) <a href="https://doi.org/ggh7t7">https://doi.org/ggh7t7</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41467-019-11415-2">10.1038/s41467-019-11415-2</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31395879">31395879</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6687785">PMC6687785</a></div></div>
</div>
<div id="ref-16yVVytBf" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">180. </div><div class="csl-right-inline"><strong>Clonal status of actionable driver events and the timing of mutational processes in cancer evolution</strong> <div class="csl-block">Nicholas McGranahan, Francesco Favero, Elza C de Bruin, Nicolai Juul Birkbak, Zoltan Szallasi, Charles Swanton</div> <em>Science Translational Medicine</em> (2015-04-15) <a href="https://doi.org/f7f83d">https://doi.org/f7f83d</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1126/scitranslmed.aaa1408">10.1126/scitranslmed.aaa1408</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25877892">25877892</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4636056">PMC4636056</a></div></div>
</div>
<div id="ref-pHDBQdQT" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">181. </div><div class="csl-right-inline"><strong>ICGC/TCGA Mutational Signatures</strong> <div class="csl-block">ICGC-TCGA Pan-Cancer Analysis Mutational Signatures Working Group</div> <em>Webpage</em> (2022) <a href="https://dcc.icgc.org/releases/PCAWG/mutational_signatures">https://dcc.icgc.org/releases/PCAWG/mutational_signatures</a></div>
</div>
<div id="ref-J4w4e1MM" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">182. </div><div class="csl-right-inline"><strong>Code for 'Widespread redundancy in -omics profiles of cancer mutation states'</strong> <div class="csl-block">Jake Crawford, Brock C. Christensen, Maria Chikina, Casey S. Greene</div> <em>Zenodo</em> (2022) <a href="https://doi.org/10.5281/zenodo.6590133">https://doi.org/10.5281/zenodo.6590133</a></div>
</div>
<div id="ref-lVGBvUEc" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">183. </div><div class="csl-right-inline"><strong>Code for 'Widespread redundancy in -omics profiles of cancer mutation states'</strong> <div class="csl-block">Jake Crawford, Brock C. Christensen, Maria Chikina, Casey S. Greene</div> <em>GitHub</em> (2022) <a href="https://github.com/greenelab/mpmp">https://github.com/greenelab/mpmp</a></div>
</div>
<div id="ref-3yukLr06" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">184. </div><div class="csl-right-inline"><strong>Widespread redundancy in -omics profiles of cancer mutation states</strong> <div class="csl-block">Jake Crawford, Brock C. Christensen, Maria Chikina, Casey S. Greene</div> <em>GitHub</em> (2022) <a href="https://github.com/greenelab/mpmp-manuscript">https://github.com/greenelab/mpmp-manuscript</a></div>
</div>
<div id="ref-30nThzUa" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">185. </div><div class="csl-right-inline"><strong>Widespread redundancy in -omics profiles of cancer mutation states</strong> <div class="csl-block">Jake Crawford, Brock C. Christensen, Maria Chikina, Casey S. Greene</div> <em>Zenodo</em> (2022) <a href="https://doi.org/10.5281/zenodo.6638831">https://doi.org/10.5281/zenodo.6638831</a></div>
</div>
<div id="ref-TIqbHsOH" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">186. </div><div class="csl-right-inline"><strong>Widespread redundancy in -omics profiles of cancer mutation states</strong> <div class="csl-block">Jake Crawford, Brock C. Christensen, Maria Chikina, Casey S. Greene</div> <em>figshare</em> (2022) <a href="https://doi.org/10.6084/m9.figshare.19576012">https://doi.org/10.6084/m9.figshare.19576012</a></div>
</div>
<div id="ref-EZnvMHZD" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">187. </div><div class="csl-right-inline"><strong>Widespread redundancy in -omics profiles of cancer mutation states</strong> <div class="csl-block">Jake Crawford, Brock C. Christensen, Maria Chikina, Casey S. Greene</div> <em>figshare</em> (2022) <a href="https://doi.org/10.6084/m9.figshare.19576048">https://doi.org/10.6084/m9.figshare.19576048</a></div>
</div>
<div id="ref-pt72xvSB" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">188. </div><div class="csl-right-inline"><strong>Effective dimension reduction methods for tumor classification
using gene expression data</strong> <div class="csl-block">A Antoniadis, S Lambert-Lacroix, F Leblanc</div> <em>Bioinformatics</em> (2003-03-22) <a href="https://doi.org/dhfzst">https://doi.org/dhfzst</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btg062">10.1093/bioinformatics/btg062</a></div></div>
</div>
<div id="ref-1Gjm0AvRK" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">189. </div><div class="csl-right-inline"><strong>Feature selection and dimension reduction for single-cell RNA-Seq based on a multinomial model</strong> <div class="csl-block">FWilliam Townes, Stephanie C Hicks, Martin J Aryee, Rafael A Irizarry</div> <em>Genome Biology</em> (2019-12) <a href="https://doi.org/ggk85t">https://doi.org/ggk85t</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-019-1861-6">10.1186/s13059-019-1861-6</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31870412">31870412</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6927135">PMC6927135</a></div></div>
</div>
<div id="ref-16Pv16qgL" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">190. </div><div class="csl-right-inline"><strong>Cancer gene expression signatures – The rise and fall?</strong> <div class="csl-block">Frederic Chibon</div> <em>European Journal of Cancer</em> (2013-05) <a href="https://doi.org/f2gtqf">https://doi.org/f2gtqf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.ejca.2013.02.021">10.1016/j.ejca.2013.02.021</a></div></div>
</div>
<div id="ref-1RajAavD" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">191. </div><div class="csl-right-inline"><strong>A Five-Gene Signature and Clinical Outcome in Non–Small-Cell Lung Cancer</strong> <div class="csl-block">Hsuan-Yu Chen, Sung-Liang Yu, Chun-Houh Chen, Gee-Chen Chang, Chih-Yi Chen, Ang Yuan, Chiou-Ling Cheng, Chien-Hsun Wang, Harn-Jing Terng, Shu-Fang Kao, … Pan-Chyr Yang</div> <em>New England Journal of Medicine</em> (2007-01-04) <a href="https://doi.org/dsnktr">https://doi.org/dsnktr</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1056/nejmoa060096">10.1056/nejmoa060096</a></div></div>
</div>
<div id="ref-dM8796ho" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">192. </div><div class="csl-right-inline"><strong>A Six-Gene Signature Predicting Breast Cancer Lung Metastasis</strong> <div class="csl-block">Thomas Landemaine, Amanda Jackson, Akeila Bellahcène, Nadia Rucci, Soraya Sin, Berta Martin Abad, Angels Sierra, Alain Boudinet, Jean-Marc Guinebretière, Enrico Ricevuto, … Keltouma Driouch</div> <em>Cancer Research</em> (2008-08-01) <a href="https://doi.org/frmj5f">https://doi.org/frmj5f</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1158/0008-5472.can-08-0436">10.1158/0008-5472.can-08-0436</a></div></div>
</div>
<div id="ref-OEMQX6hJ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">193. </div><div class="csl-right-inline"><strong>70-Gene Signature as an Aid to Treatment Decisions in Early-Stage Breast Cancer</strong> <div class="csl-block">Fatima Cardoso, Laura J van’t Veer, Jan Bogaerts, Leen Slaets, Giuseppe Viale, Suzette Delaloge, Jean-Yves Pierga, Etienne Brain, Sylvain Causeret, Mauro DeLorenzi, … Martine Piccart</div> <em>New England Journal of Medicine</em> (2016-08-25) <a href="https://doi.org/gdp988">https://doi.org/gdp988</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1056/nejmoa1602253">10.1056/nejmoa1602253</a></div></div>
</div>
<div id="ref-fdtdInsC" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">194. </div><div class="csl-right-inline"><strong>MammaPrint™ 70-gene signature: another milestone in personalized medical care for breast cancer patients</strong> <div class="csl-block">Elzbieta A Slodkowska, Jeffrey S Ross</div> <em>Expert Review of Molecular Diagnostics</em> (2009-07) <a href="https://doi.org/c8qptt">https://doi.org/c8qptt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1586/erm.09.32">10.1586/erm.09.32</a></div></div>
</div>
<div id="ref-1HDl7WHTm" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">195. </div><div class="csl-right-inline"><strong>Sorting Out Breast-Cancer Gene Signatures</strong> <div class="csl-block">Joan Massagué</div> <em>New England Journal of Medicine</em> (2007-01-18) <a href="https://doi.org/cbt3x7">https://doi.org/cbt3x7</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1056/nejme068292">10.1056/nejme068292</a></div></div>
</div>
<div id="ref-Yto9fv8i" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">196. </div><div class="csl-right-inline"><strong>Challenges translating breast cancer gene signatures into the clinic</strong> <div class="csl-block">Britta Weigelt, Lajos Pusztai, Alan Ashworth, Jorge S Reis-Filho</div> <em>Nature Reviews Clinical Oncology</em> (2011-08-30) <a href="https://doi.org/cp83ks">https://doi.org/cp83ks</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nrclinonc.2011.125">10.1038/nrclinonc.2011.125</a></div></div>
</div>
<div id="ref-kAXq3qou" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">197. </div><div class="csl-right-inline"><strong>What do we mean by validating a prognostic model?</strong> <div class="csl-block">Douglas G Altman, Patrick Royston</div> <em>Statistics in Medicine</em> (2000-02-29) <a href="https://doi.org/bhfhgd">https://doi.org/bhfhgd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1002/(sici)1097-0258(20000229)19:4&lt;453::aid-sim350&gt;3.0.co;2-5">10.1002/(sici)1097-0258(20000229)19:4&lt;453::aid-sim350&gt;3.0.co;2-5</a></div></div>
</div>
<div id="ref-wnVDbuuP" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">198. </div><div class="csl-right-inline"><strong>Evaluating Microarray-based Classifiers: An Overview</strong> <div class="csl-block">A-L Boulesteix, C Strobl, T Augustin, M Daumer</div> <em>Cancer Informatics</em> (2008-01) <a href="https://doi.org/ggsmz4">https://doi.org/ggsmz4</a> <div class="csl-block">DOI: <a href="https://doi.org/10.4137/cin.s408">10.4137/cin.s408</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/19259405">19259405</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2623308">PMC2623308</a></div></div>
</div>
<div id="ref-HmaoBsQm" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">199. </div><div class="csl-right-inline"><strong>Ten Simple Rules for Effective Statistical Practice</strong> <div class="csl-block">Robert E Kass, Brian S Caffo, Marie Davidian, Xiao-Li Meng, Bin Yu, Nancy Reid</div> <em>PLOS Computational Biology</em> (2016-06-09) <a href="https://doi.org/gcx4rn">https://doi.org/gcx4rn</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pcbi.1004961">10.1371/journal.pcbi.1004961</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27281180">27281180</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4900655">PMC4900655</a></div></div>
</div>
<div id="ref-13HbWPwpj" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">200. </div><div class="csl-right-inline"><strong>A new look at the statistical model identification</strong> <div class="csl-block">H Akaike</div> <em>IEEE Transactions on Automatic Control</em> (1974-12) <a href="https://doi.org/d98qkw">https://doi.org/d98qkw</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1109/tac.1974.1100705">10.1109/tac.1974.1100705</a></div></div>
</div>
<div id="ref-PUJO2E4L" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">201. </div><div class="csl-right-inline"><strong>Estimating the Dimension of a Model</strong> <div class="csl-block">Gideon Schwarz</div> <em>The Annals of Statistics</em> (1978-03-01) <a href="https://doi.org/d9mzdb">https://doi.org/d9mzdb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1214/aos/1176344136">10.1214/aos/1176344136</a></div></div>
</div>
<div id="ref-ZJNDbtzm" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">202. </div><div class="csl-right-inline"><strong>Few-shot learning creates predictive models of drug response that translate from high-throughput screens to individual patients</strong> <div class="csl-block">Jianzhu Ma, Samson H Fong, Yunan Luo, Christopher J Bakkenist, John Paul Shen, Soufiane Mourragui, Lodewyk FA Wessels, Marc Hafner, Roded Sharan, Jian Peng, Trey Ideker</div> <em>Nature Cancer</em> (2021-01-25) <a href="https://doi.org/gh52nt">https://doi.org/gh52nt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s43018-020-00169-2">10.1038/s43018-020-00169-2</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34223192">34223192</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8248912">PMC8248912</a></div></div>
</div>
<div id="ref-QWTf9qrW" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">203. </div><div class="csl-right-inline"><strong>Out-of-distribution generalization from labelled and unlabelled gene expression data for drug response prediction</strong> <div class="csl-block">Hossein Sharifi-Noghabi, Parsa Alamzadeh Harjandi, Olga Zolotareva, Colin C Collins, Martin Ester</div> <em>Nature Machine Intelligence</em> (2021-11-11) <a href="https://doi.org/gq32k7">https://doi.org/gq32k7</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s42256-021-00408-w">10.1038/s42256-021-00408-w</a></div></div>
</div>
<div id="ref-L3dpDrOG" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">204. </div><div class="csl-right-inline"><strong>Predicting patient response with models trained on cell lines and patient-derived xenografts by nonlinear transfer learning</strong> <div class="csl-block">Soufiane MC Mourragui, Marco Loog, Daniel J Vis, Kat Moore, Anna G Manjon, Mark A van de Wiel, Marcel JT Reinders, Lodewyk FA Wessels</div> <em>Proceedings of the National Academy of Sciences</em> (2021-12-03) <a href="https://doi.org/gshpgt">https://doi.org/gshpgt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1073/pnas.2106682118">10.1073/pnas.2106682118</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34873056">34873056</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8670522">PMC8670522</a></div></div>
</div>
<div id="ref-gOHkic9g" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">205. </div><div class="csl-right-inline"><strong>Optimizer’s dilemma: optimization strongly influences model selection in transcriptomic prediction</strong> <div class="csl-block">Jake Crawford, Maria Chikina, Casey S Greene</div> <em>Cold Spring Harbor Laboratory</em> (2023-06-26) <a href="https://doi.org/gsdsvs">https://doi.org/gsdsvs</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2023.06.26.546586">10.1101/2023.06.26.546586</a></div></div>
</div>
<div id="ref-b7RUXQvu" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">206. </div><div class="csl-right-inline"><strong>The effect of non-linear signal in classification problems using gene expression</strong> <div class="csl-block">Benjamin J Heil, Jake Crawford, Casey S Greene</div> <em>PLOS Computational Biology</em> (2023-03-27) <a href="https://doi.org/gr2q6q">https://doi.org/gr2q6q</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pcbi.1010984">10.1371/journal.pcbi.1010984</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36972227">36972227</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10079219">PMC10079219</a></div></div>
</div>
<div id="ref-Xk9rmxAA" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">207. </div><div class="csl-right-inline"><a href="https://dl.acm.org/doi/10.5555/3104322.3104425">https://dl.acm.org/doi/10.5555/3104322.3104425</a></div>
</div>
<div id="ref-iTP4h1rX" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">208. </div><div class="csl-right-inline"><strong>PyTorch: An Imperative Style, High-Performance Deep Learning Library</strong> <div class="csl-block">Adam Paszke, Sam Gross, Francisco Massa, Adam Lerer, James Bradbury, Gregory Chanan, Trevor Killeen, Zeming Lin, Natalia Gimelshein, Luca Antiga, … Soumith Chintala</div> <em>arXiv</em> (2019-12-05) <a href="https://arxiv.org/abs/1912.01703">https://arxiv.org/abs/1912.01703</a></div>
</div>
<div id="ref-qtYNZ2Q2" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">209. </div><div class="csl-right-inline"><strong>EGFR Mutations and Lung Cancer</strong> <div class="csl-block">Gilda da Cunha Santos, Frances A Shepherd, Ming Sound Tsao</div> <em>Annual Review of Pathology: Mechanisms of Disease</em> (2011-02-28) <a href="https://doi.org/dd359s">https://doi.org/dd359s</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1146/annurev-pathol-011110-130206">10.1146/annurev-pathol-011110-130206</a></div></div>
</div>
<div id="ref-rzy1RWeA" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">210. </div><div class="csl-right-inline"><strong>Spectrum of EGFR aberrations and potential clinical implications: insights from integrative pan‐cancer analysis</strong> <div class="csl-block">Haijing Liu, Bo Zhang, Zhifu Sun</div> <em>Cancer Communications</em> (2020-01) <a href="https://doi.org/ghsz4b">https://doi.org/ghsz4b</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1002/cac2.12005">10.1002/cac2.12005</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32067422">32067422</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7163653">PMC7163653</a></div></div>
</div>
<div id="ref-HQx0QuE4" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">211. </div><div class="csl-right-inline"><strong>Patient-derived cells from recurrent tumors that model the evolution of IDH-mutant glioma</strong> <div class="csl-block">Lindsey E Jones, Stephanie Hilz, Matthew R Grimmer, Tali Mazor, Chloé Najac, Joydeep Mukherjee, Andrew McKinney, Tracy Chow, Russell O Pieper, Sabrina M Ronen, … Joseph F Costello</div> <em>Neuro-Oncology Advances</em> (2020-01-01) <a href="https://doi.org/gsfw2p">https://doi.org/gsfw2p</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/noajnl/vdaa088">10.1093/noajnl/vdaa088</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32904945">32904945</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7462278">PMC7462278</a></div></div>
</div>
<div id="ref-YOm0bqVR" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">212. </div><div class="csl-right-inline"><strong>Knowledge-primed neural networks enable biologically interpretable deep learning on single-cell sequencing data</strong> <div class="csl-block">Nikolaus Fortelny, Christoph Bock</div> <em>Genome Biology</em> (2020-08-03) <a href="https://doi.org/gg8ws9">https://doi.org/gg8ws9</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-020-02100-5">10.1186/s13059-020-02100-5</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32746932">32746932</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7397672">PMC7397672</a></div></div>
</div>
<div id="ref-Mflfx7GP" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">213. </div><div class="csl-right-inline"><strong>Knowledge-guided deep learning models of drug toxicity improve interpretation</strong> <div class="csl-block">Yun Hao, Joseph D Romano, Jason H Moore</div> <em>Patterns</em> (2022-09) <a href="https://doi.org/gshk7s">https://doi.org/gshk7s</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.patter.2022.100565">10.1016/j.patter.2022.100565</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36124309">36124309</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9481960">PMC9481960</a></div></div>
</div>
<div id="ref-1HnyQJPcl" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">214. </div><div class="csl-right-inline"><strong>The Clinical Relevance of Cancer Cell Lines</strong> <div class="csl-block">J-P Gillet, S Varma, MM Gottesman</div> <em>JNCI Journal of the National Cancer Institute</em> (2013-02-21) <a href="https://doi.org/f4tstr">https://doi.org/f4tstr</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/jnci/djt007">10.1093/jnci/djt007</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23434901">23434901</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3691946">PMC3691946</a></div></div>
</div>
<div id="ref-vYqd0FZY" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">215. </div><div class="csl-right-inline"><strong>Cancer Cell Lines for Drug Discovery and Development</strong> <div class="csl-block">Jennifer L Wilding, Walter F Bodmer</div> <em>Cancer Research</em> (2014-04-30) <a href="https://doi.org/f56fwg">https://doi.org/f56fwg</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1158/0008-5472.can-13-2971">10.1158/0008-5472.can-13-2971</a></div></div>
</div>
<div id="ref-18vM3pLD6" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">216. </div><div class="csl-right-inline"><strong>A Landscape of Pharmacogenomic Interactions in Cancer</strong> <div class="csl-block">Francesco Iorio, Theo A Knijnenburg, Daniel J Vis, Graham R Bignell, Michael P Menden, Michael Schubert, Nanne Aben, Emanuel Gonçalves, Syd Barthorpe, Howard Lightfoot, … Mathew J Garnett</div> <em>Cell</em> (2016-07) <a href="https://doi.org/f8wq4s">https://doi.org/f8wq4s</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cell.2016.06.017">10.1016/j.cell.2016.06.017</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27397505">27397505</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4967469">PMC4967469</a></div></div>
</div>
<div id="ref-Inb9g7Za" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">217. </div><div class="csl-right-inline"><strong>A pitfall for machine learning methods aiming to predict across cell types</strong> <div class="csl-block">Jacob Schreiber, Ritambhara Singh, Jeffrey Bilmes, William Stafford Noble</div> <em>Genome Biology</em> (2020-11-19) <a href="https://doi.org/gshk6j">https://doi.org/gshk6j</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-020-02177-y">10.1186/s13059-020-02177-y</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33213499">33213499</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7678316">PMC7678316</a></div></div>
</div>
<div id="ref-13prIHiSM" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">218. </div><div class="csl-right-inline"><strong>Navigating the pitfalls of applying machine learning in genomics</strong> <div class="csl-block">Sean Whalen, Jacob Schreiber, William S Noble, Katherine S Pollard</div> <em>Nature Reviews Genetics</em> (2021-11-26) <a href="https://doi.org/gnm4r9">https://doi.org/gnm4r9</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41576-021-00434-9">10.1038/s41576-021-00434-9</a></div></div>
</div>
<div id="ref-cU8r76Rn" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">219. </div><div class="csl-right-inline"><strong>Classifier Technology and the Illusion of Progress</strong> <div class="csl-block">David J Hand</div> <em>Statistical Science</em> (2006-02-01) <a href="https://doi.org/fkwx3j">https://doi.org/fkwx3j</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1214/088342306000000060">10.1214/088342306000000060</a></div></div>
</div>
<div id="ref-10bvAmkln" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">220. </div><div class="csl-right-inline"><strong>Integration strategies of multi-omics data for machine learning analysis</strong> <div class="csl-block">Milan Picard, Marie-Pier Scott-Boyer, Antoine Bodein, Olivier Périn, Arnaud Droit</div> <em>Computational and Structural Biotechnology Journal</em> (2021) <a href="https://doi.org/gqwtsc">https://doi.org/gqwtsc</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.csbj.2021.06.030">10.1016/j.csbj.2021.06.030</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34285775">34285775</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8258788">PMC8258788</a></div></div>
</div>
<div id="ref-SLbB6cV8" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">221. </div><div class="csl-right-inline"><strong>Multi-omics integration in the age of million single-cell data</strong> <div class="csl-block">Zhen Miao, Benjamin D Humphreys, Andrew P McMahon, Junhyong Kim</div> <em>Nature Reviews Nephrology</em> (2021-08-20) <a href="https://doi.org/gmzwc3">https://doi.org/gmzwc3</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41581-021-00463-x">10.1038/s41581-021-00463-x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34417589">34417589</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9191639">PMC9191639</a></div></div>
</div>
<div id="ref-9lrRlyX2" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">222. </div><div class="csl-right-inline"><strong>Computational strategies for single-cell multi-omics integration</strong> <div class="csl-block">Nigatu Adossa, Sofia Khan, Kalle T Rytkönen, Laura L Elo</div> <em>Computational and Structural Biotechnology Journal</em> (2021) <a href="https://doi.org/gmh28b">https://doi.org/gmh28b</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.csbj.2021.04.060">10.1016/j.csbj.2021.04.060</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34025945">34025945</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8114078">PMC8114078</a></div></div>
</div>
<div id="ref-13Db93XWH" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">223. </div><div class="csl-right-inline"><strong>Intricacies of single-cell multi-omics data integration</strong> <div class="csl-block">Pia Rautenstrauch, Anna Hendrika Cornelia Vlot, Sepideh Saran, Uwe Ohler</div> <em>Trends in Genetics</em> (2022-02) <a href="https://doi.org/gpp6tt">https://doi.org/gpp6tt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.tig.2021.08.012">10.1016/j.tig.2021.08.012</a></div></div>
</div>
<div id="ref-g7bzDkos" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">224. </div><div class="csl-right-inline"><strong>A Variational Information Bottleneck Approach to Multi-Omics Data Integration</strong> <div class="csl-block">Changhee Lee, Mihaela van der Schaar</div> <em>arXiv</em> (2021-02-11) <a href="https://arxiv.org/abs/2102.03014">https://arxiv.org/abs/2102.03014</a></div>
</div>
<div id="ref-bcArgYmm" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">225. </div><div class="csl-right-inline"><strong>Revisiting invariances and introducing priors in Gromov-Wasserstein distances</strong> <div class="csl-block">Pinar Demetci, Quang Huy Tran, Ievgen Redko, Ritambhara Singh</div> <em>arXiv</em> (2023-07-20) <a href="https://arxiv.org/abs/2307.10093">https://arxiv.org/abs/2307.10093</a></div>
</div>
<div id="ref-hzkzGLGE" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">226. </div><div class="csl-right-inline"><strong>A multi-use deep learning method for CITE-seq and single-cell RNA-seq data integration with cell surface protein prediction and imputation</strong> <div class="csl-block">Justin Lakkis, Amelia Schroeder, Kenong Su, Michelle YY Lee, Alexander C Bashore, Muredach P Reilly, Mingyao Li</div> <em>Nature Machine Intelligence</em> (2022-10-27) <a href="https://doi.org/gq49jj">https://doi.org/gq49jj</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s42256-022-00545-w">10.1038/s42256-022-00545-w</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36873621">36873621</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9979929">PMC9979929</a></div></div>
</div>
<div id="ref-1BYOCoFwl" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">227. </div><div class="csl-right-inline"><strong>Evaluating the transcriptional fidelity of cancer models</strong> <div class="csl-block">Da Peng, Rachel Gleyzer, Wen-Hsin Tai, Pavithra Kumar, Qin Bian, Bradley Isaacs, Edroaldo Lummertz da Rocha, Stephanie Cai, Kathleen DiNapoli, Franklin W Huang, Patrick Cahan</div> <em>Genome Medicine</em> (2021-04-29) <a href="https://doi.org/gq8s52">https://doi.org/gq8s52</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13073-021-00888-w">10.1186/s13073-021-00888-w</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33926541">33926541</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8086312">PMC8086312</a></div></div>
</div>
<div id="ref-10DTHeWmL" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">228. </div><div class="csl-right-inline"><strong>Pan-cancer genome and transcriptome analyses of 1,699 paediatric leukaemias and solid tumours</strong> <div class="csl-block">Xiaotu Ma, Yu Liu, Yanling Liu, Ludmil B Alexandrov, Michael N Edmonson, Charles Gawad, Xin Zhou, Yongjin Li, Michael C Rusch, John Easton, … Jinghui Zhang</div> <em>Nature</em> (2018-02-28) <a href="https://doi.org/gc3sx2">https://doi.org/gc3sx2</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nature25795">10.1038/nature25795</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29489755">29489755</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5854542">PMC5854542</a></div></div>
</div>
<div id="ref-oRtHYf3u" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">229. </div><div class="csl-right-inline"><strong>The landscape of genomic alterations across childhood cancers</strong> <div class="csl-block">Susanne N Gröbner, Barbara C Worst, Joachim Weischenfeldt, Ivo Buchhalter, Kortine Kleinheinz, Vasilisa A Rudneva, Pascal D Johann, Gnana Prakash Balasubramanian, Maia Segura-Wang, … </div> <em>Nature</em> (2018-02-28) <a href="https://doi.org/gc3wqt">https://doi.org/gc3wqt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nature25480">10.1038/nature25480</a></div></div>
</div>
<div id="ref-142nJ47j4" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">230. </div><div class="csl-right-inline"><strong>Deletion of&lt;i&gt;IKZF1&lt;/i&gt;and Prognosis in Acute Lymphoblastic Leukemia</strong> <div class="csl-block">Charles G Mullighan, Xiaoping Su, Jinghui Zhang, Ina Radtke, Letha AA Phillips, Christopher B Miller, Jing Ma, Wei Liu, Cheng Cheng, Brenda A Schulman, … James R Downing</div> <em>New England Journal of Medicine</em> (2009-01-29) <a href="https://doi.org/d6pbqg">https://doi.org/d6pbqg</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1056/nejmoa0808253">10.1056/nejmoa0808253</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/19129520">19129520</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2674612">PMC2674612</a></div></div>
</div>
<div id="ref-5VXMHJ7N" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">231. </div><div class="csl-right-inline"><strong>OpenPBTA: The Open Pediatric Brain Tumor Atlas</strong> <div class="csl-block">Joshua A Shapiro, Krutika S Gaonkar, Stephanie J Spielman, Candace L Savonen, Chante J Bethell, Run Jin, Komal S Rathi, Yuankun Zhu, Laura E Egolf, Bailey K Farrow, … Jaclyn N Taroni</div> <em>Cell Genomics</em> (2023-07) <a href="https://doi.org/gr92p6">https://doi.org/gr92p6</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.xgen.2023.100340">10.1016/j.xgen.2023.100340</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37492101">37492101</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10363844">PMC10363844</a></div></div>
</div>
<div id="ref-SDwYl8uA" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">232. </div><div class="csl-right-inline"><strong>Genomic Profiling of Childhood Tumor Patient-Derived Xenograft Models to Enable Rational Clinical Trial Design</strong> <div class="csl-block">Jo Lynne Rokita, Komal S Rathi, Maria F Cardenas, Kristen A Upton, Joy Jayaseelan, Katherine L Cross, Jacob Pfeil, Laura E Egolf, Gregory P Way, Alvin Farrel, … John M Maris</div> <em>Cell Reports</em> (2019-11) <a href="https://doi.org/gg596n">https://doi.org/gg596n</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.celrep.2019.09.071">10.1016/j.celrep.2019.09.071</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31693904">31693904</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6880934">PMC6880934</a></div></div>
</div>
</div>
<!-- default theme -->

<style>
  /* import google fonts */
  @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600,700");
  @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

  /* -------------------------------------------------- */
  /* global */
  /* -------------------------------------------------- */

  /* all elements */
  * {
    /* force sans-serif font unless specified otherwise */
    font-family: "Open Sans", "Helvetica", sans-serif;

    /* prevent text inflation on some mobile browsers */
    -webkit-text-size-adjust: none !important;
    -moz-text-size-adjust: none !important;
    -o-text-size-adjust: none !important;
    text-size-adjust: none !important;
  }

  @media only screen {
    /* "page" element */
    body {
      position: relative;
      box-sizing: border-box;
      font-size: 12pt;
      line-height: 1.5;
      max-width: 8.5in;
      margin: 20px auto;
      padding: 40px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* "page" element */
    body {
      padding: 20px;
      margin: 0;
      border-radius: 0;
      border: none;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
      background: none;
    }
  }

  /* -------------------------------------------------- */
  /* headings */
  /* -------------------------------------------------- */

  /* all headings */
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    margin: 20px 0;
    padding: 0;
    font-weight: bold;
  }

  /* biggest heading */
  h1 {
    margin: 40px 0;
    text-align: center;
  }

  /* second biggest heading */
  h2 {
    margin-top: 30px;
    padding-bottom: 5px;
    border-bottom: solid 1px #bdbdbd;
  }

  /* heading font sizes */
  h1 {
    font-size: 2em;
  }
  h2 {
    font-size: 1.5em;
  }
  h3 {
    font-size: 1.35em;
  }
  h4 {
    font-size: 1.25em;
  }
  h5 {
    font-size: 1.15em;
  }
  h6 {
    font-size: 1em;
  }

  /* -------------------------------------------------- */
  /* manuscript header */
  /* -------------------------------------------------- */

  /* manuscript title */
  header > h1 {
    margin: 0;
  }

  /* manuscript title caption text (ie "automatically generated on") */
  header + p {
    text-align: center;
    margin-top: 10px;
  }

  /* -------------------------------------------------- */
  /* text elements */
  /* -------------------------------------------------- */

  /* links */
  a {
    color: #2196f3;
    overflow-wrap: break-word;
  }

  /* superscripts and subscripts */
  sub,
  sup {
    /* prevent from affecting line height */
    line-height: 0;
  }

  /* unordered and ordered lists*/
  ul,
  ol {
    padding-left: 20px;
  }

  /* class for styling text semibold */
  .semibold {
    font-weight: 600;
  }

  /* class for styling elements horizontally left aligned */
  .left {
    display: block;
    text-align: left;
    margin-left: auto;
    margin-right: 0;
    justify-content: left;
  }

  /* class for styling elements horizontally centered */
  .center {
    display: block;
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    justify-content: center;
  }

  /* class for styling elements horizontally right aligned */
  .right {
    display: block;
    text-align: right;
    margin-left: 0;
    margin-right: auto;
    justify-content: right;
  }

  /* -------------------------------------------------- */
  /* section elements */
  /* -------------------------------------------------- */

  /* horizontal divider line */
  hr {
    border: none;
    height: 1px;
    background: #bdbdbd;
  }

  /* paragraphs, horizontal dividers, figures, tables, code */
  p,
  hr,
  figure,
  table,
  pre {
    /* treat all as "paragraphs", with consistent vertical margins */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* figures */
  /* -------------------------------------------------- */

  /* figure */
  figure {
    max-width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure caption */
  figcaption {
    padding: 0;
    padding-top: 10px;
  }

  /* figure image element */
  figure > img,
  figure > svg {
    max-width: 100%;
    display: block;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure auto-number */
  img + figcaption > span:first-of-type,
  svg + figcaption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* tables */
  /* -------------------------------------------------- */

  /* table */
  table {
    border-collapse: collapse;
    border-spacing: 0;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* table cells */
  th,
  td {
    border: solid 1px #bdbdbd;
    padding: 10px;
    /* squash table if too wide for page by forcing line breaks */
    overflow-wrap: break-word;
    word-break: break-word;
  }

  /* header row and even rows */
  th,
  tr:nth-child(2n) {
    background-color: #fafafa;
  }

  /* odd rows */
  tr:nth-child(2n + 1) {
    background-color: #ffffff;
  }

  /* table caption */
  caption {
    text-align: left;
    padding: 0;
    padding-bottom: 10px;
  }

  /* table auto-number */
  table > caption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* code */
  /* -------------------------------------------------- */

  /* multi-line code block */
  pre {
    padding: 10px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
    break-inside: avoid;
    text-align: left;
  }

  /* inline code, ie code within normal text */
  :not(pre) > code {
    padding: 0 4px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
  }

  /* code text */
  /* apply all children, to reach syntax highlighting sub-elements */
  code,
  code * {
    /* force monospace font */
    font-family: "Source Code Pro", "Courier New", monospace;
  }

  /* -------------------------------------------------- */
  /* quotes */
  /* -------------------------------------------------- */

  /* quoted text */
  blockquote {
    margin: 0;
    padding: 0;
    border-left: 4px solid #bdbdbd;
    padding-left: 16px;
    break-inside: avoid;
  }

  /* -------------------------------------------------- */
  /* banners */
  /* -------------------------------------------------- */

  /* info banners */
  .banner {
    box-sizing: border-box;
    display: block;
    position: relative;
    width: 100%;
    margin-top: 20px;
    margin-bottom: 20px;
    padding: 20px;
    text-align: center;
  }

  /* paragraph in banner */
  .banner > p {
    margin: 0;
  }

  /* -------------------------------------------------- */
  /* highlight colors */
  /* -------------------------------------------------- */

  .white {
    background: #ffffff;
  }
  .lightgrey {
    background: #eeeeee;
  }
  .grey {
    background: #757575;
  }
  .darkgrey {
    background: #424242;
  }
  .black {
    background: #000000;
  }
  .lightred {
    background: #ffcdd2;
  }
  .lightyellow {
    background: #ffecb3;
  }
  .lightgreen {
    background: #dcedc8;
  }
  .lightblue {
    background: #e3f2fd;
  }
  .lightpurple {
    background: #f3e5f5;
  }
  .red {
    background: #f44336;
  }
  .orange {
    background: #ff9800;
  }
  .yellow {
    background: #ffeb3b;
  }
  .green {
    background: #4caf50;
  }
  .blue {
    background: #2196f3;
  }
  .purple {
    background: #9c27b0;
  }
  .white,
  .lightgrey,
  .lightred,
  .lightyellow,
  .lightgreen,
  .lightblue,
  .lightpurple,
  .orange,
  .yellow,
  .white a,
  .lightgrey a,
  .lightred a,
  .lightyellow a,
  .lightgreen a,
  .lightblue a,
  .lightpurple a,
  .orange a,
  .yellow a {
    color: #000000;
  }
  .grey,
  .darkgrey,
  .black,
  .red,
  .green,
  .blue,
  .purple,
  .grey a,
  .darkgrey a,
  .black a,
  .red a,
  .green a,
  .blue a,
  .purple a {
    color: #ffffff;
  }

  /* -------------------------------------------------- */
  /* buttons */
  /* -------------------------------------------------- */

  /* class for styling links like buttons */
  .button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    margin: 5px;
    padding: 10px 20px;
    font-size: 0.75em;
    font-weight: 600;
    text-transform: uppercase;
    text-decoration: none;
    letter-spacing: 1px;
    background: none;
    color: #2196f3;
    border: solid 1px #bdbdbd;
    border-radius: 5px;
  }

  /* buttons when hovered */
  .button:hover:not([disabled]),
  .icon_button:hover:not([disabled]) {
    cursor: pointer;
    background: #f5f5f5;
  }

  /* buttons when disabled */
  .button[disabled],
  .icon_button[disabled] {
    opacity: 0.35;
    pointer-events: none;
  }

  /* class for styling buttons containg only single icon */
  .icon_button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    text-decoration: none;
    margin: 0;
    padding: 0;
    background: none;
    border-radius: 5px;
    border: none;
    width: 20px;
    height: 20px;
    min-width: 20px;
    min-height: 20px;
  }

  /* icon button inner svg image */
  .icon_button > svg {
    height: 16px;
  }

  /* -------------------------------------------------- */
  /* icons */
  /* -------------------------------------------------- */

  /* class for styling icons inline with text */
  .inline_icon {
    height: 1em;
    position: relative;
    top: 0.125em;
  }

  /* -------------------------------------------------- */
  /* references */
  /* -------------------------------------------------- */

  .csl-entry {
    margin-top: 15px;
    margin-bottom: 15px;
  }

  /* -------------------------------------------------- */
  /* print control */
  /* -------------------------------------------------- */

  @media print {
    @page {
      /* suggested printing margin */
      margin: 0.5in;
    }

    /* document and "page" elements */
    html,
    body {
      margin: 0;
      padding: 0;
      width: 100%;
      height: 100%;
    }

    /* "page" element */
    body {
      font-size: 11pt !important;
      line-height: 1.35;
    }

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      margin: 15px 0;
    }

    /* figures and tables */
    figure,
    table {
      font-size: 0.85em;
    }

    /* table cells */
    th,
    td {
      padding: 5px;
    }

    /* shrink font awesome icons */
    i.fas,
    i.fab,
    i.far,
    i.fal {
      transform: scale(0.85);
    }

    /* decrease banner margins */
    .banner {
      margin-top: 15px;
      margin-bottom: 15px;
      padding: 15px;
    }

    /* class for centering an element vertically on its own page */
    .page_center {
      margin: auto;
      width: 100%;
      height: 100%;
      display: flex;
      align-items: center;
      vertical-align: middle;
      break-before: page;
      break-after: page;
    }

    /* always insert a page break before the element */
    .page_break_before {
      break-before: page;
    }

    /* always insert a page break after the element */
    .page_break_after {
      break-after: page;
    }

    /* avoid page break before the element */
    .page_break_before_avoid {
      break-before: avoid;
    }

    /* avoid page break after the element */
    .page_break_after_avoid {
      break-after: avoid;
    }

    /* avoid page break inside the element */
    .page_break_inside_avoid {
      break-inside: avoid;
    }
  }

  /* -------------------------------------------------- */
  /* override pandoc css quirks */
  /* -------------------------------------------------- */

  .sourceCode {
    /* prevent unsightly overflow in wide code blocks */
    overflow: auto !important;
  }

  div.sourceCode {
    /* prevent background fill on top-most code block  container */
    background: none !important;
  }

  .sourceCode * {
    /* force consistent line spacing */
    line-height: 1.5 !important;
  }

  div.sourceCode {
    /* style code block margins same as <pre> element */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* tablenos */
  /* -------------------------------------------------- */

  /* tablenos wrapper */
  .tablenos {
    width: 100%;
    margin: 20px 0;
  }

  .tablenos > table {
    /* move margins from table to table_wrapper to allow margin collapsing */
    margin: 0;
  }

  @media only screen {
    /* tablenos wrapper */
    .tablenos {
      /* show scrollbar on tables if necessary to prevent overflow */
      overflow-x: auto !important;
    }

    .tablenos th,
    .tablenos td {
      overflow-wrap: unset !important;
      word-break: unset !important;
    }

    /* table in wrapper */
    .tablenos table,
    .tablenos table * {
      /* don't break table words */
      overflow-wrap: normal !important;
    }
  }
</style>
<!-- 
    Plugin Core

    Functions needed for and shared across all first-party plugins.
-->

<script>
  // get element that is target of hash (from link element or url)
  function getHashTarget(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector(`[id="${id}"]`);
    if (!target) return;

    // if figure or table, modify target to get expected element
    if (id.indexOf("fig:") === 0) target = target.querySelector("figure");
    if (id.indexOf("tbl:") === 0) target = target.querySelector("table");

    return target;
  }

  // get position/dimensions of element or viewport
  function getRectInView(element) {
    let rect = {};
    rect.left = 0;
    rect.top = 0;
    rect.right = document.documentElement.clientWidth;
    rect.bottom = document.documentElement.clientHeight;
    let style = {};

    if (element instanceof HTMLElement) {
      rect = element.getBoundingClientRect();
      style = window.getComputedStyle(element);
    }

    const margin = {};
    margin.left = parseFloat(style.marginLeftWidth) || 0;
    margin.top = parseFloat(style.marginTopWidth) || 0;
    margin.right = parseFloat(style.marginRightWidth) || 0;
    margin.bottom = parseFloat(style.marginBottomWidth) || 0;

    const border = {};
    border.left = parseFloat(style.borderLeftWidth) || 0;
    border.top = parseFloat(style.borderTopWidth) || 0;
    border.right = parseFloat(style.borderRightWidth) || 0;
    border.bottom = parseFloat(style.borderBottomWidth) || 0;

    const newRect = {};
    newRect.left = rect.left + margin.left + border.left;
    newRect.top = rect.top + margin.top + border.top;
    newRect.right = rect.right + margin.right + border.right;
    newRect.bottom = rect.bottom + margin.bottom + border.bottom;
    newRect.width = newRect.right - newRect.left;
    newRect.height = newRect.bottom - newRect.top;

    return newRect;
  }

  // get position of element relative to page
  function getRectInPage(element) {
    const rect = getRectInView(element);
    const body = getRectInView(document.body);

    const newRect = {};
    newRect.left = rect.left - body.left;
    newRect.top = rect.top - body.top;
    newRect.right = rect.right - body.left;
    newRect.bottom = rect.bottom - body.top;
    newRect.width = rect.width;
    newRect.height = rect.height;

    return newRect;
  }

  // get closest element before specified element that matches query
  function firstBefore(element, query) {
    while (element && element !== document.body && !element.matches(query))
      element = element.previousElementSibling || element.parentNode;

    return element;
  }

  // check if element is part of collapsed heading
  function isCollapsed(element) {
    while (element && element !== document.body) {
      if (element.dataset.collapsed === "true") return true;
      element = element.parentNode;
    }
    return false;
  }

  // expand any collapsed parent containers of element if necessary
  function expandElement(element) {
    if (isCollapsed(element)) {
      // accordion plugin
      const heading = firstBefore(element, "h2");
      if (heading) heading.click();
      // details/summary HTML element
      const summary = firstBefore(element, "summary");
      if (summary) summary.click();
    }
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);

    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // get list of elements after a start element up to element matching query
  function nextUntil(element, query, exclude) {
    const elements = [];
    while (((element = element.nextElementSibling), element)) {
      if (element.matches(query)) break;
      if (!element.matches(exclude)) elements.push(element);
    }
    return elements;
  }
</script>
<!--
  Accordion Plugin

  Allows sections of content under h2 headings to be collapsible.
-->

<script type="module">
  // whether to always start expanded ('false'), always start collapsed
  // ('true'), or start collapsed when screen small ('auto')
  const startCollapsed = "auto";

  // start script
  function start() {
    // run through each <h2> heading
    const headings = document.querySelectorAll("h2");
    for (const heading of headings) {
      addArrow(heading);

      // start expanded/collapsed based on option
      if (
        startCollapsed === "true" ||
        (startCollapsed === "auto" && isSmallScreen()) ||
        heading.dataset.collapsed === "true"
      )
        collapseHeading(heading);
      else expandElement(heading);
    }

    // attach hash change listener to window
    window.addEventListener("hashchange", onHashChange);
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) goToElement(target);
  }

  // add arrow to heading
  function addArrow(heading) {
    // add arrow button
    const arrow = document.createElement("button");
    arrow.innerHTML = document.querySelector(".icon_angle_down").innerHTML;
    arrow.classList.add("icon_button", "accordion_arrow");
    heading.insertBefore(arrow, heading.firstChild);

    // attach click listener to heading and button
    heading.addEventListener("click", onHeadingClick);
    arrow.addEventListener("click", onArrowClick);
  }

  // determine if on mobile-like device with small screen
  function isSmallScreen() {
    return Math.min(window.innerWidth, window.innerHeight) < 480;
  }

  // when <h2> heading is clicked
  function onHeadingClick(event) {
    // only collapse if <h2> itself is target of click (eg, user did
    // not click on anchor within <h2>)
    if (event.target === this) toggleCollapse(this);
  }

  // when arrow button is clicked
  function onArrowClick() {
    toggleCollapse(this.parentNode);
  }

  // collapse section if expanded, expand if collapsed
  function toggleCollapse(heading) {
    if (heading.dataset.collapsed === "false") collapseHeading(heading);
    else expandElement(heading);
  }

  // elements to exclude from collapse, such as table of contents panel,
  // hypothesis panel, etc
  const exclude = "#toc_panel, div.annotator-frame, #lightbox_overlay";

  // collapse section
  function collapseHeading(heading) {
    heading.setAttribute("data-collapsed", "true");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "true");
  }

  // expand section
  function expandElement(heading) {
    heading.setAttribute("data-collapsed", "false");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "false");
  }

  // get list of elements between this <h2> and next <h2> or <h1>
  // ("children" of the <h2> section)
  function getChildren(heading) {
    return nextUntil(heading, "h2, h1", exclude);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle down icon -->

<template class="icon_angle_down">
  <!-- modified from: https://fontawesome.com/icons/angle-down -->
  <svg width="16" height="16" viewBox="0 0 448 512">
    <path
      fill="currentColor"
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* accordion arrow button */
    .accordion_arrow {
      margin-right: 10px;
    }

    /* arrow icon when <h2> data-collapsed attribute true */
    h2[data-collapsed="true"] > .accordion_arrow > svg {
      transform: rotate(-90deg);
    }

    /* all elements (except <h2>'s) when data-collapsed attribute true */
    *:not(h2)[data-collapsed="true"] {
      display: none;
    }

    /* accordion arrow button when hovered and <h2>'s when hovered */
    .accordion_arrow:hover,
    h2[data-collapsed="true"]:hover,
    h2[data-collapsed="false"]:hover {
      cursor: pointer;
    }
  }

  /* always hide accordion arrow button on print */
  @media only print {
    .accordion_arrow {
      display: none;
    }
  }
</style>
<!--
  Anchors Plugin

  Adds an anchor next to each of a certain type of element that provides a
  human-readable url to that specific item/position in the document (e.g.
  "manuscript.html#abstract"). It also makes it such that scrolling out of view
  of a target removes its identifier from the url.
-->

<script type="module">
  // which types of elements to add anchors next to, in "document.querySelector"
  // format
  const typesQuery =
    'h1, h2, h3, div[id^="fig:"], div[id^="tbl:"], span[id^="eq:"]';

  // start script
  function start() {
    // add anchor to each element of specified types
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) addAnchor(element);

    // attach scroll listener to window
    window.addEventListener("scroll", onScroll);
  }

  // when window is scrolled
  function onScroll() {
    // if url has hash and user has scrolled out of view of hash
    // target, remove hash from url
    const tolerance = 100;
    const target = getHashTarget();
    if (target) {
      if (
        target.getBoundingClientRect().top > window.innerHeight + tolerance ||
        target.getBoundingClientRect().bottom < 0 - tolerance
      )
        history.pushState(null, null, " ");
    }
  }

  // add anchor to element
  function addAnchor(element) {
    let addTo; // element to add anchor button to

    // if figure or table, modify withId and addTo to get expected
    // elements
    if (element.id.indexOf("fig:") === 0) {
      addTo = element.querySelector("figcaption");
    } else if (element.id.indexOf("tbl:") === 0) {
      addTo = element.querySelector("caption");
    } else if (element.id.indexOf("eq:") === 0) {
      addTo = element.querySelector(".eqnos-number");
    }

    addTo = addTo || element;
    const id = element.id || null;

    // do not add anchor if element doesn't have assigned id.
    // id is generated by pandoc and is assumed to be unique and
    // human-readable
    if (!id) return;

    // create anchor button
    const anchor = document.createElement("a");
    anchor.innerHTML = document.querySelector(".icon_link").innerHTML;
    anchor.title = "Link to this part of the document";
    anchor.classList.add("icon_button", "anchor");
    anchor.dataset.ignore = "true";
    anchor.href = "#" + id;
    addTo.appendChild(anchor);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- link icon -->

<template class="icon_link">
  <!-- modified from: https://fontawesome.com/icons/link -->
  <svg width="16" height="16" viewBox="0 0 512 512">
    <path
      fill="currentColor"
      d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* anchor button */
    .anchor {
      opacity: 0;
      margin-left: 5px;
    }

    /* anchor buttons within <h2>'s */
    h2 .anchor {
      margin-left: 10px;
    }

    /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
    *:hover > .anchor,
    .anchor:hover,
    .anchor:focus {
      opacity: 1;
    }

    /* anchor button when hovered */
    .anchor:hover {
      cursor: pointer;
    }
  }

  /* always show anchor button on devices with no mouse/hover ability */
  @media (hover: none) {
    .anchor {
      opacity: 1;
    }
  }

  /* always hide anchor button on print */
  @media only print {
    .anchor {
      display: none;
    }
  }
</style>
<!-- 
    Attributes Plugin

    Allows arbitrary HTML attributes to be attached to (almost) any element.
    Place an HTML comment inside or next to the desired element with the content:
    $attribute="value"
-->

<script type="module">
  // start script
  function start() {
    // get list of comments in document
    const comments = findComments();

    for (const comment of comments)
      if (comment.parentElement)
        addAttributes(comment.parentElement, comment.nodeValue.trim());
  }

  // add html attributes to specified element based on string of
  // html attributes and values
  function addAttributes(element, text) {
    // regex's for finding attribute/value pairs in the format of
    // attribute="value" or attribute='value
    const regex2 = /\$([a-zA-Z\-]+)?=\"(.+?)\"/;
    const regex1 = /\$([a-zA-Z\-]+)?=\'(.+?)\'/;

    // loop through attribute/value pairs
    let match;
    while ((match = text.match(regex2) || text.match(regex1))) {
      // get attribute and value from regex capture groups
      let attribute = match[1];
      let value = match[2];

      // remove from string
      text = text.substring(match.index + match[0].length);

      if (!attribute || !value) break;

      // set attribute of parent element
      try {
        element.setAttribute(attribute, value);
      } catch (error) {
        console.log(error);
      }

      // special case for colspan
      if (attribute === "colspan") removeTableCells(element, value);
    }
  }

  // get list of comment elements in document
  function findComments() {
    const comments = [];

    // iterate over comment nodes in document
    function acceptNode(node) {
      return NodeFilter.FILTER_ACCEPT;
    }
    const iterator = document.createNodeIterator(
      document.body,
      NodeFilter.SHOW_COMMENT,
      acceptNode
    );
    let node;
    while ((node = iterator.nextNode())) comments.push(node);

    return comments;
  }

  // remove certain number of cells after specified cell
  function removeTableCells(cell, number) {
    number = parseInt(number);
    if (!number) return;

    // remove elements
    for (; number > 1; number--) {
      if (cell.nextElementSibling) cell.nextElementSibling.remove();
    }
  }

  // start script on DOMContentLoaded instead of load to ensure this plugins
  // runs before other plugins
  window.addEventListener("DOMContentLoaded", start);
</script>
<!--
  Jump to First Plugin

  Adds a button next to each reference entry, figure, and table that jumps the
  page to the first occurrence of a link to that item in the manuscript.
-->

<script type="module">
  // whether to add buttons next to reference entries
  const references = "true";
  // whether to add buttons next to figures
  const figures = "true";
  // whether to add buttons next to tables
  const tables = "true";

  // start script
  function start() {
    if (references !== "false")
      makeButtons(`div[id^="ref-"]`, ".csl-left-margin", "reference");
    if (figures !== "false")
      makeButtons(`div[id^="fig:"]`, "figcaption", "figure");
    if (tables !== "false") makeButtons(`div[id^="tbl:"]`, "caption", "table");
  }

  // when jump button clicked
  function onButtonClick() {
    const first = getFirstOccurrence(this.dataset.id);
    if (!first) return;

    // update url hash so navigating "back" in history will return user to button
    window.location.hash = this.dataset.id;
    // scroll to link
    const timeout = function () {
      goToElement(first, window.innerHeight * 0.5);
    };
    window.setTimeout(timeout, 0);
  }

  // get first occurrence of link to item in document
  function getFirstOccurrence(id) {
    let query = "a";
    query += '[href="#' + id + '"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelector(query);
  }

  // add button next to each reference entry, figure, or table
  function makeButtons(query, containerQuery, subject) {
    const elements = document.querySelectorAll(query);
    for (const element of elements) {
      const id = element.id;
      const buttonContainer = element.querySelector(containerQuery);
      const first = getFirstOccurrence(id);

      // if can't find link to reference or place to put button, ignore
      if (!first || !buttonContainer) continue;

      // make jump button
      let button = document.createElement("button");
      button.classList.add("icon_button", "jump_arrow");
      button.title = `Jump to the first occurrence of this ${subject} in the document`;
      const icon = document.querySelector(".icon_angle_double_up");
      button.innerHTML = icon.innerHTML;
      button.dataset.id = id;
      button.dataset.ignore = "true";
      button.addEventListener("click", onButtonClick);
      buttonContainer.prepend(button);
    }
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle double up icon -->

<template class="icon_angle_double_up">
  <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
  <svg width="16" height="16" viewBox="0 0 320 512">
    <path
      fill="currentColor"
      d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* jump button */
    .jump_arrow {
      position: relative;
      top: 0.125em;
      margin-right: 5px;
    }
  }

  /* always hide jump button on print */
  @media only print {
    .jump_arrow {
      display: none;
    }
  }
</style>
<!-- 
    Lightbox Plugin

    Makes it such that when a user clicks on an image, the image fills the
    screen and the user can pan/drag/zoom the image and navigate between other
    images in the document.
-->

<script type="module">
  // list of possible zoom/scale factors
  const zooms =
    "0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1, 1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8";
  // whether to fit image to view ('fit'), display at 100% and shrink if
  // necessary ('shrink'), or always display at 100% ('100')
  const defaultZoom = "fit";
  // whether to zoom in/out toward center of view ('true') or mouse ('false')
  const centerZoom = "false";

  // start script
  function start() {
    // run through each <img> element
    const imgs = document.querySelectorAll("figure > img");
    let count = 1;
    for (const img of imgs) {
      img.classList.add("lightbox_document_img");
      img.dataset.number = count;
      img.dataset.total = imgs.length;
      img.addEventListener("click", openLightbox);
      count++;
    }

    // attach mouse and key listeners to window
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("keyup", onKeyUp);
  }

  // when mouse is moved anywhere in window
  function onWindowMouseMove(event) {
    window.mouseX = event.clientX;
    window.mouseY = event.clientY;
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("lightbox_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("lightbox_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeLightbox();
        break;
    }
  }

  // open lightbox
  function openLightbox() {
    const lightbox = makeLightbox(this);
    if (!lightbox) return;

    blurBody(lightbox);
    document.body.appendChild(lightbox);
  }

  // make lightbox
  function makeLightbox(img) {
    // delete lightbox if it exists, start fresh
    closeLightbox();

    // create screen overlay containing lightbox
    const overlay = document.createElement("div");
    overlay.id = "lightbox_overlay";

    // create image info boxes
    const numberInfo = document.createElement("div");
    const zoomInfo = document.createElement("div");
    numberInfo.id = "lightbox_number_info";
    zoomInfo.id = "lightbox_zoom_info";

    // create container for image
    const imageContainer = document.createElement("div");
    imageContainer.id = "lightbox_image_container";
    const lightboxImg = makeLightboxImg(
      img,
      imageContainer,
      numberInfo,
      zoomInfo
    );
    imageContainer.appendChild(lightboxImg);

    // create bottom container for caption and navigation buttons
    const bottomContainer = document.createElement("div");
    bottomContainer.id = "lightbox_bottom_container";
    const caption = makeCaption(img);
    const prevButton = makePrevButton(img);
    const nextButton = makeNextButton(img);
    bottomContainer.appendChild(prevButton);
    bottomContainer.appendChild(caption);
    bottomContainer.appendChild(nextButton);

    // attach top middle and bottom to overlay
    overlay.appendChild(numberInfo);
    overlay.appendChild(zoomInfo);
    overlay.appendChild(imageContainer);
    overlay.appendChild(bottomContainer);

    return overlay;
  }

  // make <img> object that is intuitively draggable and zoomable
  function makeLightboxImg(sourceImg, container, numberInfoBox, zoomInfoBox) {
    // create copy of source <img>
    const img = sourceImg.cloneNode(true);
    img.classList.remove("lightbox_document_img");
    img.removeAttribute("id");
    img.removeAttribute("width");
    img.removeAttribute("height");
    img.style.position = "unset";
    img.style.margin = "0";
    img.style.padding = "0";
    img.style.width = "";
    img.style.height = "";
    img.style.minWidth = "";
    img.style.minHeight = "";
    img.style.maxWidth = "";
    img.style.maxHeight = "";
    img.id = "lightbox_img";

    // build sorted list of zoomSteps
    const zoomSteps = zooms.split(/[^0-9.]/).map((step) => parseFloat(step));
    zoomSteps.sort((a, b) => a - b);

    // <img> object property variables
    let zoom = 1;
    let translateX = 0;
    let translateY = 0;
    let clickMouseX = undefined;
    let clickMouseY = undefined;
    let clickTranslateX = undefined;
    let clickTranslateY = undefined;

    updateNumberInfo();

    // update image numbers displayed in info box
    function updateNumberInfo() {
      numberInfoBox.innerHTML =
        sourceImg.dataset.number + " of " + sourceImg.dataset.total;
    }

    // update zoom displayed in info box
    function updateZoomInfo() {
      let zoomInfo = zoom * 100;
      if (!Number.isInteger(zoomInfo)) zoomInfo = zoomInfo.toFixed(2);
      zoomInfoBox.innerHTML = zoomInfo + "%";
    }

    // move to closest zoom step above current zoom
    const zoomIn = function () {
      for (const zoomStep of zoomSteps) {
        if (zoomStep > zoom) {
          zoom = zoomStep;
          break;
        }
      }
      updateTransform();
    };

    // move to closest zoom step above current zoom
    const zoomOut = function () {
      zoomSteps.reverse();
      for (const zoomStep of zoomSteps) {
        if (zoomStep < zoom) {
          zoom = zoomStep;
          break;
        }
      }
      zoomSteps.reverse();

      updateTransform();
    };

    // update display of <img> based on scale/translate properties
    const updateTransform = function () {
      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      // get new width/height after scale
      const rect = img.getBoundingClientRect();
      // limit translate
      translateX = Math.max(translateX, -rect.width / 2);
      translateX = Math.min(translateX, rect.width / 2);
      translateY = Math.max(translateY, -rect.height / 2);
      translateY = Math.min(translateY, rect.height / 2);

      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      updateZoomInfo();
    };

    // fit <img> to container
    const fit = function () {
      // no x/y offset, 100% zoom by default
      translateX = 0;
      translateY = 0;
      zoom = 1;

      // widths of <img> and container
      const imgWidth = img.naturalWidth;
      const imgHeight = img.naturalHeight;
      const containerWidth = parseFloat(
        window.getComputedStyle(container).width
      );
      const containerHeight = parseFloat(
        window.getComputedStyle(container).height
      );

      // how much zooming is needed to fit <img> to container
      const xRatio = imgWidth / containerWidth;
      const yRatio = imgHeight / containerHeight;
      const maxRatio = Math.max(xRatio, yRatio);
      const newZoom = 1 / maxRatio;

      // fit <img> to container according to option
      if (defaultZoom === "shrink") {
        if (maxRatio > 1) zoom = newZoom;
      } else if (defaultZoom === "fit") zoom = newZoom;

      updateTransform();
    };

    // when mouse wheel is rolled anywhere in container
    const onContainerWheel = function (event) {
      if (!event) return;

      // let ctrl + mouse wheel to zoom behave as normal
      if (event.ctrlKey) return;

      // prevent normal scroll behavior
      event.preventDefault();
      event.stopPropagation();

      // point around which to scale img
      const viewRect = container.getBoundingClientRect();
      const viewX = (viewRect.left + viewRect.right) / 2;
      const viewY = (viewRect.top + viewRect.bottom) / 2;
      const originX = centerZoom === "true" ? viewX : mouseX;
      const originY = centerZoom === "true" ? viewY : mouseY;

      // get point on image under origin
      const oldRect = img.getBoundingClientRect();
      const oldPercentX = (originX - oldRect.left) / oldRect.width;
      const oldPercentY = (originY - oldRect.top) / oldRect.height;

      // increment/decrement zoom
      if (event.deltaY < 0) zoomIn();
      if (event.deltaY > 0) zoomOut();

      // get offset between previous image point and origin
      const newRect = img.getBoundingClientRect();
      const offsetX = originX - (newRect.left + newRect.width * oldPercentX);
      const offsetY = originY - (newRect.top + newRect.height * oldPercentY);

      // translate image to keep image point under origin
      translateX += offsetX;
      translateY += offsetY;

      // perform translate
      updateTransform();
    };

    // when container is clicked
    function onContainerClick(event) {
      // if container itself is target of click, and not other
      // element above it
      if (event.target === this) closeLightbox();
    }

    // when mouse button is pressed on image
    const onImageMouseDown = function (event) {
      // store original mouse position relative to image
      clickMouseX = window.mouseX;
      clickMouseY = window.mouseY;
      clickTranslateX = translateX;
      clickTranslateY = translateY;
      event.stopPropagation();
      event.preventDefault();
    };

    // when mouse button is released anywhere in window
    const onWindowMouseUp = function (event) {
      // reset original mouse position
      clickMouseX = undefined;
      clickMouseY = undefined;
      clickTranslateX = undefined;
      clickTranslateY = undefined;

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mouseup", onWindowMouseUp);
    };

    // when mouse is moved anywhere in window
    const onWindowMouseMove = function (event) {
      if (
        clickMouseX === undefined ||
        clickMouseY === undefined ||
        clickTranslateX === undefined ||
        clickTranslateY === undefined
      )
        return;

      // offset image based on original and current mouse position
      translateX = clickTranslateX + window.mouseX - clickMouseX;
      translateY = clickTranslateY + window.mouseY - clickMouseY;
      updateTransform();
      event.preventDefault();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mousemove", onWindowMouseMove);
    };

    // when window is resized
    const onWindowResize = function (event) {
      fit();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("resize", onWindowResize);
    };

    // attach the necessary event listeners
    img.addEventListener("dblclick", fit);
    img.addEventListener("mousedown", onImageMouseDown);
    container.addEventListener("wheel", onContainerWheel);
    container.addEventListener("mousedown", onContainerClick);
    container.addEventListener("touchstart", onContainerClick);
    window.addEventListener("mouseup", onWindowMouseUp);
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("resize", onWindowResize);

    // run fit() after lightbox atttached to document and <img> Loaded
    // so needed container and img dimensions available
    img.addEventListener("load", fit);

    return img;
  }

  // make caption
  function makeCaption(img) {
    const caption = document.createElement("div");
    caption.id = "lightbox_caption";
    const captionSource = img.nextElementSibling;
    if (captionSource.tagName.toLowerCase() === "figcaption") {
      const captionCopy = makeCopy(captionSource);
      caption.innerHTML = captionCopy.innerHTML;
    }

    caption.addEventListener("touchstart", function (event) {
      event.stopPropagation();
    });

    return caption;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // make button to jump to previous image in document
  function makePrevButton(img) {
    const prevButton = document.createElement("button");
    prevButton.id = "lightbox_prev_button";
    prevButton.title = "Jump to the previous image in the document [←]";
    prevButton.classList.add("icon_button", "lightbox_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;

    // attach click listeners to button
    prevButton.addEventListener("click", function () {
      getPrevImg(img).click();
    });

    return prevButton;
  }

  // make button to jump to next image in document
  function makeNextButton(img) {
    const nextButton = document.createElement("button");
    nextButton.id = "lightbox_next_button";
    nextButton.title = "Jump to the next image in the document [→]";
    nextButton.classList.add("icon_button", "lightbox_button");
    nextButton.innerHTML = document.querySelector(
      ".icon_caret_right"
    ).innerHTML;

    // attach click listeners to button
    nextButton.addEventListener("click", function () {
      getNextImg(img).click();
    });

    return nextButton;
  }

  // get previous image in document
  function getPrevImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if < 1
    if (index - 1 >= 0) index--;
    else index = imgs.length - 1;
    return imgs[index];
  }

  // get next image in document
  function getNextImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if > total
    if (index + 1 <= imgs.length - 1) index++;
    else index = 0;
    return imgs[index];
  }

  // close lightbox
  function closeLightbox() {
    focusBody();

    const lightbox = document.getElementById("lightbox_overlay");
    if (lightbox) lightbox.remove();
  }

  // make all elements behind lightbox non-focusable
  function blurBody(overlay) {
    const all = document.querySelectorAll("*");
    for (const element of all) element.tabIndex = -1;
    document.body.classList.add("body_no_scroll");
  }

  // make all elements focusable again
  function focusBody() {
    const all = document.querySelectorAll("*");
    for (const element of all) element.removeAttribute("tabIndex");
    document.body.classList.remove("body_no_scroll");
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* regular <img> in document when hovered */
    img.lightbox_document_img:hover {
      cursor: pointer;
    }

    .body_no_scroll {
      overflow: hidden !important;
    }

    /* screen overlay */
    #lightbox_overlay {
      display: flex;
      flex-direction: column;
      position: fixed;
      left: 0;
      top: 0;
      right: 0;
      bottom: 0;
      background: rgba(0, 0, 0, 0.75);
      z-index: 3;
    }

    /* middle area containing lightbox image */
    #lightbox_image_container {
      flex-grow: 1;
      display: flex;
      justify-content: center;
      align-items: center;
      overflow: hidden;
      position: relative;
      padding: 20px;
    }

    /* bottom area containing caption */
    #lightbox_bottom_container {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 100px;
      min-height: 100px;
      max-height: 100px;
      background: rgba(0, 0, 0, 0.5);
    }

    /* image number info text box */
    #lightbox_number_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      left: 2px;
      top: 0;
      z-index: 4;
    }

    /* zoom info text box */
    #lightbox_zoom_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      right: 2px;
      top: 0;
      z-index: 4;
    }

    /* copy of image caption */
    #lightbox_caption {
      box-sizing: border-box;
      display: inline-block;
      width: 100%;
      max-height: 100%;
      padding: 10px 0;
      text-align: center;
      overflow-y: auto;
      color: #ffffff;
    }

    /* navigation previous/next button */
    .lightbox_button {
      width: 100px;
      height: 100%;
      min-width: 100px;
      min-height: 100%;
      color: #ffffff;
    }

    /* navigation previous/next button when hovered */
    .lightbox_button:hover {
      background: none !important;
    }

    /* navigation button icon */
    .lightbox_button > svg {
      height: 25px;
    }

    /* figure auto-number */
    #lightbox_caption > span:first-of-type {
      font-weight: bold;
      margin-right: 5px;
    }

    /* lightbox image when hovered */
    #lightbox_img:hover {
      cursor: grab;
    }

    /* lightbox image when grabbed */
    #lightbox_img:active {
      cursor: grabbing;
    }
  }

  /* when on screen < 480px wide */
  @media only screen and (max-width: 480px) {
    /* make navigation buttons skinnier on small screens to make more room for caption text */
    .lightbox_button {
      width: 50px;
      min-width: 50px;
    }
  }

  /* always hide lightbox on print */
  @media only print {
    #lightbox_overlay {
      display: none;
    }
  }
</style>
<!-- 
  Link Highlight Plugin

  Makes it such that when a user hovers or focuses a link, other links that have
  the same target will be highlighted. It also makes it such that when clicking
  a link, the target of the link (eg reference, figure, table) is briefly
  highlighted.
-->

<script type="module">
  // whether to also highlight links that go to external urls
  const externalLinks = "false";
  // whether user must click off to unhighlight instead of just
  // un-hovering
  const clickUnhighlight = "false";
  // whether to also highlight links that are unique
  const highlightUnique = "true";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach mouse and focus listeners to link
      link.addEventListener("mouseenter", onLinkFocus);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("mouseleave", onLinkUnhover);
    }

    // attach click and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("hashchange", onHashChange);

    // run hash change on window load in case user has navigated
    // directly to hash
    onHashChange();
  }

  // when link is focused (tabbed to) or hovered
  function onLinkFocus() {
    highlight(this);
  }

  // when link is unhovered
  function onLinkUnhover() {
    if (clickUnhighlight !== "true") unhighlightAll();
  }

  // when the mouse is clicked anywhere in window
  function onClick(event) {
    unhighlightAll();
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) glowElement(target);
  }

  // start glow sequence on an element
  function glowElement(element) {
    const startGlow = function () {
      onGlowEnd();
      element.dataset.glow = "true";
      element.addEventListener("animationend", onGlowEnd);
    };
    const onGlowEnd = function () {
      element.removeAttribute("data-glow");
      element.removeEventListener("animationend", onGlowEnd);
    };
    startGlow();
  }

  // highlight link and all others with same target
  function highlight(link) {
    // force unhighlight all to start fresh
    unhighlightAll();

    // get links with same target
    if (!link) return;
    const sameLinks = getSameLinks(link);

    // if link unique and option is off, exit and don't highlight
    if (sameLinks.length <= 1 && highlightUnique !== "true") return;

    // highlight all same links, and "select" (special highlight) this
    // one
    for (const sameLink of sameLinks) {
      if (sameLink === link) sameLink.setAttribute("data-selected", "true");
      else sameLink.setAttribute("data-highlighted", "true");
    }
  }

  // unhighlight all links
  function unhighlightAll() {
    const links = getLinks();
    for (const link of links) {
      link.setAttribute("data-selected", "false");
      link.setAttribute("data-highlighted", "false");
    }
  }

  // get links with same target
  function getSameLinks(link) {
    const results = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        results.push(otherLink);
    }
    return results;
  }

  // get all links of types we wish to handle
  function getLinks() {
    let query = "a";
    if (externalLinks !== "true") query += '[href^="#"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelectorAll(query);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<style>
  @media only screen {
    /* anything with data-highlighted attribute true */
    [data-highlighted="true"] {
      background: #ffeb3b;
    }

    /* anything with data-selected attribute true */
    [data-selected="true"] {
      background: #ff8a65 !important;
    }

    /* animation definition for glow */
    @keyframes highlight_glow {
      0% {
        background: none;
      }
      10% {
        background: #bbdefb;
      }
      100% {
        background: none;
      }
    }

    /* anything with data-glow attribute true */
    [data-glow="true"] {
      animation: highlight_glow 2s;
    }
  }
</style>
<!--
  Table of Contents Plugin

  Provides a "table of contents" (toc) panel on the side of the document that
  allows the user to conveniently navigate between sections of the document.
-->

<script type="module">
  // which types of elements to add links for, in "document.querySelector" format
  const typesQuery = "h1, h2, h3";
  // whether toc starts open. use 'true' or 'false', or 'auto' to
  // use 'true' behavior when screen wide enough and 'false' when not
  const startOpen = "false";
  // whether toc closes when clicking on toc link. use 'true' or
  // 'false', or 'auto' to use 'false' behavior when screen wide
  // enough and 'true' when not
  const clickClose = "auto";
  // if list item is more than this many characters, text will be
  // truncated
  const charLimit = "50";
  // whether or not to show bullets next to each toc item
  const bullets = "false";

  // start script
  function start() {
    // make toc panel and populate with entries (links to document
    // sections)
    const panel = makePanel();
    if (!panel) return;
    makeEntries(panel);
    // attach panel to document after making entries, so 'toc' heading
    // in panel isn't included in toc
    document.body.insertBefore(panel, document.body.firstChild);

    // initial panel state
    if (startOpen === "true" || (startOpen === "auto" && !isSmallScreen()))
      openPanel();
    else closePanel();

    // attach click, scroll, and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("scroll", onScroll);
    window.addEventListener("hashchange", onScroll);
    window.addEventListener("keyup", onKeyUp);
    onScroll();

    // add class to push document body down out of way of toc button
    document.body.classList.add("toc_body_nudge");
  }

  // determine if screen wide enough to fit toc panel
  function isSmallScreen() {
    // in default theme:
    // 816px = 8.5in = width of "page" (<body>) element
    // 260px = min width of toc panel (*2 for both sides of <body>)
    return window.innerWidth < 816 + 260 * 2;
  }

  // when mouse is clicked anywhere in window
  function onClick() {
    if (isSmallScreen()) closePanel();
  }

  // when window is scrolled or hash changed
  function onScroll() {
    highlightViewed();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    // close on esc
    if (event.key === "Escape") closePanel();
  }

  // find entry of currently viewed document section in toc and highlight
  function highlightViewed() {
    const firstId = getFirstInView(typesQuery);

    // get toc entries (links), unhighlight all, then highlight viewed
    const list = document.getElementById("toc_list");
    if (!firstId || !list) return;
    const links = list.querySelectorAll("a");
    for (const link of links) link.dataset.viewing = "false";
    const link = list.querySelector('a[href="#' + firstId + '"]');
    if (!link) return;
    link.dataset.viewing = "true";
  }

  // get first or previous toc listed element in top half of view
  function getFirstInView(query) {
    // get all elements matching query and with id
    const elements = document.querySelectorAll(query);
    const elementsWithIds = [];
    for (const element of elements) {
      if (element.id) elementsWithIds.push(element);
    }

    // get first or previous element in top half of view
    for (let i = 0; i < elementsWithIds.length; i++) {
      const element = elementsWithIds[i];
      const prevElement = elementsWithIds[Math.max(0, i - 1)];
      if (element.getBoundingClientRect().top >= 0) {
        if (element.getBoundingClientRect().top < window.innerHeight / 2)
          return element.id;
        else return prevElement.id;
      }
    }
  }

  // make panel
  function makePanel() {
    // create panel
    const panel = document.createElement("div");
    panel.id = "toc_panel";
    if (bullets === "true") panel.dataset.bullets = "true";

    // create header
    const header = document.createElement("div");
    header.id = "toc_header";

    // create toc button
    const button = document.createElement("button");
    button.id = "toc_button";
    button.innerHTML = document.querySelector(".icon_th_list").innerHTML;
    button.title = "Table of Contents";
    button.classList.add("icon_button");

    // create header text
    const text = document.createElement("h4");
    text.innerHTML = "Table of Contents";

    // create container for toc list
    const list = document.createElement("div");
    list.id = "toc_list";

    // attach click listeners
    panel.addEventListener("click", onPanelClick);
    header.addEventListener("click", onHeaderClick);
    button.addEventListener("click", onButtonClick);

    // attach elements
    header.appendChild(button);
    header.appendChild(text);
    panel.appendChild(header);
    panel.appendChild(list);

    return panel;
  }

  // create toc entries (links) to each element of the specified types
  function makeEntries(panel) {
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) {
      // do not add link if element doesn't have assigned id
      if (!element.id) continue;

      // create link/list item
      const link = document.createElement("a");
      link.classList.add("toc_link");
      switch (element.tagName.toLowerCase()) {
        case "h1":
          link.dataset.level = "1";
          break;
        case "h2":
          link.dataset.level = "2";
          break;
        case "h3":
          link.dataset.level = "3";
          break;
        case "h4":
          link.dataset.level = "4";
          break;
      }
      link.title = element.innerText;
      let text = element.innerText;
      if (text.length > charLimit) text = text.slice(0, charLimit) + "...";
      link.innerHTML = text;
      link.href = "#" + element.id;
      link.addEventListener("click", onLinkClick);

      // attach link
      panel.querySelector("#toc_list").appendChild(link);
    }
  }

  // when panel is clicked
  function onPanelClick(event) {
    // stop click from propagating to window/document and closing panel
    event.stopPropagation();
  }

  // when header itself is clicked
  function onHeaderClick(event) {
    togglePanel();
  }

  // when button is clicked
  function onButtonClick(event) {
    togglePanel();
    // stop header underneath button from also being clicked
    event.stopPropagation();
  }

  // when link is clicked
  function onLinkClick(event) {
    if (clickClose === "true" || (clickClose === "auto" && isSmallScreen()))
      closePanel();
    else openPanel();
  }

  // open panel if closed, close if opened
  function togglePanel() {
    const panel = document.getElementById("toc_panel");
    if (!panel) return;

    if (panel.dataset.open === "true") closePanel();
    else openPanel();
  }

  // open panel
  function openPanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "true";
  }

  // close panel
  function closePanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "false";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- th list icon -->

<template class="icon_th_list">
  <!-- modified from: https://fontawesome.com/icons/th-list -->
  <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
    <path
      fill="currentColor"
      d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* toc panel */
    #toc_panel {
      box-sizing: border-box;
      position: fixed;
      top: 0;
      left: 0;
      background: #ffffff;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      z-index: 2;
    }

    /* toc panel when closed */
    #toc_panel[data-open="false"] {
      min-width: 60px;
      width: 60px;
      height: 60px;
      border-right: solid 1px #bdbdbd;
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc panel when open */
    #toc_panel[data-open="true"] {
      min-width: 260px;
      max-width: 480px;
      /* keep panel edge consistent distance away from "page" edge */
      width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
      bottom: 0;
      border-right: solid 1px #bdbdbd;
    }

    /* toc panel header */
    #toc_header {
      box-sizing: border-box;
      display: flex;
      flex-direction: row;
      align-items: center;
      height: 60px;
      margin: 0;
      padding: 20px;
    }

    /* toc panel header when hovered */
    #toc_header:hover {
      cursor: pointer;
    }

    /* toc panel header when panel open */
    #toc_panel[data-open="true"] > #toc_header {
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc open/close header button */
    #toc_button {
      margin-right: 20px;
    }

    /* hide toc list and header text when closed */
    #toc_panel[data-open="false"] > #toc_header > *:not(#toc_button),
    #toc_panel[data-open="false"] > #toc_list {
      display: none;
    }

    /* toc list of entries */
    #toc_list {
      box-sizing: border-box;
      width: 100%;
      padding: 20px;
      position: absolute;
      top: calc(60px + 1px);
      bottom: 0;
      overflow: auto;
    }

    /* toc entry, link to section in document */
    .toc_link {
      display: block;
      padding: 5px;
      position: relative;
      font-weight: 600;
      text-decoration: none;
    }

    /* toc entry when hovered or when "viewed" */
    .toc_link:hover,
    .toc_link[data-viewing="true"] {
      background: #f5f5f5;
    }

    /* toc entry, level 1 indentation */
    .toc_link[data-level="1"] {
      margin-left: 0;
    }

    /* toc entry, level 2 indentation */
    .toc_link[data-level="2"] {
      margin-left: 20px;
    }

    /* toc entry, level 3 indentation */
    .toc_link[data-level="3"] {
      margin-left: 40px;
    }

    /* toc entry, level 4 indentation */
    .toc_link[data-level="4"] {
      margin-left: 60px;
    }

    /* toc entry bullets */
    #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
      position: absolute;
      left: -15px;
      top: -1px;
      font-size: 1.5em;
    }

    /* toc entry, level 2 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
      content: "\2022";
    }

    /* toc entry, level 3 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
      content: "\25AB";
    }

    /* toc entry, level 4 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
      content: "-";
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* push <body> ("page") element down to make room for toc icon */
    .toc_body_nudge {
      padding-top: 60px;
    }

    /* toc icon when panel closed and not hovered */
    #toc_panel[data-open="false"]:not(:hover) {
      background: rgba(255, 255, 255, 0.75);
    }
  }

  /* always hide toc panel on print */
  @media only print {
    #toc_panel {
      display: none;
    }
  }
</style>
<!-- 
  Tooltips Plugin

  Makes it such that when the user hovers or focuses a link to a citation or
  figure, a tooltip appears with a preview of the reference content, along with
  arrows to navigate between instances of the same reference in the document.
-->

<script type="module">
  // whether user must click off to close tooltip instead of just un-hovering
  const clickClose = "false";
  // delay (in ms) between opening and closing tooltip
  const delay = "100";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach hover and focus listeners to link
      link.addEventListener("mouseover", onLinkHover);
      link.addEventListener("mouseleave", onLinkUnhover);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("touchend", onLinkTouch);
    }

    // attach mouse, key, and resize listeners to window
    window.addEventListener("mousedown", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("keyup", onKeyUp);
    window.addEventListener("resize", onResize);
  }

  // when link is hovered
  function onLinkHover() {
    // function to open tooltip
    const delayOpenTooltip = function () {
      openTooltip(this);
    }.bind(this);

    // run open function after delay
    this.openTooltipTimer = window.setTimeout(delayOpenTooltip, delay);
  }

  // when mouse leaves link
  function onLinkUnhover() {
    // cancel opening tooltip
    window.clearTimeout(this.openTooltipTimer);

    // don't close on unhover if option specifies
    if (clickClose === "true") return;

    // function to close tooltip
    const delayCloseTooltip = function () {
      // if tooltip open and if mouse isn't over tooltip, close
      const tooltip = document.getElementById("tooltip");
      if (tooltip && !tooltip.matches(":hover")) closeTooltip();
    };

    // run close function after delay
    this.closeTooltipTimer = window.setTimeout(delayCloseTooltip, delay);
  }

  // when link is focused (tabbed to)
  function onLinkFocus(event) {
    openTooltip(this);
  }

  // when link is touched on touch screen
  function onLinkTouch(event) {
    // attempt to force hover state on first tap always, and trigger
    // regular link click (and navigation) on second tap
    if (event.target === document.activeElement) event.target.click();
    else {
      document.activeElement.blur();
      event.target.focus();
    }
    if (event.cancelable) event.preventDefault();
    event.stopPropagation();
    return false;
  }

  // when mouse is clicked anywhere in window
  function onClick(event) {
    closeTooltip();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("tooltip_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("tooltip_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeTooltip();
        break;
    }
  }

  // when window is resized or zoomed
  function onResize() {
    closeTooltip();
  }

  // get all links of types we wish to handle
  function getLinks() {
    const queries = [];
    // exclude buttons, anchor links, toc links, etc
    const exclude =
      ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    queries.push('a[href^="#ref-"]' + exclude); // citation links
    queries.push('a[href^="#fig:"]' + exclude); // figure links
    const query = queries.join(", ");
    return document.querySelectorAll(query);
  }

  // get links with same target, get index of link in set, get total
  // same links
  function getSameLinks(link) {
    const sameLinks = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        sameLinks.push(otherLink);
    }

    return {
      elements: sameLinks,
      index: sameLinks.indexOf(link),
      total: sameLinks.length,
    };
  }

  // open tooltip
  function openTooltip(link) {
    // delete tooltip if it exists, start fresh
    closeTooltip();

    // make tooltip element
    const tooltip = makeTooltip(link);

    // if source couldn't be found and tooltip not made, exit
    if (!tooltip) return;

    // make navbar elements
    const navBar = makeNavBar(link);
    if (navBar) tooltip.firstElementChild.appendChild(navBar);

    // attach tooltip to page
    document.body.appendChild(tooltip);

    // position tooltip
    const position = function () {
      positionTooltip(link);
    };
    position();

    // if tooltip contains images, position again after they've loaded
    const imgs = tooltip.querySelectorAll("img");
    for (const img of imgs) img.addEventListener("load", position);
  }

  // close (delete) tooltip
  function closeTooltip() {
    const tooltip = document.getElementById("tooltip");
    if (tooltip) tooltip.remove();
  }

  // make tooltip
  function makeTooltip(link) {
    // get target element that link points to
    const source = getSource(link);

    // if source can't be found, exit
    if (!source) return;

    // create new tooltip
    const tooltip = document.createElement("div");
    tooltip.id = "tooltip";
    const tooltipContent = document.createElement("div");
    tooltipContent.id = "tooltip_content";
    tooltip.appendChild(tooltipContent);

    // make copy of source node and put in tooltip
    const sourceCopy = makeCopy(source);
    tooltipContent.appendChild(sourceCopy);

    // attach mouse event listeners
    tooltip.addEventListener("click", onTooltipClick);
    tooltip.addEventListener("mousedown", onTooltipClick);
    tooltip.addEventListener("touchstart", onTooltipClick);
    tooltip.addEventListener("mouseleave", onTooltipUnhover);

    // (for interaction with lightbox plugin)
    // transfer click on tooltip copied img to original img
    const sourceImg = source.querySelector("img");
    const sourceCopyImg = sourceCopy.querySelector("img");
    if (sourceImg && sourceCopyImg) {
      const clickImg = function () {
        sourceImg.click();
        closeTooltip();
      };
      sourceCopyImg.addEventListener("click", clickImg);
    }

    return tooltip;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
      "class",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // when tooltip is clicked
  function onTooltipClick(event) {
    // when user clicks on tooltip, stop click from transferring
    // outside of tooltip (eg, click off to close tooltip, or eg click
    // off to unhighlight same refs)
    event.stopPropagation();
  }

  // when tooltip is unhovered
  function onTooltipUnhover(event) {
    if (clickClose === "true") return;

    // make sure new mouse/touch/focus no longer over tooltip or any
    // element within it
    const tooltip = document.getElementById("tooltip");
    if (!tooltip) return;
    if (this.contains(event.relatedTarget)) return;

    closeTooltip();
  }

  // make nav bar to go betwen prev/next instances of same reference
  function makeNavBar(link) {
    // find other links to the same source
    const sameLinks = getSameLinks(link);

    // don't show nav bar when singular reference
    if (sameLinks.total <= 1) return;

    // find prev/next links with same target
    const prevLink = getPrevLink(link, sameLinks);
    const nextLink = getNextLink(link, sameLinks);

    // create nav bar
    const navBar = document.createElement("div");
    navBar.id = "tooltip_nav_bar";
    const text = sameLinks.index + 1 + " of " + sameLinks.total;

    // create nav bar prev/next buttons
    const prevButton = document.createElement("button");
    const nextButton = document.createElement("button");
    prevButton.id = "tooltip_prev_button";
    nextButton.id = "tooltip_next_button";
    prevButton.title =
      "Jump to the previous occurence of this item in the document [←]";
    nextButton.title =
      "Jump to the next occurence of this item in the document [→]";
    prevButton.classList.add("icon_button");
    nextButton.classList.add("icon_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;
    nextButton.innerHTML =
      document.querySelector(".icon_caret_right").innerHTML;
    navBar.appendChild(prevButton);
    navBar.appendChild(document.createTextNode(text));
    navBar.appendChild(nextButton);

    // attach click listeners to buttons
    prevButton.addEventListener("click", function () {
      onPrevNextClick(link, prevLink);
    });
    nextButton.addEventListener("click", function () {
      onPrevNextClick(link, nextLink);
    });

    return navBar;
  }

  // get previous link with same target
  function getPrevLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if < 1
    let index;
    if (sameLinks.index - 1 >= 0) index = sameLinks.index - 1;
    else index = sameLinks.total - 1;
    return sameLinks.elements[index];
  }

  // get next link with same target
  function getNextLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if > total
    let index;
    if (sameLinks.index + 1 <= sameLinks.total - 1) index = sameLinks.index + 1;
    else index = 0;
    return sameLinks.elements[index];
  }

  // get element that is target of link or url hash
  function getSource(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector('[id="' + id + '"]');
    if (!target) return;

    // if ref or figure, modify target to get expected element
    if (id.indexOf("ref-") === 0) target = target.querySelector(":nth-child(2)");
    else if (id.indexOf("fig:") === 0) target = target.querySelector("figure");

    return target;
  }

  // when prev/next arrow button is clicked
  function onPrevNextClick(link, prevNextLink) {
    if (link && prevNextLink)
      goToElement(prevNextLink, window.innerHeight * 0.5);
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);
    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // determine position to place tooltip based on link position in
  // viewport and tooltip size
  function positionTooltip(link, left, top) {
    const tooltipElement = document.getElementById("tooltip");
    if (!tooltipElement) return;

    // get convenient vars for position/dimensions of
    // link/tooltip/page/view
    link = getRectInPage(link);
    const tooltip = getRectInPage(tooltipElement);
    const view = getRectInPage();

    // horizontal positioning
    if (left)
      // use explicit value
      left = left;
    else if (link.left + tooltip.width < view.right)
      // fit tooltip to right of link
      left = link.left;
    else if (link.right - tooltip.width > view.left)
      // fit tooltip to left of link
      left = link.right - tooltip.width;
    // center tooltip in view
    else left = (view.right - view.left) / 2 - tooltip.width / 2;

    // vertical positioning
    if (top)
      // use explicit value
      top = top;
    else if (link.top - tooltip.height > view.top)
      // fit tooltip above link
      top = link.top - tooltip.height;
    else if (link.bottom + tooltip.height < view.bottom)
      // fit tooltip below link
      top = link.bottom;
    else {
      // center tooltip in view
      top = view.top + view.height / 2 - tooltip.height / 2;
      // nudge off of link to left/right if possible
      if (link.right + tooltip.width < view.right) left = link.right;
      else if (link.left - tooltip.width > view.left)
        left = link.left - tooltip.width;
    }

    tooltipElement.style.left = left + "px";
    tooltipElement.style.top = top + "px";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* tooltip container */
    #tooltip {
      position: absolute;
      width: 50%;
      min-width: 240px;
      max-width: 75%;
      z-index: 1;
    }

    /* tooltip content */
    #tooltip_content {
      margin-bottom: 5px;
      padding: 20px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
      overflow-wrap: break-word;
    }

    /* tooltip copy of paragraphs and figures */
    #tooltip_content > p,
    #tooltip_content > figure {
      margin: 0;
      max-height: 320px;
      overflow-y: auto;
    }

    /* tooltip copy of <img> */
    #tooltip_content > figure > img,
    #tooltip_content > figure > svg {
      max-height: 260px;
    }

    /* navigation bar */
    #tooltip_nav_bar {
      margin-top: 10px;
      text-align: center;
    }

    /* navigation bar previous/next buton */
    #tooltip_nav_bar > .icon_button {
      position: relative;
      top: 3px;
    }

    /* navigation bar previous button */
    #tooltip_nav_bar > .icon_button:first-of-type {
      margin-right: 5px;
    }

    /* navigation bar next button */
    #tooltip_nav_bar > .icon_button:last-of-type {
      margin-left: 5px;
    }
  }

  /* always hide tooltip on print */
  @media only print {
    #tooltip {
      display: none;
    }
  }
</style>
<!--
  Analytics Plugin (third-party) 
  
  Copy and paste code from Google Analytics or similar service here.
-->
<!-- 
  Annotations Plugin

  Allows public annotation of the  manuscript. See https://web.hypothes.is/.
-->

<script type="module">
  // configuration
  window.hypothesisConfig = function () {
    return {
      branding: {
        accentColor: "#2196f3",
        appBackgroundColor: "#f8f8f8",
        ctaBackgroundColor: "#f8f8f8",
        ctaTextColor: "#000000",
        selectionFontFamily: "Open Sans, Helvetica, sans serif",
        annotationFontFamily: "Open Sans, Helvetica, sans serif",
      },
    };
  };

  // hypothesis client script
  const embed = "https://hypothes.is/embed.js";
  // hypothesis annotation count query url
  const query = "https://api.hypothes.is/api/search?limit=0&url=";

  // start script
  function start() {
    const button = makeButton();
    document.body.insertBefore(button, document.body.firstChild);
    insertCount(button);
  }

  // make button
  function makeButton() {
    // create button
    const button = document.createElement("button");
    button.id = "hypothesis_button";
    button.innerHTML = document.querySelector(".icon_hypothesis").innerHTML;
    button.title = "Hypothesis annotations";
    button.classList.add("icon_button");

    function onClick(event) {
      onButtonClick(event, button);
    }

    // attach click listeners
    button.addEventListener("click", onClick);

    return button;
  }

  // insert annotations count
  async function insertCount(button) {
    // get annotation count from Hypothesis based on url
    let count = "-";
    try {
      const canonical = document.querySelector('link[rel="canonical"]');
      const location = window.location;
      const url = encodeURIComponent((canonical || location).href);
      const response = await fetch(query + url);
      const json = await response.json();
      count = json.total || "-";
    } catch (error) {
      console.log(error);
    }

    // put count into button
    const counter = document.createElement("span");
    counter.id = "hypothesis_count";
    counter.innerHTML = count;
    button.title = "View " + count + " Hypothesis annotations";
    button.append(counter);
  }

  // when button is clicked
  function onButtonClick(event, button) {
    const script = document.createElement("script");
    script.src = embed;
    document.body.append(script);
    button.remove();
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- hypothesis icon -->

<template class="icon_hypothesis">
  <!-- modified from: https://simpleicons.org/icons/hypothesis.svg / https://git.io/Jf1VB -->
  <svg width="16" height="16" viewBox="0 0 24 24" tabindex="-1">
    <path
      fill="currentColor"
      d="M3.43 0C2.5 0 1.72 .768 1.72 1.72V18.86C1.72 19.8 2.5 20.57 3.43 20.57H9.38L12 24L14.62 20.57H20.57C21.5 20.57 22.29 19.8 22.29 18.86V1.72C22.29 .77 21.5 0 20.57 0H3.43M5.14 3.43H7.72V9.43S8.58 7.72 10.28 7.72C12 7.72 13.74 8.57 13.74 11.24V17.14H11.16V12C11.16 10.61 10.28 10.07 9.43 10.29C8.57 10.5 7.72 11.41 7.72 13.29V17.14H5.14V3.43M18 13.72C18.95 13.72 19.72 14.5 19.72 15.42A1.71 1.71 0 0 1 18 17.13A1.71 1.71 0 0 1 16.29 15.42C16.29 14.5 17.05 13.71 18 13.71Z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  /* hypothesis activation button */
  #hypothesis_button {
    box-sizing: border-box;
    position: fixed;
    top: 0;
    right: 0;
    width: 60px;
    height: 60px;
    background: #ffffff;
    border-radius: 0;
    border-left: solid 1px #bdbdbd;
    border-bottom: solid 1px #bdbdbd;
    box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
    z-index: 2;
  }

  /* hypothesis button svg */
  #hypothesis_button > svg {
    position: relative;
    top: -4px;
  }

  /* hypothesis annotation count */
  #hypothesis_count {
    position: absolute;
    left: 0;
    right: 0;
    bottom: 5px;
  }

  /* side panel */
  .annotator-frame {
    width: 280px !important;
  }

  /* match highlight color to rest of theme */
  .annotator-highlights-always-on .annotator-hl {
    background-color: #ffeb3b !important;
  }

  /* match focused color to rest of theme */
  .annotator-hl.annotator-hl-focused {
    background-color: #ff8a65 !important;
  }

  /* match bucket bar color to rest of theme */
  .annotator-bucket-bar {
    background: #f5f5f5 !important;
  }

  /* always hide button, toolbar, and tooltip on print */
  @media only print {
    #hypothesis_button {
      display: none;
    }

    .annotator-frame {
      display: none !important;
    }

    hypothesis-adder {
      display: none !important;
    }
  }
</style>
<!-- 
  Mathjax Plugin (third-party) 

  Allows the proper rendering of math/equations written in LaTeX.
  See https://www.mathjax.org/.
-->

<script type="text/x-mathjax-config">
  // configuration
  MathJax.Hub.Config({
    "CommonHTML": { linebreaks: { automatic: true } },
    "HTML-CSS": { linebreaks: { automatic: true } },
    "SVG": { linebreaks: { automatic: true } },
    "fast-preview": { disabled: true }
  });
</script>

<script
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML"
  integrity="sha512-0doc9hKxR3PYwso42RD1p5ySZpzzuDiOwMrdCEh2WdJZCjcmFKc/wEnL+z8fBQrnHoiNWbo+3fiGkOYXBdQp4A=="
  crossorigin="anonymous"
></script>

<style>
  /* mathjax containers */
  .math.display > span:not(.MathJax_Preview) {
    /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
    display: flex !important;
    overflow-x: auto !important;
    overflow-y: hidden !important;
    justify-content: center;
    align-items: center;
    margin: 0 !important;
  }

  /* right click menu */
  .MathJax_Menu {
    border-radius: 5px !important;
    border: solid 1px #bdbdbd !important;
    box-shadow: none !important;
  }

  /* equation auto-number */
  span[id^="eq:"] > span.math.display + span {
    font-weight: 600;
  }

  /* equation */
  span[id^="eq:"] > span.math.display > span {
    /* nudge to make room for equation auto-number and anchor */
    margin-right: 60px !important;
  }
</style>
</body>
</html>
